"use strict";(self.webpackChunknirvana_documentation=self.webpackChunknirvana_documentation||[]).push([[3952],{3905:(n,e,a)=>{a.d(e,{Zo:()=>m,kt:()=>g});var t=a(7294);function i(n,e,a){return e in n?Object.defineProperty(n,e,{value:a,enumerable:!0,configurable:!0,writable:!0}):n[e]=a,n}function o(n,e){var a=Object.keys(n);if(Object.getOwnPropertySymbols){var t=Object.getOwnPropertySymbols(n);e&&(t=t.filter((function(e){return Object.getOwnPropertyDescriptor(n,e).enumerable}))),a.push.apply(a,t)}return a}function l(n){for(var e=1;e<arguments.length;e++){var a=null!=arguments[e]?arguments[e]:{};e%2?o(Object(a),!0).forEach((function(e){i(n,e,a[e])})):Object.getOwnPropertyDescriptors?Object.defineProperties(n,Object.getOwnPropertyDescriptors(a)):o(Object(a)).forEach((function(e){Object.defineProperty(n,e,Object.getOwnPropertyDescriptor(a,e))}))}return n}function r(n,e){if(null==n)return{};var a,t,i=function(n,e){if(null==n)return{};var a,t,i={},o=Object.keys(n);for(t=0;t<o.length;t++)a=o[t],e.indexOf(a)>=0||(i[a]=n[a]);return i}(n,e);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(n);for(t=0;t<o.length;t++)a=o[t],e.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(n,a)&&(i[a]=n[a])}return i}var d=t.createContext({}),s=function(n){var e=t.useContext(d),a=e;return n&&(a="function"==typeof n?n(e):l(l({},e),n)),a},m=function(n){var e=s(n.components);return t.createElement(d.Provider,{value:e},n.children)},p="mdxType",c={inlineCode:"code",wrapper:function(n){var e=n.children;return t.createElement(t.Fragment,{},e)}},u=t.forwardRef((function(n,e){var a=n.components,i=n.mdxType,o=n.originalType,d=n.parentName,m=r(n,["components","mdxType","originalType","parentName"]),p=s(a),u=i,g=p["".concat(d,".").concat(u)]||p[u]||c[u]||o;return a?t.createElement(g,l(l({ref:e},m),{},{components:a})):t.createElement(g,l({ref:e},m))}));function g(n,e){var a=arguments,i=e&&e.mdxType;if("string"==typeof n||i){var o=a.length,l=new Array(o);l[0]=u;var r={};for(var d in e)hasOwnProperty.call(e,d)&&(r[d]=e[d]);r.originalType=n,r[p]="string"==typeof n?n:i,l[1]=r;for(var s=2;s<o;s++)l[s]=a[s];return t.createElement.apply(null,l)}return t.createElement.apply(null,a)}u.displayName="MDXCreateElement"},495:(n,e,a)=>{a.r(e),a.d(e,{contentTitle:()=>l,default:()=>p,frontMatter:()=>o,metadata:()=>r,toc:()=>d});var t=a(7462),i=(a(7294),a(3905));const o={title:"Getting Started with DRAGEN"},l=void 0,r={unversionedId:"introduction/getting-started-dragen",id:"version-3.25/introduction/getting-started-dragen",title:"Getting Started with DRAGEN",description:"Illumina Connected Annotations, comes bundled with DRAGEN. Users can annotate VCF files by enabling annotation on the",source:"@site/versioned_docs/version-3.25/introduction/getting-started-dragen.md",sourceDirName:"introduction",slug:"/introduction/getting-started-dragen",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/introduction/getting-started-dragen",editUrl:"https://github.com/Illumina/IlluminaConnectedAnnotationsDocumentation/edit/master/versioned_docs/version-3.25/introduction/getting-started-dragen.md",tags:[],version:"3.25",frontMatter:{title:"Getting Started with DRAGEN"},sidebar:"docs",previous:{title:"Getting Started",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/introduction/getting-started"},next:{title:"Parsing Illumina Connected Annotations JSON",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/introduction/parsing-json"}},d=[{value:"Downloading Annotation Data Files",id:"downloading-annotation-data-files",children:[{value:"Credentials",id:"credentials",children:[],level:3},{value:"Download Sources",id:"download-sources",children:[],level:3}],level:2},{value:"Annotate Files (via DRAGEN command-line)",id:"annotate-files-via-dragen-command-line",children:[],level:2},{value:"Annotate Files (via standalone Illumina Connected Annotations)",id:"annotate-files-via-standalone-illumina-connected-annotations",children:[],level:2},{value:"Output File",id:"output-file",children:[],level:2},{value:"Version History",id:"version-history",children:[],level:2}],s={toc:d},m="wrapper";function p(n){let{components:e,...a}=n;return(0,i.kt)(m,(0,t.Z)({},s,a,{components:e,mdxType:"MDXLayout"}),(0,i.kt)("p",null,"Illumina Connected Annotations, comes bundled with DRAGEN. Users can annotate VCF files by enabling annotation on the\nDRAGEN command-line or by running the standalone tool."),(0,i.kt)("p",null,"By default, the Annotations binaries are located in the ",(0,i.kt)("inlineCode",{parentName:"p"},"<INSTALL_PATH>/share/nirvana")," directory. This directory\nincludes two files:"),(0,i.kt)("ol",null,(0,i.kt)("li",{parentName:"ol"},(0,i.kt)("inlineCode",{parentName:"li"},"Nirvana"),": Illumina Connected Annotations"),(0,i.kt)("li",{parentName:"ol"},(0,i.kt)("inlineCode",{parentName:"li"},"DataManager"),": Manage annotation data sources")),(0,i.kt)("div",{className:"admonition admonition-caution alert alert--warning"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"16",height:"16",viewBox:"0 0 16 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"}))),"caution")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},(0,i.kt)("inlineCode",{parentName:"p"},"Nirvana")," and ",(0,i.kt)("inlineCode",{parentName:"p"},"DataManager")," are compatible with the following platforms:"),(0,i.kt)("ul",{parentName:"div"},(0,i.kt)("li",{parentName:"ul"},"CentOS 7, Oracle 8 and other modern Linux distributions using x64 processors.")))),(0,i.kt)("h2",{id:"downloading-annotation-data-files"},"Downloading Annotation Data Files"),(0,i.kt)("p",null,"Variant annotation in DRAGEN requires additional annotation data files that have to be downloaded prior to running any\nDRAGEN pipeline that requires variant annotation. ",(0,i.kt)("inlineCode",{parentName:"p"},"DataManager")," tool can be used to download those sources. First, the\ncredentials need to be setup to access the data sources."),(0,i.kt)("h3",{id:"credentials"},"Credentials"),(0,i.kt)("p",null,"Credentials are stored in a credentials JSON file. It will contain an Illumina API key to access the platform and DRAGEN\nserial number to access the premium data sources."),(0,i.kt)("ol",null,(0,i.kt)("li",{parentName:"ol"},"To generate Illumina API key, create an Illumina account via\n",(0,i.kt)("a",{parentName:"li",href:"https://accounts.login.illumina.com"},"https://accounts.login.illumina.com"),". For details, please\nfollow ",(0,i.kt)("a",{parentName:"li",href:"./prerequisite"},"this guide")," to obtain the ",(0,i.kt)("inlineCode",{parentName:"li"},"MyIlluminaApiKey"),"."),(0,i.kt)("li",{parentName:"ol"},"To access premium sources, get the DRAGEN serial number using the command below:")),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-shell"},"dragen_info -b | grep Serial\n")),(0,i.kt)("p",null,"Finally, put it together in a ",(0,i.kt)("inlineCode",{parentName:"p"},"credentials.json")," file:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},'{\n  "MyIlluminaApiKey": "<your Illumina account api key>",\n  "DragenSerialNo": "<your DRAGEN server serial no.>"\n}\n')),(0,i.kt)("h3",{id:"download-sources"},"Download Sources"),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"tip")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"Please see the complete ",(0,i.kt)("a",{parentName:"p",href:"../utilities/data-manager"},"DataManager guide"),"."))),(0,i.kt)("p",null,"Once the credential file is created ",(0,i.kt)("inlineCode",{parentName:"p"},"DataManager")," is raedy to be used for downloading data sources. Data sources can be\nexplicitly defined in a JSON format. DRAGEN has a predefined set of version configuration files that can be used to\ndownload the sources. The configurations are located in DRAGEN resources directory ",(0,i.kt)("inlineCode",{parentName:"p"},"<INSTALL_PATH>/resources/annotation"),"\n. There are several JSON config files in the directory, each of the describing the version of each data source that will\nbe used to annotate:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("inlineCode",{parentName:"li"},"all_annotations_GRCh37.json")," : configurations for producing full variant annotation (running DRAGEN with\nparameter ",(0,i.kt)("inlineCode",{parentName:"li"},"--enable-variant-annotation true"),") for GRCh37 assembly"),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("inlineCode",{parentName:"li"},"all_annotations_GRCh38.json")," : configurations for producing full variant annotation (running DRAGEN with\nparameter ",(0,i.kt)("inlineCode",{parentName:"li"},"--enable-variant-annotation true"),") for GRCh38 assembly"),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("inlineCode",{parentName:"li"},"germline_tagging_annotations_GRCh37.json")," : configurations required when running DRAGEN pipeline that perform\ngermline tagging step for GRCh37 assembly"),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("inlineCode",{parentName:"li"},"germline_tagging_annotations_GRCh38.json")," : configurations required when running DRAGEN pipeline that perform\ngermline tagging step for GRCh38 assembly"),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("inlineCode",{parentName:"li"},"tmb_annotations_GRCh37.json")," : configurations required when running DRAGEN pipeline that perform TMB step for GRCh37\nassembly (this file will also contain all data that are defined in ",(0,i.kt)("inlineCode",{parentName:"li"},"germline_tagging_annotations_GRCh37.json"),")"),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("inlineCode",{parentName:"li"},"tmb_annotations_GRCh38.json")," : configurations required when running DRAGEN pipeline that perform TMB step for GRCh38\nassembly (this file will also contain all data that are defined in ",(0,i.kt)("inlineCode",{parentName:"li"},"germline_tagging_annotations_GRCh38.json"),")")),(0,i.kt)("p",null,"Download the data files, for each JSON above, using the command:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-bash"},".<INSTALL_PATH>/share/nirvana/DataManager -r [assembly] --credentials-file [path to credential file] --dir [path to directory for the downloaded data] --versions-config [path to the JSON file in the resources folder]\n")),(0,i.kt)("p",null,"For the ",(0,i.kt)("inlineCode",{parentName:"p"},"--dir")," argument you can use the same directory for all config file so that it is stored in the same directory.\nThis directory path will be used for parameter ",(0,i.kt)("inlineCode",{parentName:"p"},"--variant-annotation-data")," when running DRAGEN."),(0,i.kt)("div",{className:"admonition admonition-caution alert alert--warning"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"16",height:"16",viewBox:"0 0 16 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"}))),"caution")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("ul",{parentName:"div"},(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("p",{parentName:"li"},"If DataManager was run for ",(0,i.kt)("inlineCode",{parentName:"p"},"tmb_annotations_[assembly].json"),", it is not required to run again\nfor ",(0,i.kt)("inlineCode",{parentName:"p"},"germline_tagging_annotations_[assembly].json"),".")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("p",{parentName:"li"},"Data defined in ",(0,i.kt)("inlineCode",{parentName:"p"},"tmb_annotations_[assembly].json")," are needed if DRAGEN is run with ",(0,i.kt)("inlineCode",{parentName:"p"},"--enable-tmb true")," parameter.\nWithout data defined in ",(0,i.kt)("inlineCode",{parentName:"p"},"tmb_annotations_[assembly].json")," available, running DRAGEN will result in error.")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("p",{parentName:"li"},"Data defined in ",(0,i.kt)("inlineCode",{parentName:"p"},"germline_tagging_annotations_[assembly].json")," are needed if DRAGEN is run\nwith ",(0,i.kt)("inlineCode",{parentName:"p"},"--vc-enable-germline-tagging true")," parameter. Without data defined in ",(0,i.kt)("inlineCode",{parentName:"p"},"tmb_annotations_[assembly].json"),"\navailable, running DRAGEN will result in error."))))),(0,i.kt)("p",null,"Below is the output example when running the DataManager to download using ",(0,i.kt)("inlineCode",{parentName:"p"},"all_annotations_GRCh38.json")," config file:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-shell"},"[INSTALL_PATH]/DataManager download -r GRCh38 --credentials-file [path to credential file] --dir [path to dir] --versions-config [path to resource directory]/all_annotations_GRCh38.json\n\n---------------------------------------------------------------------------\nDataManager                                         (c) 2024 Illumina, Inc.\n                                                                     3.25.0\n---------------------------------------------------------------------------\n\nListing annotation files in local directory...\nRequesting remote file information to be downloaded...\nRemote file list received!\nSyncing local files with requested files\nRequesting remote file information to be downloaded...\nStart downloading files.\nDownloading file DANN_20200205.gsa...\nDownloading file Homo_sapiens.GRCh38.Nirvana.dat...\nDownloading file Gerp_20110522.gsa...\nDownloading file PrimateAI_0.2.nsa...\nDownloading file PromoterAI_1.0.esa...\nDownloading file REVEL_20200205.nsa...\nDownloading file GRCh38.RefSeq.ndb...\nDownloading file TOPMed_freeze_5.nsa...\nDownloading file ClinGen_disease_validity_curations_20240910.nga...\nDownloading file ClinGen_Dosage_Sensitivity_Map_20240910.nga...\nDownloading file ClinGen_Dosage_Sensitivity_Map_20240910.nsi...\nDownloading file ClinGen_20160414.nsi...\nDownloading file ClinVar_20240902.nsa...\nDownloading file ClinVar_20240902.nsi...\nDownloading file ClinVarPreview_preview_20240902.nsa...\nDownloading file ClinVarPreview_preview_20240902.nsi...\nDownloading file Cosmic_Cancer_Gene_Census_99.ega...\nDownloading file COSMIC_GeneFusions_99.efj...\nDownloading file COSMIC_99.esa...\nDownloading file dbSNP_156.nsa...\nDownloading file dbSNP_151_globalMinor.nsa...\nDownloading file gnomAD_gene_scores_4.1.nga...\nDownloading file gnomAD_LCR_2.1.lcr...\nDownloading file gnomAD_4.1.nsa...\nDownloading file gnomAD_SV_4.1.nsi...\nDownloading file gnomAD-cnv_4.1.nsi...\nDownloading file gnomAD_exome_4.1.nsa...\nDownloading file MITOMAP_20200819.nsa...\nDownloading file MITOMAP_SV_20200819.nsi...\nDownloading file OMIM_20240910.ega...\nDownloading file 1000_Genomes_Project_Phase_3_v3_plus_refMinor.rma...\nDownloading file 1000_Genomes_Project_Phase_3_v3_plus.nsa...\nDownloading file 1000_Genomes_Project_(SV)_Phase_3_v5a.nsi...\nDownloading file phyloP_hg38.npd...\nDownloading file PhyloPPrimate_1.0.gsa...\nDownloading file PrimateAI-3D_1.0.esa...\nDownloading file SpliceAi_1.3.esa...\n============= Downloading ==============\n\nSpliceAi_1.3.esa (with index)           : [##################################################] 100%\nPhyloPPrimate_1.0.gsa (with index)      : [##################################################] 100%\nPrimateAI-3D_1.0.esa (with index)       : [##################################################] 100%\n======== Download Completed =========   : [##################################################] 100%\nSubmitting download usage...\nSubmitting download usage finished!\nObtaining Data License...\n\n\n---------------------------------------------------------------------------\n                            Download Summary\n---------------------------------------------------------------------------\nDownload success:\nTotal size: 63.61 GB\nDownloaded files:\n- DANN_20200205.gsa.idx\n- PrimateAI_0.2.nsa.idx\n- Gerp_20110522.gsa.idx\n- PrimateAI_0.2.nsa\n- PromoterAI_1.0.esa.idx\n- PromoterAI_1.0.esa\n- Homo_sapiens.GRCh38.Nirvana.dat\n- REVEL_20200205.nsa.idx\n- GRCh38.RefSeq.ndb.idx\n- GRCh38.RefSeq.ndb\n- REVEL_20200205.nsa\n- TOPMed_freeze_5.nsa.idx\n- ClinGen_disease_validity_curations_20240910.nga\n- ClinGen_Dosage_Sensitivity_Map_20240910.nga\n- ClinGen_Dosage_Sensitivity_Map_20240910.nsi\n- ClinGen_20160414.nsi\n- ClinVar_20240902.nsa.idx\n- ClinVar_20240902.nsa\n- ClinVar_20240902.nsi\n- ClinVarPreview_preview_20240902.nsa.idx\n- ClinVarPreview_preview_20240902.nsa\n- ClinVarPreview_preview_20240902.nsi\n- Cosmic_Cancer_Gene_Census_99.ega\n- COSMIC_GeneFusions_99.efj\n- COSMIC_99.esa.idx\n- TOPMed_freeze_5.nsa\n- COSMIC_99.esa\n- dbSNP_156.nsa.idx\n- dbSNP_151_globalMinor.nsa.idx\n- dbSNP_151_globalMinor.nsa\n- gnomAD_gene_scores_4.1.nga\n- gnomAD_LCR_2.1.lcr\n- gnomAD_4.1.nsa.idx\n- Gerp_20110522.gsa\n- gnomAD_SV_4.1.nsi\n- gnomAD-cnv_4.1.nsi\n- gnomAD_exome_4.1.nsa.idx\n- DANN_20200205.gsa\n- MITOMAP_20200819.nsa.idx\n- MITOMAP_20200819.nsa\n- MITOMAP_SV_20200819.nsi\n- OMIM_20240910.ega\n- 1000_Genomes_Project_Phase_3_v3_plus_refMinor.rma.idx\n- 1000_Genomes_Project_Phase_3_v3_plus_refMinor.rma\n- 1000_Genomes_Project_Phase_3_v3_plus.nsa.idx\n- 1000_Genomes_Project_Phase_3_v3_plus.nsa\n- 1000_Genomes_Project_(SV)_Phase_3_v5a.nsi\n- phyloP_hg38.npd.idx\n- dbSNP_156.nsa\n- PhyloPPrimate_1.0.gsa.idx\n- gnomAD_exome_4.1.nsa\n- PrimateAI-3D_1.0.esa.idx\n- phyloP_hg38.npd\n- SpliceAi_1.3.esa.idx\n- PrimateAI-3D_1.0.esa\n- SpliceAi_1.3.esa\n- PhyloPPrimate_1.0.gsa\n- gnomAD_4.1.nsa\nData license saved: /home/nirvana_resources/nirvana_bin/premium.lic\n\nPeak memory usage: 2.514 GB\nTime: 00:04:08.6\n")),(0,i.kt)("h2",{id:"annotate-files-via-dragen-command-line"},"Annotate Files (via DRAGEN command-line)"),(0,i.kt)("p",null,"To automatically annotate output VCFs when running DRAGEN, please add the following command-line arguments:"),(0,i.kt)("table",null,(0,i.kt)("thead",{parentName:"table"},(0,i.kt)("tr",{parentName:"thead"},(0,i.kt)("th",{parentName:"tr",align:null},"Argument"),(0,i.kt)("th",{parentName:"tr",align:null},"Example"),(0,i.kt)("th",{parentName:"tr",align:null},"Description"))),(0,i.kt)("tbody",{parentName:"table"},(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"--enable-variant-annotation"),(0,i.kt)("td",{parentName:"tr",align:null},"true"),(0,i.kt)("td",{parentName:"tr",align:null},"enables annotation if the pipeline supports it")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"--variant-annotation-data"),(0,i.kt)("td",{parentName:"tr",align:null},"/path/to/your/NirvanaData"),(0,i.kt)("td",{parentName:"tr",align:null},"the location where you downloaded the Nirvana annotation files")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"--variant-annotation-assembly"),(0,i.kt)("td",{parentName:"tr",align:null},"GRCh38"),(0,i.kt)("td",{parentName:"tr",align:null},"the genome assembly - either GRCh37 or GRCh38. To annotate hg19 variants, please use GRCh37")))),(0,i.kt)("p",null,"Example DRAGEN parameters for obtaining full annotation data:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-bash"},"--enable-variant-annotation true \\\n--variant-annotation-data /path/to/your/NirvanaData \\\n--variant-annotation-assembly GRCh38\n")),(0,i.kt)("h2",{id:"annotate-files-via-standalone-illumina-connected-annotations"},"Annotate Files (via standalone Illumina Connected Annotations)"),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"tip")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"Please see the complete ",(0,i.kt)("a",{parentName:"p",href:"./getting-started#the-illumina-connected-annotations-command-line"},"Annotation guide"),"."))),(0,i.kt)("p",null,"To annotate an uncompressed VCF (or bgzipped VCF) file, enter the following command:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-shell"},"<INSTALL_PATH>/share/nirvana/Nirvana \\\n -i <input_VCF> \\ \n -o <output_prefix> \\\n -c [path to data dir]/Cache \\\n -r [path to data dir]/References/Homo_sapiens.GRCh38.Nirvana.dat \\\n --sd [path to data dir]/SupplementaryAnnotation/GRCh38 \\\n -l [path to credential file] \\\n --versions-config [path to resource directory]/all_annotations_GRCh38.json \n")),(0,i.kt)("p",null,"The following are the available command line options:"),(0,i.kt)("table",null,(0,i.kt)("thead",{parentName:"table"},(0,i.kt)("tr",{parentName:"thead"},(0,i.kt)("th",{parentName:"tr",align:null},"Option"),(0,i.kt)("th",{parentName:"tr",align:null},"Value"),(0,i.kt)("th",{parentName:"tr",align:null},"Example"),(0,i.kt)("th",{parentName:"tr",align:null},"Description"))),(0,i.kt)("tbody",{parentName:"table"},(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"-i"),(0,i.kt)("td",{parentName:"tr",align:null},"path"),(0,i.kt)("td",{parentName:"tr",align:null},"<input_VCF>"),(0,i.kt)("td",{parentName:"tr",align:null},"Input VCF path")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"-o"),(0,i.kt)("td",{parentName:"tr",align:null},"prefix"),(0,i.kt)("td",{parentName:"tr",align:null},"<output_prefix>"),(0,i.kt)("td",{parentName:"tr",align:null},"Output path prefix")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"-c"),(0,i.kt)("td",{parentName:"tr",align:null},"directory"),(0,i.kt)("td",{parentName:"tr",align:null},"~/Data/Cache/"),(0,i.kt)("td",{parentName:"tr",align:null},"Cache directory")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"-r"),(0,i.kt)("td",{parentName:"tr",align:null},"directory"),(0,i.kt)("td",{parentName:"tr",align:null},"~/Data/References/Homo_sapiens.GRCh38.Nirvana.dat"),(0,i.kt)("td",{parentName:"tr",align:null},"Reference directory")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"--sd"),(0,i.kt)("td",{parentName:"tr",align:null},"directory"),(0,i.kt)("td",{parentName:"tr",align:null},"~/Data/SupplementaryAnnotation/GRCh38"),(0,i.kt)("td",{parentName:"tr",align:null},"Supplementary annotation directory")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"-l"),(0,i.kt)("td",{parentName:"tr",align:null},"file path"),(0,i.kt)("td",{parentName:"tr",align:null},"credentials.json"),(0,i.kt)("td",{parentName:"tr",align:null},"path to credential file")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"--versions-config"),(0,i.kt)("td",{parentName:"tr",align:null},"file path"),(0,i.kt)("td",{parentName:"tr",align:null},"all_annotations_GRCh38.json"),(0,i.kt)("td",{parentName:"tr",align:null},"path to versions config file")))),(0,i.kt)("p",null,"Above example generates the following output."),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-shell"},"---------------------------------------------------------------------------\nIllumina Connected Annotations                      (c) 2024 Illumina, Inc.\n                                                                     3.25.0\n---------------------------------------------------------------------------\n\nData Source              Type                     Status    Comments\n---------------------------------------------------------------------------\nDANN                     Score                    Loaded    20200205\nDECIPHER                 StructuralVariant        Loaded    201509\nEnsembl                  GeneModels               Loaded    112\nFusionCatcher            GeneFusion               Loaded    1.33\nGME                      SmallVariant             Loaded    20160618\nGenomeAssembly           GenomeAssembly           Loaded    GRCh38.p14\nGerp                     Score                    Loaded    20110522\nHGNC                     GeneModels               Loaded    20240603\nMultiZ100Way             Protein                  Loaded    20171006\nPrimateAI                SmallVariant             Loaded    0.2\nPromoterAI               SmallVariant             Loaded    1.0\nREVEL                    SmallVariant             Loaded    20200205\nRefSeq                   GeneModels               Loaded    GCF_000001405.40-RS_2023_10\nTOPMed                   SmallVariant             Loaded    freeze_5\nclingen                  Gene                     Loaded    20240910\nclingen                  StructuralVariant        Loaded    20240910\nclingen (legacy)         StructuralVariant        Loaded    20160414\nclinvar                  SmallVariant             Loaded    20240902\nclinvar                  StructuralVariant        Loaded    20240902\nclinvar-preview          SmallVariant             Loaded    20240902\nclinvar-preview          StructuralVariant        Loaded    20240902\ncosmic                   Gene                     Loaded    99\ncosmic                   GeneFusion               Loaded    99\ncosmic                   SmallVariant             Loaded    99\ndbSNP                    SmallVariant             Loaded    156\nglobalAllele             SmallVariant             Loaded    151\ngnomad                   Gene                     Loaded    4.1\ngnomad                   LowComplexityRegions     Loaded    2.1\ngnomad                   SmallVariant             Loaded    4.1\ngnomad                   StructuralVariant        Loaded    4.1\ngnomad-exome             SmallVariant             Loaded    4.1\nmitomap                  SmallVariant             Loaded    20200819\nmitomap                  StructuralVariant        Loaded    20200819\nomim                     Gene                     Loaded    20240910\noneKg                    RefMinor                 Loaded    Phase 3 v3plus\noneKg                    SmallVariant             Loaded    Phase 3 v3plus\noneKg                    StructuralVariant        Loaded    Phase 3 v5a\nphylopScore              ConservationScore        Loaded    hg38\nphylopScore              Score                    Loaded    1.0\nprimateAI-3D             SmallVariant             Loaded    1.0\nspliceAI                 SmallVariant             Loaded    1.3\n---------------------------------------------------------------------------\n\nInitialization                                         Time     Positions/s\n---------------------------------------------------------------------------\nCache                                               00:00:03.6\nSA initialization                                   00:00:08.0\nVCF Position Scan                                   00:00:00.0      115,442\n\nReference                             SA Loading    Annotation   Variants/s\n---------------------------------------------------------------------------\nchr1                                    00:00:00.5  00:00:02.0        4,980\n\nSummary                                                Time         Percent\n---------------------------------------------------------------------------\nInitialization                                      00:00:11.7       84.9 %\nAnnotation                                          00:00:02.0       14.4 %\n |--SA Loading                                      00:00:00.5        4.2 %\n |--Output writing                                  00:00:00.3        2.2 %\nLicense Check                                       00:00:00.0        0.0 %\nUsage Report                                        00:00:00.0        0.0 %\n---------------------------------------------------------------------------\nGross Annotation rate:                                  718.84 positions/sec\n\nPeak memory usage: 5.867 GB\nTime: 00:00:13.8\n")),(0,i.kt)("h2",{id:"output-file"},"Output File"),(0,i.kt)("p",null,"Illumina Connected Annotation produces an output file in JSON format by default.",(0,i.kt)("br",{parentName:"p"}),"\n","Please refer to ",(0,i.kt)("a",{parentName:"p",href:"../file-formats/illumina-annotator-json-file-format"},"Illumina Connected Annotations JSON")," for detailed\ndescription of the JSON file."),(0,i.kt)("p",null,"A VCF output format is also possible by ",(0,i.kt)("inlineCode",{parentName:"p"},"--output-format vcf")," command line parameter. VCF annotations are limited.\nPlease refer to ",(0,i.kt)("a",{parentName:"p",href:"../file-formats/illumina-annotator-vcf-file-format"},"Illumina Connected Annotations VCF")," for detailed\ndescription of the VCF file."),(0,i.kt)("h2",{id:"version-history"},"Version History"),(0,i.kt)("p",null,"Annotations binaries have been included with DRAGEN since v3.5. The table below indicates which version of Annotations\nbinaries were included with different DRAGEN releases, and their AI annotation capabilities."),(0,i.kt)("p",null,"The Annotations binaries distributed with DRAGEN can not be changed. Newer versions of Annotations are backward\ncompatible, and can therefore annotate output files from older DRAGEN releases."),(0,i.kt)("table",null,(0,i.kt)("thead",{parentName:"table"},(0,i.kt)("tr",{parentName:"thead"},(0,i.kt)("th",{parentName:"tr",align:"left"},"DRAGEN version(s)"),(0,i.kt)("th",{parentName:"tr",align:"left"},"Annotations version"),(0,i.kt)("th",{parentName:"tr",align:"left"},"AI annotations"))),(0,i.kt)("tbody",{parentName:"table"},(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:"left"},"4.4"),(0,i.kt)("td",{parentName:"tr",align:"left"},"3.25"),(0,i.kt)("td",{parentName:"tr",align:"left"},"spliceAI, primateAI3D")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:"left"},"4.3"),(0,i.kt)("td",{parentName:"tr",align:"left"},"3.23"),(0,i.kt)("td",{parentName:"tr",align:"left"},"spliceAI, primateAI3D")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:"left"},"3.9, 3.10, 4.0, 4.1, 4.2"),(0,i.kt)("td",{parentName:"tr",align:"left"},"3.16.1"),(0,i.kt)("td",{parentName:"tr",align:"left"},"spliceAI, primateAI")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:"left"},"3.8"),(0,i.kt)("td",{parentName:"tr",align:"left"},"3.14"),(0,i.kt)("td",{parentName:"tr",align:"left"},"spliceAI, primateAI")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:"left"},"3.6, 3.7"),(0,i.kt)("td",{parentName:"tr",align:"left"},"3.9.0"),(0,i.kt)("td",{parentName:"tr",align:"left"},"spliceAI, primateAI")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:"left"},"3.5"),(0,i.kt)("td",{parentName:"tr",align:"left"},"3.6.0"),(0,i.kt)("td",{parentName:"tr",align:"left"},"spliceAI, primateAI")))))}p.isMDXComponent=!0}}]);