"use strict";(self.webpackChunknirvana_documentation=self.webpackChunknirvana_documentation||[]).push([[3464,900,6498,6914,6481,8658,374,204],{3905:(t,e,n)=>{n.d(e,{Zo:()=>s,kt:()=>c});var a=n(7294);function l(t,e,n){return e in t?Object.defineProperty(t,e,{value:n,enumerable:!0,configurable:!0,writable:!0}):t[e]=n,t}function r(t,e){var n=Object.keys(t);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(t);e&&(a=a.filter((function(e){return Object.getOwnPropertyDescriptor(t,e).enumerable}))),n.push.apply(n,a)}return n}function i(t){for(var e=1;e<arguments.length;e++){var n=null!=arguments[e]?arguments[e]:{};e%2?r(Object(n),!0).forEach((function(e){l(t,e,n[e])})):Object.getOwnPropertyDescriptors?Object.defineProperties(t,Object.getOwnPropertyDescriptors(n)):r(Object(n)).forEach((function(e){Object.defineProperty(t,e,Object.getOwnPropertyDescriptor(n,e))}))}return t}function o(t,e){if(null==t)return{};var n,a,l=function(t,e){if(null==t)return{};var n,a,l={},r=Object.keys(t);for(a=0;a<r.length;a++)n=r[a],e.indexOf(n)>=0||(l[n]=t[n]);return l}(t,e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(t);for(a=0;a<r.length;a++)n=r[a],e.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(t,n)&&(l[n]=t[n])}return l}var m=a.createContext({}),p=function(t){var e=a.useContext(m),n=e;return t&&(n="function"==typeof t?t(e):i(i({},e),t)),n},s=function(t){var e=p(t.components);return a.createElement(m.Provider,{value:e},t.children)},u="mdxType",d={inlineCode:"code",wrapper:function(t){var e=t.children;return a.createElement(a.Fragment,{},e)}},N=a.forwardRef((function(t,e){var n=t.components,l=t.mdxType,r=t.originalType,m=t.parentName,s=o(t,["components","mdxType","originalType","parentName"]),u=p(n),N=l,c=u["".concat(m,".").concat(N)]||u[N]||d[N]||r;return n?a.createElement(c,i(i({ref:e},s),{},{components:n})):a.createElement(c,i({ref:e},s))}));function c(t,e){var n=arguments,l=e&&e.mdxType;if("string"==typeof t||l){var r=n.length,i=new Array(r);i[0]=N;var o={};for(var m in e)hasOwnProperty.call(e,m)&&(o[m]=e[m]);o.originalType=t,o[u]="string"==typeof t?t:l,i[1]=o;for(var p=2;p<r;p++)i[p]=n[p];return a.createElement.apply(null,i)}return a.createElement.apply(null,n)}N.displayName="MDXCreateElement"},731:(t,e,n)=>{n.r(e),n.d(e,{contentTitle:()=>i,default:()=>u,frontMatter:()=>r,metadata:()=>o,toc:()=>m});var a=n(7462),l=(n(7294),n(3905));const r={},i=void 0,o={unversionedId:"data-sources/gnomad-lof-json",id:"version-3.25/data-sources/gnomad-lof-json",title:"gnomad-lof-json",description:"| Field | Type  | Notes                                                                                                                        |",source:"@site/versioned_docs/version-3.25/data-sources/gnomad-lof-json.md",sourceDirName:"data-sources",slug:"/data-sources/gnomad-lof-json",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/gnomad-lof-json",editUrl:"https://github.com/Illumina/IlluminaConnectedAnnotationsDocumentation/edit/master/versioned_docs/version-3.25/data-sources/gnomad-lof-json.md",tags:[],version:"3.25",frontMatter:{}},m=[],p={toc:m},s="wrapper";function u(t){let{components:e,...n}=t;return(0,l.kt)(s,(0,a.Z)({},p,n,{components:e,mdxType:"MDXLayout"}),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-json"},'"gnomAD":{ \n   "pLi":1.00e0,\n   "pNull":8.94e-40,\n   "pRec":1.84e-16,\n   "synZ":-8.44e-2,\n   "misZ":5.96e-1,\n   "loeuf":1.13e0\n}\n')),(0,l.kt)("table",null,(0,l.kt)("thead",{parentName:"table"},(0,l.kt)("tr",{parentName:"thead"},(0,l.kt)("th",{parentName:"tr",align:null},"Field"),(0,l.kt)("th",{parentName:"tr",align:null},"Type"),(0,l.kt)("th",{parentName:"tr",align:null},"Notes"))),(0,l.kt)("tbody",{parentName:"table"},(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"pLi"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"probability of being intolerant of a single loss-of-function variant (like haploinsufficient genes, observed ~ 0.1*expected)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"pNull"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"probability of being completely tolerant of loss of function variation (observed = expected)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"pRec"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"probability of being intolerant of two loss of function variants (like recessive genes, observed ~ 0.5*expected)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"synZ"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"corrected synonymous Z score")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"misZ"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"corrected missense Z score")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"loeuf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"loss of function observed/expected upper bound fraction (LOEUF)")))))}u.isMDXComponent=!0},9291:(t,e,n)=>{n.r(e),n.d(e,{contentTitle:()=>i,default:()=>u,frontMatter:()=>r,metadata:()=>o,toc:()=>m});var a=n(7462),l=(n(7294),n(3905));const r={},i=void 0,o={unversionedId:"data-sources/gnomad-small-variants-json",id:"version-3.25/data-sources/gnomad-small-variants-json",title:"gnomad-small-variants-json",description:"| Field               | Type   | Notes                                                                                            |",source:"@site/versioned_docs/version-3.25/data-sources/gnomad-small-variants-json.md",sourceDirName:"data-sources",slug:"/data-sources/gnomad-small-variants-json",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/gnomad-small-variants-json",editUrl:"https://github.com/Illumina/IlluminaConnectedAnnotationsDocumentation/edit/master/versioned_docs/version-3.25/data-sources/gnomad-small-variants-json.md",tags:[],version:"3.25",frontMatter:{}},m=[],p={toc:m},s="wrapper";function u(t){let{components:e,...n}=t;return(0,l.kt)(s,(0,a.Z)({},p,n,{components:e,mdxType:"MDXLayout"}),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-json"},'"gnomad":{ \n   "coverage":20,\n   "allAf":0.190317,\n   "maleAf":0.193,\n   "femaleAf": 0.1935,\n   "afrAf":0.222876,\n   "amrAf":0.121394,\n   "easAf":0.239802,\n   "finAf":0.136833,\n   "nfeAf":0.181282,\n   "asjAf":0.258278,\n   "othAf":0.186094,\n   "allAn":30796,\n   "maleAn":15096,\n   "femaleAn":15700\n   "afrAn":8664,\n   "amrAn":832,\n   "easAn":1618,\n   "finAn":3486,\n   "nfeAn":14916,\n   "asjAn":302,\n   "othAn":978,\n   "allAc":5861,\n   "maleAc":2930,\n   "femaleAc": 2931,\n   "afrAc":1931,\n   "amrAc":101,\n   "easAc":388,\n   "finAc":477,\n   "nfeAc":2704,\n   "asjAc":78,\n   "othAc":182,\n   "allHc":561,\n   "afrHc":208,\n   "amrHc":6,\n   "easHc":42,\n   "finHc":31,\n   "nfeHc":242,\n   "asjHc":13,\n   "othHc":19,\n   "maleHc":280,\n   "femaleHc":281,\n   "controlsAllAf":0.190317,\n   "controlsAllAn":30796,\n   "controlsAllAc":5861,\n   "lowComplexityRegion":true,\n   "failedFilter":true\n}\n')),(0,l.kt)("table",null,(0,l.kt)("thead",{parentName:"table"},(0,l.kt)("tr",{parentName:"thead"},(0,l.kt)("th",{parentName:"tr",align:null},"Field"),(0,l.kt)("th",{parentName:"tr",align:null},"Type"),(0,l.kt)("th",{parentName:"tr",align:null},"Notes"))),(0,l.kt)("tbody",{parentName:"table"},(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"coverage"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"average coverage (non-negative integer values)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"allAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for all populations. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for male population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for female population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"controlsAllAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the controls subset. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"allAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for all populations. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for male population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for female population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"controlsAllAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the controls subset. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"allAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for all populations. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for male population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for female population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"controlsAllAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the controls subset. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"allHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for all populations. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for male population. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for female population. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"afrAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the African / African American population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"afrAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the African / African American population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"afrAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the African / African American population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"afrHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for African / African American population. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amrAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the Latino population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amrAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the Latino population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amrAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the Latino population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amrHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for Latino population. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"easAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the East Asian population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"easAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the East Asian population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"easAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the East Asian population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"easHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for East Asian population. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"finAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the Finnish population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"finAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the Finnish population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"finAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the Finnish population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"finHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for Finnish population. Non-negative integer")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"nfeAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the Non-Finnish European population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"nfeAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the Non-Finnish European population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"nfeAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the Non-Finnish European population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"nfeHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for Non-Finnish European population. Non-negative integer")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"othAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the Other population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"othAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the Other population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"othAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the Other population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"othHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for Other population. Non-negative integer")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"asjAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the Ashkenazi Jewish population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"asjAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the Ashkenazi Jewish population Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"asjAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the Ashkenazi Jewish population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"asjHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for the Ashkenazi Jewish population. Non-negative integer")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"sasAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the South Asian population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"sasAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the South Asian population Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"sasAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the South Asian population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"sasHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for the South Asian population. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"failedFilter"),(0,l.kt)("td",{parentName:"tr",align:null},"bool"),(0,l.kt)("td",{parentName:"tr",align:null},"True if this variant failed any filters (Note: we do not list the failed filters)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"lowComplexityRegion"),(0,l.kt)("td",{parentName:"tr",align:null},"bool"),(0,l.kt)("td",{parentName:"tr",align:null},"True if this variant is located in a low complexity region.")))))}u.isMDXComponent=!0},665:(t,e,n)=>{n.r(e),n.d(e,{contentTitle:()=>i,default:()=>u,frontMatter:()=>r,metadata:()=>o,toc:()=>m});var a=n(7462),l=(n(7294),n(3905));const r={},i=void 0,o={unversionedId:"data-sources/gnomad-structural-variants-data_description",id:"version-3.25/data-sources/gnomad-structural-variants-data_description",title:"gnomad-structural-variants-data_description",description:"Bed Example",source:"@site/versioned_docs/version-3.25/data-sources/gnomad-structural-variants-data_description.md",sourceDirName:"data-sources",slug:"/data-sources/gnomad-structural-variants-data_description",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/gnomad-structural-variants-data_description",editUrl:"https://github.com/Illumina/IlluminaConnectedAnnotationsDocumentation/edit/master/versioned_docs/version-3.25/data-sources/gnomad-structural-variants-data_description.md",tags:[],version:"3.25",frontMatter:{}},m=[{value:"Bed Example",id:"bed-example",children:[],level:4},{value:"Structural Variant Type Mapping",id:"structural-variant-type-mapping",children:[],level:4}],p={toc:m},s="wrapper";function u(t){let{components:e,...n}=t;return(0,l.kt)(s,(0,a.Z)({},p,n,{components:e,mdxType:"MDXLayout"}),(0,l.kt)("h4",{id:"bed-example"},"Bed Example"),(0,l.kt)("p",null,"The bed file was obtained from original source for GRCh37"),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-scss"},"#chrom  start   end name    svtype  ALGORITHMS  BOTHSIDES_SUPPORT   CHR2    CPX_INTERVALS   CPX_TYPE    END2    ENDEVIDENCE HIGH_SR_BACKGROUND  PCRPLUS_DEPLETED    PESR_GT_OVERDISPERSION  POS2    PROTEIN_CODING__COPY_GAIN   PROTEIN_CODING__DUP_LOF PROTEIN_CODING__DUP_PARTIAL PROTEIN_CODING__INTERGENIC  PROTEIN_CODING__INTRONIC    PROTEIN_CODING__INV_SPAN    PROTEIN_CODING__LOF PROTEIN_CODING__MSV_EXON_OVR    PROTEIN_CODING__NEAREST_TSS PROTEIN_CODING__PROMOTER    PROTEIN_CODING__UTR SOURCE  STRANDS SVLEN   SVTYPE  UNRESOLVED_TYPE UNSTABLE_AF_PCRPLUS VARIABLE_ACROSS_BATCHES AN  AC  AF  N_BI_GENOS  N_HOMREF    N_HET   N_HOMALT    FREQ_HOMREF FREQ_HET    FREQ_HOMALT MALE_AN MALE_AC MALE_AF MALE_N_BI_GENOS MALE_N_HOMREF   MALE_N_HET  MALE_N_HOMALT   MALE_FREQ_HOMREF    MALE_FREQ_HET   MALE_FREQ_HOMALT    MALE_N_HEMIREF  MALE_N_HEMIALT  MALE_FREQ_HEMIREF   MALE_FREQ_HEMIALT   PAR FEMALE_AN   FEMALE_AC   FEMALE_AF   FEMALE_N_BI_GENOS   FEMALE_N_HOMREF FEMALE_N_HET    FEMALE_N_HOMALT FEMALE_FREQ_HOMREF  FEMALE_FREQ_HET FEMALE_FREQ_HOMALT  POPMAX_AF   AFR_AN  AFR_AC  AFR_AF  AFR_N_BI_GENOS  AFR_N_HOMREF    AFR_N_HET   AFR_N_HOMALT    AFR_FREQ_HOMREF AFR_FREQ_HEAFR_FREQ_HOMALT  AFR_MALE_AN AFR_MALE_AC AFR_MALE_AF AFR_MALE_N_BI_GENOS AFR_MALE_N_HOMREF   AFR_MALE_N_HET  AFR_MALE_N_HOMALT   AFR_MALE_FREQ_HOMREF    AFR_MALE_FREQ_HET   AFR_MALE_FREQ_HOMALT    AFR_MALE_N_HEMIREF  AFR_MALE_N_HEMIALT  AFR_MALE_FREQ_HEMIREF   AFR_MALE_FREQ_HEMIALT   AFR_FEMALE_AN   AFR_FEMALE_AC   AFR_FEMALE_AF   AFR_FEMALE_N_BI_GENOS   AFR_FEMALE_N_HOMREF AFR_FEMALE_N_HET    AFR_FEMALE_N_HOMALT AFR_FEMALE_FREQ_HOMREF  AFR_FEMALE_FREQ_HET AFR_FEMALE_FREQ_HOMALT  AMR_AN  AMR_AC  AMR_AF  AMR_N_BI_GENOS  AMR_N_HOMREF    AMR_N_HET   AMR_N_HOMALT    AMR_FREQ_HOMREF AMR_FREQ_HET    AMR_FREQ_HOMALT AMR_MALE_AN AMR_MALE_AC AMR_MALE_AF AMR_MALE_N_BI_GENOS AMR_MALE_N_HOMREF   AMR_MALE_N_HET  AMR_MALE_N_HOMALT   AMR_MALE_FREQ_HOMREF    AMR_MALE_FREQ_HET   AMR_MALE_FREQ_HOMALT    AMR_MALE_N_HEMIREF  AMR_MALE_N_HEMIALT  AMR_MALE_FREQ_HEMIREF   AMR_MALE_FREQ_HEMIALT   AMR_FEMALE_AN   AMR_FEMALE_AC   AMR_FEMALE_AF   AMR_FEMALE_N_BI_GENOS   AMR_FEMALE_N_HOMREF AMR_FEMALE_N_HET    AMR_FEMALE_N_HOMALT AMR_FEMALE_FREQ_HOMREF  AMR_FEMALE_FREQ_HET AMR_FEMALE_FREQ_HOMALT  EAS_AN  EAS_AC  EAS_AF  EAS_N_BI_GENOS  EAS_N_HOMREF    EAS_N_HET   EAS_N_HOMALT    EAS_FREQ_HOMREF EAS_FREQ_HET    EAS_FREQ_HOMALT EAS_MALE_AN EAS_MALE_AC EAS_MALE_AF EAS_MALE_N_BI_GENOS EAS_MALE_N_HOMREF   EAS_MALE_N_HET  EAS_MALE_N_HOMALT   EAS_MALE_FREQ_HOMREF    EAS_MALE_FREQ_HET   EAS_MALE_FREQ_HOMALT    EAS_MALE_N_HEMIREF  EAS_MALE_N_HEMIALT  EAS_MALE_FREQ_HEMIREF   EAS_MALE_FREQ_HEMIALT   EAS_FEMALE_AN   EAS_FEMALE_AC   EAS_FEMALE_AF   EAS_FEMALE_N_BI_GENOS   EAS_FEMALE_N_HOMREF EAS_FEMALE_N_HET    EAS_FEMALE_N_HOMALT EAS_FEMALE_FREQ_HOMREF  EAS_FEMALE_FREQ_HET EAS_FEMALE_FREQ_HOMALT  EUR_AN  EUR_AC  EUR_AF  EUR_N_BI_GENOS  EUR_N_HOMREF    EUR_N_HET   EUR_N_HOMALT    EUR_FREQ_HOMREF EUR_FREQ_HET    EUR_FREQ_HOMALT EUR_MALE_AN EUR_MALE_AC EUR_MALE_AF EUR_MALE_N_BI_GENOS EUR_MALE_N_HOMREF   EUR_MALE_N_HET  EUR_MALE_N_HOMALT   EUR_MALE_FREQ_HOMREF    EUR_MALE_FREQ_HET   EUR_MALE_FREQ_HOMALT    EUR_MALE_N_HEMIREF  EUR_MALE_N_HEMIALT  EUR_MALE_FREQ_HEMIREF   EUR_MALE_FREQ_HEMIALT   EUR_FEMALE_AN   EUR_FEMALE_AC   EUR_FEMALE_AF   EUR_FEMALE_N_BI_GENOS   EUR_FEMALE_N_HOMREF EUR_FEMALE_N_HET    EUR_FEMALE_N_HOMALT EUR_FEMALE_FREQ_HOMREF  EUR_FEMALE_FREQ_HET EUR_FEMALE_FREQ_HOMALT  OTH_AN  OTH_AC  OTH_AF  OTH_N_BI_GENOS  OTH_N_HOMREF    OTH_N_HET   OTH_N_HOMALT    OTH_FREQ_HOMREF OTH_FREQ_HET    OTH_FREQ_HOMALT OTH_MALE_AN OTH_MALE_AC OTH_MALE_AF OTH_MALE_N_BI_GENOS OTH_MALE_N_HOMREF   OTH_MALE_N_HET  OTH_MALE_N_HOMALT   OTH_MALE_FREQ_HOMREF    OTH_MALE_FREQ_HET   OTH_MALE_FREQ_HOMALT    OTH_MALE_N_HEMIREF  OTH_MALE_N_HEMIALT  OTH_MALE_FREQ_HEMIREF   OTH_MALE_FREQ_HEMIALT   OTH_FEMALE_AN   OTH_FEMALE_AC   OTH_FEMALE_AF   OTH_FEMALE_N_BI_GENOS   OTH_FEMALE_N_HOMREF OTH_FEMALE_N_HET    OTH_FEMALE_N_HOMALT OTH_FEMALE_FREQ_HOMREF  OTH_FEMALE_FREQ_HET OTH_FEMALE_FREQ_HOMALT  FILTER\n1   10641   10642   gnomAD-SV_v2.1_BND_1_1  BND manta   False   15  NA  NA  10643   10643   PE,SR   False   False   True    10642   NA  NA  NA  False   NA  NA  NA  NA  NA  NA  NA  NA  NA  -1  BND SINGLE_ENDER_-- False   False   21366   145 0.006785999983549118    10683   10543   135 5   0.9868950247764587  0.012636899948120117    0.00046803298755548894  10866   69  0.00634999992325902 5433    5366    65  2   0.987667977809906   0.011963900178670883    0.000368120992789045    NA  NA  NA  NA  False   10454   76  0.007269999943673615227 5154    70  3   0.9860339760780334  0.013392000459134579    0.0005739430198445916   0.015956999734044075    93972   0.007660999894142151    4699    4629    68  2   0.9851030111312866  0.014471200294792652    0.0004256220126990229   5154    33  0.006403000093996525    2577    2544    33  0   0.9871940016746521  0.012805599719285965    0.0NA   NA  NA  NA  4232    39  0.009216000325977802    2116    2079    35  2   0.9825140237808228  0.01654059998691082 0.0009451800142414868   1910    7   0.003664999967440963    955 949 5   1   0.9937170147895813  0.00523559981957078 0.001047119963914156    950 4   0.004211000166833401    475 472 2   1   0.9936839938163757  0.00421052984893322 0.0021052600350230932   NA  NA  NA  NA  952 3   0.0031510000117123127   476473  3   0   0.9936969876289368  0.006302520167082548    0.0 2296    31  0.013501999899744987    1148    11131   0   0.9729970097541809  0.02700350061058998 0.0 1312    13  0.009909000247716904    656 643 13  0.9801830053329468  0.01981710083782673 0.0 NA  NA  NA  NA  976 18  0.018442999571561813    488470  18  0   0.9631149768829346  0.03688519820570946 0.0 7574    32  0.004224999807775021    3787    37528   2   0.9920780062675476  0.007393720094114542    0.0005281229969114065   3374    17  0.005038999952375889    1681671 15  1   0.9905160069465637  0.008891520090401173    0.000592768017668277    NA  NA  NA  NA  41815   0.003587000072002411    2091    2077    13  1   0.9933050274848938  0.006217120215296745    0.00047823999193497188  3   0.015956999734044075    94  91  3   0   0.968084990978241   0.03191490098834038 0.0 76  0.026316000148653984    38  36  2   0   0.9473680257797241  0.05263160169124603 0.0 NA  NA  NA  NA  112 1   0.008929000236093998    56  55  1   0   0.982142984867096   0.017857100814580917    0.0UNRESOLVED      \n")),(0,l.kt)("h4",{id:"structural-variant-type-mapping"},"Structural Variant Type Mapping"),(0,l.kt)("p",null,"The source files represented the structural variants with keys using various naming conventions.\nIn the Illumina Connected Annotations JSON output, these keys will be mapped according to the following. "),(0,l.kt)("table",null,(0,l.kt)("thead",{parentName:"table"},(0,l.kt)("tr",{parentName:"thead"},(0,l.kt)("th",{parentName:"tr",align:null},"Illumina Connected Annotations JSON SV Type Key"),(0,l.kt)("th",{parentName:"tr",align:null},"GRCh37 Source SV Type Key"))),(0,l.kt)("tbody",{parentName:"table"},(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"copy_number_variation"),(0,l.kt)("td",{parentName:"tr",align:null})),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"deletion"),(0,l.kt)("td",{parentName:"tr",align:null},"DEL, CN=0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"duplication"),(0,l.kt)("td",{parentName:"tr",align:null},"DUP")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"insertion"),(0,l.kt)("td",{parentName:"tr",align:null},"INS")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"inversion"),(0,l.kt)("td",{parentName:"tr",align:null},"INV")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"mobile_element_insertion"),(0,l.kt)("td",{parentName:"tr",align:null},"INS:ME")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"mobile_element_insertion"),(0,l.kt)("td",{parentName:"tr",align:null},"INS:ME:ALU")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"mobile_element_insertion"),(0,l.kt)("td",{parentName:"tr",align:null},"INS:ME:LINE1")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"mobile_element_insertion"),(0,l.kt)("td",{parentName:"tr",align:null},"INS:ME:SVA")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"structural alteration"),(0,l.kt)("td",{parentName:"tr",align:null})),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"complex_structural_alteration"),(0,l.kt)("td",{parentName:"tr",align:null},"CPX")))))}u.isMDXComponent=!0},9025:(t,e,n)=>{n.r(e),n.d(e,{contentTitle:()=>i,default:()=>u,frontMatter:()=>r,metadata:()=>o,toc:()=>m});var a=n(7462),l=(n(7294),n(3905));const r={},i=void 0,o={unversionedId:"data-sources/gnomad-structural-variants-json",id:"version-3.25/data-sources/gnomad-structural-variants-json",title:"gnomad-structural-variants-json",description:"| Field             | Type            | Notes                                                                             |",source:"@site/versioned_docs/version-3.25/data-sources/gnomad-structural-variants-json.md",sourceDirName:"data-sources",slug:"/data-sources/gnomad-structural-variants-json",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/gnomad-structural-variants-json",editUrl:"https://github.com/Illumina/IlluminaConnectedAnnotationsDocumentation/edit/master/versioned_docs/version-3.25/data-sources/gnomad-structural-variants-json.md",tags:[],version:"3.25",frontMatter:{}},m=[],p={toc:m},s="wrapper";function u(t){let{components:e,...n}=t;return(0,l.kt)(s,(0,a.Z)({},p,n,{components:e,mdxType:"MDXLayout"}),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-json"},'"gnomAD-preview": [\n  {\n    "chromosome": "1",\n    "begin": 40001,\n    "end": 47200,\n    "variantId": "gnomAD-SV_v2.1_DUP_1_1",\n    "variantType": "duplication",\n    "failedFilter": true,\n    "allAf": 0.068963,\n    "afrAf": 0.135694,\n    "amrAf": 0.022876,\n    "easAf": 0.01101,\n    "eurAf": 0.007846,\n    "othAf": 0.017544,\n    "femaleAf": 0.065288,\n    "maleAf": 0.07255,\n    "allAc": 943,\n    "afrAc": 866,\n    "amrAc": 21,\n    "easAc": 17,\n    "eurAc": 37,\n    "othAc": 2,\n    "femaleAc": 442,\n    "maleAc": 499,\n    "allAn": 13674,\n    "afrAn": 6382,\n    "amrAn": 918,\n    "easAn": 1544,\n    "eurAn": 4716,\n    "othAn": 114,\n    "femaleAn": 6770,\n    "maleAn": 6878,\n    "allHc": 91,\n    "afrHc": 90,\n    "amrHc": 1,\n    "easHc": 0,\n    "eurHc": 0,\n    "othHc": 55,\n    "femaleHc": 44,\n    "maleHc": 47,\n    "reciprocalOverlap": 0.01839,\n    "annotationOverlap": 0.16667\n  }\n]\n\n')),(0,l.kt)("table",null,(0,l.kt)("thead",{parentName:"table"},(0,l.kt)("tr",{parentName:"thead"},(0,l.kt)("th",{parentName:"tr",align:null},"Field"),(0,l.kt)("th",{parentName:"tr",align:null},"Type"),(0,l.kt)("th",{parentName:"tr",align:null},"Notes"))),(0,l.kt)("tbody",{parentName:"table"},(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"chromosome"),(0,l.kt)("td",{parentName:"tr",align:null},"string"),(0,l.kt)("td",{parentName:"tr",align:null},"chromosome number")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"begin"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"position interval start")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"end"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"position internal end")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"variantType"),(0,l.kt)("td",{parentName:"tr",align:null},"string"),(0,l.kt)("td",{parentName:"tr",align:null},"structural variant type")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"variantId"),(0,l.kt)("td",{parentName:"tr",align:null},"string"),(0,l.kt)("td",{parentName:"tr",align:null},"gnomAD ID")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"allAf"),(0,l.kt)("td",{parentName:"tr",align:null},"floating point"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for all populations. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"afrAf"),(0,l.kt)("td",{parentName:"tr",align:null},"floating point"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the African super population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amrAf"),(0,l.kt)("td",{parentName:"tr",align:null},"floating point"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the Ad Mixed American super population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"easAf"),(0,l.kt)("td",{parentName:"tr",align:null},"floating point"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the East Asian super population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"eurAf"),(0,l.kt)("td",{parentName:"tr",align:null},"floating point"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the European super population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"othAf"),(0,l.kt)("td",{parentName:"tr",align:null},"floating point"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for all other populations. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleAf"),(0,l.kt)("td",{parentName:"tr",align:null},"floating point"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for female population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleAf"),(0,l.kt)("td",{parentName:"tr",align:null},"floating point"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for male population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"allAc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for all populations.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"afrAc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the African super population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amrAc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the Ad Mixed American super population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"easAc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the East Asian super population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"eurAc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the European super population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"othAc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for all other populations.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleAc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for male population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleAc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for female population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"allAn"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for all populations.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"afrAn"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the African super population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amrAn"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the Ad Mixed American super population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"easAn"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the East Asian super population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"eurAn"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the European super population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"othAn"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for all other populations.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleAn"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for female population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleAn"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for male population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"allHc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for all populations.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"afrHc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for the African / African American population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amrHc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for the Latino population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"easHc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for the East Asian population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"eurAc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for the European super population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"othHc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for all other populations.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleHc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for male population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleHc"),(0,l.kt)("td",{parentName:"tr",align:null},"integer"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for female population.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"failedFilter"),(0,l.kt)("td",{parentName:"tr",align:null},"boolean"),(0,l.kt)("td",{parentName:"tr",align:null},"True if this variant failed any filters (Note: we do not list the failed filters)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"reciprocalOverlap"),(0,l.kt)("td",{parentName:"tr",align:null},"floating point"),(0,l.kt)("td",{parentName:"tr",align:null},"Reciprocal overlap. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"annotationOverlap"),(0,l.kt)("td",{parentName:"tr",align:null},"floating point"),(0,l.kt)("td",{parentName:"tr",align:null},"Reciprocal overlap. Range: 0 - 1.0")))),(0,l.kt)("p",null,(0,l.kt)("strong",{parentName:"p"},"Note:")," Following fields are not available in ",(0,l.kt)("em",{parentName:"p"},"GRCh38")," because the source file does not contain this information:"),(0,l.kt)("table",null,(0,l.kt)("thead",{parentName:"table"},(0,l.kt)("tr",{parentName:"thead"},(0,l.kt)("th",{parentName:"tr",align:null},"Field"))),(0,l.kt)("tbody",{parentName:"table"},(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleAf")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleAf")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleAc")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleAc")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleAn")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleAn")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"allHc")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"afrHc")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amrHc")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"easHc")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"eurAc")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"othHc")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleHc")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleHc")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"failedFilter")))))}u.isMDXComponent=!0},6857:(t,e,n)=>{n.r(e),n.d(e,{contentTitle:()=>N,default:()=>A,frontMatter:()=>d,metadata:()=>c,toc:()=>g});var a=n(7462),l=(n(7294),n(3905)),r=n(9291),i=n(6155),o=n(731),m=n(7069),p=n(9025),s=n(985),u=n(665);const d={title:"gnomAD"},N=void 0,c={unversionedId:"data-sources/gnomad",id:"version-3.25/data-sources/gnomad",title:"gnomAD",description:"Overview",source:"@site/versioned_docs/version-3.25/data-sources/gnomad.mdx",sourceDirName:"data-sources",slug:"/data-sources/gnomad",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/gnomad",editUrl:"https://github.com/Illumina/IlluminaConnectedAnnotationsDocumentation/edit/master/versioned_docs/version-3.25/data-sources/gnomad.mdx",tags:[],version:"3.25",frontMatter:{title:"gnomAD"},sidebar:"docs",previous:{title:"GME Variome",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/gme"},next:{title:"Mitochondrial Heteroplasmy",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/mito-heteroplasmy"}},g=[{value:"Overview",id:"overview",children:[],level:2},{value:"gnomAD v4.0 (GRCh38)",id:"gnomad-v40-grch38",children:[{value:"Small Variants",id:"small-variants",children:[{value:"VCF extraction",id:"vcf-extraction",children:[],level:4},{value:"JSON output",id:"json-output",children:[],level:4},{value:"Calculation",id:"calculation",children:[],level:4}],level:3},{value:"LoF Gene Metrics",id:"lof-gene-metrics",children:[{value:"Tab delimited file example",id:"tab-delimited-file-example",children:[],level:4},{value:"JSON key to TSV column mapping",id:"json-key-to-tsv-column-mapping",children:[],level:4}],level:3},{value:"Structural Variants",id:"structural-variants",children:[{value:"Structural Variant Type Mapping",id:"structural-variant-type-mapping",children:[],level:4},{value:"JSON output",id:"json-output-1",children:[],level:4}],level:3}],level:2},{value:"gnomAD v2.1 (GRCh37)",id:"gnomad-v21-grch37",children:[{value:"Small Variants",id:"small-variants-1",children:[{value:"VCF extraction",id:"vcf-extraction-1",children:[],level:4},{value:"Computation",id:"computation",children:[],level:4},{value:"Merging genomes and exomes",id:"merging-genomes-and-exomes",children:[],level:4},{value:"Filters",id:"filters",children:[],level:4},{value:"VCF download instructions",id:"vcf-download-instructions",children:[],level:4},{value:"JSON output",id:"json-output-2",children:[{value:"Building the supplementary files",id:"building-the-supplementary-files",children:[],level:5},{value:"Source data files",id:"source-data-files",children:[],level:5}],level:4}],level:3},{value:"LoF Gene Metrics",id:"lof-gene-metrics-1",children:[{value:"Tab delimited file example",id:"tab-delimited-file-example-1",children:[],level:4},{value:"JSON key to TSV column mapping",id:"json-key-to-tsv-column-mapping-1",children:[],level:4},{value:"Gene symbol update",id:"gene-symbol-update",children:[],level:4},{value:"Conflict resolution",id:"conflict-resolution",children:[],level:4},{value:"Download URL",id:"download-url",children:[],level:4},{value:"JSON output",id:"json-output-3",children:[],level:4}],level:3},{value:"Structural Variants",id:"structural-variants-1",children:[{value:"Source Files",id:"source-files",children:[],level:4},{value:"Download URLs",id:"download-urls",children:[{value:"GRCh37",id:"grch37",children:[],level:5}],level:4},{value:"JSON output",id:"json-output-4",children:[],level:4}],level:3}],level:2}],k={toc:g},f="wrapper";function A(t){let{components:e,...n}=t;return(0,l.kt)(f,(0,a.Z)({},k,n,{components:e,mdxType:"MDXLayout"}),(0,l.kt)("h2",{id:"overview"},"Overview"),(0,l.kt)("p",null,"The Genome Aggregation Database (",(0,l.kt)("a",{parentName:"p",href:"https://gnomad.broadinstitute.org/"},"gnomAD"),") is a resource developed by an international coalition of investigators, with the goal of aggregating and harmonizing both exome and genome sequencing data from a wide variety of large-scale sequencing projects, and making summary data available for the wider scientific community."),(0,l.kt)("div",{className:"admonition admonition-info alert alert--info"},(0,l.kt)("div",{parentName:"div",className:"admonition-heading"},(0,l.kt)("h5",{parentName:"div"},(0,l.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,l.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,l.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"}))),"Publication")),(0,l.kt)("div",{parentName:"div",className:"admonition-content"},(0,l.kt)("p",{parentName:"div"},"Koch, L., 2020. Exploring human genomic diversity with gnomAD. ",(0,l.kt)("em",{parentName:"p"},"Nature Reviews Genetics"),", ",(0,l.kt)("strong",{parentName:"p"},"21(8)"),", pp.448-448."))),(0,l.kt)("p",null,"Illumina Connected Analysis will support gnomAD v4.0 for GRCh38 assembly and gnomAD v2.1 for GRCh37."),(0,l.kt)("h2",{id:"gnomad-v40-grch38"},"gnomAD v4.0 (GRCh38)"),(0,l.kt)("h3",{id:"small-variants"},"Small Variants"),(0,l.kt)("p",null,"In gnomAD v4.0, like gnomAD v2.1, there are genome and exome data. Compare to gnomAD v2.1 which the data for genome and exome are merged, for gnomAD 4.0, Illumina Connected Annotation will separate them with different JSON output field.\nFor gnomAD genome, the field name would be ",(0,l.kt)("inlineCode",{parentName:"p"},"gnomad"),". For gnomAD exome, the field name would be ",(0,l.kt)("inlineCode",{parentName:"p"},"gnomad-exome"),".\nDespite this difference in the field name, the JSON data format would be identical for both genome and exome."),(0,l.kt)("h4",{id:"vcf-extraction"},"VCF extraction"),(0,l.kt)("p",null,"We currently extract the following info fields from both gnomAD genome and exome VCF files:"),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-scss"},'##INFO=<ID=AC,Number=A,Type=Integer,Description="Alternate allele count">\n##INFO=<ID=AN,Number=1,Type=Integer,Description="Total number of alleles">\n##INFO=<ID=AC_XX,Number=A,Type=Integer,Description="Alternate allele count for XX samples">\n##INFO=<ID=AN_XX,Number=1,Type=Integer,Description="Total number of alleles in XX samples">\n##INFO=<ID=nhomalt_XX,Number=A,Type=Integer,Description="Count of homozygous individuals in XX samples">\n##INFO=<ID=AC_XY,Number=A,Type=Integer,Description="Alternate allele count for XY samples">\n##INFO=<ID=AN_XY,Number=1,Type=Integer,Description="Total number of alleles in XY samples">\n##INFO=<ID=nhomalt_XY,Number=A,Type=Integer,Description="Count of homozygous individuals in XY samples">\n##INFO=<ID=nhomalt,Number=A,Type=Integer,Description="Count of homozygous individuals">\n##INFO=<ID=AC_afr,Number=A,Type=Integer,Description="Alternate allele count for samples of African/African-American ancestry">\n##INFO=<ID=AN_afr,Number=1,Type=Integer,Description="Total number of alleles in samples of African/African-American ancestry">\n##INFO=<ID=nhomalt_afr,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of African/African-American ancestry">\n##INFO=<ID=AC_amr,Number=A,Type=Integer,Description="Alternate allele count for samples of Latino ancestry">\n##INFO=<ID=AN_amr,Number=1,Type=Integer,Description="Total number of alleles in samples of Latino ancestry">\n##INFO=<ID=nhomalt_amr,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Latino ancestry">\n##INFO=<ID=AC_asj,Number=A,Type=Integer,Description="Alternate allele count for samples of Ashkenazi Jewish ancestry">\n##INFO=<ID=AN_asj,Number=1,Type=Integer,Description="Total number of alleles in samples of Ashkenazi Jewish ancestry">\n##INFO=<ID=nhomalt_asj,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Ashkenazi Jewish ancestry">\n##INFO=<ID=AC_eas,Number=A,Type=Integer,Description="Alternate allele count for samples of East Asian ancestry">\n##INFO=<ID=AN_eas,Number=1,Type=Integer,Description="Total number of alleles in samples of East Asian ancestry">\n##INFO=<ID=nhomalt_eas,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of East Asian ancestry">\n##INFO=<ID=AC_fin,Number=A,Type=Integer,Description="Alternate allele count for samples of Finnish ancestry">\n##INFO=<ID=AN_fin,Number=1,Type=Integer,Description="Total number of alleles in samples of Finnish ancestry">\n##INFO=<ID=nhomalt_fin,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Finnish ancestry">\n##INFO=<ID=AC_mid,Number=A,Type=Integer,Description="Alternate allele count for samples of Middle Eastern ancestry">\n##INFO=<ID=AN_mid,Number=1,Type=Integer,Description="Total number of alleles in samples of Middle Eastern ancestry">\n##INFO=<ID=nhomalt_mid,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Middle Eastern ancestry">\n##INFO=<ID=AC_nfe,Number=A,Type=Integer,Description="Alternate allele count for samples of Non-Finnish European ancestry">\n##INFO=<ID=AN_nfe,Number=1,Type=Integer,Description="Total number of alleles in samples of Non-Finnish European ancestry">\n##INFO=<ID=nhomalt_nfe,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Non-Finnish European ancestry">\n##INFO=<ID=AC_remaining,Number=A,Type=Integer,Description="Alternate allele count for samples of Remaining individuals ancestry">\n##INFO=<ID=AN_remaining,Number=1,Type=Integer,Description="Total number of alleles in samples of Remaining individuals ancestry">\n##INFO=<ID=nhomalt_remaining,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Remaining individuals ancestry">\n##INFO=<ID=AC_sas,Number=A,Type=Integer,Description="Alternate allele count for samples of South Asian ancestry">\n##INFO=<ID=AN_sas,Number=1,Type=Integer,Description="Total number of alleles in samples of South Asian ancestry">\n')),(0,l.kt)("h4",{id:"json-output"},"JSON output"),(0,l.kt)(i.default,{mdxType:"JSONV40"}),(0,l.kt)("h4",{id:"calculation"},"Calculation"),(0,l.kt)("p",null,"To calculate allele frequency for each group, we divide the allele count with allele number for each group."),(0,l.kt)("h3",{id:"lof-gene-metrics"},"LoF Gene Metrics"),(0,l.kt)("p",null,"In gnomAD 4.0, the gene score data for LOF is given per transcript.\nSince this is gene level data, one of the transcripts need to be chosen and value reported.\nThe transcript ID of the selected transcript will be reported.\nTranscripts are prioritized (from higher to lower) as follows:"),(0,l.kt)("ol",null,(0,l.kt)("li",{parentName:"ol"},"Ensembl Transcript has mane_select column true from source (gnomAD)."),(0,l.kt)("li",{parentName:"ol"},"Transcript is marked as Ensembl canonical in Illumina Connected Annotation cache data."),(0,l.kt)("li",{parentName:"ol"},"RefSeq transcript has mane_select column true."),(0,l.kt)("li",{parentName:"ol"},"Transcript is marked as RefSeq canonical in Illumina Connected Annotation cache data."),(0,l.kt)("li",{parentName:"ol"},"Transcript has the lowest lof.oe_ci.upper value compare to other transcript for the same gene.")),(0,l.kt)("div",{className:"admonition admonition-info alert alert--info"},(0,l.kt)("div",{parentName:"div",className:"admonition-heading"},(0,l.kt)("h5",{parentName:"div"},(0,l.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,l.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,l.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"}))),"Differences with gnomAD browser")),(0,l.kt)("div",{parentName:"div",className:"admonition-content"},(0,l.kt)("p",{parentName:"div"},"Due to difference in Ensembl version between Illumina Connected Annotation and gnomAD, there are several transcript ID that are marked as canonical in gnomAD browser but not in Illumina Connected Analysis.\nIf this is the case, the gene score shown in Illumina Connected Annotation will be different compared to the gene score shown in the gnomAD browser.\nThe ",(0,l.kt)("inlineCode",{parentName:"p"},"transcriptId")," field in the JSON output will report which transcript was used by Illumina Connected Annotation."))),(0,l.kt)("h4",{id:"tab-delimited-file-example"},"Tab delimited file example"),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-scss"},"gene    transcript  mane_select lof_hc_lc.obs   lof_hc_lc.exp   lof_hc_lc.possible  lof_hc_lc.oe    lof_hc_lc.mu    lof_hc_lc.pLI   lof_hc_lc.pNull lof_hc_lc.pRec  lof.obs lof.exp lof.possible    lof.oe  lof.mu  lof.pLI lof.pNull   lof.pRec    lof.oe_ci.lower lof.oe_ci.upper lof.z_raw   lof.z_score mis.obs mis.exp mis.possible    mis.oe  mis.mu  mis.oe_ci.lower mis.oe_ci.upper mis.z_raw   mis.z_score mis_pphen.obs   mis_pphen.exp   mis_pphen.possible  mis_pphen.oe    syn.obs syn.exp syn.possible    syn.oe  syn.mu  syn.oe_ci.lower syn.oe_ci.upper syn.z_raw   syn.z_score constraint_flags\nSCHIP1  ENST00000445224 false   8   3.0392e+01  157 2.6323e-01  3.5111e-07  9.9024e-01  5.8227e-06  9.7579e-03  8   3.0392e+01  157 2.6323e-01  3.5111e-07  9.9066e-01  5.3097e-06  9.3341e-03  1.5300e-01  4.7500e-01  4.0617e+00  3.4377e+00  193 3.0914e+02  1659    6.2431e-01  1.5780e-06  5.5400e-01  7.0300e-01  6.6055e+00  2.4115e+00  87  1.4959e+02  813 5.8160e-01  76  1.0011e+02  393 7.5914e-01  7.9269e-07  6.3000e-01  9.1900e-01  2.4099e+00  1.3142e+00  []\n")),(0,l.kt)("h4",{id:"json-key-to-tsv-column-mapping"},"JSON key to TSV column mapping"),(0,l.kt)("table",null,(0,l.kt)("thead",{parentName:"table"},(0,l.kt)("tr",{parentName:"thead"},(0,l.kt)("th",{parentName:"tr",align:null},"JSON key"),(0,l.kt)("th",{parentName:"tr",align:null},"TSV column"),(0,l.kt)("th",{parentName:"tr",align:null},"Description"))),(0,l.kt)("tbody",{parentName:"table"},(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"pLi"),(0,l.kt)("td",{parentName:"tr",align:null},"lof.pLI"),(0,l.kt)("td",{parentName:"tr",align:null},"probability of being intolerant of a single loss-of-function variant (like haploinsufficient genes, observed ~ 0.1*expected)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"pNull"),(0,l.kt)("td",{parentName:"tr",align:null},"lof.pNull"),(0,l.kt)("td",{parentName:"tr",align:null},"probability of being completely tolerant of loss of function variation (observed = expected)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"pRec"),(0,l.kt)("td",{parentName:"tr",align:null},"lof.pRec"),(0,l.kt)("td",{parentName:"tr",align:null},"probability of being intolerant of two loss of function variants (like recessive genes, observed ~ 0.5*expected)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"synZ"),(0,l.kt)("td",{parentName:"tr",align:null},"syn.z_score"),(0,l.kt)("td",{parentName:"tr",align:null},"corrected synonymous Z score")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"misZ"),(0,l.kt)("td",{parentName:"tr",align:null},"mis.z_score"),(0,l.kt)("td",{parentName:"tr",align:null},"corrected missense Z score")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"loeuf"),(0,l.kt)("td",{parentName:"tr",align:null},"lof.oe_ci.upper"),(0,l.kt)("td",{parentName:"tr",align:null},"loss of function observed/expected upper bound fraction (LOEUF)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"transcriptId"),(0,l.kt)("td",{parentName:"tr",align:null},"transcript"),(0,l.kt)("td",{parentName:"tr",align:null},"transcript ID which the values we select")))),(0,l.kt)(m.default,{mdxType:"JSONG40"}),(0,l.kt)("h3",{id:"structural-variants"},"Structural Variants"),(0,l.kt)("p",null,"Structural variants in gnomAD 4.0 is available in VCF format and has the same population data as small variants."),(0,l.kt)("h4",{id:"structural-variant-type-mapping"},"Structural Variant Type Mapping"),(0,l.kt)("p",null,"The source files represented the structural variants with keys using various naming conventions.\nIn the Illumina Connected Annotations JSON output, these keys will be mapped according to the following."),(0,l.kt)("table",null,(0,l.kt)("thead",{parentName:"table"},(0,l.kt)("tr",{parentName:"thead"},(0,l.kt)("th",{parentName:"tr",align:null},"Illumina Connected Annotations JSON SV Type Key"),(0,l.kt)("th",{parentName:"tr",align:null},"GRCh37 Source SV Type Key"))),(0,l.kt)("tbody",{parentName:"table"},(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"deletion"),(0,l.kt)("td",{parentName:"tr",align:null},"DEL, CN=0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"duplication"),(0,l.kt)("td",{parentName:"tr",align:null},"DUP")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"insertion"),(0,l.kt)("td",{parentName:"tr",align:null},"INS")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"inversion"),(0,l.kt)("td",{parentName:"tr",align:null},"INV")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"mobile_element_insertion"),(0,l.kt)("td",{parentName:"tr",align:null},"INS:ME")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"mobile_element_insertion"),(0,l.kt)("td",{parentName:"tr",align:null},"INS:ME:ALU")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"mobile_element_insertion"),(0,l.kt)("td",{parentName:"tr",align:null},"INS:ME:LINE1")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"mobile_element_insertion"),(0,l.kt)("td",{parentName:"tr",align:null},"INS:ME:SVA")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"complex_structural_alteration"),(0,l.kt)("td",{parentName:"tr",align:null},"CPX")))),(0,l.kt)("div",{className:"admonition admonition-info alert alert--info"},(0,l.kt)("div",{parentName:"div",className:"admonition-heading"},(0,l.kt)("h5",{parentName:"div"},(0,l.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,l.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,l.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"}))),"gnomAD Copy Number Variation")),(0,l.kt)("div",{parentName:"div",className:"admonition-content"},(0,l.kt)("p",{parentName:"div"},"In gnomAD 4.0 structural variants data, there are CNV data in the VCF file. Since it is not shown in the browser, we don't include CNV in our output.\nWe will evaluate in the future whether to include copy number variation from structural variation data together with new rare CNV data taht is available in gnomAD 4.0."))),(0,l.kt)("div",{className:"admonition admonition-info alert alert--info"},(0,l.kt)("div",{parentName:"div",className:"admonition-heading"},(0,l.kt)("h5",{parentName:"div"},(0,l.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,l.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,l.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"}))),"gnomAD duplication variant type")),(0,l.kt)("div",{parentName:"div",className:"admonition-content"},(0,l.kt)("p",{parentName:"div"},"In gnomAD 4.0 structural variants VCF, only DUP is shown as symbolic allele for duplication variant type.\nBased on the information in gnomAD browser, duplication variant that has split read or paired end reads evidence can be inferred as tandem duplication.\nWith this, we check the evidence data in each DUP variants entry to decide whether it can be assign tandem duplication as variant type or it is just duplication."))),(0,l.kt)("h4",{id:"json-output-1"},"JSON output"),(0,l.kt)(s.default,{mdxType:"JSONSV40"}),(0,l.kt)("h2",{id:"gnomad-v21-grch37"},"gnomAD v2.1 (GRCh37)"),(0,l.kt)("h3",{id:"small-variants-1"},"Small Variants"),(0,l.kt)("h4",{id:"vcf-extraction-1"},"VCF extraction"),(0,l.kt)("p",null,"We currently extract the following info fields from gnomAD genome and exome VCF files:"),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-scss"},'##INFO=<ID=AC,Number=A,Type=Integer,Description="Alternate allele count for samples">\n##INFO=<ID=AN,Number=A,Type=Integer,Description="Total number of alleles in samples">\n##INFO=<ID=nhomalt,Number=A,Type=Integer,Description="Count of homozygous individuals in samples">\n##INFO=<ID=DP,Number=1,Type=Integer,Description="Depth of informative coverage for each sample; reads with MQ=255 or with bad mates are filtered">\n##INFO=<ID=lcr,Number=0,Type=Flag,Description="Variant falls within a low complexity region">\n##INFO=<ID=AC_afr,Number=A,Type=Integer,Description="Alternate allele count for samples of African-American ancestry">\n##INFO=<ID=AN_afr,Number=A,Type=Integer,Description="Total number of alleles in samples of African-American ancestry">\n##INFO=<ID=AF_afr,Number=A,Type=Float,Description="Alternate allele frequency in samples of African-American ancestry">\n##INFO=<ID=nhomalt_afr,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of African-American ancestry">\n##INFO=<ID=AC_amr,Number=A,Type=Integer,Description="Alternate allele count for samples of Latino ancestry">\n##INFO=<ID=AN_amr,Number=A,Type=Integer,Description="Total number of alleles in samples of Latino ancestry">\n##INFO=<ID=nhomalt_amr,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Latino ancestry">\n##INFO=<ID=AC_eas,Number=A,Type=Integer,Description="Alternate allele count for samples of East Asian ancestry">\n##INFO=<ID=AN_eas,Number=A,Type=Integer,Description="Total number of alleles in samples of East Asian ancestry">\n##INFO=<ID=nhomalt_eas,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of East Asian ancestry">\n##INFO=<ID=AC_female,Number=A,Type=Integer,Description="Alternate allele count for female samples">\n##INFO=<ID=AN_female,Number=A,Type=Integer,Description="Total number of alleles in female samples">\n##INFO=<ID=nhomalt_female,Number=A,Type=Integer,Description="Count of homozygous individuals in female samples">\n##INFO=<ID=AC_nfe,Number=A,Type=Integer,Description="Alternate allele count for samples of non-Finnish European ancestry">\n##INFO=<ID=AN_nfe,Number=A,Type=Integer,Description="Total number of alleles in samples of non-Finnish European ancestry">\n##INFO=<ID=nhomalt_nfe,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of non-Finnish European ancestry">\n##INFO=<ID=AC_fin,Number=A,Type=Integer,Description="Alternate allele count for samples of Finnish ancestry">\n##INFO=<ID=AN_fin,Number=A,Type=Integer,Description="Total number of alleles in samples of Finnish ancestry">\n##INFO=<ID=nhomalt_fin,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Finnish ancestry">\n##INFO=<ID=AC_asj,Number=A,Type=Integer,Description="Alternate allele count for samples of Ashkenazi Jewish ancestry">\n##INFO=<ID=AN_asj,Number=A,Type=Integer,Description="Total number of alleles in samples of Ashkenazi Jewish ancestry">\n##INFO=<ID=nhomalt_asj,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Ashkenazi Jewish ancestry">\n##INFO=<ID=AC_oth,Number=A,Type=Integer,Description="Alternate allele count for samples of uncertain ancestry">\n##INFO=<ID=AN_oth,Number=A,Type=Integer,Description="Total number of alleles in samples of uncertain ancestry">\n##INFO=<ID=nhomalt_oth,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of uncertain ancestry">\n##INFO=<ID=AC_male,Number=A,Type=Integer,Description="Alternate allele count for male samples">\n##INFO=<ID=AN_male,Number=A,Type=Integer,Description="Total number of alleles in male samples">\n##INFO=<ID=nhomalt_male,Number=A,Type=Integer,Description="Count of homozygous individuals in male samples">\n##INFO=<ID=controls_AC,Number=A,Type=Integer,Description="Alternate allele count for samples in the controls subset">\n##INFO=<ID=controls_AN,Number=A,Type=Integer,Description="Total number of alleles in samples in the controls subset">\n')),(0,l.kt)("p",null,"We also extract the following extra fields from gnomAD exome VCF file:"),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-scss"},'##INFO=<ID=AC_sas,Number=A,Type=Integer,Description="Alternate allele count for samples of South Asian ancestry">\n##INFO=<ID=AN_sas,Number=A,Type=Integer,Description="Total number of alleles in samples of South Asian ancestry">\n##INFO=<ID=nhomalt_sas,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of South Asian ancestry">\n')),(0,l.kt)("h4",{id:"computation"},"Computation"),(0,l.kt)("p",null,"Using these, we compute the following:"),(0,l.kt)("ul",null,(0,l.kt)("li",{parentName:"ul"},"Coverage"),(0,l.kt)("li",{parentName:"ul"},"Allele count, Homozygous count, allele number and allele frequencies for:"),(0,l.kt)("li",{parentName:"ul"},"Global population"),(0,l.kt)("li",{parentName:"ul"},"African/African Americans"),(0,l.kt)("li",{parentName:"ul"},"Admixed Americans"),(0,l.kt)("li",{parentName:"ul"},"Ashkenazi Jews"),(0,l.kt)("li",{parentName:"ul"},"East Asians"),(0,l.kt)("li",{parentName:"ul"},"Finnish"),(0,l.kt)("li",{parentName:"ul"},"Non-Finnish Europeans"),(0,l.kt)("li",{parentName:"ul"},"South Asian"),(0,l.kt)("li",{parentName:"ul"},"Others (population not assigned)"),(0,l.kt)("li",{parentName:"ul"},"Male"),(0,l.kt)("li",{parentName:"ul"},"Female"),(0,l.kt)("li",{parentName:"ul"},"Controls")),(0,l.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,l.kt)("div",{parentName:"div",className:"admonition-heading"},(0,l.kt)("h5",{parentName:"div"},(0,l.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,l.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,l.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,l.kt)("div",{parentName:"div",className:"admonition-content"},(0,l.kt)("ul",{parentName:"div"},(0,l.kt)("li",{parentName:"ul"},"Coverage = DP / AN. Frequencies are computed using AC/AN for each population."),(0,l.kt)("li",{parentName:"ul"},"Please note that currently there is no genome sequencing data of south asian (SAS) population available in gnomAD."),(0,l.kt)("li",{parentName:"ul"},"Allele Count, Homozygous count, allele number and allele frequencies for control groups are also provided for the global population.")))),(0,l.kt)("h4",{id:"merging-genomes-and-exomes"},"Merging genomes and exomes"),(0,l.kt)("p",null,"When merging the genomes and exomes, the allele counts and allele numbers will be summed across both of the data sets."),(0,l.kt)("div",{className:"admonition admonition-info alert alert--info"},(0,l.kt)("div",{parentName:"div",className:"admonition-heading"},(0,l.kt)("h5",{parentName:"div"},(0,l.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,l.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,l.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"}))),"info")),(0,l.kt)("div",{parentName:"div",className:"admonition-content"},(0,l.kt)("ul",{parentName:"div"},(0,l.kt)("li",{parentName:"ul"},"For GRCh37, Illumina Connected Annotations currently uses gnomAD version 2.1 which contains both genomes and exomes data. Genomes and exomes data are merged in the output.")))),(0,l.kt)("h4",{id:"filters"},"Filters"),(0,l.kt)("p",null,"The following strategy will be used when there's a conflict in filter status:"),(0,l.kt)("table",null,(0,l.kt)("thead",{parentName:"table"},(0,l.kt)("tr",{parentName:"thead"},(0,l.kt)("th",{parentName:"tr",align:"center"}),(0,l.kt)("th",{parentName:"tr",align:"center"},(0,l.kt)("strong",{parentName:"th"},"Genomes PASS")),(0,l.kt)("th",{parentName:"tr",align:"center"},(0,l.kt)("strong",{parentName:"th"},"Genomes Filtered")))),(0,l.kt)("tbody",{parentName:"table"},(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:"center"},(0,l.kt)("strong",{parentName:"td"},"Exomes PASS")),(0,l.kt)("td",{parentName:"tr",align:"center"},"PASS"),(0,l.kt)("td",{parentName:"tr",align:"center"},"Only use exome data")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:"center"},(0,l.kt)("strong",{parentName:"td"},"Exomes Filtered")),(0,l.kt)("td",{parentName:"tr",align:"center"},"Only use genome data"),(0,l.kt)("td",{parentName:"tr",align:"center"},"Filtered")))),(0,l.kt)("h4",{id:"vcf-download-instructions"},"VCF download instructions"),(0,l.kt)("p",null,(0,l.kt)("a",{parentName:"p",href:"https://gnomad.broadinstitute.org/downloads"},"https://gnomad.broadinstitute.org/downloads")),(0,l.kt)("h4",{id:"json-output-2"},"JSON output"),(0,l.kt)(r.default,{mdxType:"JSONV"}),(0,l.kt)("h5",{id:"building-the-supplementary-files"},"Building the supplementary files"),(0,l.kt)("p",null,"The gnomAD ",(0,l.kt)("inlineCode",{parentName:"p"},".nsa")," for Illumina Connected Annotations can be built using the ",(0,l.kt)("inlineCode",{parentName:"p"},"SAUtils")," command's ",(0,l.kt)("inlineCode",{parentName:"p"},"gnomad")," subcommand. We will describe building gnomAD version 3.1 here."),(0,l.kt)("h5",{id:"source-data-files"},"Source data files"),(0,l.kt)("p",null,"Input VCF files (one per chromosome) and a ",(0,l.kt)("inlineCode",{parentName:"p"},".version")," file are required in a folder to build the ",(0,l.kt)("inlineCode",{parentName:"p"},".nsa")," file. For example, my directory contains:"),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-scss"},"chr10.vcf.bgz  chr22.vcf.bgz\nchr11.vcf.bgz  chr2.vcf.bgz\nchr12.vcf.bgz  chr3.vcf.bgz\nchr13.vcf.bgz  chr4.vcf.bgz\nchr14.vcf.bgz  chr5.vcf.bgz\nchr15.vcf.bgz  chr6.vcf.bgz\nchr16.vcf.bgz  chr7.vcf.bgz\nchr17.vcf.bgz  chr8.vcf.bgz\nchr18.vcf.bgz  chr9.vcf.bgz\nchr19.vcf.bgz  chrM.vcf.bgz\nchr1.vcf.bgz   chrX.vcf.bgz\nchr20.vcf.bgz  chrY.vcf.bgz\nchr21.vcf.bgz  gnomad.r3.1.version\n")),(0,l.kt)("p",null,"The version file is a text file with the following content."),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-scss"},"NAME=gnomAD\nVERSION=3.1\nDATE=2020-10-29\nDESCRIPTION=Allele frequencies from Genome Aggregation Database (gnomAD)\n")),(0,l.kt)("p",null,"The help menu for the utility is as follows:"),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-scss"},"SAUtils.dll gnomad\n---------------------------------------------------------------------------\nSAUtils                                             (c) 2021 Illumina, Inc.\nStromberg, Roy, Lajugie, Jiang, Li, and Kang                         3.17.0\n---------------------------------------------------------------------------\n\nUSAGE: dotnet SAUtils.dll gnomad [options]\nReads provided supplementary data files and populates tsv files\n\nOPTIONS:\n      --ref, -r <VALUE>      compressed reference sequence file\n      --genome, -g <VALUE>   input directory containing VCF (and .version)\n                               files with genomic frequencies\n      --exome, -e <VALUE>    input directory containing VCF (and .version)\n                               files with exomic frequencies\n      --temp, -t <VALUE>     output temp directory for intermediate (per chrom)\n                                NSA files\n      --out, -o <VALUE>      output directory for NSA file\n      --help, -h             displays the help menu\n      --version, -v          displays the version\n")),(0,l.kt)("p",null,"Here is a sample execution:"),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-scss"},"dotnet SAUtils.dll Gnomad \\\\\n--ref ~/References/7/Homo_sapiens.GRCh38.Nirvana.dat --genome genomes/ \\\\\n--out ~/SupplementaryDatabase/63/GRCh38 --temp ~/ExternalDataSources/gnomAD/3.1/GRCh38/temp\n")),(0,l.kt)("h3",{id:"lof-gene-metrics-1"},"LoF Gene Metrics"),(0,l.kt)("h4",{id:"tab-delimited-file-example-1"},"Tab delimited file example"),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-scss"},"gene transcript obs_mis exp_mis oe_mis mu_mis possible_mis obs_mis_pphen exp_mis_pphen oe_mis_pphen possible_mis_pphen obs_syn exp_syn oe_syn mu_syn possible_syn obs_lof mu_lof possible_lof exp_lof pLI pNull pRec oe_lof oe_syn_lower oe_syn_upper oe_mis_lower oe_mis_upper oe_lof_lower oe_lof_upper constraint_flag syn_zmis_z lof_z oe_lof_upper_rank oe_lof_upper_bin oe_lof_upper_bin_6 n_sites classic_caf max_af no_lofs obs_het_lof obs_hom_lof defined p exp_hom_lof classic_caf_afr classic_caf_amr classic_caf_asj classic_caf_eas classic_caf_fin classic_caf_nfe classic_caf_oth classic_caf_sas p_afr p_amr p_asj p_eas p_fin p_nfep_oth p_sas transcript_type gene_id transcript_level cds_length num_coding_exons gene_type gene_length exac_pLI exac_obs_lof exac_exp_lof exac_oe_lof brain_expression chromosome start_positionend_position\nMED13 ENST00000397786 871 1.1178e+03 7.7921e-01 5.5598e-05 14195 314 5.2975e+02 5.9273e-01 6708 422 3.8753e+02 1.0890e+00 1.9097e-05 4248 0 4.9203e-06 1257 9.8429e+01 1.0000e+00 8.9436e-40 1.8383e-16 0.0000e+00 1.0050e+00 1.1800e+00 7.3600e-01 8.2400e-01 0.0000e+00 3.0000e-02 -1.3765e+00 2.6232e+00 9.1935e+00 0 0 0 2 1.2058e-05 8.0492e-06 124782 3 0 124785 1.2021e-05 1.8031e-05 0.0000e+00 0.0000e+00 0.0000e+00 0.0000e+00 9.2812e-05 8.8571e-06 0.0000e+00 0.0000e+00 0.0000e+00 0.0000e+00 0.0000e+00 0.0000e+00 9.2760e-05 8.8276e-06 0.0000e+00 0.0000e+00 protein_coding ENSG00000108510 2 6522 30 protein_coding 122678 1.0000e+00 0 6.4393e+01 0.0000e+00 NA 17 60019966 60142643\n")),(0,l.kt)("h4",{id:"json-key-to-tsv-column-mapping-1"},"JSON key to TSV column mapping"),(0,l.kt)("table",null,(0,l.kt)("thead",{parentName:"table"},(0,l.kt)("tr",{parentName:"thead"},(0,l.kt)("th",{parentName:"tr",align:null},"JSON key"),(0,l.kt)("th",{parentName:"tr",align:null},"TSV column"),(0,l.kt)("th",{parentName:"tr",align:null},"Description"))),(0,l.kt)("tbody",{parentName:"table"},(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"pLi"),(0,l.kt)("td",{parentName:"tr",align:null},"pLI"),(0,l.kt)("td",{parentName:"tr",align:null},"probability of being intolerant of a single loss-of-function variant (like haploinsufficient genes, observed ~ 0.1*expected)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"pNull"),(0,l.kt)("td",{parentName:"tr",align:null},"pNull"),(0,l.kt)("td",{parentName:"tr",align:null},"probability of being completely tolerant of loss of function variation (observed = expected)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"pRec"),(0,l.kt)("td",{parentName:"tr",align:null},"pRec"),(0,l.kt)("td",{parentName:"tr",align:null},"probability of being intolerant of two loss of function variants (like recessive genes, observed ~ 0.5*expected)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"synZ"),(0,l.kt)("td",{parentName:"tr",align:null},"syn_z"),(0,l.kt)("td",{parentName:"tr",align:null},"corrected synonymous Z score")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"misZ"),(0,l.kt)("td",{parentName:"tr",align:null},"mis_z"),(0,l.kt)("td",{parentName:"tr",align:null},"corrected missense Z score")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"loeuf"),(0,l.kt)("td",{parentName:"tr",align:null},"oe_lof_upper"),(0,l.kt)("td",{parentName:"tr",align:null},"loss of function observed/expected upper bound fraction (LOEUF)")))),(0,l.kt)("h4",{id:"gene-symbol-update"},"Gene symbol update"),(0,l.kt)("p",null,"The input file provides Ensembl gene ids for each entry. We observed that they were unique while gene symbols may be repeated (multiple lines may have the same gene symbol). Since Ensembl gene Ids are more stable, and Illumina Connected Annotations transcript cache data contains Ensembl gene ids, we use these ids to extract the gene symbols from the transcript cache. For example, if ENSG0001 has gene symbol GENE1 in the input but Illumina Connected Annotations cache say ENSG0001 maps to GENE2, we use GENE2 as the gene symbol for that entry."),(0,l.kt)("h4",{id:"conflict-resolution"},"Conflict resolution"),(0,l.kt)("p",null,"gnomAD uses Ensembl GeneID as unique identifiers in the ",(0,l.kt)("a",{parentName:"p",href:"https://storage.googleapis.com/gnomad-public/release/2.1.1/constraint/gnomad.v2.1.1.lof_metrics.by_gene.txt.bgz"},"source file")," but Illumina Connected Annotations uses HGNC gene symbols. Multiple Ensembl GeneIDs can map to the same HGNC symbol and therefore may result is conflict."),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-scss"},"MDGA2   ENST00000426342 306 4.0043e+02  7.6419e-01  2.1096e-05  4724    78  1.6525e+02  4.7202e-01  1923    125 1.3737e+02  9.0993e-01  7.1973e-06  1413    4   2.0926e-06  453 3.8316e+01  9.9922e-01  8.6490e-12  7.8128e-04  1.0440e-01  7.8600e-01  1.0560e+00  6.9500e-01  8.4000e-01  5.0000e-02  2.3900e-01      8.2988e-01  1.6769e+00  5.1372e+00  1529    0   0   7   2.8103e-05  4.0317e-06  124784  7   0   124791  2.8047e-05  9.8167e-05  0.0000e+00  2.8962e-05  0.0000e+00  0.0000e+00  0.0000e+00  3.5391e-05  1.6672e-04  3.2680e-05  0.0000e+00  2.8962e-05  0.0000e+00  0.0000e+00  0.0000e+00  3.5308e-05  1.6492e-04  3.2678e-05  protein_coding  ENSG00000139915 2   2181    13  protein_coding  835332  9.9322e-01  3   2.7833e+01  1.0779e-01  NA  14  47308826    48144157\nMDGA2   ENST00000439988 438 5.5311e+02  7.9189e-01  2.9490e-05  6608    105 2.0496e+02  5.1228e-01  2386    180 1.9491e+02  9.2351e-01  9.8371e-06  2048    11  2.8074e-06  627 5.1882e+01  6.6457e-01  5.5841e-10  3.3543e-01  2.1202e-01  8.1700e-01  1.0450e+00  7.3100e-01  8.5700e-01  1.3200e-01  3.5100e-01      8.3940e-01  1.7393e+00  5.2595e+00  2989    1   0   9   3.6173e-05  4.0463e-06  124782  9   0   124791  3.6061e-05  1.6228e-04  6.4986e-05  2.8962e-05  0.0000e+00  0.0000e+00  0.0000e+00  4.4275e-05  1.6672e-04  3.2680e-05  6.4577e-05  2.8962e-05  0.0000e+00  0.0000e+00  0.0000e+00  4.4135e-05  1.6492e-04  3.2678e-05  protein_coding  ENSG00000272781 3   3075    17  protein_coding  832866  NA  NA  NA  NA  NA  14  47311134    48143999\n")),(0,l.kt)("p",null,'In such cases, Illumina Connected Annotations chooses the entry with the smallest "LOEUF" value. The reason for choosing this value can be highlighted by the following table:'),(0,l.kt)("table",null,(0,l.kt)("thead",{parentName:"table"},(0,l.kt)("tr",{parentName:"thead"},(0,l.kt)("th",{parentName:"tr",align:"right"},"LOEUF decile"),(0,l.kt)("th",{parentName:"tr",align:"right"},"Haplo-insufficient"),(0,l.kt)("th",{parentName:"tr",align:"right"},"Autosomal Dominant"),(0,l.kt)("th",{parentName:"tr",align:"right"},"Autosomal Recessive"),(0,l.kt)("th",{parentName:"tr",align:"right"},"Olfactory Genes"))),(0,l.kt)("tbody",{parentName:"table"},(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:"right"},"0-10%"),(0,l.kt)("td",{parentName:"tr",align:"right"},"104"),(0,l.kt)("td",{parentName:"tr",align:"right"},"140"),(0,l.kt)("td",{parentName:"tr",align:"right"},"36"),(0,l.kt)("td",{parentName:"tr",align:"right"},"0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:"right"},"10-20%"),(0,l.kt)("td",{parentName:"tr",align:"right"},"47"),(0,l.kt)("td",{parentName:"tr",align:"right"},"128"),(0,l.kt)("td",{parentName:"tr",align:"right"},"72"),(0,l.kt)("td",{parentName:"tr",align:"right"},"1")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:"right"},"20-30%"),(0,l.kt)("td",{parentName:"tr",align:"right"},"17"),(0,l.kt)("td",{parentName:"tr",align:"right"},"86"),(0,l.kt)("td",{parentName:"tr",align:"right"},"112"),(0,l.kt)("td",{parentName:"tr",align:"right"},"0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:"right"},"30-40%"),(0,l.kt)("td",{parentName:"tr",align:"right"},"8"),(0,l.kt)("td",{parentName:"tr",align:"right"},"80"),(0,l.kt)("td",{parentName:"tr",align:"right"},"173"),(0,l.kt)("td",{parentName:"tr",align:"right"},"4")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:"right"},"40-50%"),(0,l.kt)("td",{parentName:"tr",align:"right"},"7"),(0,l.kt)("td",{parentName:"tr",align:"right"},"65"),(0,l.kt)("td",{parentName:"tr",align:"right"},"206"),(0,l.kt)("td",{parentName:"tr",align:"right"},"8")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:"right"},"50-60%"),(0,l.kt)("td",{parentName:"tr",align:"right"},"4"),(0,l.kt)("td",{parentName:"tr",align:"right"},"54"),(0,l.kt)("td",{parentName:"tr",align:"right"},"207"),(0,l.kt)("td",{parentName:"tr",align:"right"},"6")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:"right"},"60-70%"),(0,l.kt)("td",{parentName:"tr",align:"right"},"0"),(0,l.kt)("td",{parentName:"tr",align:"right"},"46"),(0,l.kt)("td",{parentName:"tr",align:"right"},"154"),(0,l.kt)("td",{parentName:"tr",align:"right"},"18")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:"right"},"70-80%"),(0,l.kt)("td",{parentName:"tr",align:"right"},"2"),(0,l.kt)("td",{parentName:"tr",align:"right"},"49"),(0,l.kt)("td",{parentName:"tr",align:"right"},"120"),(0,l.kt)("td",{parentName:"tr",align:"right"},"49")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:"right"},"80-90%"),(0,l.kt)("td",{parentName:"tr",align:"right"},"0"),(0,l.kt)("td",{parentName:"tr",align:"right"},"34"),(0,l.kt)("td",{parentName:"tr",align:"right"},"58"),(0,l.kt)("td",{parentName:"tr",align:"right"},"96")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:"right"},"90-100%"),(0,l.kt)("td",{parentName:"tr",align:"right"},"0"),(0,l.kt)("td",{parentName:"tr",align:"right"},"26"),(0,l.kt)("td",{parentName:"tr",align:"right"},"40"),(0,l.kt)("td",{parentName:"tr",align:"right"},"174")))),(0,l.kt)("div",{className:"admonition admonition-info alert alert--info"},(0,l.kt)("div",{parentName:"div",className:"admonition-heading"},(0,l.kt)("h5",{parentName:"div"},(0,l.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,l.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,l.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"}))),"Note")),(0,l.kt)("div",{parentName:"div",className:"admonition-content"},(0,l.kt)("ul",{parentName:"div"},(0,l.kt)("li",{parentName:"ul"},"Table source: ",(0,l.kt)("a",{parentName:"li",href:"https://www.biorxiv.org/content/biorxiv/early/2019/01/28/531210.full-text.pdf"},"https://www.biorxiv.org/content/biorxiv/early/2019/01/28/531210.full-text.pdf")),(0,l.kt)("li",{parentName:"ul"},"This table indicates that lower LOEUF scores have more deleterious effect on genes."),(0,l.kt)("li",{parentName:"ul"},"Only 15 out of 19685 genes have conflicting entries.")))),(0,l.kt)("p",null,(0,l.kt)("strong",{parentName:"p"},"List of genes with conflicting entries")),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-scss"},'MDGA2:\n {"pLI":9.99e-1,"pRec":7.81e-4,"pNull":8.65e-12,"synZ":8.30e-1,"misZ":1.68e0,"loeuf":2.39e-1}\n {"pLI":6.65e-1,"pRec":3.35e-1,"pNull":5.58e-10,"synZ":8.39e-1,"misZ":1.74e0,"loeuf":3.51e-1}\nCRYBG3:\n {"pLI":9.27e-5,"pRec":1.00e0,"pNull":1.88e-7,"synZ":1.82e0,"misZ":4.68e-1,"loeuf":4.93e-1}\n {"pLI":2.69e-4,"pRec":1.00e0,"pNull":1.20e-4,"synZ":2.63e0,"misZ":9.80e-1,"loeuf":5.98e-1}\nCHTF8:\n {"pLI":8.29e-1,"pRec":1.67e-1,"pNull":3.21e-3,"synZ":1.94e0,"misZ":9.48e-1,"loeuf":5.13e-1}\n {"pLI":3.73e-1,"pRec":5.84e-1,"pNull":4.29e-2,"synZ":3.33e-1,"misZ":2.91e-1,"loeuf":9.92e-1}\nSEPT1:\n {"pLI":6.77e-8,"pRec":8.90e-1,"pNull":1.10e-1,"synZ":1.58e-1,"misZ":1.57e0,"loeuf":9.68e-1}\n {"pLI":1.96e-8,"pRec":6.71e-1,"pNull":3.29e-1,"synZ":1.68e-1,"misZ":1.41e0,"loeuf":1.08e0}\nARL14EPL:\n {"pLI":3.48e-2,"pRec":8.38e-1,"pNull":1.28e-1,"synZ":3.56e-1,"misZ":-1.87e-1,"loeuf":1.23e0}\n {"pLI":3.23e-2,"pRec":8.29e-1,"pNull":1.38e-1,"synZ":1.15e0,"misZ":-4.05e-1,"loeuf":1.26e0}\nUGT2A1:\n {"pLI":2.90e-13,"pRec":1.40e-1,"pNull":8.60e-1,"synZ":-1.29e0,"misZ":-1.77e0,"loeuf":1.18e0}\n {"pLI":3.88e-17,"pRec":2.87e-3,"pNull":9.97e-1,"synZ":-8.00e-1,"misZ":-1.40e0,"loeuf":1.53e0}\nLTB4R2:\n {"pLI":4.39e-4,"pRec":6.71e-1,"pNull":3.29e-1,"synZ":-5.24e-1,"misZ":-2.96e-1,"loeuf":1.40e0}\n {"pLI":1.38e-5,"pRec":4.12e-1,"pNull":5.88e-1,"synZ":-4.58e-1,"misZ":-2.02e-1,"loeuf":1.54e0}\nCDRT1:\n {"pLI":4.98e-14,"pRec":5.31e-1,"pNull":4.69e-1,"synZ":8.18e-1,"misZ":6.57e-1,"loeuf":1.00e0}\n {"pLI":3.50e-3,"pRec":6.37e-1,"pNull":3.59e-1,"synZ":4.89e-1,"misZ":6.90e-1,"loeuf":1.63e0}\nMUC3A:\n {"pLI":1.48e-10,"pRec":5.76e-1,"pNull":4.24e-1,"synZ":5.81e-2,"misZ":-6.01e-1,"loeuf":1.06e0}\n {"pLI":4.03e-1,"pRec":4.79e-1,"pNull":1.17e-1,"synZ":4.05e-2,"misZ":-1.60e-1,"loeuf":1.70e0}\nCOG8:\n {"pLI":2.97e-9,"pRec":5.04e-1,"pNull":4.96e-1,"synZ":-1.35e0,"misZ":-9.37e-2,"loeuf":1.13e0}\n {"pLI":2.31e-3,"pRec":5.47e-1,"pNull":4.50e-1,"synZ":-4.94e-1,"misZ":-1.48e-1,"loeuf":1.76e0}\nAC006486.1:\n {"pLI":9.37e-1,"pRec":6.27e-2,"pNull":2.47e-4,"synZ":1.44e0,"misZ":2.12e0,"loeuf":3.41e-1}\n {"pLI":1.14e-1,"pRec":6.16e-1,"pNull":2.70e-1,"synZ":-7.57e-2,"misZ":8.33e-2,"loeuf":1.84e0}\nAL645922.1:\n {"pLI":4.67e-16,"pRec":1.00e0,"pNull":4.15e-5,"synZ":7.99e-1,"misZ":1.61e0,"loeuf":6.92e-1}\n {"pLI":1.60e-3,"pRec":2.78e-1,"pNull":7.21e-1,"synZ":-7.30e-2,"misZ":3.21e-1,"loeuf":1.96e0}\nNBPF20:\n {"pLI":1.42e-7,"pRec":3.40e-2,"pNull":9.66e-1,"synZ":-1.86e0,"misZ":-2.88e0,"loeuf":1.97e0}\n {"pLI":1.92e-22,"pRec":7.96e-6,"pNull":1.00e0,"synZ":-9.73e0,"misZ":-7.67e0,"loeuf":1.97e0}\nPRAMEF11:\n {"pLI":6.16e-4,"pRec":7.42e-1,"pNull":2.58e-1,"synZ":-4.02e0,"misZ":-3.69e0,"loeuf":1.31e0}\n {"synZ":-3.33e0,"misZ":-2.59e0}\nFAM231D:\n {"synZ":-1.98e0,"misZ":-1.44e0}\n {"synZ":1.07e0,"misZ":3.13e-1}\n')),(0,l.kt)("p",null,(0,l.kt)("strong",{parentName:"p"},"Conflict resolution")),(0,l.kt)("ul",null,(0,l.kt)("li",{parentName:"ul"},"Pick the entry with the lowest LOEUF score"),(0,l.kt)("li",{parentName:"ul"},"If the same, pick the lowest pLI"),(0,l.kt)("li",{parentName:"ul"},"Otherwise pick the entry with the max absolute value of synZ + misZ")),(0,l.kt)("h4",{id:"download-url"},"Download URL"),(0,l.kt)("p",null,(0,l.kt)("a",{parentName:"p",href:"https://storage.googleapis.com/gnomad-public/release/2.1.1/constraint/gnomad.v2.1.1.lof_metrics.by_gene.txt.bgz"},"https://storage.googleapis.com/gnomad-public/release/2.1.1/constraint/gnomad.v2.1.1.lof_metrics.by_gene.txt.bgz")),(0,l.kt)("h4",{id:"json-output-3"},"JSON output"),(0,l.kt)(o.default,{mdxType:"JSONG"}),(0,l.kt)("h3",{id:"structural-variants-1"},"Structural Variants"),(0,l.kt)("div",{className:"admonition admonition-info alert alert--info"},(0,l.kt)("div",{parentName:"div",className:"admonition-heading"},(0,l.kt)("h5",{parentName:"div"},(0,l.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,l.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,l.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"}))),"Publication")),(0,l.kt)("div",{parentName:"div",className:"admonition-content"},(0,l.kt)("p",{parentName:"div"},"Collins, R.L., Brand, H., Karczewski, K.J. et al. 2020. A structural variation reference for medical and population genetics. ",(0,l.kt)("em",{parentName:"p"},"Nature")," ",(0,l.kt)("strong",{parentName:"p"},"581"),", pp.444\u2013451. ",(0,l.kt)("a",{parentName:"p",href:"https://doi.org/10.1038/s41586-020-2287-8"},"https://doi.org/10.1038/s41586-020-2287-8")))),(0,l.kt)("p",null,(0,l.kt)("strong",{parentName:"p"},"Note"),"\nThe gnomAD structural variant annotations are in a preview stage at the moment.\nCurrently, the annotations do not include translocation breakends.\nFuture updates will include a better way of annotating the structural variants."),(0,l.kt)("h4",{id:"source-files"},"Source Files"),(0,l.kt)(u.default,{mdxType:"SVDATADESCRIPTION"}),(0,l.kt)("h4",{id:"download-urls"},"Download URLs"),(0,l.kt)("h5",{id:"grch37"},"GRCh37"),(0,l.kt)("p",null,"The GRCh37 file was downloaded from the original source. Following table gives some essential data metrics:"),(0,l.kt)("p",null,(0,l.kt)("a",{parentName:"p",href:"https://storage.googleapis.com/gcp-public-data--gnomad/papers/2019-sv/gnomad_v2.1_sv.sites.bed.gz"},"https://storage.googleapis.com/gcp-public-data--gnomad/papers/2019-sv/gnomad_v2.1_sv.sites.bed.gz")),(0,l.kt)("h4",{id:"json-output-4"},"JSON output"),(0,l.kt)(p.default,{mdxType:"JSONSV"}))}A.isMDXComponent=!0},7069:(t,e,n)=>{n.r(e),n.d(e,{contentTitle:()=>i,default:()=>u,frontMatter:()=>r,metadata:()=>o,toc:()=>m});var a=n(7462),l=(n(7294),n(3905));const r={},i=void 0,o={unversionedId:"data-sources/gnomad4.0-lof-json",id:"version-3.25/data-sources/gnomad4.0-lof-json",title:"gnomad4.0-lof-json",description:"",source:"@site/versioned_docs/version-3.25/data-sources/gnomad4.0-lof-json.md",sourceDirName:"data-sources",slug:"/data-sources/gnomad4.0-lof-json",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/gnomad4.0-lof-json",editUrl:"https://github.com/Illumina/IlluminaConnectedAnnotationsDocumentation/edit/master/versioned_docs/version-3.25/data-sources/gnomad4.0-lof-json.md",tags:[],version:"3.25",frontMatter:{}},m=[],p={toc:m},s="wrapper";function u(t){let{components:e,...n}=t;return(0,l.kt)(s,(0,a.Z)({},p,n,{components:e,mdxType:"MDXLayout"}),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-json"},'"gnomAD": {\n    "pLi": 0.00000122,\n    "pRec": 0.32,\n    "pNull": 0.68,\n    "synZ": 0.0117,\n    "misZ": 0.162,\n    "loeuf": 1.94,\n    "transcriptId": "ENST00000360525"\n}\n')))}u.isMDXComponent=!0},6155:(t,e,n)=>{n.r(e),n.d(e,{contentTitle:()=>i,default:()=>u,frontMatter:()=>r,metadata:()=>o,toc:()=>m});var a=n(7462),l=(n(7294),n(3905));const r={},i=void 0,o={unversionedId:"data-sources/gnomad4.0-small-variants-json",id:"version-3.25/data-sources/gnomad4.0-small-variants-json",title:"gnomad4.0-small-variants-json",description:"| Field        | Type   | Notes                                                                                            |",source:"@site/versioned_docs/version-3.25/data-sources/gnomad4.0-small-variants-json.md",sourceDirName:"data-sources",slug:"/data-sources/gnomad4.0-small-variants-json",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/gnomad4.0-small-variants-json",editUrl:"https://github.com/Illumina/IlluminaConnectedAnnotationsDocumentation/edit/master/versioned_docs/version-3.25/data-sources/gnomad4.0-small-variants-json.md",tags:[],version:"3.25",frontMatter:{}},m=[],p={toc:m},s="wrapper";function u(t){let{components:e,...n}=t;return(0,l.kt)(s,(0,a.Z)({},p,n,{components:e,mdxType:"MDXLayout"}),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-json"},'"gnomad": {\n    "coverage": 154,\n    "failedFilter": true,\n    "allAf": 0.5,\n    "allAn": 152428,\n    "allAc": 76214,\n    "allHc": 0,\n    "afrAf": 0.5,\n    "afrAn": 41608,\n    "afrAc": 20804,\n    "afrHc": 0,\n    "amiAf": 0.5,\n    "amiAn": 912,\n    "amiAc": 456,\n    "amiHc": 0,\n    "amrAf": 0.5,\n    "amrAn": 15314,\n    "amrAc": 7657,\n    "amrHc": 0,\n    "easAf": 0.5,\n    "easAn": 5196,\n    "easAc": 2598,\n    "easHc": 0,\n    "finAf": 0.5,\n    "finAn": 10632,\n    "finAc": 5316,\n    "finHc": 0,\n    "nfeAf": 0.5,\n    "nfeAn": 68050,\n    "nfeAc": 34025,\n    "nfeHc": 0,\n    "asjAf": 0.5,\n    "asjAn": 3472,\n    "asjAc": 1736,\n    "asjHc": 0,\n    "sasAf": 0.5,\n    "sasAn": 4834,\n    "sasAc": 2417,\n    "sasHc": 0,\n    "midAf": 0.5,\n    "midAn": 294,\n    "midAc": 147,\n    "midHc": 0,\n    "remainingAf": 0.5,\n    "remainingAn": 2116,\n    "remainingAc": 1058,\n    "remainingHc": 0,\n    "maleAf": 0.5,\n    "maleAn": 74544,\n    "maleAc": 37272,\n    "maleHc": 0,\n    "femaleAf": 0.5,\n    "femaleAn": 77884,\n    "femaleAc": 38942,\n    "femaleHc": 0\n}\n')),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-json"},'"gnomad-exome": {\n    "coverage": 53,\n    "allAf": 0.495074,\n    "allAn": 4060,\n    "allAc": 2010,\n    "allHc": 11,\n    "afrAf": 0.5,\n    "afrAn": 86,\n    "afrAc": 43,\n    "afrHc": 0,\n    "amrAf": 0.5,\n    "amrAn": 46,\n    "amrAc": 23,\n    "amrHc": 0,\n    "easAf": 0.491071,\n    "easAn": 112,\n    "easAc": 55,\n    "easHc": 0,\n    "finAf": 0.5,\n    "finAn": 306,\n    "finAc": 153,\n    "finHc": 0,\n    "nfeAf": 0.49503,\n    "nfeAn": 3018,\n    "nfeAc": 1494,\n    "nfeHc": 11,\n    "asjAf": 0.461538,\n    "asjAn": 26,\n    "asjAc": 12,\n    "asjHc": 0,\n    "sasAf": 0.486111,\n    "sasAn": 72,\n    "sasAc": 35,\n    "sasHc": 0,\n    "midAf": 0.5,\n    "midAn": 68,\n    "midAc": 34,\n    "midHc": 0,\n    "remainingAf": 0.493865,\n    "remainingAn": 326,\n    "remainingAc": 161,\n    "remainingHc": 0,\n    "maleAf": 0.495212,\n    "maleAn": 2924,\n    "maleAc": 1448,\n    "maleHc": 9,\n    "femaleAf": 0.494718,\n    "femaleAn": 1136,\n    "femaleAc": 562,\n    "femaleHc": 2\n}\n')),(0,l.kt)("table",null,(0,l.kt)("thead",{parentName:"table"},(0,l.kt)("tr",{parentName:"thead"},(0,l.kt)("th",{parentName:"tr",align:null},"Field"),(0,l.kt)("th",{parentName:"tr",align:null},"Type"),(0,l.kt)("th",{parentName:"tr",align:null},"Notes"))),(0,l.kt)("tbody",{parentName:"table"},(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"coverage"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"average coverage (non-negative integer values)")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for male population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for male population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for male population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"maleHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for male population. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for female population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for female population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for female population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"femaleHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for female population. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"remainingAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the Other population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"remainingAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the Other population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"remainingAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the Other population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"remainingHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for Other population. Non-negative integer")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"allAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for all populations. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"allAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for all populations. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"allAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for all populations. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"allHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for all populations. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"afrAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the African / African American population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"afrAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the African / African American population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"afrAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the African / African American population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"afrHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for African / African American population. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amiAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for Amish populations. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amiAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for Amish populations. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amiAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for Amish populations. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amiHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for Amish populations. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amrAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the Latino population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amrAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the Latino population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amrAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the Latino population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"amrHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for Latino population. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"easAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the East Asian population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"easAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the East Asian population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"easAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the East Asian population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"easHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for East Asian population. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"finAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the Finnish population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"finAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the Finnish population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"finAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the Finnish population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"finHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for Finnish population. Non-negative integer")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"nfeAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the Non-Finnish European population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"nfeAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the Non-Finnish European population. Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"nfeAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the Non-Finnish European population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"nfeHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for Non-Finnish European population. Non-negative integer")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"asjAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the Ashkenazi Jewish population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"asjAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the Ashkenazi Jewish population Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"asjAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the Ashkenazi Jewish population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"asjHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for the Ashkenazi Jewish population. Non-negative integer")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"sasAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the South Asian population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"sasAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the South Asian population Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"sasAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the South Asian population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"sasHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for the South Asian population. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"midAf"),(0,l.kt)("td",{parentName:"tr",align:null},"float"),(0,l.kt)("td",{parentName:"tr",align:null},"allele frequency for the Middle Eastern population. Range: 0 - 1.0")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"midAc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele count for the iddle Eastern population Integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"midAn"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"allele number for the iddle Eastern population. Non-zero integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"midHc"),(0,l.kt)("td",{parentName:"tr",align:null},"int"),(0,l.kt)("td",{parentName:"tr",align:null},"count of homozygous individuals for the iddle Eastern population. Non-negative integer.")),(0,l.kt)("tr",{parentName:"tbody"},(0,l.kt)("td",{parentName:"tr",align:null},"failedFilter"),(0,l.kt)("td",{parentName:"tr",align:null},"bool"),(0,l.kt)("td",{parentName:"tr",align:null},"True if this variant failed any filters (Note: we do not list the failed filters)")))))}u.isMDXComponent=!0},985:(t,e,n)=>{n.r(e),n.d(e,{contentTitle:()=>i,default:()=>u,frontMatter:()=>r,metadata:()=>o,toc:()=>m});var a=n(7462),l=(n(7294),n(3905));const r={},i=void 0,o={unversionedId:"data-sources/gnomad40-structural-variants-json",id:"version-3.25/data-sources/gnomad40-structural-variants-json",title:"gnomad40-structural-variants-json",description:"",source:"@site/versioned_docs/version-3.25/data-sources/gnomad40-structural-variants-json.md",sourceDirName:"data-sources",slug:"/data-sources/gnomad40-structural-variants-json",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/gnomad40-structural-variants-json",editUrl:"https://github.com/Illumina/IlluminaConnectedAnnotationsDocumentation/edit/master/versioned_docs/version-3.25/data-sources/gnomad40-structural-variants-json.md",tags:[],version:"3.25",frontMatter:{}},m=[],p={toc:m},s="wrapper";function u(t){let{components:e,...n}=t;return(0,l.kt)(s,(0,a.Z)({},p,n,{components:e,mdxType:"MDXLayout"}),(0,l.kt)("pre",null,(0,l.kt)("code",{parentName:"pre",className:"language-json"},'"gnomad": [\n    {\n      "chromosome": "1",\n      "begin": 1769047,\n      "end": 78686496,\n      "variantId": "gnomAD-SV_v3_CPX_chr1_4787cfba",\n      "variantType": "complex_structural_alteration",\n      "failedFilter": true,\n      "allAf": 0.51192,\n      "afrAf": 0.491986,\n      "amiAf": 0.559382,\n      "amrAf": 0.499444,\n      "asjAf": 0.505975,\n      "easAf": 0.51924,\n      "midAf": 0.53125,\n      "finAf": 0.542619,\n      "nfeAf": 0.521916,\n      "othAf": 0.492366,\n      "sasAf": 0.516568,\n      "femaleAf": 0.509225,\n      "maleAf": 0.514861,\n      "allAc": 64549,\n      "afrAc": 16637,\n      "amiAc": 471,\n      "amrAc": 6290,\n      "asjAc": 1609,\n      "easAc": 2105,\n      "midAc": 34,\n      "finAc": 3514,\n      "nfeAc": 30839,\n      "othAc": 774,\n      "sasAc": 2276,\n      "femaleAc": 33507,\n      "maleAc": 31042,\n      "allAn": 126092,\n      "afrAn": 33816,\n      "amiAn": 842,\n      "amrAn": 12594,\n      "asjAn": 3180,\n      "easAn": 4054,\n      "midAn": 64,\n      "finAn": 6476,\n      "nfeAn": 59088,\n      "othAn": 1572,\n      "sasAn": 4406,\n      "femaleAn": 65800,\n      "maleAn": 60292,\n      "allHc": 3167,\n      "afrHc": 413,\n      "amiHc": 54,\n      "amrHc": 238,\n      "asjHc": 49,\n      "easHc": 97,\n      "midHc": 2,\n      "finHc": 368,\n      "nfeHc": 1807,\n      "othHc": 23,\n      "sasHc": 116,\n      "femaleHc": 1407,\n      "maleHc": 1760\n    }\n]\n')))}u.isMDXComponent=!0}}]);