"use strict";(self.webpackChunknirvana_documentation=self.webpackChunknirvana_documentation||[]).push([[9922,9972],{5680:(e,n,a)=>{a.d(n,{xA:()=>d,yg:()=>y});var t=a(6540);function i(e,n,a){return n in e?Object.defineProperty(e,n,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[n]=a,e}function l(e,n){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var t=Object.getOwnPropertySymbols(e);n&&(t=t.filter((function(n){return Object.getOwnPropertyDescriptor(e,n).enumerable}))),a.push.apply(a,t)}return a}function r(e){for(var n=1;n<arguments.length;n++){var a=null!=arguments[n]?arguments[n]:{};n%2?l(Object(a),!0).forEach((function(n){i(e,n,a[n])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):l(Object(a)).forEach((function(n){Object.defineProperty(e,n,Object.getOwnPropertyDescriptor(a,n))}))}return e}function o(e,n){if(null==e)return{};var a,t,i=function(e,n){if(null==e)return{};var a,t,i={},l=Object.keys(e);for(t=0;t<l.length;t++)a=l[t],n.indexOf(a)>=0||(i[a]=e[a]);return i}(e,n);if(Object.getOwnPropertySymbols){var l=Object.getOwnPropertySymbols(e);for(t=0;t<l.length;t++)a=l[t],n.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(i[a]=e[a])}return i}var s=t.createContext({}),g=function(e){var n=t.useContext(s),a=n;return e&&(a="function"==typeof e?e(n):r(r({},n),e)),a},d=function(e){var n=g(e.components);return t.createElement(s.Provider,{value:n},e.children)},p="mdxType",m={inlineCode:"code",wrapper:function(e){var n=e.children;return t.createElement(t.Fragment,{},n)}},c=t.forwardRef((function(e,n){var a=e.components,i=e.mdxType,l=e.originalType,s=e.parentName,d=o(e,["components","mdxType","originalType","parentName"]),p=g(a),c=i,y=p["".concat(s,".").concat(c)]||p[c]||m[c]||l;return a?t.createElement(y,r(r({ref:n},d),{},{components:a})):t.createElement(y,r({ref:n},d))}));function y(e,n){var a=arguments,i=n&&n.mdxType;if("string"==typeof e||i){var l=a.length,r=new Array(l);r[0]=c;var o={};for(var s in n)hasOwnProperty.call(n,s)&&(o[s]=n[s]);o.originalType=e,o[p]="string"==typeof e?e:i,r[1]=o;for(var g=2;g<l;g++)r[g]=a[g];return t.createElement.apply(null,r)}return t.createElement.apply(null,a)}c.displayName="MDXCreateElement"},2123:(e,n,a)=>{a.r(n),a.d(n,{contentTitle:()=>r,default:()=>p,frontMatter:()=>l,metadata:()=>o,toc:()=>s});var t=a(8168),i=(a(6540),a(5680));const l={},r=void 0,o={unversionedId:"data-sources/clinvar-preview-json",id:"version-3.25/data-sources/clinvar-preview-json",title:"clinvar-preview-json",description:"small variants:",source:"@site/versioned_docs/version-3.25/data-sources/clinvar-preview-json.md",sourceDirName:"data-sources",slug:"/data-sources/clinvar-preview-json",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/clinvar-preview-json",editUrl:"https://github.com/Illumina/IlluminaConnectedAnnotationsDocumentation/edit/master/versioned_docs/version-3.25/data-sources/clinvar-preview-json.md",tags:[],version:"3.25",frontMatter:{}},s=[{value:"Variant Types",id:"variant-types",children:[],level:4},{value:"Review Statuses",id:"review-statuses",children:[],level:4},{value:"classification",id:"classification",children:[],level:4}],g={toc:s},d="wrapper";function p(e){let{components:n,...a}=e;return(0,i.yg)(d,(0,t.A)({},g,a,{components:n,mdxType:"MDXLayout"}),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"},"small variants:")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-json"},'{\n  "clinvar-preview": [\n    {\n      "altAllele": "A",\n      "refAllele": "G",\n      "variantType": "SNV",\n      "accession": "VCV000437934",\n      "version": "1",\n      "recordType": "classified",\n      "dateLastUpdated": "2023-08-06",\n      "rcvs": [\n        {\n          "accession": "RCV000505090",\n          "version": "1",\n          "classifications": {\n            "germlineClassification": {\n              "reviewStatus": "no assertion criteria provided",\n              "descriptions": [\n                {\n                  "dateLastEvaluated": "2016-08-31",\n                  "classification": "Pathogenic"\n                }\n              ]\n            }\n          },\n          "classifiedConditions": [\n            {\n              "condition": "Cleidocranial dysostosis",\n              "db": "MedGen",\n              "id": "C0008928"\n            }\n          ]\n        }\n      ],\n      "classifications": {\n        "germlineClassification": {\n          "reviewStatus": "no assertion criteria provided",\n          "classification": "Pathogenic",\n          "dateLastEvaluated": "2016-08-31",\n          "mostRecentSubmission": "2017-09-09",\n          "conditions": [\n            {\n              "type": "Disease",\n              "contributesToAggregateClassification": true,\n              "traits": [\n                {\n                  "id": "820",\n                  "name": {\n                    "xRefs": [\n                      {\n                        "db": "Genetic Alliance",\n                        "id": "Cleidocranial+Dysplasia/1683"\n                      },\n                      {\n                        "db": "SNOMED CT",\n                        "id": "65976001"\n                      }\n                    ],\n                    "value": "Cleidocranial dysostosis"\n                  }\n                }\n              ]\n            }\n          ]\n        }\n      },\n      "clinicalAssertions": [\n        {\n          "accession": "SCV000598565"\n        }\n      ]\n    }\n  ]\n}\n')),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"},"large variants:")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-json"},'{\n  "clinvar-preview": [\n    {\n      "chromosome": "17",\n      "begin": 150732,\n      "end": 14764202,\n      "variantType": "copy_number_gain",\n      "accession": "VCV000154089",\n      "version": "2",\n      "recordType": "classified",\n      "dateLastUpdated": "2023-10-15",\n      "rcvs": [\n        {\n          "accession": "RCV000142236",\n          "version": "6",\n          "classifications": {\n            "germlineClassification": {\n              "reviewStatus": "no assertion criteria provided",\n              "descriptions": [\n                {\n                  "dateLastEvaluated": "2014-03-10",\n                  "classification": "Pathogenic"\n                }\n              ]\n            }\n          },\n          "classifiedConditions": [\n            {\n              "condition": "See cases"\n            }\n          ]\n        }\n      ],\n      "classifications": {\n        "germlineClassification": {\n          "reviewStatus": "no assertion criteria provided",\n          "classification": "Pathogenic",\n          "dateLastEvaluated": "2014-03-10",\n          "mostRecentSubmission": "2015-07-13",\n          "conditions": [\n            {\n              "type": "PhenotypeInstruction",\n              "contributesToAggregateClassification": true,\n              "traits": [\n                {\n                  "id": "18728",\n                  "name": {\n                    "value": "See cases"\n                  }\n                }\n              ]\n            }\n          ]\n        }\n      },\n      "clinicalAssertions": [\n        {\n          "accession": "SCV000183512"\n        }\n      ]\n    }\n  ]\n}\n')),(0,i.yg)("table",null,(0,i.yg)("thead",{parentName:"table"},(0,i.yg)("tr",{parentName:"thead"},(0,i.yg)("th",{parentName:"tr",align:"left"},"Field"),(0,i.yg)("th",{parentName:"tr",align:"center"},"Type"),(0,i.yg)("th",{parentName:"tr",align:"left"},"Notes"))),(0,i.yg)("tbody",{parentName:"table"},(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"chromosome"),(0,i.yg)("td",{parentName:"tr",align:"center"},"string"),(0,i.yg)("td",{parentName:"tr",align:"left"},"Chromosome")),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"begin"),(0,i.yg)("td",{parentName:"tr",align:"center"},"integer"),(0,i.yg)("td",{parentName:"tr",align:"left"},"start position of variant")),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"end"),(0,i.yg)("td",{parentName:"tr",align:"center"},"integer"),(0,i.yg)("td",{parentName:"tr",align:"left"},"end of position of variant")),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"refAllele"),(0,i.yg)("td",{parentName:"tr",align:"center"},"string"),(0,i.yg)("td",{parentName:"tr",align:"left"})),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"altAllele"),(0,i.yg)("td",{parentName:"tr",align:"center"},"string"),(0,i.yg)("td",{parentName:"tr",align:"left"})),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"accession"),(0,i.yg)("td",{parentName:"tr",align:"center"},"string"),(0,i.yg)("td",{parentName:"tr",align:"left"},"ClinVar ID")),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"version"),(0,i.yg)("td",{parentName:"tr",align:"center"},"string"),(0,i.yg)("td",{parentName:"tr",align:"left"},"ClinVar version")),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"variantType"),(0,i.yg)("td",{parentName:"tr",align:"center"},"string"),(0,i.yg)("td",{parentName:"tr",align:"left"},"variant type")),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"recordType"),(0,i.yg)("td",{parentName:"tr",align:"center"},"string"),(0,i.yg)("td",{parentName:"tr",align:"left"},"record type")),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"dateLastUpdated"),(0,i.yg)("td",{parentName:"tr",align:"center"},"string"),(0,i.yg)("td",{parentName:"tr",align:"left"},"yyyy-MM-dd")),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"rcvs"),(0,i.yg)("td",{parentName:"tr",align:"center"},"array"),(0,i.yg)("td",{parentName:"tr",align:"left"},"RCV objects associated to this VCV")),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"classifications"),(0,i.yg)("td",{parentName:"tr",align:"center"},"array"),(0,i.yg)("td",{parentName:"tr",align:"left"},"classifications for this VCV")),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"clinicalAssertions"),(0,i.yg)("td",{parentName:"tr",align:"center"},"array"),(0,i.yg)("td",{parentName:"tr",align:"left"},"SCV objects associated to this VCV")),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:"left"},"isAlleleSpecific"),(0,i.yg)("td",{parentName:"tr",align:"center"},"bool"),(0,i.yg)("td",{parentName:"tr",align:"left"},"true when the current variant alternate allele matches the ClinVar alternate allele")))),(0,i.yg)("h4",{id:"variant-types"},"Variant Types"),(0,i.yg)("ul",null,(0,i.yg)("li",{parentName:"ul"},"copy_number_gain"),(0,i.yg)("li",{parentName:"ul"},"copy_number_loss"),(0,i.yg)("li",{parentName:"ul"},"deletion"),(0,i.yg)("li",{parentName:"ul"},"delins"),(0,i.yg)("li",{parentName:"ul"},"duplication"),(0,i.yg)("li",{parentName:"ul"},"insertion"),(0,i.yg)("li",{parentName:"ul"},"inversion"),(0,i.yg)("li",{parentName:"ul"},"SNV"),(0,i.yg)("li",{parentName:"ul"},"tandem_duplication")),(0,i.yg)("h4",{id:"review-statuses"},"Review Statuses"),(0,i.yg)("ul",null,(0,i.yg)("li",{parentName:"ul"},"criteria provided, conflicting classifications"),(0,i.yg)("li",{parentName:"ul"},"criteria provided, multiple submitters, no conflicts"),(0,i.yg)("li",{parentName:"ul"},"criteria provided, single submitter"),(0,i.yg)("li",{parentName:"ul"},"no assertion criteria provided"),(0,i.yg)("li",{parentName:"ul"},"no classification provided"),(0,i.yg)("li",{parentName:"ul"},"practice guideline"),(0,i.yg)("li",{parentName:"ul"},"reviewed by expert panel")),(0,i.yg)("h4",{id:"classification"},"classification"),(0,i.yg)("ul",null,(0,i.yg)("li",{parentName:"ul"},"Benign"),(0,i.yg)("li",{parentName:"ul"},"Likely benign"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic"),(0,i.yg)("li",{parentName:"ul"},"Uncertain significance"),(0,i.yg)("li",{parentName:"ul"},"Likely pathogenic"),(0,i.yg)("li",{parentName:"ul"},"Benign/Likely benign"),(0,i.yg)("li",{parentName:"ul"},"not provided"),(0,i.yg)("li",{parentName:"ul"},"conflicting data from submitters"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic/Likely pathogenic"),(0,i.yg)("li",{parentName:"ul"},"association"),(0,i.yg)("li",{parentName:"ul"},"Conflicting classifications of pathogenicity"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic; risk factor"),(0,i.yg)("li",{parentName:"ul"},"risk factor"),(0,i.yg)("li",{parentName:"ul"},"other"),(0,i.yg)("li",{parentName:"ul"},"drug response"),(0,i.yg)("li",{parentName:"ul"},"Uncertain significance; Pathogenic/Likely pathogenic"),(0,i.yg)("li",{parentName:"ul"},"Likely pathogenic, low penetrance"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic; Affects"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic, low penetrance"),(0,i.yg)("li",{parentName:"ul"},"protective"),(0,i.yg)("li",{parentName:"ul"},"Affects"),(0,i.yg)("li",{parentName:"ul"},"Benign; other"),(0,i.yg)("li",{parentName:"ul"},"Conflicting classifications of pathogenicity; other"),(0,i.yg)("li",{parentName:"ul"},"Conflicting classifications of pathogenicity; association"),(0,i.yg)("li",{parentName:"ul"},"Uncertain risk allele"),(0,i.yg)("li",{parentName:"ul"},"Uncertain significance; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Likely pathogenic; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Likely benign; association"),(0,i.yg)("li",{parentName:"ul"},"Likely risk allele"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic/Likely pathogenic; other"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic; other"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic/Likely pathogenic/Pathogenic, low penetrance"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic/Likely pathogenic; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Benign/Likely benign; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Uncertain significance/Uncertain risk allele"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic; association; protective"),(0,i.yg)("li",{parentName:"ul"},"protective; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Benign/Likely benign; other; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Benign/Likely benign; association"),(0,i.yg)("li",{parentName:"ul"},"Benign; association"),(0,i.yg)("li",{parentName:"ul"},"Affects; association; other"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic; protective"),(0,i.yg)("li",{parentName:"ul"},"Conflicting classifications of pathogenicity; drug response; other"),(0,i.yg)("li",{parentName:"ul"},"Conflicting classifications of pathogenicity; drug response"),(0,i.yg)("li",{parentName:"ul"},"Benign; drug response"),(0,i.yg)("li",{parentName:"ul"},"Likely pathogenic; other"),(0,i.yg)("li",{parentName:"ul"},"Conflicting classifications of pathogenicity; protective"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic/Likely pathogenic; drug response"),(0,i.yg)("li",{parentName:"ul"},"Benign/Likely benign; other"),(0,i.yg)("li",{parentName:"ul"},"Likely pathogenic/Likely risk allele"),(0,i.yg)("li",{parentName:"ul"},"Uncertain risk allele; protective"),(0,i.yg)("li",{parentName:"ul"},"association not found"),(0,i.yg)("li",{parentName:"ul"},"Affects; association"),(0,i.yg)("li",{parentName:"ul"},"Uncertain significance; association"),(0,i.yg)("li",{parentName:"ul"},"Likely benign; other"),(0,i.yg)("li",{parentName:"ul"},"Uncertain significance; other"),(0,i.yg)("li",{parentName:"ul"},"Conflicting classifications of pathogenicity; association; risk factor Pathogenic;"),(0,i.yg)("li",{parentName:"ul"},"association"),(0,i.yg)("li",{parentName:"ul"},"Benign; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Conflicting classifications of pathogenicity; other; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic/Likely risk allele; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Uncertain significance; drug response"),(0,i.yg)("li",{parentName:"ul"},"Conflicting classifications of pathogenicity; risk factor"),(0,i.yg)("li",{parentName:"ul"},"other; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic/Likely pathogenic/Likely risk allele"),(0,i.yg)("li",{parentName:"ul"},"Likely pathogenic; drug response"),(0,i.yg)("li",{parentName:"ul"},"Conflicting classifications of pathogenicity; Affects"),(0,i.yg)("li",{parentName:"ul"},"association; drug response; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic; drug response"),(0,i.yg)("li",{parentName:"ul"},"Affects; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic; drug response; other"),(0,i.yg)("li",{parentName:"ul"},"Likely pathogenic; protective"),(0,i.yg)("li",{parentName:"ul"},"confers sensitivity"),(0,i.yg)("li",{parentName:"ul"},"Likely pathogenic; association"),(0,i.yg)("li",{parentName:"ul"},"Benign; Affects"),(0,i.yg)("li",{parentName:"ul"},"Likely pathogenic; Affects"),(0,i.yg)("li",{parentName:"ul"},"Uncertain risk allele; risk factor"),(0,i.yg)("li",{parentName:"ul"},"drug response; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic/Likely risk allele"),(0,i.yg)("li",{parentName:"ul"},"Likely benign; drug response; other"),(0,i.yg)("li",{parentName:"ul"},"Benign/Likely benign; drug response"),(0,i.yg)("li",{parentName:"ul"},"Benign/Likely benign; drug response; other"),(0,i.yg)("li",{parentName:"ul"},"drug response; other"),(0,i.yg)("li",{parentName:"ul"},"association; drug response"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic; confers sensitivity"),(0,i.yg)("li",{parentName:"ul"},"association; risk factor"),(0,i.yg)("li",{parentName:"ul"},"Pathogenic/Pathogenic, low penetrance; other"),(0,i.yg)("li",{parentName:"ul"},"Benign; confers sensitivity"),(0,i.yg)("li",{parentName:"ul"},"confers sensitivity; other"),(0,i.yg)("li",{parentName:"ul"},"Likely pathogenic/Pathogenic, low penetrance"),(0,i.yg)("li",{parentName:"ul"},"Likely benign; risk factor")))}p.isMDXComponent=!0},6988:(e,n,a)=>{a.r(n),a.d(n,{contentTitle:()=>o,default:()=>m,frontMatter:()=>r,metadata:()=>s,toc:()=>g});var t=a(8168),i=(a(6540),a(5680)),l=a(2123);const r={title:"ClinVar Preview"},o=void 0,s={unversionedId:"data-sources/clinvar-preview",id:"version-3.25/data-sources/clinvar-preview",title:"ClinVar Preview",description:"Overview",source:"@site/versioned_docs/version-3.25/data-sources/clinvar-preview.mdx",sourceDirName:"data-sources",slug:"/data-sources/clinvar-preview",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/clinvar-preview",editUrl:"https://github.com/Illumina/IlluminaConnectedAnnotationsDocumentation/edit/master/versioned_docs/version-3.25/data-sources/clinvar-preview.mdx",tags:[],version:"3.25",frontMatter:{title:"ClinVar Preview"},sidebar:"docs",previous:{title:"ClinVar",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/clinvar"},next:{title:"COSMIC",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/data-sources/cosmic"}},g=[{value:"Overview",id:"overview",children:[],level:2},{value:"Parsing",id:"parsing",children:[{value:"Overall XML to JSON mapping",id:"overall-xml-to-json-mapping",children:[],level:3},{value:"Variation fields",id:"variation-fields",children:[],level:3},{value:"Location fields",id:"location-fields",children:[],level:3},{value:"RCVs",id:"rcvs",children:[{value:"Classifications",id:"classifications",children:[{value:"Germline Classification",id:"germline-classification",children:[],level:5},{value:"Classified Conditions",id:"classified-conditions",children:[],level:5}],level:4}],level:3},{value:"Classifications",id:"classifications-1",children:[{value:"Germline Classification",id:"germline-classification-1",children:[{value:"Conditions",id:"conditions",children:[],level:6}],level:5}],level:3},{value:"Clinical Assertions",id:"clinical-assertions",children:[],level:3}],level:2},{value:"Known Issues",id:"known-issues",children:[],level:2},{value:"Download URLs",id:"download-urls",children:[],level:2},{value:"JSON Output",id:"json-output",children:[],level:2},{value:"Building the supplementary files",id:"building-the-supplementary-files",children:[{value:"Using <code>clinvar</code> subcommands and source data files",id:"using-clinvar-subcommands-and-source-data-files",children:[],level:3}],level:2}],d={toc:g},p="wrapper";function m(e){let{components:n,...a}=e;return(0,i.yg)(p,(0,t.A)({},d,a,{components:n,mdxType:"MDXLayout"}),(0,i.yg)("h2",{id:"overview"},"Overview"),(0,i.yg)("p",null,"ClinVar is a freely accessible, public archive of reports of the relationships among human variations and phenotypes, with supporting evidence. ClinVar thus facilitates access to and communication about the relationships asserted between human variation and observed health status, and the history of that interpretation."),(0,i.yg)("div",{className:"admonition admonition-info alert alert--info"},(0,i.yg)("div",{parentName:"div",className:"admonition-heading"},(0,i.yg)("h5",{parentName:"div"},(0,i.yg)("span",{parentName:"h5",className:"admonition-icon"},(0,i.yg)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,i.yg)("path",{parentName:"svg",fillRule:"evenodd",d:"M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"}))),"Publication")),(0,i.yg)("div",{parentName:"div",className:"admonition-content"},(0,i.yg)("p",{parentName:"div"},"Melissa J Landrum, Jennifer M Lee, Mark Benson, Garth R Brown, Chen Chao, Shanmuga Chitipiralla, Baoshan Gu, Jennifer Hart, Douglas Hoffman, Wonhee Jang, Karen Karapetyan, Kenneth Katz, Chunlei Liu, Zenith Maddipatla, Adriana Malheiro, Kurt McDaniel, Michael Ovetsky, George Riley, George Zhou, J Bradley Holmes, Brandi L Kattman, Donna R Maglott, ClinVar: improving access to variant interpretations and supporting evidence, ",(0,i.yg)("em",{parentName:"p"},"Nucleic Acids Research"),", ",(0,i.yg)("strong",{parentName:"p"},"46"),", Issue D1, 4 January 2018, Pages D1062\u2013D1067, ",(0,i.yg)("a",{parentName:"p",href:"https://doi.org/10.1093/nar/gkx1153"},"https://doi.org/10.1093/nar/gkx1153")))),(0,i.yg)("p",null,"ClinVar Preview relates to the new ClinVar XML format introduced in 2024.\nFollowing sections describe the parsing and subsequent json format provided by Illumina Connected Annotations."),(0,i.yg)("h2",{id:"parsing"},"Parsing"),(0,i.yg)("p",null,"ClinVar ",(0,i.yg)("a",{parentName:"p",href:"https://github.com/ncbi/clinvar/blob/master/FTPSiteXmlChanges.md"},"recommends")," using the VCV XML file because it contains comprehensive information."),(0,i.yg)("p",null,"Parsing is simplified by using the XSD file generation.\nCommand for generating XSD file"),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-shell"},"xsd ClinVar_VCV.xsd /n:VariationArchive /c\n")),(0,i.yg)("h3",{id:"overall-xml-to-json-mapping"},"Overall XML to JSON mapping"),(0,i.yg)("table",null,(0,i.yg)("thead",{parentName:"table"},(0,i.yg)("tr",{parentName:"thead"},(0,i.yg)("th",{parentName:"tr",align:null},"key"),(0,i.yg)("th",{parentName:"tr",align:null},"type"),(0,i.yg)("th",{parentName:"tr",align:null},"description"),(0,i.yg)("th",{parentName:"tr",align:null},"XML path"))),(0,i.yg)("tbody",{parentName:"table"},(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"variantType")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"sequence ontology"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"VariationArchive.VariationType"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"accession")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"VCV Id from ClinVar"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"VariationArchive.Accession"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"version")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"VCV Id version"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"VariationArchive.Version"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"recordType")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"classified")),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"VariationArchive.RecordType"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"dateLastUpdated")),(0,i.yg)("td",{parentName:"tr",align:null},"date time"),(0,i.yg)("td",{parentName:"tr",align:null},"date VCV was last updated"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"VariationArchive.DateLastUpdated"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"chromosome")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"chromosome (large variants only)"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"VariationArchive.ClassifiedRecord.SimpleAllele.Location.SequenceLocation.Chr"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"begin")),(0,i.yg)("td",{parentName:"tr",align:null},"number"),(0,i.yg)("td",{parentName:"tr",align:null},"start position of the variant (large variants only)"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"VariationArchive.ClassifiedRecord.SimpleAllele.Location.SequenceLocation.positionVCF"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"end")),(0,i.yg)("td",{parentName:"tr",align:null},"number"),(0,i.yg)("td",{parentName:"tr",align:null},"end position of the variant (large variants only)"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"VariationArchive.ClassifiedRecord.SimpleAllele.Location.SequenceLocation.displayStop")," or calculated")),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"refAllele")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"reference alleles (small variants only)"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"VariationArchive.ClassifiedRecord.SimpleAllele.Location.SequenceLocation.referenceAlleleVCF"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"altAllele")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"alternate alleles (small variants only)"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"VariationArchive.ClassifiedRecord.SimpleAllele.Location.SequenceLocation.alternateAlleleVCF"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"rcvs")),(0,i.yg)("td",{parentName:"tr",align:null},"list"),(0,i.yg)("td",{parentName:"tr",align:null},"list of RCV objects"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"VariationArchive.ClassifiedRecord.RCVList"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"classifications")),(0,i.yg)("td",{parentName:"tr",align:null},"list"),(0,i.yg)("td",{parentName:"tr",align:null},"list of classification objects"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"VariationArchive.ClassifiedRecord.Classifications"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"clinicalAssertions")),(0,i.yg)("td",{parentName:"tr",align:null},"list"),(0,i.yg)("td",{parentName:"tr",align:null},"list of clinicalAssertion objects"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"VariationArchive.ClassifiedRecord.ClinicalAssertionList"))))),(0,i.yg)("h3",{id:"variation-fields"},"Variation fields"),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," XML ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-xml",metastring:"{4-8}","{4-8}":!0},'<VariationArchive\n    VariationID="1381081"\n    VariationName="NM_003000.3(SDHB):c.19_41dup (p.Pro14_Ala15insSerProTer)"\n    VariationType="Indel"\n    Accession="VCV001381081"\n    Version="3"\n    RecordType="classified"\n    DateLastUpdated="2024-01-26"\n    NumberOfSubmissions="1"\n    NumberOfSubmitters="1"\n    DateCreated="2022-03-28"\n    MostRecentSubmission="2023-02-07"\n>\n...\n')),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," JSON ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-json"},'{\n  "variantType": "delins",\n  "accession": "VCV001381081",\n  "version": "3",\n  "recordType": "classified",\n  "dateLastUpdated": "2024-01-26",\n  ...\n}\n')),(0,i.yg)("h3",{id:"location-fields"},"Location fields"),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-xml",metastring:"{9-15}","{9-15}":!0},'<SimpleAllele\n    AlleleID="196495"\n    VariationID="1381081"\n    >\n    <Location>\n        <CytogeneticLocation>1p36.13</CytogeneticLocation>\n        <SequenceLocation\n            Accession="NC_000001.11"\n            Chr="1"\n            Assembly="GRCh38"\n            positionVCF="17053978"\n            referenceAlleleVCF="C"\n            alternateAlleleVCF="CGGCAACCGGCGCCTCAAGGAGAG"\n            display_start="17053978"\n            display_stop="17053979"\n            AssemblyAccessionVersion="GCF_000001405.38"\n            forDisplay="true"\n            AssemblyStatus="current"\n            start="17053978"\n            stop="17053979"\n            variantLength="23"\n        />\n        <SequenceLocation Assembly="GRCh37" AssemblyAccessionVersion="GCF_000001405.25"\n                          AssemblyStatus="previous" Chr="1" Accession="NC_000001.10" start="17380473"\n                          stop="17380474" display_start="17380473" display_stop="17380474"\n                          variantLength="23" positionVCF="17380473" referenceAlleleVCF="C"\n                          alternateAlleleVCF="CGGCAACCGGCGCCTCAAGGAGAG"/>\n    </Location>\n    ...\n    </SimpleAllele>\n')),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," JSON Small Variant")),(0,i.yg)("p",null,"note the alleles are trimmed"),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-json"},'{\n  "altAllele": "GGCAACCGGCGCCTCAAGGAGAG",\n  "refAllele": "-",\n  ...\n}\n')),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," JSON Large Variant")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-json"},'{\n  "chromosome": "17",\n  "begin": 150732,\n  "end": 14764202,\n  ...\n}\n')),(0,i.yg)("h3",{id:"rcvs"},"RCVs"),(0,i.yg)("p",null,"RCV Object from XML path ",(0,i.yg)("inlineCode",{parentName:"p"},"VariationArchive.ClassifiedRecord.RCVList")),(0,i.yg)("table",null,(0,i.yg)("thead",{parentName:"table"},(0,i.yg)("tr",{parentName:"thead"},(0,i.yg)("th",{parentName:"tr",align:null},"key"),(0,i.yg)("th",{parentName:"tr",align:null},"type"),(0,i.yg)("th",{parentName:"tr",align:null},"description"),(0,i.yg)("th",{parentName:"tr",align:null},"XML sub-path"))),(0,i.yg)("tbody",{parentName:"table"},(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"accession")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"VCV Id from ClinVar"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"RCVList.RCVAccession.Accession"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"version")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"VCV Id version"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"RCVList.RCVAccession.Accession"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"classifications")),(0,i.yg)("td",{parentName:"tr",align:null},"list"),(0,i.yg)("td",{parentName:"tr",align:null},"list of classification objects"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"RCVList.RCVAccession.RCVClassifications"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"classifiedConditions")),(0,i.yg)("td",{parentName:"tr",align:null},"list"),(0,i.yg)("td",{parentName:"tr",align:null},"list of classified conditions"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"RCVList.RCVAccession.ClassifiedConditionList"))))),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," XML ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-xml"},'<RCVList>\n    <RCVAccession\n        Title="NM_003000.3(SDHB):c.19_41dup (p.Pro14_Ala15insSerProTer) AND multiple conditions"\n        Accession="RCV001921860"\n        Version="3">\n        <ClassifiedConditionList TraitSetID="23696">\n            ...\n        </ClassifiedConditionList>\n        <RCVClassifications>\n            ...\n        </RCVClassifications>\n    </RCVAccession>\n</RCVList>\n...\n')),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," JSON ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-json"},'{\n  "rcvs": [\n    {\n      "accession": "RCV001921860",\n      "version": "3",\n      "classifications": {\n        ...\n      },\n      "classifiedConditions": [\n        ...\n      ]\n    }\n  ]\n}\n')),(0,i.yg)("h4",{id:"classifications"},"Classifications"),(0,i.yg)("p",null,"Classification object from XML path ",(0,i.yg)("inlineCode",{parentName:"p"},"VariationArchive.ClassifiedRecord.RCVList.RCVAccession.RCVClassifications"),"\n",(0,i.yg)("inlineCode",{parentName:"p"},"classification")," can be of following types:"),(0,i.yg)("ol",null,(0,i.yg)("li",{parentName:"ol"},(0,i.yg)("inlineCode",{parentName:"li"},"germlineClassification")),(0,i.yg)("li",{parentName:"ol"},(0,i.yg)("inlineCode",{parentName:"li"},"somaticClinicalImpact")),(0,i.yg)("li",{parentName:"ol"},(0,i.yg)("inlineCode",{parentName:"li"},"oncogenicityClassification"))),(0,i.yg)("h5",{id:"germline-classification"},"Germline Classification"),(0,i.yg)("p",null,"Classification object from XML path ",(0,i.yg)("inlineCode",{parentName:"p"},"VariationArchive.ClassifiedRecord.RCVList.RCVAccession.RCVClassifications.GermlineClassification")),(0,i.yg)("table",null,(0,i.yg)("thead",{parentName:"table"},(0,i.yg)("tr",{parentName:"thead"},(0,i.yg)("th",{parentName:"tr",align:null},"key"),(0,i.yg)("th",{parentName:"tr",align:null},"type"),(0,i.yg)("th",{parentName:"tr",align:null},"description"),(0,i.yg)("th",{parentName:"tr",align:null},"XML sub-path"))),(0,i.yg)("tbody",{parentName:"table"},(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"reviewStatus")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"review status"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"GermlineClassification.ReviewStatus"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"descriptions")),(0,i.yg)("td",{parentName:"tr",align:null},"list"),(0,i.yg)("td",{parentName:"tr",align:null},"list of classification objects"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"GermlineClassification.Description"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"descriptions[].classification")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"classification"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"GermlineClassification.Description.Value"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"descriptions[].dateLastEvaluated")),(0,i.yg)("td",{parentName:"tr",align:null},"date"),(0,i.yg)("td",{parentName:"tr",align:null},"date last evaluated"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"GermlineClassification.Description.DateLastEvaluated"))))),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," XML ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-xml"},'<RCVClassifications>\n    <GermlineClassification>\n        <ReviewStatus>criteria provided, single submitter</ReviewStatus>\n        <Description DateLastEvaluated="2021-08-04" SubmissionCount="1">Pathogenic</Description>\n    </GermlineClassification>\n</RCVClassifications>\n')),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," JSON ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-json"},'{\n  "classifications": {\n    "germlineClassification": {\n      "reviewStatus": "criteria provided, single submitter",\n      "descriptions": [\n        {\n          "dateLastEvaluated": "2021-08-04",\n          "classification": "Pathogenic"\n        }\n      ]\n    }\n  }\n}\n')),(0,i.yg)("h5",{id:"classified-conditions"},"Classified Conditions"),(0,i.yg)("p",null,"Classified conditions object from XML path ",(0,i.yg)("inlineCode",{parentName:"p"},"VariationArchive.ClassifiedRecord.RCVList.RCVAccession.ClassifiedConditionList")),(0,i.yg)("table",null,(0,i.yg)("thead",{parentName:"table"},(0,i.yg)("tr",{parentName:"thead"},(0,i.yg)("th",{parentName:"tr",align:null},"key"),(0,i.yg)("th",{parentName:"tr",align:null},"type"),(0,i.yg)("th",{parentName:"tr",align:null},"description"),(0,i.yg)("th",{parentName:"tr",align:null},"XML sub-path"))),(0,i.yg)("tbody",{parentName:"table"},(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"condition")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"VCV Id from ClinVar"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ClassifiedConditionList.ClassifiedCondition.Value"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"db")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"list of classification objects"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ClassifiedConditionList.ClassifiedCondition.DB"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"id")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"classification"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ClassifiedConditionList.ClassifiedCondition.ID"))))),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," XML ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-xml"},'<ClassifiedConditionList TraitSetID="23696">\n    <ClassifiedCondition DB="MedGen" ID="C0238198">Gastrointestinal stromal tumor</ClassifiedCondition>\n    <ClassifiedCondition DB="MedGen" ID="C1861848">Paragangliomas 4</ClassifiedCondition>\n    <ClassifiedCondition DB="MedGen" ID="C0031511">Pheochromocytoma</ClassifiedCondition>\n</ClassifiedConditionList>\n')),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," JSON ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-json"},'{\n  "classifiedConditions": [\n    {\n      "condition": "Gastrointestinal stromal tumor",\n      "db": "MedGen",\n      "id": "C0238198"\n    },\n    {\n      "condition": "Paragangliomas 4",\n      "db": "MedGen",\n      "id": "C1861848"\n    },\n    {\n      "condition": "Pheochromocytoma",\n      "db": "MedGen",\n      "id": "C0031511"\n    }\n  ]\n}\n')),(0,i.yg)("h3",{id:"classifications-1"},"Classifications"),(0,i.yg)("p",null,"Classification object from XML path ",(0,i.yg)("inlineCode",{parentName:"p"},"VariationArchive.ClassifiedRecord.Classifications"),"\n",(0,i.yg)("inlineCode",{parentName:"p"},"classification")," can be of following types:"),(0,i.yg)("ol",null,(0,i.yg)("li",{parentName:"ol"},(0,i.yg)("inlineCode",{parentName:"li"},"germlineClassification")),(0,i.yg)("li",{parentName:"ol"},(0,i.yg)("inlineCode",{parentName:"li"},"somaticClinicalImpact")),(0,i.yg)("li",{parentName:"ol"},(0,i.yg)("inlineCode",{parentName:"li"},"oncogenicityClassification"))),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," XML ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-xml"},'<Classifications>\n    <GermlineClassification DateLastEvaluated="2021-08-04" NumberOfSubmissions="1" NumberOfSubmitters="1"\n                            DateCreated="2022-03-28" MostRecentSubmission="2023-02-07">\n    ...\n    </GermlineClassification>\n</Classifications>\n')),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," JSON ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-json"},'"classifications": {\n    "germlineClassification": {...}\n}\n')),(0,i.yg)("h5",{id:"germline-classification-1"},"Germline Classification"),(0,i.yg)("p",null,"Classification object from XML path ",(0,i.yg)("inlineCode",{parentName:"p"},"VariationArchive.ClassifiedRecord.Classifications.GermlineClassification")),(0,i.yg)("table",null,(0,i.yg)("thead",{parentName:"table"},(0,i.yg)("tr",{parentName:"thead"},(0,i.yg)("th",{parentName:"tr",align:null},"key"),(0,i.yg)("th",{parentName:"tr",align:null},"type"),(0,i.yg)("th",{parentName:"tr",align:null},"description"),(0,i.yg)("th",{parentName:"tr",align:null},"XML sub-path"))),(0,i.yg)("tbody",{parentName:"table"},(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"classification")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"classification"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"GermlineClassification.Description"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"reviewStatus")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"review status"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"GermlineClassification.ReviewStatus"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"dateLastEvaluated")),(0,i.yg)("td",{parentName:"tr",align:null},"date"),(0,i.yg)("td",{parentName:"tr",align:null},"date last evaluated"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"GermlineClassification.DateLastEvaluated"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"mostRecentSubmission")),(0,i.yg)("td",{parentName:"tr",align:null},"date"),(0,i.yg)("td",{parentName:"tr",align:null},"date last evaluated"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"GermlineClassification.MostRecentSubmission"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"pubMedIds")),(0,i.yg)("td",{parentName:"tr",align:null},"list"),(0,i.yg)("td",{parentName:"tr",align:null},"list of PubMedIds"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"GermlineClassification.Citation.ID.Value"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"conditions")),(0,i.yg)("td",{parentName:"tr",align:null},"list"),(0,i.yg)("td",{parentName:"tr",align:null},"list of conditions"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"GermlineClassification.ConditionList"))))),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," XML ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-xml"},'<Classifications>\n    <GermlineClassification DateLastEvaluated="2021-08-04" NumberOfSubmissions="1" NumberOfSubmitters="1"\n                            DateCreated="2022-03-28" MostRecentSubmission="2023-02-07">\n        <ReviewStatus>criteria provided, single submitter</ReviewStatus>\n        <Description>Pathogenic</Description>\n        <Citation Type="general">\n            <ID Source="PubMed">19454582</ID>\n        </Citation>\n        <Citation Type="general">\n            <ID Source="PubMed">19802898</ID>\n        </Citation>\n        <ConditionList>\n            ...\n        </ConditionList>\n    </GermlineClassification>\n</Classifications>\n')),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," JSON ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-json"},'{\n  "classifications": {\n    "germlineClassification": {\n      "classification": "Pathogenic",\n      "reviewStatus": "criteria provided, single submitter",\n      "dateLastEvaluated": "2021-08-04",\n      "mostRecentSubmission": "2023-02-07",\n      "conditions": [...],\n      "pubMedIds": [\n        "19454582",\n        "19802898"\n      ]\n    }\n  }\n}\n')),(0,i.yg)("h6",{id:"conditions"},"Conditions"),(0,i.yg)("p",null,"Conditions object from XML path ",(0,i.yg)("inlineCode",{parentName:"p"},"VariationArchive.ClassifiedRecord.Classifications.GermlineClassification.ConditionList")),(0,i.yg)("table",null,(0,i.yg)("thead",{parentName:"table"},(0,i.yg)("tr",{parentName:"thead"},(0,i.yg)("th",{parentName:"tr",align:null},"key"),(0,i.yg)("th",{parentName:"tr",align:null},"type"),(0,i.yg)("th",{parentName:"tr",align:null},"description"),(0,i.yg)("th",{parentName:"tr",align:null},"XML sub-path"))),(0,i.yg)("tbody",{parentName:"table"},(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"type")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"classification"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ConditionList.TraitSet.Type"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"contributesToAggregateClassification")),(0,i.yg)("td",{parentName:"tr",align:null},"True or blank"),(0,i.yg)("td",{parentName:"tr",align:null},"contributes to aggregate classifcation"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ConditionList.TraitSet.ContributesToAggregateClassification"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"traits")),(0,i.yg)("td",{parentName:"tr",align:null},"list"),(0,i.yg)("td",{parentName:"tr",align:null},"trait objects"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ConditionList.TraitSet.Trait"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"traits[].id")),(0,i.yg)("td",{parentName:"tr",align:null},"date"),(0,i.yg)("td",{parentName:"tr",align:null},"date last evaluated"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ConditionList.TraitSet.Trait"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"traits[].name")),(0,i.yg)("td",{parentName:"tr",align:null},"object"),(0,i.yg)("td",{parentName:"tr",align:null},"trait name object"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ConditionList.TraitSet.Trait"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"traits[].name.value")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"preferred trait name"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ConditionList.TraitSet.Trait.Name.ElementValue.Type"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"traits[].name.xRefs")),(0,i.yg)("td",{parentName:"tr",align:null},"list"),(0,i.yg)("td",{parentName:"tr",align:null},"list of cross references"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ConditionList.TraitSet.Trait.Name.XRef"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"traits[].name.xRefs[].db")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"preferred name cross reference database"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ConditionList.TraitSet.Trait.Name.XRef.DB"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"traits[].name.xRefs[].id")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"preferred name cross reference identifier"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ConditionList.TraitSet.Trait.Name.XRef.ID"))))),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," XML ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-xml"},'<Classifications>\n    <GermlineClassification DateLastEvaluated="2021-08-04" NumberOfSubmissions="1" NumberOfSubmitters="1"\n                            DateCreated="2022-03-28" MostRecentSubmission="2023-02-07">\n        <ConditionList>\n            <TraitSet ID="23696" Type="Disease" ContributesToAggregateClassification="true">\n                <Trait ID="3796" Type="Disease">\n                    <Name>\n                        <ElementValue Type="Preferred">Pheochromocytoma</ElementValue>\n                        <XRef ID="Pheochromocytoma/5718" DB="Genetic Alliance"/>\n                        <XRef ID="HP:0002666" DB="Human Phenotype Ontology"/>\n                        <XRef ID="MONDO:0008233" DB="MONDO"/>\n                    </Name>\n                    <Name>\n                        <ElementValue Type="Alternate">Chromaffinoma</ElementValue>\n                    </Name>\n                    ...\n                </Trait>\n            </TraitSet>\n        </ConditionList>\n    </GermlineClassification>\n</Classifications>\n')),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," JSON ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-json"},'{\n  "classifications": {\n    "germlineClassification": {\n      "classification": "Pathogenic",\n      "reviewStatus": "criteria provided, single submitter",\n      "dateLastEvaluated": "2021-08-04",\n      "mostRecentSubmission": "2023-02-07",\n      "conditions": [\n        {\n          "type": "Disease",\n          "contributesToAggregateClassification": true,\n          "traits": [\n            {\n              "id": "3796",\n              "name": {\n                "xRefs": [\n                  {\n                    "db": "Genetic Alliance",\n                    "id": "Pheochromocytoma/5718"\n                  },\n                  {\n                    "db": "Human Phenotype Ontology",\n                    "id": "HP:0002666"\n                  },\n                  {\n                    "db": "MONDO",\n                    "id": "MONDO:0008233"\n                  }\n                ],\n                "value": "Pheochromocytoma"\n              }\n            }\n          ]\n        }\n      ],\n      "pubMedIds": [\n        "19454582",\n        "19802898"\n      ]\n    }\n  }\n}\n')),(0,i.yg)("h3",{id:"clinical-assertions"},"Clinical Assertions"),(0,i.yg)("p",null,"Conditions object from XML path ",(0,i.yg)("inlineCode",{parentName:"p"},"VariationArchive.ClassifiedRecord.ClinicalAssertionList")),(0,i.yg)("table",null,(0,i.yg)("thead",{parentName:"table"},(0,i.yg)("tr",{parentName:"thead"},(0,i.yg)("th",{parentName:"tr",align:null},"key"),(0,i.yg)("th",{parentName:"tr",align:null},"type"),(0,i.yg)("th",{parentName:"tr",align:null},"description"),(0,i.yg)("th",{parentName:"tr",align:null},"XML sub-path"))),(0,i.yg)("tbody",{parentName:"table"},(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"accession")),(0,i.yg)("td",{parentName:"tr",align:null},"string"),(0,i.yg)("td",{parentName:"tr",align:null},"SCV Id from ClinVar"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ClinicalAssertionList.ClinVarAccession.Accession"))),(0,i.yg)("tr",{parentName:"tbody"},(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"pubMedIds")),(0,i.yg)("td",{parentName:"tr",align:null},"list"),(0,i.yg)("td",{parentName:"tr",align:null},"list of PubMedIds"),(0,i.yg)("td",{parentName:"tr",align:null},(0,i.yg)("inlineCode",{parentName:"td"},"ClinicalAssertionList.ClinicalAssertion.AttributeSet.Citation.ID.Value"))))),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," XML ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-xml"},'<ClinicalAssertionList>\n    <ClinicalAssertion ID="4172562" SubmissionDate="2023-01-11" DateLastUpdated="2023-02-07"\n                       DateCreated="2022-03-28">\n        <ClinVarSubmissionID localKey="12475853|MedGen:C0238198;C1861848;C0031511"\n                             submittedAssembly="GRCh37"/>\n        <ClinVarAccession\n            Accession="SCV002152762"\n            DateUpdated="2023-02-07"\n            DateCreated="2022-03-28"\n            Type="SCV"\n            Version="2"\n            SubmitterName="Invitae"\n            OrgID="500031"\n            OrganizationCategory="laboratory"\n            OrgAbbreviation="Invitae"\n        />\n        <RecordStatus>current</RecordStatus>\n        <Classification DateLastEvaluated="2021-08-04">\n            ...\n        </Classification>\n        <Assertion>variation to disease</Assertion>\n        <AttributeSet>\n            <Attribute Type="AssertionMethod">Invitae Variant Classification Sherloc (09022015)</Attribute>\n            <Citation>\n                <ID Source="PubMed">28492532</ID>\n            </Citation>\n        </AttributeSet>\n        <ObservedInList>\n            ...\n        </ObservedInList>\n        <SimpleAllele>\n            ...\n        </SimpleAllele>\n        <TraitSet Type="Disease">\n            ...\n        </TraitSet>\n        <SubmissionNameList>\n            ...\n        </SubmissionNameList>\n    </ClinicalAssertion>\n</ClinicalAssertionList>\n')),(0,i.yg)("p",null,(0,i.yg)("strong",{parentName:"p"}," JSON ")),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-json"},'{\n"clinicalAssertions": [\n    {\n      "accession": "SCV002152762",\n      "pubMedIds": [\n        "28492532"\n      ]\n    }\n  ]\n}\n')),(0,i.yg)("h2",{id:"known-issues"},"Known Issues"),(0,i.yg)("div",{className:"admonition admonition-caution alert alert--warning"},(0,i.yg)("div",{parentName:"div",className:"admonition-heading"},(0,i.yg)("h5",{parentName:"div"},(0,i.yg)("span",{parentName:"h5",className:"admonition-icon"},(0,i.yg)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"16",height:"16",viewBox:"0 0 16 16"},(0,i.yg)("path",{parentName:"svg",fillRule:"evenodd",d:"M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"}))),"Known Issues")),(0,i.yg)("div",{parentName:"div",className:"admonition-content"},(0,i.yg)("p",{parentName:"div"},"Entries with following missing/incorrect information are skipped"),(0,i.yg)("ol",{parentName:"div"},(0,i.yg)("li",{parentName:"ol"},"Invalid Ref Allele (example  ",(0,i.yg)("inlineCode",{parentName:"li"},"VCV000437934"),")"),(0,i.yg)("li",{parentName:"ol"},"Invalid Alt Allele (example  ",(0,i.yg)("inlineCode",{parentName:"li"},"VCV000006637"),")"),(0,i.yg)("li",{parentName:"ol"},"Following variant types are not supported:",(0,i.yg)("ol",{parentName:"li"},(0,i.yg)("li",{parentName:"ol"},(0,i.yg)("inlineCode",{parentName:"li"},"Variation")," (example  ",(0,i.yg)("inlineCode",{parentName:"li"},"VCV000001101"),")"),(0,i.yg)("li",{parentName:"ol"},(0,i.yg)("inlineCode",{parentName:"li"},"fusion")," (example  ",(0,i.yg)("inlineCode",{parentName:"li"},"VCV000015269"),")"),(0,i.yg)("li",{parentName:"ol"},(0,i.yg)("inlineCode",{parentName:"li"},"unknown")," (example  ",(0,i.yg)("inlineCode",{parentName:"li"},"VCV000017564"),")"),(0,i.yg)("li",{parentName:"ol"},(0,i.yg)("inlineCode",{parentName:"li"},"protein only")," (example  ",(0,i.yg)("inlineCode",{parentName:"li"},"VCV000132152"),")"),(0,i.yg)("li",{parentName:"ol"},(0,i.yg)("inlineCode",{parentName:"li"},"Complex")," (example  ",(0,i.yg)("inlineCode",{parentName:"li"},"VCV000221337"),")"),(0,i.yg)("li",{parentName:"ol"},(0,i.yg)("inlineCode",{parentName:"li"},"Translocation")," (example  ",(0,i.yg)("inlineCode",{parentName:"li"},"VCV000267801"),")"),(0,i.yg)("li",{parentName:"ol"},(0,i.yg)("inlineCode",{parentName:"li"},"no_sequence_alteration")," (example  ",(0,i.yg)("inlineCode",{parentName:"li"},"VCV000010504"),")"))),(0,i.yg)("li",{parentName:"ol"},"Only records of type ",(0,i.yg)("inlineCode",{parentName:"li"},"classified")," are included ","[VCV with type ",(0,i.yg)("inlineCode",{parentName:"li"},"included")," is skipped (example  ",(0,i.yg)("inlineCode",{parentName:"li"},"VCV000431749"),")]"),(0,i.yg)("li",{parentName:"ol"},"Records with missing genomic location are skipped (example  ",(0,i.yg)("inlineCode",{parentName:"li"},"VCV000000254"),")")))),(0,i.yg)("h2",{id:"download-urls"},"Download URLs"),(0,i.yg)("p",null,(0,i.yg)("a",{parentName:"p",href:"https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_00-latest.xml.gz"},"https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_00-latest.xml.gz")),(0,i.yg)("h2",{id:"json-output"},"JSON Output"),(0,i.yg)(l.default,{mdxType:"JSON"}),(0,i.yg)("h2",{id:"building-the-supplementary-files"},"Building the supplementary files"),(0,i.yg)("p",null,"There are 2 ways of building your own OMIM supplementary files using ",(0,i.yg)("inlineCode",{parentName:"p"},"SAUtils"),"."),(0,i.yg)("p",null,"The first way is to use ",(0,i.yg)("inlineCode",{parentName:"p"},"SAUtils")," command's subcommands ",(0,i.yg)("inlineCode",{parentName:"p"},"clinvar"),".\nThe ClinVar ",(0,i.yg)("inlineCode",{parentName:"p"},".nsa")," and ",(0,i.yg)("inlineCode",{parentName:"p"},".nsi")," for Illumina Connected Annotations can be built using the ",(0,i.yg)("inlineCode",{parentName:"p"},"SAUtils")," command's ",(0,i.yg)("inlineCode",{parentName:"p"},"clinvar")," subcommand."),(0,i.yg)("p",null,"The second way is to use ",(0,i.yg)("inlineCode",{parentName:"p"},"SAUtils")," command's subcommands ",(0,i.yg)("inlineCode",{parentName:"p"},"AutoDownloadGenerate"),". To use ",(0,i.yg)("inlineCode",{parentName:"p"},"AutoDownloadGenerate"),", read more in ",(0,i.yg)("inlineCode",{parentName:"p"},"SAUtils")," section."),(0,i.yg)("h3",{id:"using-clinvar-subcommands-and-source-data-files"},"Using ",(0,i.yg)("inlineCode",{parentName:"h3"},"clinvar")," subcommands and source data files"),(0,i.yg)("p",null,"Two input ",(0,i.yg)("inlineCode",{parentName:"p"},".xml")," files and a ",(0,i.yg)("inlineCode",{parentName:"p"},".version")," file are required in order to build the ",(0,i.yg)("inlineCode",{parentName:"p"},".nsa")," and ",(0,i.yg)("inlineCode",{parentName:"p"},".nsi")," file. You should have the following files:"),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-scss"},"ClinVarVCVRelease_00-latest.xml.gz\nClinVarVCVRelease_00-latest.xml.gz.version\n")),(0,i.yg)("p",null,"The version file is a json file with the following format."),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-scss"},'{\n  "name": "ClinVar",\n  "version": "20240501",\n  "description": "A freely accessible, public archive of reports of the relationships among human variations and phenotypes, with supporting evidence",\n  "releaseDate": "2024-05-01"\n}\n')),(0,i.yg)("p",null,"You have to adjust the version and release date according to the actual date of the ClinVar."),(0,i.yg)("p",null,"Here is a sample execution:"),(0,i.yg)("pre",null,(0,i.yg)("code",{parentName:"pre",className:"language-shell"},"dotnet SAUtils ClinVarPreview \\\n--r ~/References/7/Homo_sapiens.GRCh38.Nirvana.dat\\\n--vcv ClinVarVCVRelease_00-latest.xml.gz\\\n--o output\n---------------------------------------------------------------------------\nSAUtils                                             (c) 2024 Illumina, Inc.\n                                                                     3.25.0\n---------------------------------------------------------------------------\n\nParsing XML completed in 14.7 mins.\nSorting and adjusting completed in 4.7 mins.\nWriting 2351609 Small Varaints\nChromosome 1 completed in 00:00:57.1\nChromosome 2 completed in 00:01:30.8\nChromosome 3 completed in 00:00:32.9\nChromosome 4 completed in 00:00:21.2\nChromosome 5 completed in 00:00:31.7\nChromosome 6 completed in 00:00:34.6\nChromosome 7 completed in 00:00:27.9\nChromosome 8 completed in 00:00:17.9\nChromosome 9 completed in 00:00:34.0\nChromosome 10 completed in 00:00:26.6\nChromosome 11 completed in 00:00:35.4\nChromosome 12 completed in 00:00:31.5\nChromosome 13 completed in 00:00:22.7\nChromosome 14 completed in 00:00:22.7\nChromosome 15 completed in 00:00:23.7\nChromosome 16 completed in 00:00:39.6\nChromosome 17 completed in 00:00:46.7\nChromosome 18 completed in 00:00:10.2\nChromosome 19 completed in 00:00:32.9\nChromosome 20 completed in 00:00:10.7\nChromosome 21 completed in 00:00:05.3\nChromosome 22 completed in 00:00:11.0\nChromosome X completed in 00:00:19.6\nChromosome Y completed in 00:00:00.1\nChromosome MT completed in 00:00:00.3\nMaximum bp shifted for any variant:1\nNSA writing completed in 11.5 mins.\nWriting 76122 Large Varaints\nWriting 76122 intervals to database...\nNSI writing completed in 1.1 mins.\n\nTime: 00:32:10.9\nProcess finished with exit code 0.\n\n\n")))}m.isMDXComponent=!0}}]);