"use strict";(self.webpackChunknirvana_documentation=self.webpackChunknirvana_documentation||[]).push([[6348],{3905:(t,e,a)=>{a.d(e,{Zo:()=>c,kt:()=>k});var n=a(7294);function i(t,e,a){return e in t?Object.defineProperty(t,e,{value:a,enumerable:!0,configurable:!0,writable:!0}):t[e]=a,t}function l(t,e){var a=Object.keys(t);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(t);e&&(n=n.filter((function(e){return Object.getOwnPropertyDescriptor(t,e).enumerable}))),a.push.apply(a,n)}return a}function r(t){for(var e=1;e<arguments.length;e++){var a=null!=arguments[e]?arguments[e]:{};e%2?l(Object(a),!0).forEach((function(e){i(t,e,a[e])})):Object.getOwnPropertyDescriptors?Object.defineProperties(t,Object.getOwnPropertyDescriptors(a)):l(Object(a)).forEach((function(e){Object.defineProperty(t,e,Object.getOwnPropertyDescriptor(a,e))}))}return t}function o(t,e){if(null==t)return{};var a,n,i=function(t,e){if(null==t)return{};var a,n,i={},l=Object.keys(t);for(n=0;n<l.length;n++)a=l[n],e.indexOf(a)>=0||(i[a]=t[a]);return i}(t,e);if(Object.getOwnPropertySymbols){var l=Object.getOwnPropertySymbols(t);for(n=0;n<l.length;n++)a=l[n],e.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(t,a)&&(i[a]=t[a])}return i}var s=n.createContext({}),d=function(t){var e=n.useContext(s),a=e;return t&&(a="function"==typeof t?t(e):r(r({},e),t)),a},c=function(t){var e=d(t.components);return n.createElement(s.Provider,{value:e},t.children)},u="mdxType",p={inlineCode:"code",wrapper:function(t){var e=t.children;return n.createElement(n.Fragment,{},e)}},m=n.forwardRef((function(t,e){var a=t.components,i=t.mdxType,l=t.originalType,s=t.parentName,c=o(t,["components","mdxType","originalType","parentName"]),u=d(a),m=i,k=u["".concat(s,".").concat(m)]||u[m]||p[m]||l;return a?n.createElement(k,r(r({ref:e},c),{},{components:a})):n.createElement(k,r({ref:e},c))}));function k(t,e){var a=arguments,i=e&&e.mdxType;if("string"==typeof t||i){var l=a.length,r=new Array(l);r[0]=m;var o={};for(var s in e)hasOwnProperty.call(e,s)&&(o[s]=e[s]);o.originalType=t,o[u]="string"==typeof t?t:i,r[1]=o;for(var d=2;d<l;d++)r[d]=a[d];return n.createElement.apply(null,r)}return n.createElement.apply(null,a)}m.displayName="MDXCreateElement"},291:(t,e,a)=>{a.r(e),a.d(e,{contentTitle:()=>r,default:()=>u,frontMatter:()=>l,metadata:()=>o,toc:()=>s});var n=a(7462),i=(a(7294),a(3905));const l={title:"Variant Classification"},r=void 0,o={unversionedId:"core-functionality/variant-classification",id:"core-functionality/variant-classification",title:"Variant Classification",description:"Introduction",source:"@site/docs/core-functionality/variant-classification.md",sourceDirName:"core-functionality",slug:"/core-functionality/variant-classification",permalink:"/IlluminaConnectedAnnotationsDocumentation/core-functionality/variant-classification",editUrl:"https://github.com/Illumina/IlluminaConnectedAnnotationsDocumentation/edit/master/docs/core-functionality/variant-classification.md",tags:[],version:"current",frontMatter:{title:"Variant Classification"},sidebar:"docs",previous:{title:"Variant IDs",permalink:"/IlluminaConnectedAnnotationsDocumentation/core-functionality/variant-ids"},next:{title:"Jasix",permalink:"/IlluminaConnectedAnnotationsDocumentation/utilities/jasix"}},s=[{value:"Introduction",id:"introduction",children:[{value:"Classifying Small and Structural Variants",id:"classifying-small-and-structural-variants",children:[],level:3}],level:2},{value:"Examples",id:"examples",children:[{value:"Small Variants",id:"small-variants",children:[{value:"SNP",id:"snp",children:[{value:"Input Data:",id:"input-data",children:[],level:5},{value:"Classification:",id:"classification",children:[],level:5}],level:4},{value:"Short Indel",id:"short-indel",children:[{value:"Input Data:",id:"input-data-1",children:[],level:5},{value:"Classification:",id:"classification-1",children:[],level:5}],level:4},{value:"Multiallelic SNP",id:"multiallelic-snp",children:[{value:"Input Data:",id:"input-data-2",children:[],level:5},{value:"Classification:",id:"classification-2",children:[],level:5}],level:4}],level:3},{value:"Structural Variants",id:"structural-variants",children:[{value:"Deletion",id:"deletion",children:[{value:"Input Data:",id:"input-data-3",children:[],level:5},{value:"Classification:",id:"classification-3",children:[],level:5}],level:4},{value:"Deletion with explicit alleles",id:"deletion-with-explicit-alleles",children:[{value:"Input Data:",id:"input-data-4",children:[],level:5},{value:"Classification:",id:"classification-4",children:[],level:5}],level:4},{value:"Run Of Homozygosity",id:"run-of-homozygosity",children:[{value:"Input Data:",id:"input-data-5",children:[],level:5},{value:"Classification:",id:"classification-5",children:[],level:5}],level:4},{value:"Repeat Expansion",id:"repeat-expansion",children:[{value:"Input Data:",id:"input-data-6",children:[],level:5},{value:"Classification:",id:"classification-6",children:[],level:5}],level:4},{value:"Translocation",id:"translocation",children:[{value:"Input Data:",id:"input-data-7",children:[],level:5},{value:"Classification:",id:"classification-7",children:[],level:5}],level:4},{value:"Translocation",id:"translocation-1",children:[{value:"Input Data:",id:"input-data-8",children:[],level:5},{value:"Classification:",id:"classification-8",children:[],level:5}],level:4}],level:3}],level:2}],d={toc:s},c="wrapper";function u(t){let{components:e,...a}=t;return(0,i.kt)(c,(0,n.Z)({},d,a,{components:e,mdxType:"MDXLayout"}),(0,i.kt)("h2",{id:"introduction"},"Introduction"),(0,i.kt)("p",null,"Illumina Connected Annotation follows the ",(0,i.kt)("a",{parentName:"p",href:"https://samtools.github.io/hts-specs/VCFv4.5.pdf"},"VCF Specification"),"\nwhen parsing the VCF files. VCF files classify the variants based on alleles and certain ",(0,i.kt)("inlineCode",{parentName:"p"},"INFO")," fields. "),(0,i.kt)("p",null,"While the VCF variant type does not depend on the size of the variant, (both a ",(0,i.kt)("inlineCode",{parentName:"p"},"ATG->A")," and ",(0,i.kt)("inlineCode",{parentName:"p"},"A-><DEL>")," are ",(0,i.kt)("inlineCode",{parentName:"p"},"deletion"),") for the purpose of annotation, it is useful to classify variants into small and structural (large) variants. This is due to the realization that reporting annotations for a deletion like ",(0,i.kt)("inlineCode",{parentName:"p"},"1-1000-ATG-A")," from ",(0,i.kt)("inlineCode",{parentName:"p"},"ClinVar")," for a 20kbp ",(0,i.kt)("inlineCode",{parentName:"p"},"<DEL>")," at that same location is not helpful and vice versa.\nFollowing guidelines are used by Illumina Connected Annotations."),(0,i.kt)("div",{className:"admonition admonition-info alert alert--info"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"}))),"info")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"Structural variants are treated as intervals when providing supplementary annotations such as ClinVar, gnomAD etc."))),(0,i.kt)("h3",{id:"classifying-small-and-structural-variants"},"Classifying Small and Structural Variants"),(0,i.kt)("table",null,(0,i.kt)("thead",{parentName:"table"},(0,i.kt)("tr",{parentName:"thead"},(0,i.kt)("th",{parentName:"tr",align:null},"Variant Classification"),(0,i.kt)("th",{parentName:"tr",align:null},"Criteria"),(0,i.kt)("th",{parentName:"tr",align:null},"Example"))),(0,i.kt)("tbody",{parentName:"table"},(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Small Variant"),(0,i.kt)("td",{parentName:"tr",align:null},"Single nucleotide changes"),(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"G -> A")," SNP or ",(0,i.kt)("inlineCode",{parentName:"td"},"T -> TAA")," insertion")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Small Variant"),(0,i.kt)("td",{parentName:"tr",align:null},"Short insertions/deletions"),(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"T -> TAA")," insertion")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Structural Variant"),(0,i.kt)("td",{parentName:"tr",align:null},"Explicit alleles with ",(0,i.kt)("inlineCode",{parentName:"td"},"SVTYPE"),","),(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"ACCCCT....")," with ",(0,i.kt)("inlineCode",{parentName:"td"},"SVLEN=200")," or ",(0,i.kt)("inlineCode",{parentName:"td"},"SVTYPE=DEL"))),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Structural Variant"),(0,i.kt)("td",{parentName:"tr",align:null},"Symbolic notation (",(0,i.kt)("inlineCode",{parentName:"td"},"<DEL>"),", ",(0,i.kt)("inlineCode",{parentName:"td"},"<DUP>"),", etc.),"),(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"<DEL>"))),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Structural Variant"),(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"IMPRECISE")," in ",(0,i.kt)("inlineCode",{parentName:"td"},"INFO")," field"),(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"IMPRECISE")," in ",(0,i.kt)("inlineCode",{parentName:"td"},"INFO")," field")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Structural Variant"),(0,i.kt)("td",{parentName:"tr",align:null},"Breakend notation (",(0,i.kt)("inlineCode",{parentName:"td"},"[ ]"),")"),(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"N[chr10:1234567[")," translocation")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Structural Variant"),(0,i.kt)("td",{parentName:"tr",align:null},"Expansion of short tandem repeats (STRs)"),(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"<STR>")," with ",(0,i.kt)("inlineCode",{parentName:"td"},"SVTYPE=STR"))),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Structural Variant"),(0,i.kt)("td",{parentName:"tr",align:null},"Copy number variation (CNV) or loss of heterozygosity (LOH)"),(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"SVTYPE=CNV")," or ",(0,i.kt)("inlineCode",{parentName:"td"},"SVTYPE=LOH"))))),(0,i.kt)("h2",{id:"examples"},"Examples"),(0,i.kt)("h3",{id:"small-variants"},"Small Variants"),(0,i.kt)("h4",{id:"snp"},"SNP"),(0,i.kt)("h5",{id:"input-data"},"Input Data:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-vcf"},"chr1    1234567    .    G    A    .    PASS    .\n")),(0,i.kt)("h5",{id:"classification"},"Classification:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"This is a single nucleotide polymorphism (SNP)"),(0,i.kt)("li",{parentName:"ul"},"It does not contain breakend notation or symbolic alleles"),(0,i.kt)("li",{parentName:"ul"},"It does not contain ",(0,i.kt)("inlineCode",{parentName:"li"},"SVTYPE")," or ",(0,i.kt)("inlineCode",{parentName:"li"},"SVLEN")," in the ",(0,i.kt)("inlineCode",{parentName:"li"},"INFO")," field")),(0,i.kt)("h4",{id:"short-indel"},"Short Indel"),(0,i.kt)("h5",{id:"input-data-1"},"Input Data:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-vcf"},"chr2    8901234    .    T    TAA    .    PASS    .\n")),(0,i.kt)("h5",{id:"classification-1"},"Classification:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"This is an insertion of 2 base pairs"),(0,i.kt)("li",{parentName:"ul"},"The length of change is small, so it falls under small variant"),(0,i.kt)("li",{parentName:"ul"},"It does not contain ",(0,i.kt)("inlineCode",{parentName:"li"},"SVTYPE")," or ",(0,i.kt)("inlineCode",{parentName:"li"},"SVLEN")," in the ",(0,i.kt)("inlineCode",{parentName:"li"},"INFO")," field")),(0,i.kt)("h4",{id:"multiallelic-snp"},"Multiallelic SNP"),(0,i.kt)("h5",{id:"input-data-2"},"Input Data:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-vcf"},"chr7    2345678    .    G    A,C    .    PASS    .\n")),(0,i.kt)("h5",{id:"classification-2"},"Classification:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"This represents a multiallelic SNP, where two different alternative alleles exist (",(0,i.kt)("inlineCode",{parentName:"li"},"A")," and ",(0,i.kt)("inlineCode",{parentName:"li"},"C"),")"),(0,i.kt)("li",{parentName:"ul"},"Despite multiple alleles, it does not involve large rearrangements"),(0,i.kt)("li",{parentName:"ul"},"It does not contain ",(0,i.kt)("inlineCode",{parentName:"li"},"SVTYPE")," or ",(0,i.kt)("inlineCode",{parentName:"li"},"SVLEN")," in the ",(0,i.kt)("inlineCode",{parentName:"li"},"INFO")," field")),(0,i.kt)("h3",{id:"structural-variants"},"Structural Variants"),(0,i.kt)("h4",{id:"deletion"},"Deletion"),(0,i.kt)("h5",{id:"input-data-3"},"Input Data:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-vcf"},"chr3    3456789    .    N    <DEL>    .    PASS    SVTYPE=DEL;END=3457890\n")),(0,i.kt)("h5",{id:"classification-3"},"Classification:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"The alternate allele (",(0,i.kt)("inlineCode",{parentName:"li"},"<DEL>"),") is symbolic"),(0,i.kt)("li",{parentName:"ul"},"The ",(0,i.kt)("inlineCode",{parentName:"li"},"SVTYPE=DEL")," confirms it is a structural variant")),(0,i.kt)("h4",{id:"deletion-with-explicit-alleles"},"Deletion with explicit alleles"),(0,i.kt)("h5",{id:"input-data-4"},"Input Data:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-vcf"},"chr1    934064    .    AGGGGAGGCGGCTGCGTTACAGGTGGGCAGGGGAGGCGGCTCCGTTACAGGTGGGCAGGGGAGGCGGCTGCGTTACA    A    .    .    END=934904;SVTYPE=DEL\n")),(0,i.kt)("h5",{id:"classification-4"},"Classification:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"The alternate and reference allels are precise but  ",(0,i.kt)("inlineCode",{parentName:"li"},"SVTYPE=DEL")," indicates it is a structural variant")),(0,i.kt)("h4",{id:"run-of-homozygosity"},"Run Of Homozygosity"),(0,i.kt)("h5",{id:"input-data-5"},"Input Data:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-vcf"},"1   4000000    .    N    <ROH>    .    ROHLC    SVTYPE=ROH;END=4001000    GT    .    .    1\n")),(0,i.kt)("h5",{id:"classification-5"},"Classification:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"The alternate allele (",(0,i.kt)("inlineCode",{parentName:"li"},"<ROH>"),") is symbolic"),(0,i.kt)("li",{parentName:"ul"},"The ",(0,i.kt)("inlineCode",{parentName:"li"},"SVTYPE=ROH")," confirms it is a structural variant")),(0,i.kt)("h4",{id:"repeat-expansion"},"Repeat Expansion"),(0,i.kt)("h5",{id:"input-data-6"},"Input Data:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-vcf"},"chrX    9876543    .    CAG    <STR>    .    PASS    SVTYPE=STR;REPEAT_UNIT=CAG\n")),(0,i.kt)("h5",{id:"classification-6"},"Classification:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"The alternate allele ",(0,i.kt)("inlineCode",{parentName:"li"},"<STR>")," is symbolic, indicating a repeat expansion"),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("inlineCode",{parentName:"li"},"SVTYPE=STR")," confirms it is a structural variant related to repeat instability")),(0,i.kt)("h4",{id:"translocation"},"Translocation"),(0,i.kt)("h5",{id:"input-data-7"},"Input Data:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-vcf"},"chr7    110541589    .    N    <DEL>    27    cnvLength    SVTYPE=CNV;END=110548681;REFLEN=7092\n")),(0,i.kt)("h5",{id:"classification-7"},"Classification:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"The alternate allele ",(0,i.kt)("inlineCode",{parentName:"li"},"<DEL>")," is symbolic"),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("inlineCode",{parentName:"li"},"SVTYPE=CNV")," confirms it is a structural variant (copy number loss)")),(0,i.kt)("h4",{id:"translocation-1"},"Translocation"),(0,i.kt)("h5",{id:"input-data-8"},"Input Data:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-vcf"},"chr4    5678901    .    N    N[chr10:1234567[    .    PASS    SVTYPE=BND\n")),(0,i.kt)("h5",{id:"classification-8"},"Classification:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"The alternate allele contains ",(0,i.kt)("inlineCode",{parentName:"li"},"[")," and ",(0,i.kt)("inlineCode",{parentName:"li"},"]"),", signifying a breakend"),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("inlineCode",{parentName:"li"},"SVTYPE=BND")," confirms this is a structural variant (translocation)")))}u.isMDXComponent=!0}}]);