"use strict";(self.webpackChunknirvana_documentation=self.webpackChunknirvana_documentation||[]).push([[7910],{3905:(t,e,n)=>{n.d(e,{Zo:()=>p,kt:()=>g});var a=n(7294);function r(t,e,n){return e in t?Object.defineProperty(t,e,{value:n,enumerable:!0,configurable:!0,writable:!0}):t[e]=n,t}function l(t,e){var n=Object.keys(t);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(t);e&&(a=a.filter((function(e){return Object.getOwnPropertyDescriptor(t,e).enumerable}))),n.push.apply(n,a)}return n}function i(t){for(var e=1;e<arguments.length;e++){var n=null!=arguments[e]?arguments[e]:{};e%2?l(Object(n),!0).forEach((function(e){r(t,e,n[e])})):Object.getOwnPropertyDescriptors?Object.defineProperties(t,Object.getOwnPropertyDescriptors(n)):l(Object(n)).forEach((function(e){Object.defineProperty(t,e,Object.getOwnPropertyDescriptor(n,e))}))}return t}function o(t,e){if(null==t)return{};var n,a,r=function(t,e){if(null==t)return{};var n,a,r={},l=Object.keys(t);for(a=0;a<l.length;a++)n=l[a],e.indexOf(n)>=0||(r[n]=t[n]);return r}(t,e);if(Object.getOwnPropertySymbols){var l=Object.getOwnPropertySymbols(t);for(a=0;a<l.length;a++)n=l[a],e.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(t,n)&&(r[n]=t[n])}return r}var s=a.createContext({}),d=function(t){var e=a.useContext(s),n=e;return t&&(n="function"==typeof t?t(e):i(i({},e),t)),n},p=function(t){var e=d(t.components);return a.createElement(s.Provider,{value:e},t.children)},m="mdxType",c={inlineCode:"code",wrapper:function(t){var e=t.children;return a.createElement(a.Fragment,{},e)}},u=a.forwardRef((function(t,e){var n=t.components,r=t.mdxType,l=t.originalType,s=t.parentName,p=o(t,["components","mdxType","originalType","parentName"]),m=d(n),u=r,g=m["".concat(s,".").concat(u)]||m[u]||c[u]||l;return n?a.createElement(g,i(i({ref:e},p),{},{components:n})):a.createElement(g,i({ref:e},p))}));function g(t,e){var n=arguments,r=e&&e.mdxType;if("string"==typeof t||r){var l=n.length,i=new Array(l);i[0]=u;var o={};for(var s in e)hasOwnProperty.call(e,s)&&(o[s]=e[s]);o.originalType=t,o[m]="string"==typeof t?t:r,i[1]=o;for(var d=2;d<l;d++)i[d]=n[d];return a.createElement.apply(null,i)}return a.createElement.apply(null,n)}u.displayName="MDXCreateElement"},5255:(t,e,n)=>{n.r(e),n.d(e,{contentTitle:()=>i,default:()=>m,frontMatter:()=>l,metadata:()=>o,toc:()=>s});var a=n(7462),r=(n(7294),n(3905));const l={id:"introduction",title:"Introduction",description:"translational research-grade variant annotation",hide_title:!0,slug:"/"},i=void 0,o={unversionedId:"introduction/introduction",id:"version-3.25/introduction/introduction",title:"Introduction",description:"translational research-grade variant annotation",source:"@site/versioned_docs/version-3.25/introduction/introduction.mdx",sourceDirName:"introduction",slug:"/",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/",editUrl:"https://github.com/Illumina/IlluminaConnectedAnnotationsDocumentation/edit/master/versioned_docs/version-3.25/introduction/introduction.mdx",tags:[],version:"3.25",frontMatter:{id:"introduction",title:"Introduction",description:"translational research-grade variant annotation",hide_title:!0,slug:"/"},sidebar:"docs",next:{title:"Licensed Content",permalink:"/IlluminaConnectedAnnotationsDocumentation/3.25/introduction/licensedContent"}},s=[{value:"What does Illumina Connected Annotations annotate?",id:"what-does-illumina-connected-annotations-annotate",children:[{value:"Reference genome",id:"reference-genome",children:[],level:3},{value:"Transcript and Gene Models",id:"transcript-and-gene-models",children:[],level:3},{value:"Supplementary Annotation",id:"supplementary-annotation",children:[],level:3},{value:"Download manager",id:"download-manager",children:[],level:3}],level:2},{value:"Download",id:"download",children:[],level:2}],d={toc:s},p="wrapper";function m(t){let{components:e,...l}=t;return(0,r.kt)(p,(0,a.Z)({},d,l,{components:e,mdxType:"MDXLayout"}),(0,r.kt)("p",null,(0,r.kt)("img",{src:n(9073).Z})),(0,r.kt)("p",null,"Illumina Connected Annotations provides translational research-grade annotation of genomic variants (SNVs, MNVs, insertions, deletions, indels, STRs, gene fusions, and SVs (including CNVs). It can be run as a stand-alone package, or integrated into larger software tools that require variant annotation."),(0,r.kt)("p",null,"The input to Illumina Connected Annotations are VCFs and the output is a structured JSON representation of all annotation and sample information (as extracted from the VCF). Illumina Connected Annotations handles multiple alternate alleles and multiple samples with ease."),(0,r.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,r.kt)("div",{parentName:"div",className:"admonition-heading"},(0,r.kt)("h5",{parentName:"div"},(0,r.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,r.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,r.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"tip")),(0,r.kt)("div",{parentName:"div",className:"admonition-content"},(0,r.kt)("p",{parentName:"div"},"We have also added supporting VCF output with limited annotation sources. See ",(0,r.kt)("a",{parentName:"p",href:"/IlluminaConnectedAnnotationsDocumentation/3.25/file-formats/illumina-annotator-vcf-file-format"},"Illumina Connected Annotations VCF File Format")))),(0,r.kt)("p",null,"The software is being developed under a rigorous SDLC and testing process to ensure accuracy of the results and enable embedding in other software. Illumina Connected Annotations uses a continuous integration pipeline where millions of variant annotations are monitored against baseline values daily."),(0,r.kt)("h2",{id:"what-does-illumina-connected-annotations-annotate"},"What does Illumina Connected Annotations annotate?"),(0,r.kt)("p",null,"We use Sequence Ontology consequences to describe how each variant impacts a given transcript:"),(0,r.kt)("p",null,(0,r.kt)("img",{src:n(5845).Z})),(0,r.kt)("h3",{id:"reference-genome"},"Reference genome"),(0,r.kt)("p",null,"For GRCh37, our latest reference genome is using version GRCh37.p13. The list of chromosome and contigs in this version can be checked from this ",(0,r.kt)("a",{parentName:"p",href:"https://ftp.ncbi.nlm.nih.gov/genomes/all/annotation_releases/9606/105.20220307/GCF_000001405.25_GRCh37.p13/GCF_000001405.25_GRCh37.p13_assembly_report.txt"},"file"),"\nFor GRCh38, our latest reference genome is using version GRCh38.p14. The list of chromosome and contigs in this version can be checked from this ","[file]","(",(0,r.kt)("a",{parentName:"p",href:"https://ftp.ncbi.nlm.nih."},"https://ftp.ncbi.nlm.nih."),"\ngov/genomes/all/annotation_releases/9606/GCF_000001405.40-RS_2023_10/GCF_000001405.40_GRCh38.p14_assembly_report.txt)"),(0,r.kt)("div",{className:"admonition admonition-caution alert alert--warning"},(0,r.kt)("div",{parentName:"div",className:"admonition-heading"},(0,r.kt)("h5",{parentName:"div"},(0,r.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,r.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"16",height:"16",viewBox:"0 0 16 16"},(0,r.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"}))),"caution")),(0,r.kt)("div",{parentName:"div",className:"admonition-content"},(0,r.kt)("p",{parentName:"div"},"There was a bug in our previous genome reference. Previously, we generate genome reference based on GRCh38.p13 version ",(0,r.kt)("a",{parentName:"p",href:"https://ftp.ncbi.nlm.nih.gov/genomes/all/annotation_releases/9606/109.20190607/GCF_000001405.39_GRCh38.p13/GCF_000001405.39_GRCh38.p13_assembly_report.txt"},"109.20190607")," for chromosomes and contigs name.\nDespite that, we use GRCh38.p12 version ",(0,r.kt)("a",{parentName:"p",href:"https://ftp.ncbi.nlm.nih.gov/genomes/all/annotation_releases/9606/109/GCF_000001405.38_GRCh38.p12/"},"109")," for the FASTA sequence.\nThis resulted in mismatch between some alt contigs name and its FASTA sequence. This mismatch can cause some minor issue on the variant ID and HGVS g. notation if the variant is on the alt contigs that were affected.\nThis issue is not present for variant that occurs on main chromosome (chromosome 1-22, X, and Y)."),(0,r.kt)("p",{parentName:"div"},"With release GRCh38.p14, we have fixed this issue and the GRCh38.p14 release will also be backward compatible with previous release. Old transcript and gene model files and all of\nsupplementary annotation data will not have any issue with the new genome reference version.\nIt is advisable to update to the current genome reference version."))),(0,r.kt)("h3",{id:"transcript-and-gene-models"},"Transcript and Gene Models"),(0,r.kt)("p",null,"The transcript and gene models are obtained from ",(0,r.kt)("a",{parentName:"p",href:"https://ftp.ncbi.nlm.nih.gov/genomes/all/annotation_releases/9606/"},"RefSeq")," and ",(0,r.kt)("a",{parentName:"p",href:"https://ftp.ensembl.org/pub/"},"Ensembl"),".\nThe current officially supported versions for GRCh38 are:"),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Data Source"),(0,r.kt)("th",{parentName:"tr",align:null},"Version"),(0,r.kt)("th",{parentName:"tr",align:null},"Release Date"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"RefSeq"),(0,r.kt)("td",{parentName:"tr",align:null},"GCF_000001405.40-RS_2023_10"),(0,r.kt)("td",{parentName:"tr",align:null},"2023-10-07")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"RefSeq"),(0,r.kt)("td",{parentName:"tr",align:null},"GCF_000001405.40-RS_2023_03"),(0,r.kt)("td",{parentName:"tr",align:null},"2023-03-21")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Ensembl"),(0,r.kt)("td",{parentName:"tr",align:null},"112"),(0,r.kt)("td",{parentName:"tr",align:null},"2024-05-14")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Ensembl"),(0,r.kt)("td",{parentName:"tr",align:null},"110"),(0,r.kt)("td",{parentName:"tr",align:null},"2023-04-27")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Ensembl"),(0,r.kt)("td",{parentName:"tr",align:null},"108"),(0,r.kt)("td",{parentName:"tr",align:null},"2022-10-20")))),(0,r.kt)("p",null,"For GRCh37:"),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Data Source"),(0,r.kt)("th",{parentName:"tr",align:null},"Version"),(0,r.kt)("th",{parentName:"tr",align:null},"Release Date"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"RefSeq"),(0,r.kt)("td",{parentName:"tr",align:null},"105.20220307"),(0,r.kt)("td",{parentName:"tr",align:null},"2022-03-10")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Ensembl"),(0,r.kt)("td",{parentName:"tr",align:null},"110"),(0,r.kt)("td",{parentName:"tr",align:null},"2023-02-08")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Ensembl"),(0,r.kt)("td",{parentName:"tr",align:null},"108"),(0,r.kt)("td",{parentName:"tr",align:null},"2022-10-20")))),(0,r.kt)("div",{className:"admonition admonition-note alert alert--secondary"},(0,r.kt)("div",{parentName:"div",className:"admonition-heading"},(0,r.kt)("h5",{parentName:"div"},(0,r.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,r.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,r.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"}))),"note")),(0,r.kt)("div",{parentName:"div",className:"admonition-content"},(0,r.kt)("p",{parentName:"div"},"For GRCh37 Ensembl release, despite the latest version being 110 (with existing newer release versions 111 and 112), the annotation data is effectively the same as version 87 which was released back in 2017."),(0,r.kt)("p",{parentName:"div"},"For GRCh37 Refseq release, NCBI has stopped releasing new gene annotation and version 105.20220307 is the last release version for GRCh37."))),(0,r.kt)("p",null,"For gene symbols that we support for the transcript model above, the latest data is from ",(0,r.kt)("a",{parentName:"p",href:"https://ftp.ebi.ac.uk/pub/databases/genenames/hgnc/json/hgnc_complete_set.json"},"HGNC"),"\nwebsite as of 2024-06-03."),(0,r.kt)("p",null,"More supported versions can be listed and downloaded using our ",(0,r.kt)("a",{parentName:"p",href:"/IlluminaConnectedAnnotationsDocumentation/3.25/utilities/data-manager"},"DataManager Manager"),"."),(0,r.kt)("h3",{id:"supplementary-annotation"},"Supplementary Annotation"),(0,r.kt)("p",null,"In addition, it uses external data sources to provide additional context for each variant.\nIllumina Connected Annotations provides annotations from the following sources divided into 2 tiers: Professional and basic.\nThe basic tier can be accessed free of charge. The professional tier requires a license. Please see ",(0,r.kt)("a",{parentName:"p",href:"./introduction/licensedContent"},"Licensed Content")," for details. For access, please contact ",(0,r.kt)("a",{parentName:"p",href:"mailto:annotation_support@illumina.com"},"annotation_support@illumina.com"),"."),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Data Source"),(0,r.kt)("th",{parentName:"tr",align:null},"Availability"),(0,r.kt)("th",{parentName:"tr",align:null},"Latest Supported Version"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"COSMIC"),(0,r.kt)("td",{parentName:"tr",align:null},"Professional"),(0,r.kt)("td",{parentName:"tr",align:null},"99")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"OMIM"),(0,r.kt)("td",{parentName:"tr",align:null},"Professional"),(0,r.kt)("td",{parentName:"tr",align:null},"20240807")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Primate AI-3D"),(0,r.kt)("td",{parentName:"tr",align:null},"Professional"),(0,r.kt)("td",{parentName:"tr",align:null},"1.0")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Promoter AI"),(0,r.kt)("td",{parentName:"tr",align:null},"Professional"),(0,r.kt)("td",{parentName:"tr",align:null},"1.0")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Splice AI"),(0,r.kt)("td",{parentName:"tr",align:null},"Professional"),(0,r.kt)("td",{parentName:"tr",align:null},"1.3")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"1000 Genomes Project"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"Phase 3 v3plus")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Cancer Hotspots"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"2017")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"ClinGen"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"20240807")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"ClinVar"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"20240730")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"DANN"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"20200205")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"dbSNP"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"156")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"DECIPHER"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"201509")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"FusionCatcher"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"1.33")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"GERP"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"20110522")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"GME Variome"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"20160618")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"gnomAD (GRCh37)"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"2.1")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"gnomAD (GRCh38)"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"4.1")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"MITOMAP"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"20200819")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"MultiZ 100 way"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"20171006")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"REVEL"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"20200205")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"TOPMed"),(0,r.kt)("td",{parentName:"tr",align:null},"Basic"),(0,r.kt)("td",{parentName:"tr",align:null},"freeze 5")))),(0,r.kt)("h3",{id:"download-manager"},"Download manager"),(0,r.kt)("p",null,"To effectively download all of data, we have provided download manager. Please go to ",(0,r.kt)("a",{parentName:"p",href:"./utilities/data-manager"},"DataManager")," page to read more."),(0,r.kt)("h2",{id:"download"},"Download"),(0,r.kt)("p",null,"Please visit ",(0,r.kt)("a",{parentName:"p",href:"https://developer.illumina.com/illumina-connected-annotations"},"Illumina Connected Annotations"),"."))}m.isMDXComponent=!0},9073:(t,e,n)=>{n.d(e,{Z:()=>a});const a=n.p+"assets/images/ICAnnotations-966475fab8adae0519d1667d592ad4b2.png"},5845:(t,e,n)=>{n.d(e,{Z:()=>a});const a=n.p+"assets/images/TranscriptConsequences-60ca1c43a36dacf896fecdabf09ce02c.svg"}}]);