---
title: gnomAD
---

import JSONV from './gnomad-small-variants-json.md'
import JSONV40 from './gnomad4.0-small-variants-json.md'
import JSONG from './gnomad-lof-json.md'
import JSONG40 from './gnomad4.0-lof-json.md'
import JSONSV from './gnomad-structural-variants-json.md'
import JSONSV40 from './gnomad40-structural-variants-json.md'
import SVDATADESCRIPTION from './gnomad-structural-variants-data_description.md'


## Overview

The Genome Aggregation Database ([gnomAD](https://gnomad.broadinstitute.org/)) is a resource developed by an international coalition of investigators, with the goal of aggregating and harmonizing both exome and genome sequencing data from a wide variety of large-scale sequencing projects, and making summary data available for the wider scientific community.

:::info Publication
Koch, L., 2020. Exploring human genomic diversity with gnomAD. *Nature Reviews Genetics*, **21(8)**, pp.448-448.
:::

Illumina Connected Analysis will support gnomAD v4.0 for GRCh38 assembly and gnomAD v2.1 for GRCh37.


## gnomAD v4.0 (GRCh38)

### Small Variants
In gnomAD v4.0, like gnomAD v2.1, there are genome and exome data. Compare to gnomAD v2.1 which the data for genome and exome are merged, for gnomAD 4.0, Illumina Connected Annotation will separate them with different JSON output field.
For gnomAD genome, the field name would be `gnomad`. For gnomAD exome, the field name would be `gnomad-exome`.
Despite this difference in the field name, the JSON data format would be identical for both genome and exome.

#### VCF extraction
We currently extract the following info fields from both gnomAD genome and exome VCF files:
```scss
##INFO=<ID=AC,Number=A,Type=Integer,Description="Alternate allele count">
##INFO=<ID=AN,Number=1,Type=Integer,Description="Total number of alleles">
##INFO=<ID=AC_XX,Number=A,Type=Integer,Description="Alternate allele count for XX samples">
##INFO=<ID=AN_XX,Number=1,Type=Integer,Description="Total number of alleles in XX samples">
##INFO=<ID=nhomalt_XX,Number=A,Type=Integer,Description="Count of homozygous individuals in XX samples">
##INFO=<ID=AC_XY,Number=A,Type=Integer,Description="Alternate allele count for XY samples">
##INFO=<ID=AN_XY,Number=1,Type=Integer,Description="Total number of alleles in XY samples">
##INFO=<ID=nhomalt_XY,Number=A,Type=Integer,Description="Count of homozygous individuals in XY samples">
##INFO=<ID=nhomalt,Number=A,Type=Integer,Description="Count of homozygous individuals">
##INFO=<ID=AC_afr,Number=A,Type=Integer,Description="Alternate allele count for samples of African/African-American ancestry">
##INFO=<ID=AN_afr,Number=1,Type=Integer,Description="Total number of alleles in samples of African/African-American ancestry">
##INFO=<ID=nhomalt_afr,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of African/African-American ancestry">
##INFO=<ID=AC_amr,Number=A,Type=Integer,Description="Alternate allele count for samples of Latino ancestry">
##INFO=<ID=AN_amr,Number=1,Type=Integer,Description="Total number of alleles in samples of Latino ancestry">
##INFO=<ID=nhomalt_amr,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Latino ancestry">
##INFO=<ID=AC_asj,Number=A,Type=Integer,Description="Alternate allele count for samples of Ashkenazi Jewish ancestry">
##INFO=<ID=AN_asj,Number=1,Type=Integer,Description="Total number of alleles in samples of Ashkenazi Jewish ancestry">
##INFO=<ID=nhomalt_asj,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Ashkenazi Jewish ancestry">
##INFO=<ID=AC_eas,Number=A,Type=Integer,Description="Alternate allele count for samples of East Asian ancestry">
##INFO=<ID=AN_eas,Number=1,Type=Integer,Description="Total number of alleles in samples of East Asian ancestry">
##INFO=<ID=nhomalt_eas,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of East Asian ancestry">
##INFO=<ID=AC_fin,Number=A,Type=Integer,Description="Alternate allele count for samples of Finnish ancestry">
##INFO=<ID=AN_fin,Number=1,Type=Integer,Description="Total number of alleles in samples of Finnish ancestry">
##INFO=<ID=nhomalt_fin,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Finnish ancestry">
##INFO=<ID=AC_mid,Number=A,Type=Integer,Description="Alternate allele count for samples of Middle Eastern ancestry">
##INFO=<ID=AN_mid,Number=1,Type=Integer,Description="Total number of alleles in samples of Middle Eastern ancestry">
##INFO=<ID=nhomalt_mid,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Middle Eastern ancestry">
##INFO=<ID=AC_nfe,Number=A,Type=Integer,Description="Alternate allele count for samples of Non-Finnish European ancestry">
##INFO=<ID=AN_nfe,Number=1,Type=Integer,Description="Total number of alleles in samples of Non-Finnish European ancestry">
##INFO=<ID=nhomalt_nfe,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Non-Finnish European ancestry">
##INFO=<ID=AC_remaining,Number=A,Type=Integer,Description="Alternate allele count for samples of Remaining individuals ancestry">
##INFO=<ID=AN_remaining,Number=1,Type=Integer,Description="Total number of alleles in samples of Remaining individuals ancestry">
##INFO=<ID=nhomalt_remaining,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Remaining individuals ancestry">
##INFO=<ID=AC_sas,Number=A,Type=Integer,Description="Alternate allele count for samples of South Asian ancestry">
##INFO=<ID=AN_sas,Number=1,Type=Integer,Description="Total number of alleles in samples of South Asian ancestry">
```

#### JSON output

<JSONV40/>

#### Calculation
To calculate allele frequency for each group, we divide the allele count with allele number for each group.

### LoF Gene Metrics
In gnomAD 4.0, the gene score data for LOF is given per transcript. Since this is gene level data, we have to select one of the value.
We will show the transcript ID that we select in the JSON output.
The rule to select transcript ID is based on priority. Below is the priority list:
1. Transcript ID is Ensembl transcript ID and has mane_select column true
2. Transcript ID is Ensembl transcript ID and it is marked as canonical in Illumina Connected Annotation cache data
3. Transcript ID is RefSeq transcript ID and has mane_select column true
4. Transcript ID is RefSeq transcript ID and it is marked as canonical in Illumina Connected Annotation cache data
5. Transcript ID has the lowest lof.oe_ci.upper value compare to other transcript for the same gene

:::info Differences with gnomAD browser
Due to difference in Ensembl version between Illumina Connected Annotation and gnomAD, there are several transcript ID that are marked as canonical in gnomAD browser but not in Illumina Connected Analysis.
If this is thh case, the gene score that is shown in Illumina Connected Annotation will be different compare to gene score that is shown in the gnomAD browser.
The transcriptId field in the JSON output will tell you which transcript is selected by Illumina Connected Annotation.
:::

#### Tab delimited file example

```scss
gene	transcript	mane_select	lof_hc_lc.obs	lof_hc_lc.exp	lof_hc_lc.possible	lof_hc_lc.oe	lof_hc_lc.mu	lof_hc_lc.pLI	lof_hc_lc.pNull	lof_hc_lc.pRec	lof.obs	lof.exp	lof.possible	lof.oe	lof.mu	lof.pLI	lof.pNull	lof.pRec	lof.oe_ci.lower	lof.oe_ci.upper	lof.z_raw	lof.z_score	mis.obs	mis.exp	mis.possible	mis.oe	mis.mu	mis.oe_ci.lower	mis.oe_ci.upper	mis.z_raw	mis.z_score	mis_pphen.obs	mis_pphen.exp	mis_pphen.possible	mis_pphen.oe	syn.obs	syn.exp	syn.possible	syn.oe	syn.mu	syn.oe_ci.lower	syn.oe_ci.upper	syn.z_raw	syn.z_score	constraint_flags
SCHIP1	ENST00000445224	false	8	3.0392e+01	157	2.6323e-01	3.5111e-07	9.9024e-01	5.8227e-06	9.7579e-03	8	3.0392e+01	157	2.6323e-01	3.5111e-07	9.9066e-01	5.3097e-06	9.3341e-03	1.5300e-01	4.7500e-01	4.0617e+00	3.4377e+00	193	3.0914e+02	1659	6.2431e-01	1.5780e-06	5.5400e-01	7.0300e-01	6.6055e+00	2.4115e+00	87	1.4959e+02	813	5.8160e-01	76	1.0011e+02	393	7.5914e-01	7.9269e-07	6.3000e-01	9.1900e-01	2.4099e+00	1.3142e+00	[]
```

#### JSON key to TSV column mapping
| JSON key      | TSV column      | Description                                                                                                                  |
|---------------|-----------------|------------------------------------------------------------------------------------------------------------------------------|
| pLi           | lof.pLI         | probability of being intolerant of a single loss-of-function variant (like haploinsufficient genes, observed ~ 0.1*expected) |
| pNull         | lof.pNull       | probability of being completely tolerant of loss of function variation (observed = expected)                                 |
| pRec          | lof.pRec        | probability of being intolerant of two loss of function variants (like recessive genes, observed ~ 0.5*expected)             |
| synZ          | syn.z_score     | corrected synonymous Z score                                                                                                 |
| misZ          | mis.z_score     | corrected missense Z score                                                                                                   |
| loeuf         | lof.oe_ci.upper | loss of function observed/expected upper bound fraction (LOEUF)                                                              |
| transcriptId  | transcript      | transcript ID which the values we select                                                                                     |

<JSONG40/>

### Structural Variants

Structural variants in gnomAD 4.0 is available in VCF format and has the same population data as small variants.

#### Structural Variant Type Mapping
The source files represented the structural variants with keys using various naming conventions.
In the Illumina Connected Annotations JSON output, these keys will be mapped according to the following.

| Illumina Connected Annotations JSON SV Type Key | GRCh37 Source SV Type Key |
|-------------------------------------------------|---------------------------|
| deletion                                        | DEL, CN=0                 |
| duplication                                     | DUP                       |
| insertion                                       | INS                       |
| inversion                                       | INV                       |
| mobile_element_insertion                        | INS:ME                    |
| mobile_element_insertion                        | INS:ME:ALU                |
| mobile_element_insertion                        | INS:ME:LINE1              |
| mobile_element_insertion                        | INS:ME:SVA                |
| complex_structural_alteration                   | CPX                       |

:::info gnomAD Copy Number Variation
In gnomAD 4.0 structural variants data, there are CNV data in the VCF file. Since it is not shown in the browser, we don't include CNV in our output.
We will evaluate in the future whether to include copy number variation from structural variation data together with new rare CNV data taht is available in gnomAD 4.0.
:::

:::info gnomAD duplication variant type
In gnomAD 4.0 structural variants VCF, only DUP is shown as symbolic allele for duplication variant type.
Based on the information in gnomAD browser, duplication variant that has split read or paired end reads evidence can be inferred as tandem duplication.
With this, we check the evidence data in each DUP variants entry to decide whether it can be assign tandem duplication as variant type or it is just duplication.
:::

#### JSON output

<JSONSV40/>

## gnomAD v2.1 (GRCh37)

### Small Variants

#### VCF extraction

We currently extract the following info fields from gnomAD genome and exome VCF files:

```scss
##INFO=<ID=AC,Number=A,Type=Integer,Description="Alternate allele count for samples">
##INFO=<ID=AN,Number=A,Type=Integer,Description="Total number of alleles in samples">
##INFO=<ID=nhomalt,Number=A,Type=Integer,Description="Count of homozygous individuals in samples">
##INFO=<ID=DP,Number=1,Type=Integer,Description="Depth of informative coverage for each sample; reads with MQ=255 or with bad mates are filtered">
##INFO=<ID=lcr,Number=0,Type=Flag,Description="Variant falls within a low complexity region">
##INFO=<ID=AC_afr,Number=A,Type=Integer,Description="Alternate allele count for samples of African-American ancestry">
##INFO=<ID=AN_afr,Number=A,Type=Integer,Description="Total number of alleles in samples of African-American ancestry">
##INFO=<ID=AF_afr,Number=A,Type=Float,Description="Alternate allele frequency in samples of African-American ancestry">
##INFO=<ID=nhomalt_afr,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of African-American ancestry">
##INFO=<ID=AC_amr,Number=A,Type=Integer,Description="Alternate allele count for samples of Latino ancestry">
##INFO=<ID=AN_amr,Number=A,Type=Integer,Description="Total number of alleles in samples of Latino ancestry">
##INFO=<ID=nhomalt_amr,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Latino ancestry">
##INFO=<ID=AC_eas,Number=A,Type=Integer,Description="Alternate allele count for samples of East Asian ancestry">
##INFO=<ID=AN_eas,Number=A,Type=Integer,Description="Total number of alleles in samples of East Asian ancestry">
##INFO=<ID=nhomalt_eas,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of East Asian ancestry">
##INFO=<ID=AC_female,Number=A,Type=Integer,Description="Alternate allele count for female samples">
##INFO=<ID=AN_female,Number=A,Type=Integer,Description="Total number of alleles in female samples">
##INFO=<ID=nhomalt_female,Number=A,Type=Integer,Description="Count of homozygous individuals in female samples">
##INFO=<ID=AC_nfe,Number=A,Type=Integer,Description="Alternate allele count for samples of non-Finnish European ancestry">
##INFO=<ID=AN_nfe,Number=A,Type=Integer,Description="Total number of alleles in samples of non-Finnish European ancestry">
##INFO=<ID=nhomalt_nfe,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of non-Finnish European ancestry">
##INFO=<ID=AC_fin,Number=A,Type=Integer,Description="Alternate allele count for samples of Finnish ancestry">
##INFO=<ID=AN_fin,Number=A,Type=Integer,Description="Total number of alleles in samples of Finnish ancestry">
##INFO=<ID=nhomalt_fin,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Finnish ancestry">
##INFO=<ID=AC_asj,Number=A,Type=Integer,Description="Alternate allele count for samples of Ashkenazi Jewish ancestry">
##INFO=<ID=AN_asj,Number=A,Type=Integer,Description="Total number of alleles in samples of Ashkenazi Jewish ancestry">
##INFO=<ID=nhomalt_asj,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of Ashkenazi Jewish ancestry">
##INFO=<ID=AC_oth,Number=A,Type=Integer,Description="Alternate allele count for samples of uncertain ancestry">
##INFO=<ID=AN_oth,Number=A,Type=Integer,Description="Total number of alleles in samples of uncertain ancestry">
##INFO=<ID=nhomalt_oth,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of uncertain ancestry">
##INFO=<ID=AC_male,Number=A,Type=Integer,Description="Alternate allele count for male samples">
##INFO=<ID=AN_male,Number=A,Type=Integer,Description="Total number of alleles in male samples">
##INFO=<ID=nhomalt_male,Number=A,Type=Integer,Description="Count of homozygous individuals in male samples">
##INFO=<ID=controls_AC,Number=A,Type=Integer,Description="Alternate allele count for samples in the controls subset">
##INFO=<ID=controls_AN,Number=A,Type=Integer,Description="Total number of alleles in samples in the controls subset">
```

We also extract the following extra fields from gnomAD exome VCF file:

```scss
##INFO=<ID=AC_sas,Number=A,Type=Integer,Description="Alternate allele count for samples of South Asian ancestry">
##INFO=<ID=AN_sas,Number=A,Type=Integer,Description="Total number of alleles in samples of South Asian ancestry">
##INFO=<ID=nhomalt_sas,Number=A,Type=Integer,Description="Count of homozygous individuals in samples of South Asian ancestry">
```

#### Computation

Using these, we compute the following:
* Coverage
* Allele count, Homozygous count, allele number and allele frequencies for:
* Global population
* African/African Americans
* Admixed Americans
* Ashkenazi Jews
* East Asians
* Finnish
* Non-Finnish Europeans
* South Asian
* Others (population not assigned)
* Male
* Female
* Controls

:::tip Note
* Coverage = DP / AN. Frequencies are computed using AC/AN for each population.
* Please note that currently there is no genome sequencing data of south asian (SAS) population available in gnomAD.
* Allele Count, Homozygous count, allele number and allele frequencies for control groups are also provided for the global population.
:::

#### Merging genomes and exomes

When merging the genomes and exomes, the allele counts and allele numbers will be summed across both of the data sets.

:::info
* For GRCh37, Illumina Connected Annotations currently uses gnomAD version 2.1 which contains both genomes and exomes data. Genomes and exomes data are merged in the output.
:::

#### Filters

The following strategy will be used when there's a conflict in filter status:

|                     |   **Genomes PASS**   | **Genomes Filtered** |
|:-------------------:|:--------------------:|:--------------------:|
|   **Exomes PASS**   |         PASS         | Only use exome data  |
| **Exomes Filtered** | Only use genome data |       Filtered       |

#### VCF download instructions
[https://gnomad.broadinstitute.org/downloads](https://gnomad.broadinstitute.org/downloads)

#### JSON output

<JSONV/>

##### Building the supplementary files
The gnomAD ``.nsa`` for Illumina Connected Annotations can be built using the ``SAUtils`` command's ``gnomad`` subcommand. We will describe building gnomAD version 3.1 here.
##### Source data files
Input VCF files (one per chromosome) and a ``.version`` file are required in a folder to build the ``.nsa`` file. For example, my directory contains:
```scss
chr10.vcf.bgz  chr22.vcf.bgz
chr11.vcf.bgz  chr2.vcf.bgz
chr12.vcf.bgz  chr3.vcf.bgz
chr13.vcf.bgz  chr4.vcf.bgz
chr14.vcf.bgz  chr5.vcf.bgz
chr15.vcf.bgz  chr6.vcf.bgz
chr16.vcf.bgz  chr7.vcf.bgz
chr17.vcf.bgz  chr8.vcf.bgz
chr18.vcf.bgz  chr9.vcf.bgz
chr19.vcf.bgz  chrM.vcf.bgz
chr1.vcf.bgz   chrX.vcf.bgz
chr20.vcf.bgz  chrY.vcf.bgz
chr21.vcf.bgz  gnomad.r3.1.version
```
The version file is a text file with the following content.
```scss
NAME=gnomAD
VERSION=3.1
DATE=2020-10-29
DESCRIPTION=Allele frequencies from Genome Aggregation Database (gnomAD)
```
The help menu for the utility is as follows:
```scss
SAUtils.dll gnomad
---------------------------------------------------------------------------
SAUtils                                             (c) 2021 Illumina, Inc.
Stromberg, Roy, Lajugie, Jiang, Li, and Kang                         3.17.0
---------------------------------------------------------------------------

USAGE: dotnet SAUtils.dll gnomad [options]
Reads provided supplementary data files and populates tsv files

OPTIONS:
      --ref, -r <VALUE>      compressed reference sequence file
      --genome, -g <VALUE>   input directory containing VCF (and .version)
                               files with genomic frequencies
      --exome, -e <VALUE>    input directory containing VCF (and .version)
                               files with exomic frequencies
      --temp, -t <VALUE>     output temp directory for intermediate (per chrom)
                                NSA files
      --out, -o <VALUE>      output directory for NSA file
      --help, -h             displays the help menu
      --version, -v          displays the version
```
Here is a sample execution:
```scss
dotnet SAUtils.dll Gnomad \\
--ref ~/References/7/Homo_sapiens.GRCh38.Nirvana.dat --genome genomes/ \\
--out ~/SupplementaryDatabase/63/GRCh38 --temp ~/ExternalDataSources/gnomAD/3.1/GRCh38/temp
```
### LoF Gene Metrics

#### Tab delimited file example

```scss
gene transcript obs_mis exp_mis oe_mis mu_mis possible_mis obs_mis_pphen exp_mis_pphen oe_mis_pphen possible_mis_pphen obs_syn exp_syn oe_syn mu_syn possible_syn obs_lof mu_lof possible_lof exp_lof pLI pNull pRec oe_lof oe_syn_lower oe_syn_upper oe_mis_lower oe_mis_upper oe_lof_lower oe_lof_upper constraint_flag syn_zmis_z lof_z oe_lof_upper_rank oe_lof_upper_bin oe_lof_upper_bin_6 n_sites classic_caf max_af no_lofs obs_het_lof obs_hom_lof defined p exp_hom_lof classic_caf_afr classic_caf_amr classic_caf_asj classic_caf_eas classic_caf_fin classic_caf_nfe classic_caf_oth classic_caf_sas p_afr p_amr p_asj p_eas p_fin p_nfep_oth p_sas transcript_type gene_id transcript_level cds_length num_coding_exons gene_type gene_length exac_pLI exac_obs_lof exac_exp_lof exac_oe_lof brain_expression chromosome start_positionend_position
MED13 ENST00000397786 871 1.1178e+03 7.7921e-01 5.5598e-05 14195 314 5.2975e+02 5.9273e-01 6708 422 3.8753e+02 1.0890e+00 1.9097e-05 4248 0 4.9203e-06 1257 9.8429e+01 1.0000e+00 8.9436e-40 1.8383e-16 0.0000e+00 1.0050e+00 1.1800e+00 7.3600e-01 8.2400e-01 0.0000e+00 3.0000e-02 -1.3765e+00 2.6232e+00 9.1935e+00 0 0 0 2 1.2058e-05 8.0492e-06 124782 3 0 124785 1.2021e-05 1.8031e-05 0.0000e+00 0.0000e+00 0.0000e+00 0.0000e+00 9.2812e-05 8.8571e-06 0.0000e+00 0.0000e+00 0.0000e+00 0.0000e+00 0.0000e+00 0.0000e+00 9.2760e-05 8.8276e-06 0.0000e+00 0.0000e+00 protein_coding ENSG00000108510 2 6522 30 protein_coding 122678 1.0000e+00 0 6.4393e+01 0.0000e+00 NA 17 60019966 60142643
```

#### JSON key to TSV column mapping
| JSON key | TSV column   | Description                                                                                                                  |
|----------|--------------|------------------------------------------------------------------------------------------------------------------------------|
| pLi      | pLI          | probability of being intolerant of a single loss-of-function variant (like haploinsufficient genes, observed ~ 0.1*expected) |
| pNull    | pNull        | probability of being completely tolerant of loss of function variation (observed = expected)                                 |
| pRec     | pRec         | probability of being intolerant of two loss of function variants (like recessive genes, observed ~ 0.5*expected)             |
| synZ     | syn_z        | corrected synonymous Z score                                                                                                 |
| misZ     | mis_z        | corrected missense Z score                                                                                                   |
| loeuf    | oe_lof_upper | loss of function observed/expected upper bound fraction (LOEUF)                                                              |

#### Gene symbol update

The input file provides Ensembl gene ids for each entry. We observed that they were unique while gene symbols may be repeated (multiple lines may have the same gene symbol). Since Ensembl gene Ids are more stable, and Illumina Connected Annotations transcript cache data contains Ensembl gene ids, we use these ids to extract the gene symbols from the transcript cache. For example, if ENSG0001 has gene symbol GENE1 in the input but Illumina Connected Annotations cache say ENSG0001 maps to GENE2, we use GENE2 as the gene symbol for that entry.

#### Conflict resolution

gnomAD uses Ensembl GeneID as unique identifiers in the [source file](https://storage.googleapis.com/gnomad-public/release/2.1.1/constraint/gnomad.v2.1.1.lof_metrics.by_gene.txt.bgz) but Illumina Connected Annotations uses HGNC gene symbols. Multiple Ensembl GeneIDs can map to the same HGNC symbol and therefore may result is conflict.

```scss
MDGA2   ENST00000426342 306 4.0043e+02  7.6419e-01  2.1096e-05  4724    78  1.6525e+02  4.7202e-01  1923    125 1.3737e+02  9.0993e-01  7.1973e-06  1413    4   2.0926e-06  453 3.8316e+01  9.9922e-01  8.6490e-12  7.8128e-04  1.0440e-01  7.8600e-01  1.0560e+00  6.9500e-01  8.4000e-01  5.0000e-02  2.3900e-01      8.2988e-01  1.6769e+00  5.1372e+00  1529    0   0   7   2.8103e-05  4.0317e-06  124784  7   0   124791  2.8047e-05  9.8167e-05  0.0000e+00  2.8962e-05  0.0000e+00  0.0000e+00  0.0000e+00  3.5391e-05  1.6672e-04  3.2680e-05  0.0000e+00  2.8962e-05  0.0000e+00  0.0000e+00  0.0000e+00  3.5308e-05  1.6492e-04  3.2678e-05  protein_coding  ENSG00000139915 2   2181    13  protein_coding  835332  9.9322e-01  3   2.7833e+01  1.0779e-01  NA  14  47308826    48144157
MDGA2   ENST00000439988 438 5.5311e+02  7.9189e-01  2.9490e-05  6608    105 2.0496e+02  5.1228e-01  2386    180 1.9491e+02  9.2351e-01  9.8371e-06  2048    11  2.8074e-06  627 5.1882e+01  6.6457e-01  5.5841e-10  3.3543e-01  2.1202e-01  8.1700e-01  1.0450e+00  7.3100e-01  8.5700e-01  1.3200e-01  3.5100e-01      8.3940e-01  1.7393e+00  5.2595e+00  2989    1   0   9   3.6173e-05  4.0463e-06  124782  9   0   124791  3.6061e-05  1.6228e-04  6.4986e-05  2.8962e-05  0.0000e+00  0.0000e+00  0.0000e+00  4.4275e-05  1.6672e-04  3.2680e-05  6.4577e-05  2.8962e-05  0.0000e+00  0.0000e+00  0.0000e+00  4.4135e-05  1.6492e-04  3.2678e-05  protein_coding  ENSG00000272781 3   3075    17  protein_coding  832866  NA  NA  NA  NA  NA  14  47311134    48143999
```

In such cases, Illumina Connected Annotations chooses the entry with the smallest "LOEUF" value. The reason for choosing this value can be highlighted by the following table:

| LOEUF decile | Haplo-insufficient | Autosomal Dominant | Autosomal Recessive | Olfactory Genes |
|-------------:|-------------------:|-------------------:|--------------------:|----------------:|
|        0-10% |                104 |                140 |                  36 |               0 |
|       10-20% |                 47 |                128 |                  72 |               1 |
|       20-30% |                 17 |                 86 |                 112 |               0 |
|       30-40% |                  8 |                 80 |                 173 |               4 |
|       40-50% |                  7 |                 65 |                 206 |               8 |
|       50-60% |                  4 |                 54 |                 207 |               6 |
|       60-70% |                  0 |                 46 |                 154 |              18 |
|       70-80% |                  2 |                 49 |                 120 |              49 |
|       80-90% |                  0 |                 34 |                  58 |              96 |
|      90-100% |                  0 |                 26 |                  40 |             174 |

:::info Note
* Table source: [https://www.biorxiv.org/content/biorxiv/early/2019/01/28/531210.full-text.pdf](https://www.biorxiv.org/content/biorxiv/early/2019/01/28/531210.full-text.pdf)
* This table indicates that lower LOEUF scores have more deleterious effect on genes.
* Only 15 out of 19685 genes have conflicting entries.
:::

**List of genes with conflicting entries**
```scss
MDGA2:
 {"pLI":9.99e-1,"pRec":7.81e-4,"pNull":8.65e-12,"synZ":8.30e-1,"misZ":1.68e0,"loeuf":2.39e-1}
 {"pLI":6.65e-1,"pRec":3.35e-1,"pNull":5.58e-10,"synZ":8.39e-1,"misZ":1.74e0,"loeuf":3.51e-1}
CRYBG3:
 {"pLI":9.27e-5,"pRec":1.00e0,"pNull":1.88e-7,"synZ":1.82e0,"misZ":4.68e-1,"loeuf":4.93e-1}
 {"pLI":2.69e-4,"pRec":1.00e0,"pNull":1.20e-4,"synZ":2.63e0,"misZ":9.80e-1,"loeuf":5.98e-1}
CHTF8:
 {"pLI":8.29e-1,"pRec":1.67e-1,"pNull":3.21e-3,"synZ":1.94e0,"misZ":9.48e-1,"loeuf":5.13e-1}
 {"pLI":3.73e-1,"pRec":5.84e-1,"pNull":4.29e-2,"synZ":3.33e-1,"misZ":2.91e-1,"loeuf":9.92e-1}
SEPT1:
 {"pLI":6.77e-8,"pRec":8.90e-1,"pNull":1.10e-1,"synZ":1.58e-1,"misZ":1.57e0,"loeuf":9.68e-1}
 {"pLI":1.96e-8,"pRec":6.71e-1,"pNull":3.29e-1,"synZ":1.68e-1,"misZ":1.41e0,"loeuf":1.08e0}
ARL14EPL:
 {"pLI":3.48e-2,"pRec":8.38e-1,"pNull":1.28e-1,"synZ":3.56e-1,"misZ":-1.87e-1,"loeuf":1.23e0}
 {"pLI":3.23e-2,"pRec":8.29e-1,"pNull":1.38e-1,"synZ":1.15e0,"misZ":-4.05e-1,"loeuf":1.26e0}
UGT2A1:
 {"pLI":2.90e-13,"pRec":1.40e-1,"pNull":8.60e-1,"synZ":-1.29e0,"misZ":-1.77e0,"loeuf":1.18e0}
 {"pLI":3.88e-17,"pRec":2.87e-3,"pNull":9.97e-1,"synZ":-8.00e-1,"misZ":-1.40e0,"loeuf":1.53e0}
LTB4R2:
 {"pLI":4.39e-4,"pRec":6.71e-1,"pNull":3.29e-1,"synZ":-5.24e-1,"misZ":-2.96e-1,"loeuf":1.40e0}
 {"pLI":1.38e-5,"pRec":4.12e-1,"pNull":5.88e-1,"synZ":-4.58e-1,"misZ":-2.02e-1,"loeuf":1.54e0}
CDRT1:
 {"pLI":4.98e-14,"pRec":5.31e-1,"pNull":4.69e-1,"synZ":8.18e-1,"misZ":6.57e-1,"loeuf":1.00e0}
 {"pLI":3.50e-3,"pRec":6.37e-1,"pNull":3.59e-1,"synZ":4.89e-1,"misZ":6.90e-1,"loeuf":1.63e0}
MUC3A:
 {"pLI":1.48e-10,"pRec":5.76e-1,"pNull":4.24e-1,"synZ":5.81e-2,"misZ":-6.01e-1,"loeuf":1.06e0}
 {"pLI":4.03e-1,"pRec":4.79e-1,"pNull":1.17e-1,"synZ":4.05e-2,"misZ":-1.60e-1,"loeuf":1.70e0}
COG8:
 {"pLI":2.97e-9,"pRec":5.04e-1,"pNull":4.96e-1,"synZ":-1.35e0,"misZ":-9.37e-2,"loeuf":1.13e0}
 {"pLI":2.31e-3,"pRec":5.47e-1,"pNull":4.50e-1,"synZ":-4.94e-1,"misZ":-1.48e-1,"loeuf":1.76e0}
AC006486.1:
 {"pLI":9.37e-1,"pRec":6.27e-2,"pNull":2.47e-4,"synZ":1.44e0,"misZ":2.12e0,"loeuf":3.41e-1}
 {"pLI":1.14e-1,"pRec":6.16e-1,"pNull":2.70e-1,"synZ":-7.57e-2,"misZ":8.33e-2,"loeuf":1.84e0}
AL645922.1:
 {"pLI":4.67e-16,"pRec":1.00e0,"pNull":4.15e-5,"synZ":7.99e-1,"misZ":1.61e0,"loeuf":6.92e-1}
 {"pLI":1.60e-3,"pRec":2.78e-1,"pNull":7.21e-1,"synZ":-7.30e-2,"misZ":3.21e-1,"loeuf":1.96e0}
NBPF20:
 {"pLI":1.42e-7,"pRec":3.40e-2,"pNull":9.66e-1,"synZ":-1.86e0,"misZ":-2.88e0,"loeuf":1.97e0}
 {"pLI":1.92e-22,"pRec":7.96e-6,"pNull":1.00e0,"synZ":-9.73e0,"misZ":-7.67e0,"loeuf":1.97e0}
PRAMEF11:
 {"pLI":6.16e-4,"pRec":7.42e-1,"pNull":2.58e-1,"synZ":-4.02e0,"misZ":-3.69e0,"loeuf":1.31e0}
 {"synZ":-3.33e0,"misZ":-2.59e0}
FAM231D:
 {"synZ":-1.98e0,"misZ":-1.44e0}
 {"synZ":1.07e0,"misZ":3.13e-1}
```

**Conflict resolution**
* Pick the entry with the lowest LOEUF score
* If the same, pick the lowest pLI
* Otherwise pick the entry with the max absolute value of synZ + misZ

#### Download URL
[https://storage.googleapis.com/gnomad-public/release/2.1.1/constraint/gnomad.v2.1.1.lof_metrics.by_gene.txt.bgz](https://storage.googleapis.com/gnomad-public/release/2.1.1/constraint/gnomad.v2.1.1.lof_metrics.by_gene.txt.bgz)

#### JSON output

<JSONG/>

### Structural Variants
:::info Publication
Collins, R.L., Brand, H., Karczewski, K.J. et al. 2020. A structural variation reference for medical and population genetics. *Nature* **581**, pp.444–451. https://doi.org/10.1038/s41586-020-2287-8
:::

**Note**
The gnomAD structural variant annotations are in a preview stage at the moment.
Currently, the annotations do not include translocation breakends.
Future updates will include a better way of annotating the structural variants.

#### Source Files
<SVDATADESCRIPTION/>

#### Download URLs
##### GRCh37
The GRCh37 file was downloaded from the original source. Following table gives some essential data metrics:

[https://storage.googleapis.com/gcp-public-data--gnomad/papers/2019-sv/gnomad_v2.1_sv.sites.bed.gz](https://storage.googleapis.com/gcp-public-data--gnomad/papers/2019-sv/gnomad_v2.1_sv.sites.bed.gz)

#### JSON output
<JSONSV/>
