{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Parse Nirvana JSON output in python\n",
    "\n",
    "Nirvana outputs a single JSON annotation file for a single input VCF file. The output file contains a single [JSON object](https://www.w3schools.com/js/js_json_objects.asp) to represent the annotations of all input VCF variants. The JSON file format can be found in the [documentation](https://illumina.github.io/NirvanaDocumentation/file-formats/nirvana-json-file-format).\n",
    "\n",
    "A [sample JSON file](https://github.com/Illumina/NirvanaDocumentation/blob/master/static/files/ceph_trio_test.json.gz) of the CEPH trio (NA12878, NA12891, and NA12892) can be downloaded from the [NirvanaDocumention](https://github.com/Illumina/NirvanaDocumentation/tree/master/static/files) git repo.\n",
    "\n",
    "This notebook demonstrates how you can parse this sample JSON file in python and retrieves the annotation data.\n",
    "\n",
    "## Read Nirvana JSON output by lines\n",
    "\n",
    "Even though the Nirvana JSON output is a single JSON object, different JSON object fields are written in different lines for memory efficient reading.\n",
    "\n",
    "The first line in the JSON file is the `header` line:\n",
    "\n",
    "```json\n",
    "{\"header\":{\"annotator\":\"Nirvana\n",
    "...\n",
    ",\"positions\":[\n",
    "```\n",
    "Followed by the `positions` lines:\n",
    "\n",
    "```json\n",
    "{\"chromosome\":\"chr21\",\"position\":9975027,\"refAllele\":\"C\",\"altAlleles\":[\"G\"],\"quality\":102.47,\"filters\":[\"PASS\"],\"fisherStrandBias\":0.727,\"mappingQuality\":43.11,\"cytogeneticBand\":\"21p11.2\",\"samples\":,\n",
    "...\n",
    "```\n",
    "\n",
    "After the `positions` lines, there are the `genes` lines, which are optional if there is no overlapping gene of the input VCF variants (3 lines for 2 genes in the example):\n",
    "\n",
    "```json\n",
    "],\"genes\":[\n",
    "{\"name\":\"ABCC13\",\"omim\":[{\"mimNumber\":608835,\"geneName\":\"ATP-binding cassette, subfamily C, member 13\",\"description\":\"ABCC13 belongs to a large family of ATP-binding cassette (ABC) transporters that play important roles as membrane transporters or ion channel modulators. However, ABCC13 is a truncated protein that lacks critical ATP-binding motifs and is unlikely to be a functional transporter (Yabuuchi et al., 2002).\"}]},\n",
    "```\n",
    "\n",
    "Finally, the last line of the JSON file are two brackets to complete the JSON object structure:\n",
    "\n",
    "```json\n",
    "]}\n",
    "```\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Install requirements"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pip3 install pydantic\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "import gzip\n",
    "import json\n",
    "import functools\n",
    "\n",
    "\n",
    "import pandas as pd\n",
    "import pydantic\n",
    "\n",
    "\n",
    "class AnnotatedData:\n",
    "    def __init__(self, filename: str):\n",
    "        with gzip.open(filename, 'r') as f:\n",
    "            self._annotated_data = json.load(f)\n",
    "        \n",
    "        for key in (\"annotator\", \"genomeAssembly\", \"creationTime\"):\n",
    "            print(f\"{key}: {self.header[key]}\")     \n",
    "        \n",
    "        print()\n",
    "        \n",
    "        for chromosome in self.chromosomes:\n",
    "            chr_positions = self._split_by_chromosome[chromosome]\n",
    "            print(\n",
    "                f\"chromsome={chromosome}, \"\n",
    "                f\"positions={len(chr_positions)}, \"\n",
    "                f\"min={chr_positions[0]['position']}, \"\n",
    "                f\"max={chr_positions[-1]['position']}\"\n",
    "            )\n",
    "        \n",
    "    @property\n",
    "    def header(self) -> dict:\n",
    "        return self._annotated_data[\"header\"]\n",
    "    \n",
    "    @property\n",
    "    def data_sources(self):\n",
    "        return pd.DataFrame(self.header[\"dataSources\"]).set_index(\"name\").sort_index()\n",
    "        \n",
    "    @property\n",
    "    def genes(self) -> pd.DataFrame:\n",
    "        return pd.DataFrame(self._annotated_data[\"genes\"])\n",
    "        \n",
    "    @property\n",
    "    def positions(self) -> list:\n",
    "        return self._annotated_data[\"positions\"]\n",
    "    \n",
    "    @property\n",
    "    def chromosomes(self) -> list:\n",
    "        return sorted(self._split_by_chromosome.keys())\n",
    "    \n",
    "    @functools.cached_property\n",
    "    def _split_by_chromosome(self) -> dict:\n",
    "        split_by_chromosome = {}\n",
    "        for position in self.positions:\n",
    "            chromosome = position[\"chromosome\"]\n",
    "            if chromosome not in split_by_chromosome:\n",
    "                split_by_chromosome[chromosome] = []\n",
    "                \n",
    "            split_by_chromosome[chromosome].append(position)\n",
    "            \n",
    "        \n",
    "        \n",
    "        return split_by_chromosome\n",
    "    \n",
    "    def get_annotation(self, chromosome: str, position: int) -> dict:\n",
    "        if chromosome not in self.chromosomes:\n",
    "            return None\n",
    "        \n",
    "        return next(\n",
    "            (\n",
    "                position_item for position_item in self._split_by_chromosome[chromosome]\n",
    "            ),\n",
    "            None\n",
    "        )\n",
    "    \n",
    "class Position(pydantic.BaseModel):\n",
    "    model_config = pydantic.ConfigDict(extra='allow') \n",
    "    chromosome: str\n",
    "    position: int\n",
    "    refAllele: str\n",
    "    altAlleles: list[str]\n",
    "    filters: list\n",
    "    mappingQuality: float\n",
    "    cytogeneticBand: str\n",
    "    vcfInfo: dict\n",
    "    samples: list\n",
    "    variants: list\n",
    "        \n",
    "    def get_top_level(self):\n",
    "        return dict(zip(\n",
    "            (\"chromosome\", \"position\", \"refAllele\", \"altAlleles\", \"filters\", \"mappingQuality\", \"cyatogeneticBands\", \"vcfInfo\"),\n",
    "            (self.chromosome, self.position, self.refAllele, self.altAlleles, self.filters, self.mappingQuality, self.cytogeneticBand, self.vcfInfo)\n",
    "        ))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Header"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotator: Illumina Connected Annotations 3.22.0\n",
      "genomeAssembly: GRCh38\n",
      "creationTime: 2023-12-07 14:15:54\n",
      "\n",
      "chromsome=chr21, positions=1819, min=5222289, max=46678074\n"
     ]
    }
   ],
   "source": [
    "annotated_data = AnnotatedData(filename=\"annotated_38.json.gz\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>version</th>\n",
       "      <th>description</th>\n",
       "      <th>releaseDate</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1000 Genomes Project</th>\n",
       "      <td>Phase 3 v3plus</td>\n",
       "      <td>A public catalogue of human variation and geno...</td>\n",
       "      <td>2013-05-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000 Genomes Project (SV)</th>\n",
       "      <td>Phase 3 v5a</td>\n",
       "      <td>A public catalogue of human variation and geno...</td>\n",
       "      <td>2013-05-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COSMIC</th>\n",
       "      <td>96</td>\n",
       "      <td>resource for exploring the impact of somatic m...</td>\n",
       "      <td>2022-05-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COSMIC gene fusions</th>\n",
       "      <td>96</td>\n",
       "      <td>manually curated somatic gene fusions</td>\n",
       "      <td>2023-11-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CancerHotspots</th>\n",
       "      <td>2017</td>\n",
       "      <td>A resouce for statistically significant mutati...</td>\n",
       "      <td>2017-01-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ClinGen</th>\n",
       "      <td>20160414</td>\n",
       "      <td>None</td>\n",
       "      <td>2016-04-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ClinGen Dosage Sensitivity Map</th>\n",
       "      <td>20231105</td>\n",
       "      <td>Dosage sensitivity map from ClinGen (dbVar)</td>\n",
       "      <td>2023-11-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ClinGen disease validity curations</th>\n",
       "      <td>20231105</td>\n",
       "      <td>Disease validity curations from ClinGen (dbVar)</td>\n",
       "      <td>2023-11-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ClinVar</th>\n",
       "      <td>20231028</td>\n",
       "      <td>A freely accessible, public archive of reports...</td>\n",
       "      <td>2023-11-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cosmic Cancer Gene Census</th>\n",
       "      <td>97</td>\n",
       "      <td>Cosmic Cancer Gene Census catalogs genes with ...</td>\n",
       "      <td>2022-11-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DANN</th>\n",
       "      <td>20200205</td>\n",
       "      <td>Pathogenicity scores of missense variants pred...</td>\n",
       "      <td>2020-02-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DECIPHER</th>\n",
       "      <td>201509</td>\n",
       "      <td>Common copy-number variants and their frequenc...</td>\n",
       "      <td>2015-09-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ensembl</th>\n",
       "      <td>110</td>\n",
       "      <td>Ensembl GRCh38 Release</td>\n",
       "      <td>2023-04-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FusionCatcher</th>\n",
       "      <td>1.33</td>\n",
       "      <td>known germline and somatic gene fusions</td>\n",
       "      <td>2020-12-22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GME</th>\n",
       "      <td>20160618</td>\n",
       "      <td>Greater Middle Eastern Variome allele frequenc...</td>\n",
       "      <td>2016-06-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gerp</th>\n",
       "      <td>20110522</td>\n",
       "      <td>GERP identifies constrained elements in multip...</td>\n",
       "      <td>2011-05-22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MITOMAP</th>\n",
       "      <td>20200819</td>\n",
       "      <td>Small variants in the MITOMAP human mitochondr...</td>\n",
       "      <td>2020-08-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MITOMAP_SV</th>\n",
       "      <td>20200819</td>\n",
       "      <td>Large structural variants in the MITOMAP human...</td>\n",
       "      <td>2020-08-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MitochondrialHeteroplasmy</th>\n",
       "      <td>20180410</td>\n",
       "      <td>Variant read frequency percentiles for the Mit...</td>\n",
       "      <td>2020-05-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OMIM</th>\n",
       "      <td>20231105</td>\n",
       "      <td>An Online Catalog of Human Genes and Genetic D...</td>\n",
       "      <td>2023-11-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PhyloPPrimate</th>\n",
       "      <td>1.0</td>\n",
       "      <td>PhyloPPrimate identifies primate-specific DNA ...</td>\n",
       "      <td>2023-11-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PrimateAI-3D</th>\n",
       "      <td>1.0</td>\n",
       "      <td>PrimateAI-3D is a deep-learning network that i...</td>\n",
       "      <td>2023-06-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REVEL</th>\n",
       "      <td>20200205</td>\n",
       "      <td>Pathogenicity scores of missense variants pred...</td>\n",
       "      <td>2020-02-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RefSeq</th>\n",
       "      <td>GCF_000001405.40-RS_2023_03</td>\n",
       "      <td>NCBI Homo sapiens Annotation Release</td>\n",
       "      <td>2023-03-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SpliceAI</th>\n",
       "      <td>1.3</td>\n",
       "      <td>SpliceAI scores, distances, etc.</td>\n",
       "      <td>2019-10-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TOPMed</th>\n",
       "      <td>freeze_5</td>\n",
       "      <td>Allele frequencies from TOPMed data</td>\n",
       "      <td>2017-08-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbSNP</th>\n",
       "      <td>156</td>\n",
       "      <td>Identifiers for observed variants</td>\n",
       "      <td>2023-09-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gnomAD</th>\n",
       "      <td>3.1.2</td>\n",
       "      <td>Allele frequencies from Genome Aggregation Dat...</td>\n",
       "      <td>2021-10-22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gnomAD_LCR</th>\n",
       "      <td>2.1</td>\n",
       "      <td>Low complexity regions from gnomAD.</td>\n",
       "      <td>2019-04-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gnomAD_SV</th>\n",
       "      <td>2.1</td>\n",
       "      <td>A structural variation reference from Genome A...</td>\n",
       "      <td>2020-08-20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gnomAD_gene_scores</th>\n",
       "      <td>2.1</td>\n",
       "      <td>Gene Loss of function scores from Genome Aggre...</td>\n",
       "      <td>2018-10-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phyloP</th>\n",
       "      <td>hg38</td>\n",
       "      <td>20 way conservation score between humans and 1...</td>\n",
       "      <td>2015-04-17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        version   \n",
       "name                                                              \n",
       "1000 Genomes Project                             Phase 3 v3plus  \\\n",
       "1000 Genomes Project (SV)                           Phase 3 v5a   \n",
       "COSMIC                                                       96   \n",
       "COSMIC gene fusions                                          96   \n",
       "CancerHotspots                                             2017   \n",
       "ClinGen                                                20160414   \n",
       "ClinGen Dosage Sensitivity Map                         20231105   \n",
       "ClinGen disease validity curations                     20231105   \n",
       "ClinVar                                                20231028   \n",
       "Cosmic Cancer Gene Census                                    97   \n",
       "DANN                                                   20200205   \n",
       "DECIPHER                                                 201509   \n",
       "Ensembl                                                     110   \n",
       "FusionCatcher                                              1.33   \n",
       "GME                                                    20160618   \n",
       "Gerp                                                   20110522   \n",
       "MITOMAP                                                20200819   \n",
       "MITOMAP_SV                                             20200819   \n",
       "MitochondrialHeteroplasmy                              20180410   \n",
       "OMIM                                                   20231105   \n",
       "PhyloPPrimate                                               1.0   \n",
       "PrimateAI-3D                                                1.0   \n",
       "REVEL                                                  20200205   \n",
       "RefSeq                              GCF_000001405.40-RS_2023_03   \n",
       "SpliceAI                                                    1.3   \n",
       "TOPMed                                                 freeze_5   \n",
       "dbSNP                                                       156   \n",
       "gnomAD                                                    3.1.2   \n",
       "gnomAD_LCR                                                  2.1   \n",
       "gnomAD_SV                                                   2.1   \n",
       "gnomAD_gene_scores                                          2.1   \n",
       "phyloP                                                     hg38   \n",
       "\n",
       "                                                                          description   \n",
       "name                                                                                    \n",
       "1000 Genomes Project                A public catalogue of human variation and geno...  \\\n",
       "1000 Genomes Project (SV)           A public catalogue of human variation and geno...   \n",
       "COSMIC                              resource for exploring the impact of somatic m...   \n",
       "COSMIC gene fusions                             manually curated somatic gene fusions   \n",
       "CancerHotspots                      A resouce for statistically significant mutati...   \n",
       "ClinGen                                                                          None   \n",
       "ClinGen Dosage Sensitivity Map            Dosage sensitivity map from ClinGen (dbVar)   \n",
       "ClinGen disease validity curations    Disease validity curations from ClinGen (dbVar)   \n",
       "ClinVar                             A freely accessible, public archive of reports...   \n",
       "Cosmic Cancer Gene Census           Cosmic Cancer Gene Census catalogs genes with ...   \n",
       "DANN                                Pathogenicity scores of missense variants pred...   \n",
       "DECIPHER                            Common copy-number variants and their frequenc...   \n",
       "Ensembl                                                        Ensembl GRCh38 Release   \n",
       "FusionCatcher                                 known germline and somatic gene fusions   \n",
       "GME                                 Greater Middle Eastern Variome allele frequenc...   \n",
       "Gerp                                GERP identifies constrained elements in multip...   \n",
       "MITOMAP                             Small variants in the MITOMAP human mitochondr...   \n",
       "MITOMAP_SV                          Large structural variants in the MITOMAP human...   \n",
       "MitochondrialHeteroplasmy           Variant read frequency percentiles for the Mit...   \n",
       "OMIM                                An Online Catalog of Human Genes and Genetic D...   \n",
       "PhyloPPrimate                       PhyloPPrimate identifies primate-specific DNA ...   \n",
       "PrimateAI-3D                        PrimateAI-3D is a deep-learning network that i...   \n",
       "REVEL                               Pathogenicity scores of missense variants pred...   \n",
       "RefSeq                                           NCBI Homo sapiens Annotation Release   \n",
       "SpliceAI                                             SpliceAI scores, distances, etc.   \n",
       "TOPMed                                            Allele frequencies from TOPMed data   \n",
       "dbSNP                                               Identifiers for observed variants   \n",
       "gnomAD                              Allele frequencies from Genome Aggregation Dat...   \n",
       "gnomAD_LCR                                        Low complexity regions from gnomAD.   \n",
       "gnomAD_SV                           A structural variation reference from Genome A...   \n",
       "gnomAD_gene_scores                  Gene Loss of function scores from Genome Aggre...   \n",
       "phyloP                              20 way conservation score between humans and 1...   \n",
       "\n",
       "                                   releaseDate  \n",
       "name                                            \n",
       "1000 Genomes Project                2013-05-27  \n",
       "1000 Genomes Project (SV)           2013-05-27  \n",
       "COSMIC                              2022-05-31  \n",
       "COSMIC gene fusions                 2023-11-07  \n",
       "CancerHotspots                      2017-01-01  \n",
       "ClinGen                             2016-04-14  \n",
       "ClinGen Dosage Sensitivity Map      2023-11-05  \n",
       "ClinGen disease validity curations  2023-11-05  \n",
       "ClinVar                             2023-11-05  \n",
       "Cosmic Cancer Gene Census           2022-11-29  \n",
       "DANN                                2020-02-05  \n",
       "DECIPHER                            2015-09-01  \n",
       "Ensembl                             2023-04-27  \n",
       "FusionCatcher                       2020-12-22  \n",
       "GME                                 2016-06-18  \n",
       "Gerp                                2011-05-22  \n",
       "MITOMAP                             2020-08-19  \n",
       "MITOMAP_SV                          2020-08-19  \n",
       "MitochondrialHeteroplasmy           2020-05-21  \n",
       "OMIM                                2023-11-05  \n",
       "PhyloPPrimate                       2023-11-29  \n",
       "PrimateAI-3D                        2023-06-02  \n",
       "REVEL                               2020-02-05  \n",
       "RefSeq                              2023-03-21  \n",
       "SpliceAI                            2019-10-04  \n",
       "TOPMed                              2017-08-28  \n",
       "dbSNP                               2023-09-28  \n",
       "gnomAD                              2021-10-22  \n",
       "gnomAD_LCR                          2019-04-10  \n",
       "gnomAD_SV                           2020-08-20  \n",
       "gnomAD_gene_scores                  2018-10-17  \n",
       "phyloP                              2015-04-17  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.data_sources"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>clingenDosageSensitivityMap</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AATBC</td>\n",
       "      <td>51526.0</td>\n",
       "      <td>284837</td>\n",
       "      <td>ENSG00000215458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000291052</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>16022.0</td>\n",
       "      <td>150000</td>\n",
       "      <td>ENSG00000243064</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCG1</td>\n",
       "      <td>73.0</td>\n",
       "      <td>9619</td>\n",
       "      <td>ENSG00000160179</td>\n",
       "      <td>[{'mimNumber': 603076, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>{'pLi': 0.112, 'pRec': 0.888, 'pNull': 9.34e-0...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ADARB1</td>\n",
       "      <td>226.0</td>\n",
       "      <td>104</td>\n",
       "      <td>ENSG00000197381</td>\n",
       "      <td>[{'mimNumber': 601218, 'geneName': 'Adenosine ...</td>\n",
       "      <td>{'pLi': 0.803, 'pRec': 0.197, 'pNull': 8.32e-0...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>YBEY</td>\n",
       "      <td>1299.0</td>\n",
       "      <td>54059</td>\n",
       "      <td>ENSG00000182362</td>\n",
       "      <td>[{'mimNumber': 617461, 'geneName': 'YBEY metal...</td>\n",
       "      <td>{'pLi': 1.34e-07, 'pRec': 0.0625, 'pNull': 0.9...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>YRDCP3</td>\n",
       "      <td>39921.0</td>\n",
       "      <td>100861429</td>\n",
       "      <td>ENSG00000230859</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>ZBTB21</td>\n",
       "      <td>13083.0</td>\n",
       "      <td>49854</td>\n",
       "      <td>ENSG00000173276</td>\n",
       "      <td>[{'mimNumber': 616485, 'geneName': 'Zinc finge...</td>\n",
       "      <td>{'pLi': 0.998, 'pRec': 0.00222, 'pNull': 5.45e...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>ZNF295-AS1</td>\n",
       "      <td>23130.0</td>\n",
       "      <td>150142</td>\n",
       "      <td>ENSG00000237232</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>ZNF355P</td>\n",
       "      <td>17218.0</td>\n",
       "      <td>100505852</td>\n",
       "      <td>ENSG00000168122</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>442 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "0         AATBC  51526.0     284837  ENSG00000215458  \\\n",
       "1        ABCC13      NaN        NaN  ENSG00000291052   \n",
       "2        ABCC13  16022.0     150000  ENSG00000243064   \n",
       "3         ABCG1     73.0       9619  ENSG00000160179   \n",
       "4        ADARB1    226.0        104  ENSG00000197381   \n",
       "..          ...      ...        ...              ...   \n",
       "437        YBEY   1299.0      54059  ENSG00000182362   \n",
       "438      YRDCP3  39921.0  100861429  ENSG00000230859   \n",
       "439      ZBTB21  13083.0      49854  ENSG00000173276   \n",
       "440  ZNF295-AS1  23130.0     150142  ENSG00000237232   \n",
       "441     ZNF355P  17218.0  100505852  ENSG00000168122   \n",
       "\n",
       "                                                  omim   \n",
       "0                                                  NaN  \\\n",
       "1    [{'mimNumber': 608835, 'geneName': 'ATP-bindin...   \n",
       "2    [{'mimNumber': 608835, 'geneName': 'ATP-bindin...   \n",
       "3    [{'mimNumber': 603076, 'geneName': 'ATP-bindin...   \n",
       "4    [{'mimNumber': 601218, 'geneName': 'Adenosine ...   \n",
       "..                                                 ...   \n",
       "437  [{'mimNumber': 617461, 'geneName': 'YBEY metal...   \n",
       "438                                                NaN   \n",
       "439  [{'mimNumber': 616485, 'geneName': 'Zinc finge...   \n",
       "440                                                NaN   \n",
       "441                                                NaN   \n",
       "\n",
       "                                                gnomAD   \n",
       "0                                                  NaN  \\\n",
       "1                                                  NaN   \n",
       "2                                                  NaN   \n",
       "3    {'pLi': 0.112, 'pRec': 0.888, 'pNull': 9.34e-0...   \n",
       "4    {'pLi': 0.803, 'pRec': 0.197, 'pNull': 8.32e-0...   \n",
       "..                                                 ...   \n",
       "437  {'pLi': 1.34e-07, 'pRec': 0.0625, 'pNull': 0.9...   \n",
       "438                                                NaN   \n",
       "439  {'pLi': 0.998, 'pRec': 0.00222, 'pNull': 5.45e...   \n",
       "440                                                NaN   \n",
       "441                                                NaN   \n",
       "\n",
       "    clingenDosageSensitivityMap clingenGeneValidity cosmic  \n",
       "0                           NaN                 NaN    NaN  \n",
       "1                           NaN                 NaN    NaN  \n",
       "2                           NaN                 NaN    NaN  \n",
       "3                           NaN                 NaN    NaN  \n",
       "4                           NaN                 NaN    NaN  \n",
       "..                          ...                 ...    ...  \n",
       "437                         NaN                 NaN    NaN  \n",
       "438                         NaN                 NaN    NaN  \n",
       "439                         NaN                 NaN    NaN  \n",
       "440                         NaN                 NaN    NaN  \n",
       "441                         NaN                 NaN    NaN  \n",
       "\n",
       "[442 rows x 9 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Positions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'chromosome': 'chr21',\n",
       " 'position': 5222289,\n",
       " 'refAllele': 'C',\n",
       " 'altAlleles': ['T'],\n",
       " 'filters': ['mapping_quality', 'weak_evidence'],\n",
       " 'mappingQuality': 49.7,\n",
       " 'cyatogeneticBands': '21p12',\n",
       " 'vcfInfo': {'DP': '749'}}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position = Position.model_validate(annotated_data.get_annotation(\"chr21\", 5222289))\n",
    "position.get_top_level()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'genotype': '0/0',\n",
       "  'variantFrequencies': [0.0219],\n",
       "  'totalDepth': 137,\n",
       "  'alleleDepths': [134, 3],\n",
       "  'somaticQuality': 0,\n",
       "  'vcfSampleInfo': {'F1R2': '77,3', 'F2R1': '57,0'}},\n",
       " {'genotype': '0/1',\n",
       "  'variantFrequencies': [0.0196],\n",
       "  'totalDepth': 562,\n",
       "  'alleleDepths': [551, 11],\n",
       "  'somaticQuality': 7,\n",
       "  'vcfSampleInfo': {'F1R2': '286,6', 'F2R1': '265,5'}}]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position.samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'vid': '21-5222289-C-T',\n",
       "  'chromosome': 'chr21',\n",
       "  'begin': 5222289,\n",
       "  'end': 5222289,\n",
       "  'refAllele': 'C',\n",
       "  'altAllele': 'T',\n",
       "  'variantType': 'SNV',\n",
       "  'hgvsg': 'NC_000021.9:g.5222289C>T',\n",
       "  'phylopScore': -0.2,\n",
       "  'dbsnp': ['rs1366179382'],\n",
       "  'gnomad': {'coverage': 3,\n",
       "   'allAf': 0.000502,\n",
       "   'allAn': 149274,\n",
       "   'allAc': 75,\n",
       "   'allHc': 0,\n",
       "   'afrAf': 0.001704,\n",
       "   'afrAn': 40502,\n",
       "   'afrAc': 69,\n",
       "   'afrHc': 0,\n",
       "   'amrAf': 6.7e-05,\n",
       "   'amrAn': 14966,\n",
       "   'amrAc': 1,\n",
       "   'amrHc': 0,\n",
       "   'easAf': 0,\n",
       "   'easAn': 5022,\n",
       "   'easAc': 0,\n",
       "   'easHc': 0,\n",
       "   'finAf': 0,\n",
       "   'finAn': 10430,\n",
       "   'finAc': 0,\n",
       "   'finHc': 0,\n",
       "   'nfeAf': 7.5e-05,\n",
       "   'nfeAn': 66950,\n",
       "   'nfeAc': 5,\n",
       "   'nfeHc': 0,\n",
       "   'asjAf': 0,\n",
       "   'asjAn': 3438,\n",
       "   'asjAc': 0,\n",
       "   'asjHc': 0,\n",
       "   'sasAf': 0,\n",
       "   'sasAn': 4734,\n",
       "   'sasAc': 0,\n",
       "   'sasHc': 0,\n",
       "   'othAf': 0,\n",
       "   'othAn': 2024,\n",
       "   'othAc': 0,\n",
       "   'othHc': 0,\n",
       "   'maleAf': 0.000411,\n",
       "   'maleAn': 73006,\n",
       "   'maleAc': 30,\n",
       "   'maleHc': 0,\n",
       "   'femaleAf': 0.00059,\n",
       "   'femaleAn': 76268,\n",
       "   'femaleAc': 45,\n",
       "   'femaleHc': 0,\n",
       "   'controlsAllAf': 0.000565,\n",
       "   'controlsAllAn': 31850,\n",
       "   'controlsAllAc': 18},\n",
       "  'topmed': {'allAf': 0.001091,\n",
       "   'allAn': 125568,\n",
       "   'allAc': 137,\n",
       "   'allHc': 0,\n",
       "   'failedFilter': True},\n",
       "  'phyloPPrimateScore': 0.074}]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position.variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Parsing and Filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "class Parser:\n",
    "    def __init__(self, annotated_data: AnnotatedData):\n",
    "        self.annotated_data = annotated_data\n",
    "        \n",
    "    def get_variants_above_gnomad_freq(\n",
    "        self,\n",
    "        frequency_key: str,\n",
    "        frequency_threshold_low=float(\"-inf\"),\n",
    "        frequency_threshold_high=float(\"inf\")\n",
    "    ) -> list:\n",
    "        positions = [\n",
    "            Position.model_validate(position)\n",
    "            for position in self.annotated_data.positions\n",
    "            for variant in position.get(\"variants\", {})\n",
    "            if (freq := variant.get(\"gnomad\", {}).get(frequency_key, None)) \\\n",
    "            and frequency_threshold_low < freq < frequency_threshold_high\n",
    "        ]\n",
    "        return positions\n",
    "    \n",
    "    def get_positions_with_cannonical_transcripts(self):   \n",
    "        positions = [\n",
    "            Position.model_validate(position)\n",
    "            for position in self.annotated_data.positions\n",
    "            for variant in position.get(\"variants\", {})\n",
    "            for transcript in variant.get(\"transcripts\", [])\n",
    "            if transcript.get(\"isCanonical\")\n",
    "        ]\n",
    "\n",
    "        return positions\n",
    "    \n",
    "    def filter_transcripts_by_consequence(self, include=[], exclude=[]):\n",
    "        positions = [\n",
    "            Position.model_validate(position)\n",
    "            for position in self.annotated_data.positions\n",
    "            for variant in position.get(\"variants\", {})\n",
    "            for transcript in variant.get(\"transcripts\", [])\n",
    "            for consequence in transcript.get(\"consequence\", [])\n",
    "            if (not bool(include) or consequence in include) and consequence not in exclude\n",
    "        ]\n",
    "        return positions\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Positions with Cannonical Transcripts only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = Parser(annotated_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2472,\n",
       " {'chromosome': 'chr21',\n",
       "  'position': 5228221,\n",
       "  'refAllele': 'G',\n",
       "  'altAlleles': ['T'],\n",
       "  'filters': ['PASS'],\n",
       "  'mappingQuality': 75.75,\n",
       "  'cytogeneticBand': '21p12',\n",
       "  'vcfInfo': {'DP': '1004'},\n",
       "  'samples': [{'genotype': '0/0',\n",
       "    'variantFrequencies': [0.0046],\n",
       "    'totalDepth': 219,\n",
       "    'alleleDepths': [218, 1],\n",
       "    'somaticQuality': 0,\n",
       "    'vcfSampleInfo': {'F1R2': '108,0', 'F2R1': '110,1'}},\n",
       "   {'genotype': '0/1',\n",
       "    'variantFrequencies': [0.1227],\n",
       "    'totalDepth': 709,\n",
       "    'alleleDepths': [622, 87],\n",
       "    'somaticQuality': 86.7,\n",
       "    'vcfSampleInfo': {'F1R2': '309,38', 'F2R1': '313,49'}}],\n",
       "  'variants': [{'vid': '21-5228221-G-T',\n",
       "    'chromosome': 'chr21',\n",
       "    'begin': 5228221,\n",
       "    'end': 5228221,\n",
       "    'refAllele': 'G',\n",
       "    'altAllele': 'T',\n",
       "    'variantType': 'SNV',\n",
       "    'hgvsg': 'NC_000021.9:g.5228221G>T',\n",
       "    'phylopScore': 0.2,\n",
       "    'phyloPPrimateScore': 0,\n",
       "    'transcripts': [{'transcript': 'ENST00000623753.1',\n",
       "      'source': 'Ensembl',\n",
       "      'bioType': 'lncRNA',\n",
       "      'geneId': 'ENSG00000279669',\n",
       "      'hgnc': 'ENSG00000279669',\n",
       "      'consequence': ['downstream_gene_variant'],\n",
       "      'impact': 'modifier',\n",
       "      'isCanonical': True}]}]})"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = parser.get_positions_with_cannonical_transcripts()\n",
    "len(positions), positions[0].model_dump()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'transcript': 'ENST00000623753.1',\n",
       "  'source': 'Ensembl',\n",
       "  'bioType': 'lncRNA',\n",
       "  'geneId': 'ENSG00000279669',\n",
       "  'hgnc': 'ENSG00000279669',\n",
       "  'consequence': ['downstream_gene_variant'],\n",
       "  'impact': 'modifier',\n",
       "  'isCanonical': True}]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions[0].variants[0].get(\"transcripts\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter by Consequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = Parser(annotated_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(147,\n",
       " {'chromosome': 'chr21',\n",
       "  'position': 5232869,\n",
       "  'refAllele': 'T',\n",
       "  'altAlleles': ['G'],\n",
       "  'filters': ['base_quality', 'weak_evidence'],\n",
       "  'mappingQuality': 58.1,\n",
       "  'cytogeneticBand': '21p12',\n",
       "  'vcfInfo': {'DP': '799'},\n",
       "  'samples': [{'genotype': '0/0',\n",
       "    'variantFrequencies': [0.0552],\n",
       "    'totalDepth': 181,\n",
       "    'alleleDepths': [171, 10],\n",
       "    'somaticQuality': 0,\n",
       "    'vcfSampleInfo': {'F1R2': '96,6', 'F2R1': '75,4'}},\n",
       "   {'genotype': '0/1',\n",
       "    'variantFrequencies': [0.0677],\n",
       "    'totalDepth': 591,\n",
       "    'alleleDepths': [551, 40],\n",
       "    'somaticQuality': 8.6,\n",
       "    'vcfSampleInfo': {'F1R2': '280,23', 'F2R1': '271,17'}}],\n",
       "  'variants': [{'vid': '21-5232869-T-G',\n",
       "    'chromosome': 'chr21',\n",
       "    'begin': 5232869,\n",
       "    'end': 5232869,\n",
       "    'refAllele': 'T',\n",
       "    'altAllele': 'G',\n",
       "    'variantType': 'SNV',\n",
       "    'hgvsg': 'NC_000021.9:g.5232869T>G',\n",
       "    'phylopScore': 0.5,\n",
       "    'transcripts': [{'transcript': 'ENST00000623753.1',\n",
       "      'source': 'Ensembl',\n",
       "      'bioType': 'lncRNA',\n",
       "      'cdnaPos': '187',\n",
       "      'exons': '2/2',\n",
       "      'geneId': 'ENSG00000279669',\n",
       "      'hgnc': 'ENSG00000279669',\n",
       "      'consequence': ['non_coding_transcript_exon_variant'],\n",
       "      'impact': 'modifier',\n",
       "      'hgvsc': 'ENST00000623753.1:n.187A>C',\n",
       "      'isCanonical': True}]}]})"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = parser.filter_transcripts_by_consequence(\n",
    "    include=[\"non_coding_transcript_exon_variant\"]\n",
    ")\n",
    "len(positions), positions[0].model_dump()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(11723,\n",
       " {'chromosome': 'chr21',\n",
       "  'position': 5232869,\n",
       "  'refAllele': 'T',\n",
       "  'altAlleles': ['G'],\n",
       "  'filters': ['base_quality', 'weak_evidence'],\n",
       "  'mappingQuality': 58.1,\n",
       "  'cytogeneticBand': '21p12',\n",
       "  'vcfInfo': {'DP': '799'},\n",
       "  'samples': [{'genotype': '0/0',\n",
       "    'variantFrequencies': [0.0552],\n",
       "    'totalDepth': 181,\n",
       "    'alleleDepths': [171, 10],\n",
       "    'somaticQuality': 0,\n",
       "    'vcfSampleInfo': {'F1R2': '96,6', 'F2R1': '75,4'}},\n",
       "   {'genotype': '0/1',\n",
       "    'variantFrequencies': [0.0677],\n",
       "    'totalDepth': 591,\n",
       "    'alleleDepths': [551, 40],\n",
       "    'somaticQuality': 8.6,\n",
       "    'vcfSampleInfo': {'F1R2': '280,23', 'F2R1': '271,17'}}],\n",
       "  'variants': [{'vid': '21-5232869-T-G',\n",
       "    'chromosome': 'chr21',\n",
       "    'begin': 5232869,\n",
       "    'end': 5232869,\n",
       "    'refAllele': 'T',\n",
       "    'altAllele': 'G',\n",
       "    'variantType': 'SNV',\n",
       "    'hgvsg': 'NC_000021.9:g.5232869T>G',\n",
       "    'phylopScore': 0.5,\n",
       "    'transcripts': [{'transcript': 'ENST00000623753.1',\n",
       "      'source': 'Ensembl',\n",
       "      'bioType': 'lncRNA',\n",
       "      'cdnaPos': '187',\n",
       "      'exons': '2/2',\n",
       "      'geneId': 'ENSG00000279669',\n",
       "      'hgnc': 'ENSG00000279669',\n",
       "      'consequence': ['non_coding_transcript_exon_variant'],\n",
       "      'impact': 'modifier',\n",
       "      'hgvsc': 'ENST00000623753.1:n.187A>C',\n",
       "      'isCanonical': True}]}]})"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = parser.filter_transcripts_by_consequence(\n",
    "    exclude=[\"downstream_gene_variant\", \"upstream_gene_variant\"]\n",
    ")\n",
    "len(positions), positions[0].model_dump()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter by gnomad frequency\n",
    "Possible values\n",
    "\n",
    "'coverage', 'failedFilter', 'allAf', 'allAn', 'allAc', 'allHc', 'afrAf', 'afrAn', 'afrAc', 'afrHc', 'amrAf', 'amrAn', 'amrAc', 'amrHc', 'easAf', 'easAn', 'easAc', 'easHc', 'finAf', 'finAn', 'finAc', 'finHc', 'nfeAf', 'nfeAn', 'nfeAc', 'nfeHc', 'asjAf', 'asjAn', 'asjAc', 'asjHc', 'sasAf', 'sasAn', 'sasAc', 'sasHc', 'othAf', 'othAn', 'othAc', 'othHc', 'maleAf', 'maleAn', 'maleAc', 'maleHc', 'femaleAf', 'femaleAn', 'femaleAc', 'femaleHc', 'controlsAllAf', 'controlsAllAn', 'controlsAllAc'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = Parser(annotated_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(851,\n",
       " {'chromosome': 'chr21',\n",
       "  'position': 5222289,\n",
       "  'refAllele': 'C',\n",
       "  'altAlleles': ['T'],\n",
       "  'filters': ['mapping_quality', 'weak_evidence'],\n",
       "  'mappingQuality': 49.7,\n",
       "  'cytogeneticBand': '21p12',\n",
       "  'vcfInfo': {'DP': '749'},\n",
       "  'samples': [{'genotype': '0/0',\n",
       "    'variantFrequencies': [0.0219],\n",
       "    'totalDepth': 137,\n",
       "    'alleleDepths': [134, 3],\n",
       "    'somaticQuality': 0,\n",
       "    'vcfSampleInfo': {'F1R2': '77,3', 'F2R1': '57,0'}},\n",
       "   {'genotype': '0/1',\n",
       "    'variantFrequencies': [0.0196],\n",
       "    'totalDepth': 562,\n",
       "    'alleleDepths': [551, 11],\n",
       "    'somaticQuality': 7,\n",
       "    'vcfSampleInfo': {'F1R2': '286,6', 'F2R1': '265,5'}}],\n",
       "  'variants': [{'vid': '21-5222289-C-T',\n",
       "    'chromosome': 'chr21',\n",
       "    'begin': 5222289,\n",
       "    'end': 5222289,\n",
       "    'refAllele': 'C',\n",
       "    'altAllele': 'T',\n",
       "    'variantType': 'SNV',\n",
       "    'hgvsg': 'NC_000021.9:g.5222289C>T',\n",
       "    'phylopScore': -0.2,\n",
       "    'dbsnp': ['rs1366179382'],\n",
       "    'gnomad': {'coverage': 3,\n",
       "     'allAf': 0.000502,\n",
       "     'allAn': 149274,\n",
       "     'allAc': 75,\n",
       "     'allHc': 0,\n",
       "     'afrAf': 0.001704,\n",
       "     'afrAn': 40502,\n",
       "     'afrAc': 69,\n",
       "     'afrHc': 0,\n",
       "     'amrAf': 6.7e-05,\n",
       "     'amrAn': 14966,\n",
       "     'amrAc': 1,\n",
       "     'amrHc': 0,\n",
       "     'easAf': 0,\n",
       "     'easAn': 5022,\n",
       "     'easAc': 0,\n",
       "     'easHc': 0,\n",
       "     'finAf': 0,\n",
       "     'finAn': 10430,\n",
       "     'finAc': 0,\n",
       "     'finHc': 0,\n",
       "     'nfeAf': 7.5e-05,\n",
       "     'nfeAn': 66950,\n",
       "     'nfeAc': 5,\n",
       "     'nfeHc': 0,\n",
       "     'asjAf': 0,\n",
       "     'asjAn': 3438,\n",
       "     'asjAc': 0,\n",
       "     'asjHc': 0,\n",
       "     'sasAf': 0,\n",
       "     'sasAn': 4734,\n",
       "     'sasAc': 0,\n",
       "     'sasHc': 0,\n",
       "     'othAf': 0,\n",
       "     'othAn': 2024,\n",
       "     'othAc': 0,\n",
       "     'othHc': 0,\n",
       "     'maleAf': 0.000411,\n",
       "     'maleAn': 73006,\n",
       "     'maleAc': 30,\n",
       "     'maleHc': 0,\n",
       "     'femaleAf': 0.00059,\n",
       "     'femaleAn': 76268,\n",
       "     'femaleAc': 45,\n",
       "     'femaleHc': 0,\n",
       "     'controlsAllAf': 0.000565,\n",
       "     'controlsAllAn': 31850,\n",
       "     'controlsAllAc': 18},\n",
       "    'topmed': {'allAf': 0.001091,\n",
       "     'allAn': 125568,\n",
       "     'allAc': 137,\n",
       "     'allHc': 0,\n",
       "     'failedFilter': True},\n",
       "    'phyloPPrimateScore': 0.074}]})"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = parser.get_variants_above_gnomad_freq(frequency_key=\"allAf\", frequency_threshold_high=0.1)\n",
    "len(positions), positions[0].model_dump()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(302,\n",
       " {'chromosome': 'chr21',\n",
       "  'position': 5227548,\n",
       "  'refAllele': 'C',\n",
       "  'altAlleles': ['T'],\n",
       "  'filters': ['PASS'],\n",
       "  'mappingQuality': 93.13,\n",
       "  'cytogeneticBand': '21p12',\n",
       "  'vcfInfo': {'DP': '314'},\n",
       "  'samples': [{'genotype': '0|0',\n",
       "    'variantFrequencies': [0.125],\n",
       "    'totalDepth': 40,\n",
       "    'alleleDepths': [35, 5],\n",
       "    'somaticQuality': 0,\n",
       "    'vcfSampleInfo': {'F1R2': '16,4', 'F2R1': '19,1'}},\n",
       "   {'genotype': '0|1',\n",
       "    'variantFrequencies': [0.3432],\n",
       "    'totalDepth': 169,\n",
       "    'alleleDepths': [111, 58],\n",
       "    'somaticQuality': 28.1,\n",
       "    'vcfSampleInfo': {'F1R2': '60,29', 'F2R1': '51,29'}}],\n",
       "  'variants': [{'vid': '21-5227548-C-T',\n",
       "    'chromosome': 'chr21',\n",
       "    'begin': 5227548,\n",
       "    'end': 5227548,\n",
       "    'refAllele': 'C',\n",
       "    'altAllele': 'T',\n",
       "    'variantType': 'SNV',\n",
       "    'hgvsg': 'NC_000021.9:g.5227548C>T',\n",
       "    'phylopScore': 0.2,\n",
       "    'inLowComplexityRegion': True,\n",
       "    'dbsnp': ['rs1219135472'],\n",
       "    'gnomad': {'coverage': 15,\n",
       "     'allAf': 0.348842,\n",
       "     'allAn': 65554,\n",
       "     'allAc': 22868,\n",
       "     'allHc': 4287,\n",
       "     'afrAf': 0.195536,\n",
       "     'afrAn': 16130,\n",
       "     'afrAc': 3154,\n",
       "     'afrHc': 157,\n",
       "     'amrAf': 0.38877,\n",
       "     'amrAn': 5628,\n",
       "     'amrAc': 2188,\n",
       "     'amrHc': 454,\n",
       "     'easAf': 0.258838,\n",
       "     'easAn': 2376,\n",
       "     'easAc': 615,\n",
       "     'easHc': 84,\n",
       "     'finAf': 0.511703,\n",
       "     'finAn': 2478,\n",
       "     'finAc': 1268,\n",
       "     'finHc': 372,\n",
       "     'nfeAf': 0.406416,\n",
       "     'nfeAn': 33980,\n",
       "     'nfeAc': 13810,\n",
       "     'nfeHc': 2911,\n",
       "     'asjAf': 0.357451,\n",
       "     'asjAn': 1852,\n",
       "     'asjAc': 662,\n",
       "     'asjHc': 100,\n",
       "     'sasAf': 0.371722,\n",
       "     'sasAn': 1754,\n",
       "     'sasAc': 652,\n",
       "     'sasHc': 107,\n",
       "     'othAf': 0.363636,\n",
       "     'othAn': 880,\n",
       "     'othAc': 320,\n",
       "     'othHc': 60,\n",
       "     'maleAf': 0.348586,\n",
       "     'maleAn': 30136,\n",
       "     'maleAc': 10505,\n",
       "     'maleHc': 2007,\n",
       "     'femaleAf': 0.34906,\n",
       "     'femaleAn': 35418,\n",
       "     'femaleAc': 12363,\n",
       "     'femaleHc': 2280,\n",
       "     'controlsAllAf': 0.338529,\n",
       "     'controlsAllAn': 11804,\n",
       "     'controlsAllAc': 3996},\n",
       "    'topmed': {'allAf': 0.499148,\n",
       "     'allAn': 125568,\n",
       "     'allAc': 62677,\n",
       "     'allHc': 0,\n",
       "     'failedFilter': True},\n",
       "    'phyloPPrimateScore': 0}]})"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = parser.get_variants_above_gnomad_freq(frequency_key=\"allAf\", frequency_threshold_low=0.1)\n",
    "len(positions), positions[0].model_dump()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(114,\n",
       " {'chromosome': 'chr21',\n",
       "  'position': 5278225,\n",
       "  'refAllele': 'A',\n",
       "  'altAlleles': ['T'],\n",
       "  'filters': ['alt_allele_in_normal',\n",
       "   'filtered_reads',\n",
       "   'mapping_quality',\n",
       "   'non_homref_normal',\n",
       "   'weak_evidence'],\n",
       "  'mappingQuality': 12.91,\n",
       "  'cytogeneticBand': '21p12',\n",
       "  'vcfInfo': {'DP': '113'},\n",
       "  'samples': [{'genotype': '0/0',\n",
       "    'variantFrequencies': [0.25],\n",
       "    'totalDepth': 12,\n",
       "    'alleleDepths': [9, 3],\n",
       "    'somaticQuality': 0.5,\n",
       "    'vcfSampleInfo': {'F1R2': '5,2', 'F2R1': '4,1'}},\n",
       "   {'genotype': '0/1',\n",
       "    'variantFrequencies': [0.7931],\n",
       "    'totalDepth': 29,\n",
       "    'alleleDepths': [6, 23],\n",
       "    'somaticQuality': 7.5,\n",
       "    'vcfSampleInfo': {'F1R2': '4,9', 'F2R1': '2,14'}}],\n",
       "  'variants': [{'vid': '21-5278225-A-T',\n",
       "    'chromosome': 'chr21',\n",
       "    'begin': 5278225,\n",
       "    'end': 5278225,\n",
       "    'refAllele': 'A',\n",
       "    'altAllele': 'T',\n",
       "    'variantType': 'SNV',\n",
       "    'hgvsg': 'NC_000021.9:g.5278225A>T',\n",
       "    'dbsnp': ['rs1171728286'],\n",
       "    'gnomad': {'coverage': 7,\n",
       "     'failedFilter': True,\n",
       "     'allAf': 0.148668,\n",
       "     'allAn': 46856,\n",
       "     'allAc': 6966,\n",
       "     'allHc': 80,\n",
       "     'afrAf': 0.181794,\n",
       "     'afrAn': 26244,\n",
       "     'afrAc': 4771,\n",
       "     'afrHc': 46,\n",
       "     'amrAf': 0.110039,\n",
       "     'amrAn': 2590,\n",
       "     'amrAc': 285,\n",
       "     'amrHc': 5,\n",
       "     'easAf': 0.000608,\n",
       "     'easAn': 1644,\n",
       "     'easAc': 1,\n",
       "     'easHc': 0,\n",
       "     'finAf': 0.050104,\n",
       "     'finAn': 958,\n",
       "     'finAc': 48,\n",
       "     'finHc': 0,\n",
       "     'nfeAf': 0.122751,\n",
       "     'nfeAn': 12448,\n",
       "     'nfeAc': 1528,\n",
       "     'nfeHc': 26,\n",
       "     'asjAf': 0.225636,\n",
       "     'asjAn': 944,\n",
       "     'asjAc': 213,\n",
       "     'asjHc': 2,\n",
       "     'sasAf': 0.022059,\n",
       "     'sasAn': 1224,\n",
       "     'sasAc': 27,\n",
       "     'sasHc': 0,\n",
       "     'othAf': 0.116838,\n",
       "     'othAn': 582,\n",
       "     'othAc': 68,\n",
       "     'othHc': 0,\n",
       "     'maleAf': 0.145535,\n",
       "     'maleAn': 22084,\n",
       "     'maleAc': 3214,\n",
       "     'maleHc': 34,\n",
       "     'femaleAf': 0.151461,\n",
       "     'femaleAn': 24772,\n",
       "     'femaleAc': 3752,\n",
       "     'femaleHc': 46,\n",
       "     'controlsAllAf': 0.128554,\n",
       "     'controlsAllAn': 9848,\n",
       "     'controlsAllAc': 1266},\n",
       "    'topmed': {'allAf': 0.084209,\n",
       "     'allAn': 125568,\n",
       "     'allAc': 10574,\n",
       "     'allHc': 0,\n",
       "     'failedFilter': True}}]})"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = parser.get_variants_above_gnomad_freq(frequency_key=\"allAf\", frequency_threshold_low=0.1, frequency_threshold_high=0.2)\n",
    "len(positions), positions[0].model_dump()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene Filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>clingenDosageSensitivityMap</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AATBC</td>\n",
       "      <td>51526.0</td>\n",
       "      <td>284837</td>\n",
       "      <td>ENSG00000215458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000291052</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>16022.0</td>\n",
       "      <td>150000</td>\n",
       "      <td>ENSG00000243064</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCG1</td>\n",
       "      <td>73.0</td>\n",
       "      <td>9619</td>\n",
       "      <td>ENSG00000160179</td>\n",
       "      <td>[{'mimNumber': 603076, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>{'pLi': 0.112, 'pRec': 0.888, 'pNull': 9.34e-0...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ADARB1</td>\n",
       "      <td>226.0</td>\n",
       "      <td>104</td>\n",
       "      <td>ENSG00000197381</td>\n",
       "      <td>[{'mimNumber': 601218, 'geneName': 'Adenosine ...</td>\n",
       "      <td>{'pLi': 0.803, 'pRec': 0.197, 'pNull': 8.32e-0...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>YBEY</td>\n",
       "      <td>1299.0</td>\n",
       "      <td>54059</td>\n",
       "      <td>ENSG00000182362</td>\n",
       "      <td>[{'mimNumber': 617461, 'geneName': 'YBEY metal...</td>\n",
       "      <td>{'pLi': 1.34e-07, 'pRec': 0.0625, 'pNull': 0.9...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>YRDCP3</td>\n",
       "      <td>39921.0</td>\n",
       "      <td>100861429</td>\n",
       "      <td>ENSG00000230859</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>ZBTB21</td>\n",
       "      <td>13083.0</td>\n",
       "      <td>49854</td>\n",
       "      <td>ENSG00000173276</td>\n",
       "      <td>[{'mimNumber': 616485, 'geneName': 'Zinc finge...</td>\n",
       "      <td>{'pLi': 0.998, 'pRec': 0.00222, 'pNull': 5.45e...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>ZNF295-AS1</td>\n",
       "      <td>23130.0</td>\n",
       "      <td>150142</td>\n",
       "      <td>ENSG00000237232</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>ZNF355P</td>\n",
       "      <td>17218.0</td>\n",
       "      <td>100505852</td>\n",
       "      <td>ENSG00000168122</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>442 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "0         AATBC  51526.0     284837  ENSG00000215458  \\\n",
       "1        ABCC13      NaN        NaN  ENSG00000291052   \n",
       "2        ABCC13  16022.0     150000  ENSG00000243064   \n",
       "3         ABCG1     73.0       9619  ENSG00000160179   \n",
       "4        ADARB1    226.0        104  ENSG00000197381   \n",
       "..          ...      ...        ...              ...   \n",
       "437        YBEY   1299.0      54059  ENSG00000182362   \n",
       "438      YRDCP3  39921.0  100861429  ENSG00000230859   \n",
       "439      ZBTB21  13083.0      49854  ENSG00000173276   \n",
       "440  ZNF295-AS1  23130.0     150142  ENSG00000237232   \n",
       "441     ZNF355P  17218.0  100505852  ENSG00000168122   \n",
       "\n",
       "                                                  omim   \n",
       "0                                                  NaN  \\\n",
       "1    [{'mimNumber': 608835, 'geneName': 'ATP-bindin...   \n",
       "2    [{'mimNumber': 608835, 'geneName': 'ATP-bindin...   \n",
       "3    [{'mimNumber': 603076, 'geneName': 'ATP-bindin...   \n",
       "4    [{'mimNumber': 601218, 'geneName': 'Adenosine ...   \n",
       "..                                                 ...   \n",
       "437  [{'mimNumber': 617461, 'geneName': 'YBEY metal...   \n",
       "438                                                NaN   \n",
       "439  [{'mimNumber': 616485, 'geneName': 'Zinc finge...   \n",
       "440                                                NaN   \n",
       "441                                                NaN   \n",
       "\n",
       "                                                gnomAD   \n",
       "0                                                  NaN  \\\n",
       "1                                                  NaN   \n",
       "2                                                  NaN   \n",
       "3    {'pLi': 0.112, 'pRec': 0.888, 'pNull': 9.34e-0...   \n",
       "4    {'pLi': 0.803, 'pRec': 0.197, 'pNull': 8.32e-0...   \n",
       "..                                                 ...   \n",
       "437  {'pLi': 1.34e-07, 'pRec': 0.0625, 'pNull': 0.9...   \n",
       "438                                                NaN   \n",
       "439  {'pLi': 0.998, 'pRec': 0.00222, 'pNull': 5.45e...   \n",
       "440                                                NaN   \n",
       "441                                                NaN   \n",
       "\n",
       "    clingenDosageSensitivityMap clingenGeneValidity cosmic  \n",
       "0                           NaN                 NaN    NaN  \n",
       "1                           NaN                 NaN    NaN  \n",
       "2                           NaN                 NaN    NaN  \n",
       "3                           NaN                 NaN    NaN  \n",
       "4                           NaN                 NaN    NaN  \n",
       "..                          ...                 ...    ...  \n",
       "437                         NaN                 NaN    NaN  \n",
       "438                         NaN                 NaN    NaN  \n",
       "439                         NaN                 NaN    NaN  \n",
       "440                         NaN                 NaN    NaN  \n",
       "441                         NaN                 NaN    NaN  \n",
       "\n",
       "[442 rows x 9 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>clingenDosageSensitivityMap</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000291052</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>16022.0</td>\n",
       "      <td>150000</td>\n",
       "      <td>ENSG00000243064</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "1  ABCC13      NaN        NaN  ENSG00000291052  \\\n",
       "2  ABCC13  16022.0     150000  ENSG00000243064   \n",
       "\n",
       "                                                omim gnomAD   \n",
       "1  [{'mimNumber': 608835, 'geneName': 'ATP-bindin...    NaN  \\\n",
       "2  [{'mimNumber': 608835, 'geneName': 'ATP-bindin...    NaN   \n",
       "\n",
       "  clingenDosageSensitivityMap clingenGeneValidity cosmic  \n",
       "1                         NaN                 NaN    NaN  \n",
       "2                         NaN                 NaN    NaN  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.genes[annotated_data.genes[\"name\"] == \"ABCC13\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>clingenDosageSensitivityMap</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>16022.0</td>\n",
       "      <td>150000</td>\n",
       "      <td>ENSG00000243064</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>ZBTB21</td>\n",
       "      <td>13083.0</td>\n",
       "      <td>49854</td>\n",
       "      <td>ENSG00000173276</td>\n",
       "      <td>[{'mimNumber': 616485, 'geneName': 'Zinc finge...</td>\n",
       "      <td>{'pLi': 0.998, 'pRec': 0.00222, 'pNull': 5.45e...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "2    ABCC13  16022.0     150000  ENSG00000243064  \\\n",
       "439  ZBTB21  13083.0      49854  ENSG00000173276   \n",
       "\n",
       "                                                  omim   \n",
       "2    [{'mimNumber': 608835, 'geneName': 'ATP-bindin...  \\\n",
       "439  [{'mimNumber': 616485, 'geneName': 'Zinc finge...   \n",
       "\n",
       "                                                gnomAD   \n",
       "2                                                  NaN  \\\n",
       "439  {'pLi': 0.998, 'pRec': 0.00222, 'pNull': 5.45e...   \n",
       "\n",
       "    clingenDosageSensitivityMap clingenGeneValidity cosmic  \n",
       "2                           NaN                 NaN    NaN  \n",
       "439                         NaN                 NaN    NaN  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.genes[annotated_data.genes[\"ensemblGeneId\"].isin([\"ENSG00000173276\", \"ENSG00000243064\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>clingenDosageSensitivityMap</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>ERG</td>\n",
       "      <td>3446.0</td>\n",
       "      <td>2078</td>\n",
       "      <td>ENSG00000157554</td>\n",
       "      <td>[{'mimNumber': 165080, 'geneName': 'ETS transc...</td>\n",
       "      <td>{'pLi': 0.964, 'pRec': 0.0359, 'pNull': 5.03e-...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{'roleInCancer': ['oncogene', 'fusion']}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>TMPRSS2</td>\n",
       "      <td>11876.0</td>\n",
       "      <td>7113</td>\n",
       "      <td>ENSG00000184012</td>\n",
       "      <td>[{'mimNumber': 602060, 'geneName': 'Transmembr...</td>\n",
       "      <td>{'pLi': 2.4e-10, 'pRec': 0.88, 'pNull': 0.12, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{'roleInCancer': ['fusion']}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "198      ERG   3446.0       2078  ENSG00000157554  \\\n",
       "418  TMPRSS2  11876.0       7113  ENSG00000184012   \n",
       "\n",
       "                                                  omim   \n",
       "198  [{'mimNumber': 165080, 'geneName': 'ETS transc...  \\\n",
       "418  [{'mimNumber': 602060, 'geneName': 'Transmembr...   \n",
       "\n",
       "                                                gnomAD   \n",
       "198  {'pLi': 0.964, 'pRec': 0.0359, 'pNull': 5.03e-...  \\\n",
       "418  {'pLi': 2.4e-10, 'pRec': 0.88, 'pNull': 0.12, ...   \n",
       "\n",
       "    clingenDosageSensitivityMap clingenGeneValidity   \n",
       "198                         NaN                 NaN  \\\n",
       "418                         NaN                 NaN   \n",
       "\n",
       "                                       cosmic  \n",
       "198  {'roleInCancer': ['oncogene', 'fusion']}  \n",
       "418              {'roleInCancer': ['fusion']}  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.genes.dropna(subset=[\"cosmic\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
