{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Parse Nirvana JSON output in python\n",
    "\n",
    "Nirvana outputs a single JSON annotation file for a single input VCF file. The output file contains a single [JSON object](https://www.w3schools.com/js/js_json_objects.asp) to represent the annotations of all input VCF variants. The JSON file format can be found in the [documentation](https://illumina.github.io/NirvanaDocumentation/file-formats/nirvana-json-file-format).\n",
    "\n",
    "A [sample JSON file](https://github.com/Illumina/NirvanaDocumentation/blob/master/static/files/ceph_trio_test.json.gz) of the CEPH trio (NA12878, NA12891, and NA12892) can be downloaded from the [NirvanaDocumention](https://github.com/Illumina/NirvanaDocumentation/tree/master/static/files) git repo.\n",
    "\n",
    "This notebook demonstrates how you can parse this sample JSON file in python and retrieves the annotation data.\n",
    "\n",
    "## Read Nirvana JSON output by lines\n",
    "\n",
    "Even though the Nirvana JSON output is a single JSON object, different JSON object fields are written in different lines for memory efficient reading.\n",
    "\n",
    "The first line in the JSON file is the `header` line:\n",
    "\n",
    "```json\n",
    "{\"header\":{\"annotator\":\"Nirvana\n",
    "...\n",
    ",\"positions\":[\n",
    "```\n",
    "Followed by the `positions` lines:\n",
    "\n",
    "```json\n",
    "{\"chromosome\":\"chr21\",\"position\":9975027,\"refAllele\":\"C\",\"altAlleles\":[\"G\"],\"quality\":102.47,\"filters\":[\"PASS\"],\"fisherStrandBias\":0.727,\"mappingQuality\":43.11,\"cytogeneticBand\":\"21p11.2\",\"samples\":,\n",
    "...\n",
    "```\n",
    "\n",
    "After the `positions` lines, there are the `genes` lines, which are optional if there is no overlapping gene of the input VCF variants (3 lines for 2 genes in the example):\n",
    "\n",
    "```json\n",
    "],\"genes\":[\n",
    "{\"name\":\"ABCC13\",\"omim\":[{\"mimNumber\":608835,\"geneName\":\"ATP-binding cassette, subfamily C, member 13\",\"description\":\"ABCC13 belongs to a large family of ATP-binding cassette (ABC) transporters that play important roles as membrane transporters or ion channel modulators. However, ABCC13 is a truncated protein that lacks critical ATP-binding motifs and is unlikely to be a functional transporter (Yabuuchi et al., 2002).\"}]},\n",
    "```\n",
    "\n",
    "Finally, the last line of the JSON file are two brackets to complete the JSON object structure:\n",
    "\n",
    "```json\n",
    "]}\n",
    "```\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Install requirements"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 364,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pydantic in /usr/local/lib/python3.11/site-packages (2.5.2)\n",
      "Requirement already satisfied: ijson in /usr/local/lib/python3.11/site-packages (3.2.3)\n",
      "Collecting polars\n",
      "  Downloading polars-0.19.19-cp38-abi3-macosx_10_12_x86_64.whl (26.6 MB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m26.6/26.6 MB\u001b[0m \u001b[31m22.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0mm eta \u001b[36m0:00:01\u001b[0m[36m0:00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: annotated-types>=0.4.0 in /usr/local/lib/python3.11/site-packages (from pydantic) (0.6.0)\n",
      "Requirement already satisfied: pydantic-core==2.14.5 in /usr/local/lib/python3.11/site-packages (from pydantic) (2.14.5)\n",
      "Requirement already satisfied: typing-extensions>=4.6.1 in /usr/local/lib/python3.11/site-packages (from pydantic) (4.8.0)\n",
      "Installing collected packages: polars\n",
      "Successfully installed polars-0.19.19\n",
      "\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip available: \u001b[0m\u001b[31;49m22.3.1\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m23.3.1\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpython3.11 -m pip install --upgrade pip\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "# !pip3 install pydantic ijson polars\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 389,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "import gzip\n",
    "import ijson\n",
    "import functools\n",
    "\n",
    "from typing import Any, Dict, List, Optional\n",
    "\n",
    "\n",
    "import polars as pd\n",
    "import pydantic\n",
    "\n",
    "# pd.set_option('display.max_columns', None)\n",
    "pd.Config(tbl_rows=50)\n",
    "\n",
    "class AnnotatedData:\n",
    "    def __init__(self, filename: str):\n",
    "        self._filename = filename\n",
    "        \n",
    "        \n",
    "        for key in (\"annotator\", \"genomeAssembly\", \"creationTime\"):\n",
    "            print(f\"{key}: {self.header[key]}\")     \n",
    "        \n",
    "    @property\n",
    "    def header(self) -> dict:\n",
    "        with gzip.open(self._filename, 'r') as f:\n",
    "            return next(ijson.items(f, \"header\"))\n",
    "    \n",
    "    @property\n",
    "    def data_sources(self):\n",
    "        return pd.DataFrame(self.header[\"dataSources\"]).sort(by=\"name\")\n",
    "        \n",
    "    @property\n",
    "    def genes(self) -> pd.DataFrame:\n",
    "        with gzip.open(self._filename, 'r') as f:\n",
    "            # return pd.json_normalize(ijson.items(f, \"genes.item\"))\n",
    "            return pd.DataFrame(ijson.items(f, \"genes.item\"))\n",
    "        \n",
    "    @property\n",
    "    def positions(self) -> list:\n",
    "        f =  gzip.open(self._filename, 'r')\n",
    "        return ijson.items(f, \"positions.item\")\n",
    "\n",
    "\n",
    "    def get_annotation(self, chromosome: str, position: int) -> dict:\n",
    "        \n",
    "        return next(\n",
    "            (\n",
    "                position_item for position_item in self.positions\n",
    "                if chromosome == position_item.get(\"chromosome\") and \\\n",
    "                position == position_item.get(\"position\")\n",
    "            ),\n",
    "            None\n",
    "        )\n",
    "    \n",
    "    @staticmethod\n",
    "    def multiple_to_df(items: list, key: str = \"\"):\n",
    "        return pd.concat((item.to_df(key) for item in items))\n",
    "    \n",
    "class BaseClass(pydantic.BaseModel):\n",
    "    @staticmethod\n",
    "    def _get_nested_value(dictionary, keys):\n",
    "        keys = keys.split('.')\n",
    "        for key in keys:\n",
    "            print(dictionary)\n",
    "            if isinstance(dictionary, dict):\n",
    "                dictionary = dictionary.get(key)\n",
    "            else:\n",
    "                return None\n",
    "        return dictionary\n",
    "        \n",
    "    def get_top_level(self):\n",
    "        return pd.DataFrame(self.get_top_level_dict()).unnest()\n",
    "    \n",
    "    def get_top_level_dict(self):\n",
    "        raise NotImplementedError\n",
    "    \n",
    "    def to_df(self, key: str = \"\") -> pd.DataFrame:\n",
    "        if not key:\n",
    "            return pd.DataFrame(self.model_dump()).unnest()\n",
    "                        \n",
    "        # values = self._get_nested_value(self.model_dump(), key)\n",
    "        values = self.model_dump().get(key)\n",
    "        \n",
    "        if isinstance(values, list):\n",
    "            merged = [self.get_top_level_dict() | value for value in values]\n",
    "        else:\n",
    "            merged = [self.get_top_level_dict() | {key: values}]\n",
    "        \n",
    "        # return pd.json_normalize(merged)\n",
    "        return pd.DataFrame(merged).unnest()\n",
    "    \n",
    "class Transcript(BaseClass):\n",
    "    model_config = pydantic.ConfigDict(extra='allow') \n",
    "    transcript: str\n",
    "    source: str\n",
    "    bioType: Optional[str]\n",
    "    geneId: Optional[str]\n",
    "    hgnc: Optional[str]\n",
    "    consequence: Optional[List[str]]\n",
    "    impact: Optional[str]\n",
    "    isCanonical: Optional[bool]\n",
    "    \n",
    "    def get_top_level_dict(self):\n",
    "        return dict(zip(\n",
    "            (\"transcript\", \"isCanonical\"),\n",
    "            (self.transcript, self.isCanonical)\n",
    "        ))\n",
    "    \n",
    "    \n",
    "class Variant(BaseClass):\n",
    "    model_config = pydantic.ConfigDict(extra='allow') \n",
    "    vid: str\n",
    "    chromosome: str\n",
    "    begin: int\n",
    "    end: int\n",
    "    refAllele: str\n",
    "    altAllele: str\n",
    "    variantType: Optional[str]\n",
    "    hgvsg: Optional[str]\n",
    "    phylopScore: Optional[float]\n",
    "    phyloPPrimateScore: Optional[float]\n",
    "    transcripts: Optional[List[Transcript]] = None\n",
    "        \n",
    "    def get_top_level_dict(self):\n",
    "        return dict(zip(\n",
    "            (\"chromosome\", \"begin\", \"end\", \"refAllele\", \"altAllele\", \"hgvsg\"),\n",
    "            (self.chromosome, self.begin, self.end, self.refAllele, self.altAllele, self.hgvsg)\n",
    "        ))\n",
    "    \n",
    "    \n",
    "class Position(BaseClass):\n",
    "    model_config = pydantic.ConfigDict(extra='allow') \n",
    "    chromosome: str\n",
    "    position: int\n",
    "    refAllele: str\n",
    "    altAlleles: List[str]\n",
    "    filters: List[str]\n",
    "    mappingQuality: float\n",
    "    cytogeneticBand: str\n",
    "    vcfInfo: Dict[str, Any]\n",
    "    samples: List\n",
    "    variants: List[Variant]\n",
    "    \n",
    "    def get_top_level_dict(self):\n",
    "        return dict(zip(\n",
    "            (\"chromosome\", \"position\", \"refAllele\", \"altAlleles\", \"filters\", \"mappingQuality\", \"cyatogeneticBand\", \"vcfInfo\"),\n",
    "            (self.chromosome, self.position, self.refAllele, self.altAlleles, self.filters, self.mappingQuality, self.cytogeneticBand, self.vcfInfo)\n",
    "        ))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Header"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 390,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotator: Illumina Connected Annotations 3.22.0\n",
      "genomeAssembly: GRCh38\n",
      "creationTime: 2023-12-07 14:15:54\n"
     ]
    }
   ],
   "source": [
    "filename = \"annotated_38.json.gz\"\n",
    "annotated_data = AnnotatedData(filename=filename)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 391,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><style>\n",
       ".dataframe > thead > tr,\n",
       ".dataframe > tbody > tr {\n",
       "  text-align: right;\n",
       "  white-space: pre-wrap;\n",
       "}\n",
       "</style>\n",
       "<small>shape: (32, 4)</small><table border=\"1\" class=\"dataframe\"><thead><tr><th>name</th><th>version</th><th>description</th><th>releaseDate</th></tr><tr><td>str</td><td>str</td><td>str</td><td>str</td></tr></thead><tbody><tr><td>&quot;1000 Genomes P…</td><td>&quot;Phase 3 v3plus…</td><td>&quot;A public catal…</td><td>&quot;2013-05-27&quot;</td></tr><tr><td>&quot;1000 Genomes P…</td><td>&quot;Phase 3 v5a&quot;</td><td>&quot;A public catal…</td><td>&quot;2013-05-27&quot;</td></tr><tr><td>&quot;COSMIC&quot;</td><td>&quot;96&quot;</td><td>&quot;resource for e…</td><td>&quot;2022-05-31&quot;</td></tr><tr><td>&quot;COSMIC gene fu…</td><td>&quot;96&quot;</td><td>&quot;manually curat…</td><td>&quot;2023-11-07&quot;</td></tr><tr><td>&quot;CancerHotspots…</td><td>&quot;2017&quot;</td><td>&quot;A resouce for …</td><td>&quot;2017-01-01&quot;</td></tr><tr><td>&quot;ClinGen&quot;</td><td>&quot;20160414&quot;</td><td>null</td><td>&quot;2016-04-14&quot;</td></tr><tr><td>&quot;ClinGen Dosage…</td><td>&quot;20231105&quot;</td><td>&quot;Dosage sensiti…</td><td>&quot;2023-11-05&quot;</td></tr><tr><td>&quot;ClinGen diseas…</td><td>&quot;20231105&quot;</td><td>&quot;Disease validi…</td><td>&quot;2023-11-05&quot;</td></tr><tr><td>&quot;ClinVar&quot;</td><td>&quot;20231028&quot;</td><td>&quot;A freely acces…</td><td>&quot;2023-11-05&quot;</td></tr><tr><td>&quot;Cosmic Cancer …</td><td>&quot;97&quot;</td><td>&quot;Cosmic Cancer …</td><td>&quot;2022-11-29&quot;</td></tr><tr><td>&quot;DANN&quot;</td><td>&quot;20200205&quot;</td><td>&quot;Pathogenicity …</td><td>&quot;2020-02-05&quot;</td></tr><tr><td>&quot;DECIPHER&quot;</td><td>&quot;201509&quot;</td><td>&quot;Common copy-nu…</td><td>&quot;2015-09-01&quot;</td></tr><tr><td>&quot;Ensembl&quot;</td><td>&quot;110&quot;</td><td>&quot;Ensembl GRCh38…</td><td>&quot;2023-04-27&quot;</td></tr><tr><td>&quot;FusionCatcher&quot;</td><td>&quot;1.33&quot;</td><td>&quot;known germline…</td><td>&quot;2020-12-22&quot;</td></tr><tr><td>&quot;GME&quot;</td><td>&quot;20160618&quot;</td><td>&quot;Greater Middle…</td><td>&quot;2016-06-18&quot;</td></tr><tr><td>&quot;Gerp&quot;</td><td>&quot;20110522&quot;</td><td>&quot;GERP identifie…</td><td>&quot;2011-05-22&quot;</td></tr><tr><td>&quot;MITOMAP&quot;</td><td>&quot;20200819&quot;</td><td>&quot;Small variants…</td><td>&quot;2020-08-19&quot;</td></tr><tr><td>&quot;MITOMAP_SV&quot;</td><td>&quot;20200819&quot;</td><td>&quot;Large structur…</td><td>&quot;2020-08-19&quot;</td></tr><tr><td>&quot;MitochondrialH…</td><td>&quot;20180410&quot;</td><td>&quot;Variant read f…</td><td>&quot;2020-05-21&quot;</td></tr><tr><td>&quot;OMIM&quot;</td><td>&quot;20231105&quot;</td><td>&quot;An Online Cata…</td><td>&quot;2023-11-05&quot;</td></tr><tr><td>&quot;PhyloPPrimate&quot;</td><td>&quot;1.0&quot;</td><td>&quot;PhyloPPrimate …</td><td>&quot;2023-11-29&quot;</td></tr><tr><td>&quot;PrimateAI-3D&quot;</td><td>&quot;1.0&quot;</td><td>&quot;PrimateAI-3D i…</td><td>&quot;2023-06-02&quot;</td></tr><tr><td>&quot;REVEL&quot;</td><td>&quot;20200205&quot;</td><td>&quot;Pathogenicity …</td><td>&quot;2020-02-05&quot;</td></tr><tr><td>&quot;RefSeq&quot;</td><td>&quot;GCF_000001405.…</td><td>&quot;NCBI Homo sapi…</td><td>&quot;2023-03-21&quot;</td></tr><tr><td>&quot;SpliceAI&quot;</td><td>&quot;1.3&quot;</td><td>&quot;SpliceAI score…</td><td>&quot;2019-10-04&quot;</td></tr><tr><td>&quot;TOPMed&quot;</td><td>&quot;freeze_5&quot;</td><td>&quot;Allele frequen…</td><td>&quot;2017-08-28&quot;</td></tr><tr><td>&quot;dbSNP&quot;</td><td>&quot;156&quot;</td><td>&quot;Identifiers fo…</td><td>&quot;2023-09-28&quot;</td></tr><tr><td>&quot;gnomAD&quot;</td><td>&quot;3.1.2&quot;</td><td>&quot;Allele frequen…</td><td>&quot;2021-10-22&quot;</td></tr><tr><td>&quot;gnomAD_LCR&quot;</td><td>&quot;2.1&quot;</td><td>&quot;Low complexity…</td><td>&quot;2019-04-10&quot;</td></tr><tr><td>&quot;gnomAD_SV&quot;</td><td>&quot;2.1&quot;</td><td>&quot;A structural v…</td><td>&quot;2020-08-20&quot;</td></tr><tr><td>&quot;gnomAD_gene_sc…</td><td>&quot;2.1&quot;</td><td>&quot;Gene Loss of f…</td><td>&quot;2018-10-17&quot;</td></tr><tr><td>&quot;phyloP&quot;</td><td>&quot;hg38&quot;</td><td>&quot;20 way conserv…</td><td>&quot;2015-04-17&quot;</td></tr></tbody></table></div>"
      ],
      "text/plain": [
       "shape: (32, 4)\n",
       "┌────────────────────────────┬───────────────────────────┬───────────────────────────┬─────────────┐\n",
       "│ name                       ┆ version                   ┆ description               ┆ releaseDate │\n",
       "│ ---                        ┆ ---                       ┆ ---                       ┆ ---         │\n",
       "│ str                        ┆ str                       ┆ str                       ┆ str         │\n",
       "╞════════════════════════════╪═══════════════════════════╪═══════════════════════════╪═════════════╡\n",
       "│ 1000 Genomes Project       ┆ Phase 3 v3plus            ┆ A public catalogue of     ┆ 2013-05-27  │\n",
       "│                            ┆                           ┆ human vari…               ┆             │\n",
       "│ 1000 Genomes Project (SV)  ┆ Phase 3 v5a               ┆ A public catalogue of     ┆ 2013-05-27  │\n",
       "│                            ┆                           ┆ human vari…               ┆             │\n",
       "│ COSMIC                     ┆ 96                        ┆ resource for exploring    ┆ 2022-05-31  │\n",
       "│                            ┆                           ┆ the impac…                ┆             │\n",
       "│ COSMIC gene fusions        ┆ 96                        ┆ manually curated somatic  ┆ 2023-11-07  │\n",
       "│                            ┆                           ┆ gene fu…                  ┆             │\n",
       "│ CancerHotspots             ┆ 2017                      ┆ A resouce for             ┆ 2017-01-01  │\n",
       "│                            ┆                           ┆ statistically sign…       ┆             │\n",
       "│ ClinGen                    ┆ 20160414                  ┆ null                      ┆ 2016-04-14  │\n",
       "│ ClinGen Dosage Sensitivity ┆ 20231105                  ┆ Dosage sensitivity map    ┆ 2023-11-05  │\n",
       "│ Map                        ┆                           ┆ from Clin…                ┆             │\n",
       "│ ClinGen disease validity   ┆ 20231105                  ┆ Disease validity          ┆ 2023-11-05  │\n",
       "│ curatio…                   ┆                           ┆ curations from …          ┆             │\n",
       "│ ClinVar                    ┆ 20231028                  ┆ A freely accessible,      ┆ 2023-11-05  │\n",
       "│                            ┆                           ┆ public arch…              ┆             │\n",
       "│ Cosmic Cancer Gene Census  ┆ 97                        ┆ Cosmic Cancer Gene Census ┆ 2022-11-29  │\n",
       "│                            ┆                           ┆ catalo…                   ┆             │\n",
       "│ DANN                       ┆ 20200205                  ┆ Pathogenicity scores of   ┆ 2020-02-05  │\n",
       "│                            ┆                           ┆ missense…                 ┆             │\n",
       "│ DECIPHER                   ┆ 201509                    ┆ Common copy-number        ┆ 2015-09-01  │\n",
       "│                            ┆                           ┆ variants and …            ┆             │\n",
       "│ Ensembl                    ┆ 110                       ┆ Ensembl GRCh38 Release    ┆ 2023-04-27  │\n",
       "│ FusionCatcher              ┆ 1.33                      ┆ known germline and        ┆ 2020-12-22  │\n",
       "│                            ┆                           ┆ somatic gene …            ┆             │\n",
       "│ GME                        ┆ 20160618                  ┆ Greater Middle Eastern    ┆ 2016-06-18  │\n",
       "│                            ┆                           ┆ Variome a…                ┆             │\n",
       "│ Gerp                       ┆ 20110522                  ┆ GERP identifies           ┆ 2011-05-22  │\n",
       "│                            ┆                           ┆ constrained elem…         ┆             │\n",
       "│ MITOMAP                    ┆ 20200819                  ┆ Small variants in the     ┆ 2020-08-19  │\n",
       "│                            ┆                           ┆ MITOMAP hu…               ┆             │\n",
       "│ MITOMAP_SV                 ┆ 20200819                  ┆ Large structural variants ┆ 2020-08-19  │\n",
       "│                            ┆                           ┆ in the…                   ┆             │\n",
       "│ MitochondrialHeteroplasmy  ┆ 20180410                  ┆ Variant read frequency    ┆ 2020-05-21  │\n",
       "│                            ┆                           ┆ percentil…                ┆             │\n",
       "│ OMIM                       ┆ 20231105                  ┆ An Online Catalog of      ┆ 2023-11-05  │\n",
       "│                            ┆                           ┆ Human Genes…              ┆             │\n",
       "│ PhyloPPrimate              ┆ 1.0                       ┆ PhyloPPrimate identifies  ┆ 2023-11-29  │\n",
       "│                            ┆                           ┆ primate…                  ┆             │\n",
       "│ PrimateAI-3D               ┆ 1.0                       ┆ PrimateAI-3D is a         ┆ 2023-06-02  │\n",
       "│                            ┆                           ┆ deep-learning …           ┆             │\n",
       "│ REVEL                      ┆ 20200205                  ┆ Pathogenicity scores of   ┆ 2020-02-05  │\n",
       "│                            ┆                           ┆ missense…                 ┆             │\n",
       "│ RefSeq                     ┆ GCF_000001405.40-RS_2023_ ┆ NCBI Homo sapiens         ┆ 2023-03-21  │\n",
       "│                            ┆ 03                        ┆ Annotation Rel…           ┆             │\n",
       "│ SpliceAI                   ┆ 1.3                       ┆ SpliceAI scores,          ┆ 2019-10-04  │\n",
       "│                            ┆                           ┆ distances, etc.           ┆             │\n",
       "│ TOPMed                     ┆ freeze_5                  ┆ Allele frequencies from   ┆ 2017-08-28  │\n",
       "│                            ┆                           ┆ TOPMed d…                 ┆             │\n",
       "│ dbSNP                      ┆ 156                       ┆ Identifiers for observed  ┆ 2023-09-28  │\n",
       "│                            ┆                           ┆ variant…                  ┆             │\n",
       "│ gnomAD                     ┆ 3.1.2                     ┆ Allele frequencies from   ┆ 2021-10-22  │\n",
       "│                            ┆                           ┆ Genome A…                 ┆             │\n",
       "│ gnomAD_LCR                 ┆ 2.1                       ┆ Low complexity regions    ┆ 2019-04-10  │\n",
       "│                            ┆                           ┆ from gnom…                ┆             │\n",
       "│ gnomAD_SV                  ┆ 2.1                       ┆ A structural variation    ┆ 2020-08-20  │\n",
       "│                            ┆                           ┆ reference…                ┆             │\n",
       "│ gnomAD_gene_scores         ┆ 2.1                       ┆ Gene Loss of function     ┆ 2018-10-17  │\n",
       "│                            ┆                           ┆ scores fro…               ┆             │\n",
       "│ phyloP                     ┆ hg38                      ┆ 20 way conservation score ┆ 2015-04-17  │\n",
       "│                            ┆                           ┆ betwee…                   ┆             │\n",
       "└────────────────────────────┴───────────────────────────┴───────────────────────────┴─────────────┘"
      ]
     },
     "execution_count": 391,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.data_sources"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 392,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "thread '<unnamed>' panicked at crates/polars-core/src/frame/row/av_buffer.rs:249:85:\n",
      "called `Result::unwrap()` on an `Err` value: ComputeError(ErrString(\"unable to losslessly convert any-value of scale 60 to scale 3\"))\n"
     ]
    },
    {
     "ename": "PanicException",
     "evalue": "called `Result::unwrap()` on an `Err` value: ComputeError(ErrString(\"unable to losslessly convert any-value of scale 60 to scale 3\"))",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mPanicException\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[392], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mannotated_data\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mgenes\u001b[49m\n",
      "Cell \u001b[0;32mIn[389], line 35\u001b[0m, in \u001b[0;36mAnnotatedData.genes\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     31\u001b[0m \u001b[38;5;129m@property\u001b[39m\n\u001b[1;32m     32\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mgenes\u001b[39m(\u001b[38;5;28mself\u001b[39m) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame:\n\u001b[1;32m     33\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m gzip\u001b[38;5;241m.\u001b[39mopen(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_filename, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124m'\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m f:\n\u001b[1;32m     34\u001b[0m         \u001b[38;5;66;03m# return pd.json_normalize(ijson.items(f, \"genes.item\"))\u001b[39;00m\n\u001b[0;32m---> 35\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mDataFrame\u001b[49m\u001b[43m(\u001b[49m\u001b[43mijson\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mitems\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mgenes.item\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/site-packages/polars/dataframe/frame.py:400\u001b[0m, in \u001b[0;36mDataFrame.__init__\u001b[0;34m(self, data, schema, schema_overrides, orient, infer_schema_length, nan_to_null)\u001b[0m\n\u001b[1;32m    395\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_df \u001b[38;5;241m=\u001b[39m pandas_to_pydf(\n\u001b[1;32m    396\u001b[0m         data, schema\u001b[38;5;241m=\u001b[39mschema, schema_overrides\u001b[38;5;241m=\u001b[39mschema_overrides\n\u001b[1;32m    397\u001b[0m     )\n\u001b[1;32m    399\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(data, Sized) \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(data, (Generator, Iterable)):\n\u001b[0;32m--> 400\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_df \u001b[38;5;241m=\u001b[39m \u001b[43miterable_to_pydf\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    401\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    402\u001b[0m \u001b[43m        \u001b[49m\u001b[43mschema\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mschema\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    403\u001b[0m \u001b[43m        \u001b[49m\u001b[43mschema_overrides\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mschema_overrides\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    404\u001b[0m \u001b[43m        \u001b[49m\u001b[43morient\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43morient\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    405\u001b[0m \u001b[43m        \u001b[49m\u001b[43minfer_schema_length\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43minfer_schema_length\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    406\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    407\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    408\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m(\n\u001b[1;32m    409\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mDataFrame constructor called with unsupported type \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mtype\u001b[39m(data)\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__name__\u001b[39m\u001b[38;5;132;01m!r}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    410\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m for the `data` parameter\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    411\u001b[0m     )\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/site-packages/polars/utils/_construction.py:1664\u001b[0m, in \u001b[0;36miterable_to_pydf\u001b[0;34m(data, schema, schema_overrides, orient, chunk_size, infer_schema_length)\u001b[0m\n\u001b[1;32m   1662\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m values:\n\u001b[1;32m   1663\u001b[0m     \u001b[38;5;28;01mbreak\u001b[39;00m\n\u001b[0;32m-> 1664\u001b[0m frame_chunk \u001b[38;5;241m=\u001b[39m \u001b[43mto_frame_chunk\u001b[49m\u001b[43m(\u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43moriginal_schema\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1665\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m df \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   1666\u001b[0m     df \u001b[38;5;241m=\u001b[39m frame_chunk\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/site-packages/polars/utils/_construction.py:1638\u001b[0m, in \u001b[0;36miterable_to_pydf.<locals>.to_frame_chunk\u001b[0;34m(values, schema)\u001b[0m\n\u001b[1;32m   1637\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mto_frame_chunk\u001b[39m(values: \u001b[38;5;28mlist\u001b[39m[Any], schema: SchemaDefinition \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m DataFrame:\n\u001b[0;32m-> 1638\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mpl\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mDataFrame\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1639\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdata\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mvalues\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1640\u001b[0m \u001b[43m        \u001b[49m\u001b[43mschema\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mschema\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1641\u001b[0m \u001b[43m        \u001b[49m\u001b[43morient\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mrow\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1642\u001b[0m \u001b[43m        \u001b[49m\u001b[43minfer_schema_length\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43minfer_schema_length\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1643\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/site-packages/polars/dataframe/frame.py:368\u001b[0m, in \u001b[0;36mDataFrame.__init__\u001b[0;34m(self, data, schema, schema_overrides, orient, infer_schema_length, nan_to_null)\u001b[0m\n\u001b[1;32m    360\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_df \u001b[38;5;241m=\u001b[39m dict_to_pydf(\n\u001b[1;32m    361\u001b[0m         data,\n\u001b[1;32m    362\u001b[0m         schema\u001b[38;5;241m=\u001b[39mschema,\n\u001b[1;32m    363\u001b[0m         schema_overrides\u001b[38;5;241m=\u001b[39mschema_overrides,\n\u001b[1;32m    364\u001b[0m         nan_to_null\u001b[38;5;241m=\u001b[39mnan_to_null,\n\u001b[1;32m    365\u001b[0m     )\n\u001b[1;32m    367\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(data, (\u001b[38;5;28mlist\u001b[39m, \u001b[38;5;28mtuple\u001b[39m, Sequence)):\n\u001b[0;32m--> 368\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_df \u001b[38;5;241m=\u001b[39m \u001b[43msequence_to_pydf\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    369\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    370\u001b[0m \u001b[43m        \u001b[49m\u001b[43mschema\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mschema\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    371\u001b[0m \u001b[43m        \u001b[49m\u001b[43mschema_overrides\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mschema_overrides\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    372\u001b[0m \u001b[43m        \u001b[49m\u001b[43morient\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43morient\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    373\u001b[0m \u001b[43m        \u001b[49m\u001b[43minfer_schema_length\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43minfer_schema_length\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    374\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    375\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(data, pl\u001b[38;5;241m.\u001b[39mSeries):\n\u001b[1;32m    376\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_df \u001b[38;5;241m=\u001b[39m series_to_pydf(\n\u001b[1;32m    377\u001b[0m         data, schema\u001b[38;5;241m=\u001b[39mschema, schema_overrides\u001b[38;5;241m=\u001b[39mschema_overrides\n\u001b[1;32m    378\u001b[0m     )\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/site-packages/polars/utils/_construction.py:940\u001b[0m, in \u001b[0;36msequence_to_pydf\u001b[0;34m(data, schema, schema_overrides, orient, infer_schema_length)\u001b[0m\n\u001b[1;32m    937\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(data) \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m    938\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m dict_to_pydf({}, schema\u001b[38;5;241m=\u001b[39mschema, schema_overrides\u001b[38;5;241m=\u001b[39mschema_overrides)\n\u001b[0;32m--> 940\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_sequence_to_pydf_dispatcher\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    941\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdata\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    942\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdata\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    943\u001b[0m \u001b[43m    \u001b[49m\u001b[43mschema\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mschema\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    944\u001b[0m \u001b[43m    \u001b[49m\u001b[43mschema_overrides\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mschema_overrides\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    945\u001b[0m \u001b[43m    \u001b[49m\u001b[43morient\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43morient\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    946\u001b[0m \u001b[43m    \u001b[49m\u001b[43minfer_schema_length\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43minfer_schema_length\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    947\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/functools.py:909\u001b[0m, in \u001b[0;36msingledispatch.<locals>.wrapper\u001b[0;34m(*args, **kw)\u001b[0m\n\u001b[1;32m    905\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m args:\n\u001b[1;32m    906\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mfuncname\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m requires at least \u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m    907\u001b[0m                     \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m1 positional argument\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m--> 909\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mdispatch\u001b[49m\u001b[43m(\u001b[49m\u001b[43margs\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[38;5;18;43m__class__\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkw\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/site-packages/polars/utils/_construction.py:1159\u001b[0m, in \u001b[0;36m_sequence_of_dict_to_pydf\u001b[0;34m(first_element, data, schema, schema_overrides, infer_schema_length, **kwargs)\u001b[0m\n\u001b[1;32m   1151\u001b[0m column_names, schema_overrides \u001b[38;5;241m=\u001b[39m _unpack_schema(\n\u001b[1;32m   1152\u001b[0m     schema, schema_overrides\u001b[38;5;241m=\u001b[39mschema_overrides\n\u001b[1;32m   1153\u001b[0m )\n\u001b[1;32m   1154\u001b[0m dicts_schema \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m   1155\u001b[0m     include_unknowns(schema_overrides, column_names \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mlist\u001b[39m(schema_overrides))\n\u001b[1;32m   1156\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m column_names\n\u001b[1;32m   1157\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m   1158\u001b[0m )\n\u001b[0;32m-> 1159\u001b[0m pydf \u001b[38;5;241m=\u001b[39m \u001b[43mPyDataFrame\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_dicts\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1160\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43minfer_schema_length\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdicts_schema\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mschema_overrides\u001b[49m\n\u001b[1;32m   1161\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1163\u001b[0m \u001b[38;5;66;03m# TODO: we can remove this `schema_overrides` block completely\u001b[39;00m\n\u001b[1;32m   1164\u001b[0m \u001b[38;5;66;03m#  once https://github.com/pola-rs/polars/issues/11044 is fixed\u001b[39;00m\n\u001b[1;32m   1165\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m schema_overrides:\n",
      "\u001b[0;31mPanicException\u001b[0m: called `Result::unwrap()` on an `Err` value: ComputeError(ErrString(\"unable to losslessly convert any-value of scale 60 to scale 3\"))"
     ]
    }
   ],
   "source": [
    "annotated_data.genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Positions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cytogeneticBand</th>\n",
       "      <th>samples</th>\n",
       "      <th>variants</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5228221</td>\n",
       "      <td>G</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[PASS]</td>\n",
       "      <td>75.75</td>\n",
       "      <td>21p12</td>\n",
       "      <td>[{'genotype': '0/0', 'variantFrequencies': [0....</td>\n",
       "      <td>[{'vid': '21-5228221-G-T', 'chromosome': 'chr2...</td>\n",
       "      <td>1004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles filters  mappingQuality   \n",
       "0      chr21   5228221         G        [T]  [PASS]           75.75  \\\n",
       "\n",
       "  cytogeneticBand                                            samples   \n",
       "0           21p12  [{'genotype': '0/0', 'variantFrequencies': [0....  \\\n",
       "\n",
       "                                            variants vcfInfo.DP  \n",
       "0  [{'vid': '21-5228221-G-T', 'chromosome': 'chr2...       1004  "
      ]
     },
     "execution_count": 354,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "position = Position.model_validate(annotated_data.get_annotation(\"chr21\", 5228221))\n",
    "position.to_df()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 355,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>vid</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>phylopScore</th>\n",
       "      <th>phyloPPrimateScore</th>\n",
       "      <th>transcripts</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5228221</td>\n",
       "      <td>G</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[PASS]</td>\n",
       "      <td>75.75</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5228221-G-T</td>\n",
       "      <td>5228221</td>\n",
       "      <td>5228221</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5228221G&gt;T</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[{'transcript': 'ENST00000623753.1', 'source':...</td>\n",
       "      <td>1004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles filters  mappingQuality   \n",
       "0      chr21   5228221         G        [T]  [PASS]           75.75  \\\n",
       "\n",
       "  cyatogeneticBand             vid    begin      end altAllele variantType   \n",
       "0            21p12  21-5228221-G-T  5228221  5228221         T         SNV  \\\n",
       "\n",
       "                      hgvsg  phylopScore  phyloPPrimateScore   \n",
       "0  NC_000021.9:g.5228221G>T          0.2                 0.0  \\\n",
       "\n",
       "                                         transcripts vcfInfo.DP  \n",
       "0  [{'transcript': 'ENST00000623753.1', 'source':...       1004  "
      ]
     },
     "execution_count": 355,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position.to_df(key=\"variants\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>transcript</th>\n",
       "      <th>source</th>\n",
       "      <th>bioType</th>\n",
       "      <th>geneId</th>\n",
       "      <th>hgnc</th>\n",
       "      <th>consequence</th>\n",
       "      <th>impact</th>\n",
       "      <th>isCanonical</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5228221</td>\n",
       "      <td>5228221</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000021.9:g.5228221G&gt;T</td>\n",
       "      <td>ENST00000623753.1</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENSG00000279669</td>\n",
       "      <td>ENSG00000279669</td>\n",
       "      <td>[downstream_gene_variant]</td>\n",
       "      <td>modifier</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome    begin      end refAllele altAllele                     hgvsg   \n",
       "0      chr21  5228221  5228221         G         T  NC_000021.9:g.5228221G>T  \\\n",
       "\n",
       "          transcript   source bioType           geneId             hgnc   \n",
       "0  ENST00000623753.1  Ensembl  lncRNA  ENSG00000279669  ENSG00000279669  \\\n",
       "\n",
       "                 consequence    impact  isCanonical  \n",
       "0  [downstream_gene_variant]  modifier         True  "
      ]
     },
     "execution_count": 356,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position.variants[0].to_df(\"transcripts\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 357,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>transcript</th>\n",
       "      <th>source</th>\n",
       "      <th>bioType</th>\n",
       "      <th>geneId</th>\n",
       "      <th>hgnc</th>\n",
       "      <th>consequence</th>\n",
       "      <th>impact</th>\n",
       "      <th>isCanonical</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5228221</td>\n",
       "      <td>5228221</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000021.9:g.5228221G&gt;T</td>\n",
       "      <td>ENST00000623753.1</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENSG00000279669</td>\n",
       "      <td>ENSG00000279669</td>\n",
       "      <td>[downstream_gene_variant]</td>\n",
       "      <td>modifier</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome    begin      end refAllele altAllele                     hgvsg   \n",
       "0      chr21  5228221  5228221         G         T  NC_000021.9:g.5228221G>T  \\\n",
       "\n",
       "          transcript   source bioType           geneId             hgnc   \n",
       "0  ENST00000623753.1  Ensembl  lncRNA  ENSG00000279669  ENSG00000279669  \\\n",
       "\n",
       "                 consequence    impact  isCanonical  \n",
       "0  [downstream_gene_variant]  modifier         True  "
      ]
     },
     "execution_count": 357,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AnnotatedData.multiple_to_df(position.variants, key=\"transcripts\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 358,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5222289</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[mapping_quality, weak_evidence]</td>\n",
       "      <td>49.7</td>\n",
       "      <td>21p12</td>\n",
       "      <td>749</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles                           filters   \n",
       "0      chr21   5222289         C        [T]  [mapping_quality, weak_evidence]  \\\n",
       "\n",
       "   mappingQuality cyatogeneticBand vcfInfo.DP  \n",
       "0            49.7            21p12        749  "
      ]
     },
     "execution_count": 358,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position = Position.model_validate(annotated_data.get_annotation(\"chr21\", 5222289))\n",
    "position.get_top_level()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cytogeneticBand</th>\n",
       "      <th>samples</th>\n",
       "      <th>variants</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5222289</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[mapping_quality, weak_evidence]</td>\n",
       "      <td>49.7</td>\n",
       "      <td>21p12</td>\n",
       "      <td>[{'genotype': '0/0', 'variantFrequencies': [0....</td>\n",
       "      <td>[{'vid': '21-5222289-C-T', 'chromosome': 'chr2...</td>\n",
       "      <td>749</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles                           filters   \n",
       "0      chr21   5222289         C        [T]  [mapping_quality, weak_evidence]  \\\n",
       "\n",
       "   mappingQuality cytogeneticBand   \n",
       "0            49.7           21p12  \\\n",
       "\n",
       "                                             samples   \n",
       "0  [{'genotype': '0/0', 'variantFrequencies': [0....  \\\n",
       "\n",
       "                                            variants vcfInfo.DP  \n",
       "0  [{'vid': '21-5222289-C-T', 'chromosome': 'chr2...        749  "
      ]
     },
     "execution_count": 359,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position.to_df()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 360,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>cytogeneticBand</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5222289</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[mapping_quality, weak_evidence]</td>\n",
       "      <td>49.7</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21p12</td>\n",
       "      <td>749</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles                           filters   \n",
       "0      chr21   5222289         C        [T]  [mapping_quality, weak_evidence]  \\\n",
       "\n",
       "   mappingQuality cyatogeneticBand cytogeneticBand vcfInfo.DP  \n",
       "0            49.7            21p12           21p12        749  "
      ]
     },
     "execution_count": 360,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position.to_df(\"cytogeneticBand\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 361,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>vid</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>phylopScore</th>\n",
       "      <th>phyloPPrimateScore</th>\n",
       "      <th>transcripts</th>\n",
       "      <th>dbsnp</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "      <th>gnomad.coverage</th>\n",
       "      <th>gnomad.allAf</th>\n",
       "      <th>gnomad.allAn</th>\n",
       "      <th>gnomad.allAc</th>\n",
       "      <th>gnomad.allHc</th>\n",
       "      <th>gnomad.afrAf</th>\n",
       "      <th>gnomad.afrAn</th>\n",
       "      <th>gnomad.afrAc</th>\n",
       "      <th>gnomad.afrHc</th>\n",
       "      <th>gnomad.amrAf</th>\n",
       "      <th>gnomad.amrAn</th>\n",
       "      <th>gnomad.amrAc</th>\n",
       "      <th>gnomad.amrHc</th>\n",
       "      <th>gnomad.easAf</th>\n",
       "      <th>gnomad.easAn</th>\n",
       "      <th>gnomad.easAc</th>\n",
       "      <th>gnomad.easHc</th>\n",
       "      <th>gnomad.finAf</th>\n",
       "      <th>gnomad.finAn</th>\n",
       "      <th>gnomad.finAc</th>\n",
       "      <th>gnomad.finHc</th>\n",
       "      <th>gnomad.nfeAf</th>\n",
       "      <th>gnomad.nfeAn</th>\n",
       "      <th>gnomad.nfeAc</th>\n",
       "      <th>gnomad.nfeHc</th>\n",
       "      <th>gnomad.asjAf</th>\n",
       "      <th>gnomad.asjAn</th>\n",
       "      <th>gnomad.asjAc</th>\n",
       "      <th>gnomad.asjHc</th>\n",
       "      <th>gnomad.sasAf</th>\n",
       "      <th>gnomad.sasAn</th>\n",
       "      <th>gnomad.sasAc</th>\n",
       "      <th>gnomad.sasHc</th>\n",
       "      <th>gnomad.othAf</th>\n",
       "      <th>gnomad.othAn</th>\n",
       "      <th>gnomad.othAc</th>\n",
       "      <th>gnomad.othHc</th>\n",
       "      <th>gnomad.maleAf</th>\n",
       "      <th>gnomad.maleAn</th>\n",
       "      <th>gnomad.maleAc</th>\n",
       "      <th>gnomad.maleHc</th>\n",
       "      <th>gnomad.femaleAf</th>\n",
       "      <th>gnomad.femaleAn</th>\n",
       "      <th>gnomad.femaleAc</th>\n",
       "      <th>gnomad.femaleHc</th>\n",
       "      <th>gnomad.controlsAllAf</th>\n",
       "      <th>gnomad.controlsAllAn</th>\n",
       "      <th>gnomad.controlsAllAc</th>\n",
       "      <th>topmed.allAf</th>\n",
       "      <th>topmed.allAn</th>\n",
       "      <th>topmed.allAc</th>\n",
       "      <th>topmed.allHc</th>\n",
       "      <th>topmed.failedFilter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5222289</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[mapping_quality, weak_evidence]</td>\n",
       "      <td>49.7</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5222289-C-T</td>\n",
       "      <td>5222289</td>\n",
       "      <td>5222289</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5222289C&gt;T</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.074</td>\n",
       "      <td>None</td>\n",
       "      <td>[rs1366179382]</td>\n",
       "      <td>749</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000502</td>\n",
       "      <td>149274</td>\n",
       "      <td>75</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001704</td>\n",
       "      <td>40502</td>\n",
       "      <td>69</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>14966</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5022</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10430</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000075</td>\n",
       "      <td>66950</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3438</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4734</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2024</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000411</td>\n",
       "      <td>73006</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00059</td>\n",
       "      <td>76268</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000565</td>\n",
       "      <td>31850</td>\n",
       "      <td>18</td>\n",
       "      <td>0.001091</td>\n",
       "      <td>125568</td>\n",
       "      <td>137</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles                           filters   \n",
       "0      chr21   5222289         C        [T]  [mapping_quality, weak_evidence]  \\\n",
       "\n",
       "   mappingQuality cyatogeneticBand             vid    begin      end   \n",
       "0            49.7            21p12  21-5222289-C-T  5222289  5222289  \\\n",
       "\n",
       "  altAllele variantType                     hgvsg  phylopScore   \n",
       "0         T         SNV  NC_000021.9:g.5222289C>T         -0.2  \\\n",
       "\n",
       "   phyloPPrimateScore transcripts           dbsnp vcfInfo.DP  gnomad.coverage   \n",
       "0               0.074        None  [rs1366179382]        749                3  \\\n",
       "\n",
       "  gnomad.allAf  gnomad.allAn  gnomad.allAc  gnomad.allHc gnomad.afrAf   \n",
       "0     0.000502        149274            75             0     0.001704  \\\n",
       "\n",
       "   gnomad.afrAn  gnomad.afrAc  gnomad.afrHc gnomad.amrAf  gnomad.amrAn   \n",
       "0         40502            69             0     0.000067         14966  \\\n",
       "\n",
       "   gnomad.amrAc  gnomad.amrHc  gnomad.easAf  gnomad.easAn  gnomad.easAc   \n",
       "0             1             0             0          5022             0  \\\n",
       "\n",
       "   gnomad.easHc  gnomad.finAf  gnomad.finAn  gnomad.finAc  gnomad.finHc   \n",
       "0             0             0         10430             0             0  \\\n",
       "\n",
       "  gnomad.nfeAf  gnomad.nfeAn  gnomad.nfeAc  gnomad.nfeHc  gnomad.asjAf   \n",
       "0     0.000075         66950             5             0             0  \\\n",
       "\n",
       "   gnomad.asjAn  gnomad.asjAc  gnomad.asjHc  gnomad.sasAf  gnomad.sasAn   \n",
       "0          3438             0             0             0          4734  \\\n",
       "\n",
       "   gnomad.sasAc  gnomad.sasHc  gnomad.othAf  gnomad.othAn  gnomad.othAc   \n",
       "0             0             0             0          2024             0  \\\n",
       "\n",
       "   gnomad.othHc gnomad.maleAf  gnomad.maleAn  gnomad.maleAc  gnomad.maleHc   \n",
       "0             0      0.000411          73006             30              0  \\\n",
       "\n",
       "  gnomad.femaleAf  gnomad.femaleAn  gnomad.femaleAc  gnomad.femaleHc   \n",
       "0         0.00059            76268               45                0  \\\n",
       "\n",
       "  gnomad.controlsAllAf  gnomad.controlsAllAn  gnomad.controlsAllAc   \n",
       "0             0.000565                 31850                    18  \\\n",
       "\n",
       "  topmed.allAf  topmed.allAn  topmed.allAc  topmed.allHc  topmed.failedFilter  \n",
       "0     0.001091        125568           137             0                 True  "
      ]
     },
     "execution_count": 361,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position.to_df(\"variants\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Parsing and Filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "metadata": {},
   "outputs": [],
   "source": [
    "class Parser:\n",
    "    def __init__(self, annotated_data: AnnotatedData):\n",
    "        self.annotated_data = annotated_data\n",
    "        \n",
    "    def get_variants_above_gnomad_freq(\n",
    "        self,\n",
    "        frequency_key: str,\n",
    "        frequency_threshold_low=float(\"-inf\"),\n",
    "        frequency_threshold_high=float(\"inf\")\n",
    "    ) -> list:\n",
    "        positions = [\n",
    "            Position.model_validate(position)\n",
    "            for position in self.annotated_data.positions\n",
    "            for variant in position.get(\"variants\", {})\n",
    "            if (freq := variant.get(\"gnomad\", {}).get(frequency_key, None)) \\\n",
    "            and frequency_threshold_low < freq < frequency_threshold_high\n",
    "        ]\n",
    "        return positions\n",
    "    \n",
    "    def get_positions_with_cannonical_transcripts(self):   \n",
    "        positions = [\n",
    "            Position.model_validate(position)\n",
    "            for position in self.annotated_data.positions\n",
    "            for variant in position.get(\"variants\", {})\n",
    "            for transcript in variant.get(\"transcripts\", [])\n",
    "            if transcript.get(\"isCanonical\")\n",
    "        ]\n",
    "\n",
    "        return positions\n",
    "    \n",
    "    def filter_transcripts_by_consequence(self, include=[], exclude=[]):\n",
    "        positions = [\n",
    "            Position.model_validate(position)\n",
    "            for position in self.annotated_data.positions\n",
    "            for variant in position.get(\"variants\", {})\n",
    "            for transcript in variant.get(\"transcripts\", [])\n",
    "            for consequence in transcript.get(\"consequence\", [])\n",
    "            if (not bool(include) or consequence in include) and consequence not in exclude\n",
    "        ]\n",
    "        return positions\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Positions with Cannonical Transcripts only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = Parser(annotated_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2472"
      ]
     },
     "execution_count": 243,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = parser.get_positions_with_cannonical_transcripts()\n",
    "len(list(positions))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>vid</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>phylopScore</th>\n",
       "      <th>phyloPPrimateScore</th>\n",
       "      <th>transcripts</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "      <th>regulatoryRegions</th>\n",
       "      <th>gnomad.coverage</th>\n",
       "      <th>gnomad.failedFilter</th>\n",
       "      <th>gnomad.allAf</th>\n",
       "      <th>gnomad.allAn</th>\n",
       "      <th>gnomad.allAc</th>\n",
       "      <th>gnomad.allHc</th>\n",
       "      <th>gnomad.afrAf</th>\n",
       "      <th>gnomad.afrAn</th>\n",
       "      <th>gnomad.afrAc</th>\n",
       "      <th>gnomad.afrHc</th>\n",
       "      <th>gnomad.amrAf</th>\n",
       "      <th>gnomad.amrAn</th>\n",
       "      <th>gnomad.amrAc</th>\n",
       "      <th>gnomad.amrHc</th>\n",
       "      <th>gnomad.easAf</th>\n",
       "      <th>gnomad.easAn</th>\n",
       "      <th>gnomad.easAc</th>\n",
       "      <th>gnomad.easHc</th>\n",
       "      <th>gnomad.finAf</th>\n",
       "      <th>gnomad.finAn</th>\n",
       "      <th>gnomad.finAc</th>\n",
       "      <th>gnomad.finHc</th>\n",
       "      <th>gnomad.nfeAf</th>\n",
       "      <th>gnomad.nfeAn</th>\n",
       "      <th>gnomad.nfeAc</th>\n",
       "      <th>gnomad.nfeHc</th>\n",
       "      <th>gnomad.asjAf</th>\n",
       "      <th>gnomad.asjAn</th>\n",
       "      <th>gnomad.asjAc</th>\n",
       "      <th>gnomad.asjHc</th>\n",
       "      <th>gnomad.sasAf</th>\n",
       "      <th>gnomad.sasAn</th>\n",
       "      <th>gnomad.sasAc</th>\n",
       "      <th>gnomad.sasHc</th>\n",
       "      <th>gnomad.othAf</th>\n",
       "      <th>gnomad.othAn</th>\n",
       "      <th>gnomad.othAc</th>\n",
       "      <th>gnomad.othHc</th>\n",
       "      <th>gnomad.maleAf</th>\n",
       "      <th>gnomad.maleAn</th>\n",
       "      <th>gnomad.maleAc</th>\n",
       "      <th>gnomad.maleHc</th>\n",
       "      <th>gnomad.femaleAf</th>\n",
       "      <th>gnomad.femaleAn</th>\n",
       "      <th>gnomad.femaleAc</th>\n",
       "      <th>gnomad.femaleHc</th>\n",
       "      <th>gnomad.controlsAllAf</th>\n",
       "      <th>gnomad.controlsAllAn</th>\n",
       "      <th>gnomad.controlsAllAc</th>\n",
       "      <th>dbsnp</th>\n",
       "      <th>topmed.allAf</th>\n",
       "      <th>topmed.allAn</th>\n",
       "      <th>topmed.allAc</th>\n",
       "      <th>topmed.allHc</th>\n",
       "      <th>topmed.failedFilter</th>\n",
       "      <th>inLowComplexityRegion</th>\n",
       "      <th>dannScore</th>\n",
       "      <th>oneKg.allAf</th>\n",
       "      <th>oneKg.afrAf</th>\n",
       "      <th>oneKg.amrAf</th>\n",
       "      <th>oneKg.easAf</th>\n",
       "      <th>oneKg.eurAf</th>\n",
       "      <th>oneKg.sasAf</th>\n",
       "      <th>oneKg.allAn</th>\n",
       "      <th>oneKg.afrAn</th>\n",
       "      <th>oneKg.amrAn</th>\n",
       "      <th>oneKg.easAn</th>\n",
       "      <th>oneKg.eurAn</th>\n",
       "      <th>oneKg.sasAn</th>\n",
       "      <th>oneKg.allAc</th>\n",
       "      <th>oneKg.afrAc</th>\n",
       "      <th>oneKg.amrAc</th>\n",
       "      <th>oneKg.easAc</th>\n",
       "      <th>oneKg.eurAc</th>\n",
       "      <th>oneKg.sasAc</th>\n",
       "      <th>cosmic</th>\n",
       "      <th>gerpScore</th>\n",
       "      <th>primateAI-3D</th>\n",
       "      <th>revel.score</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>spliceAI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5228221</td>\n",
       "      <td>G</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[PASS]</td>\n",
       "      <td>75.75</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5228221-G-T</td>\n",
       "      <td>5228221</td>\n",
       "      <td>5228221</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5228221G&gt;T</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0</td>\n",
       "      <td>[{'transcript': 'ENST00000623753.1', 'source':...</td>\n",
       "      <td>1004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5232869</td>\n",
       "      <td>T</td>\n",
       "      <td>[G]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>58.10</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5232869-T-G</td>\n",
       "      <td>5232869</td>\n",
       "      <td>5232869</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5232869T&gt;G</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'transcript': 'ENST00000623753.1', 'source':...</td>\n",
       "      <td>799</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5243664</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[mapping_quality, no_reliable_supporting_read,...</td>\n",
       "      <td>58.13</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5243664-C-T</td>\n",
       "      <td>5243664</td>\n",
       "      <td>5243664</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5243664C&gt;T</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.074</td>\n",
       "      <td>[{'transcript': 'ENST00000623753.1', 'source':...</td>\n",
       "      <td>644</td>\n",
       "      <td>[{'id': 'ENSR00000140073', 'type': 'TF_binding...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>152310.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>41488.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>15294.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>5208.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>10632.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68056.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3472.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4838.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2094.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>74416.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>77894.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>32928.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5243812</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[PASS]</td>\n",
       "      <td>108.19</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5243812-C-T</td>\n",
       "      <td>5243812</td>\n",
       "      <td>5243812</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5243812C&gt;T</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.074</td>\n",
       "      <td>[{'transcript': 'ENST00000623753.1', 'source':...</td>\n",
       "      <td>514</td>\n",
       "      <td>[{'id': 'ENSR00000140073', 'type': 'TF_binding...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>152296.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>41480.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>15294.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>5206.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>10628.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68058.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3472.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4836.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2094.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>74406.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>77890.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>32922.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>6367612</td>\n",
       "      <td>-</td>\n",
       "      <td>[CAATG]</td>\n",
       "      <td>[filtered_reads]</td>\n",
       "      <td>21.70</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-6367612-C-CAATG</td>\n",
       "      <td>6367613</td>\n",
       "      <td>6367612</td>\n",
       "      <td>AATG</td>\n",
       "      <td>insertion</td>\n",
       "      <td>NC_000021.9:g.6367616_6367619dup</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'transcript': 'ENST00000615262.1', 'source':...</td>\n",
       "      <td>375</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.139262</td>\n",
       "      <td>20860.0</td>\n",
       "      <td>2905.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065393</td>\n",
       "      <td>2722.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.110881</td>\n",
       "      <td>1930.0</td>\n",
       "      <td>214.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.108384</td>\n",
       "      <td>978.0</td>\n",
       "      <td>106.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.132432</td>\n",
       "      <td>370.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.172472</td>\n",
       "      <td>12518.0</td>\n",
       "      <td>2159.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.087379</td>\n",
       "      <td>824.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.059756</td>\n",
       "      <td>820.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.16</td>\n",
       "      <td>300.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.127778</td>\n",
       "      <td>8640.0</td>\n",
       "      <td>1104.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.147381</td>\n",
       "      <td>12220.0</td>\n",
       "      <td>1801.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.123314</td>\n",
       "      <td>5190.0</td>\n",
       "      <td>640.0</td>\n",
       "      <td>[rs1234944247]</td>\n",
       "      <td>0.078897</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>9907.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46597114</td>\n",
       "      <td>C</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[PASS]</td>\n",
       "      <td>219.26</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46597114-C-A</td>\n",
       "      <td>46597114</td>\n",
       "      <td>46597114</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46597114C&gt;A</td>\n",
       "      <td>0.5</td>\n",
       "      <td>-0.072</td>\n",
       "      <td>[{'transcript': 'ENST00000291700.9', 'source':...</td>\n",
       "      <td>143</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.75</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.635</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46597227</td>\n",
       "      <td>A</td>\n",
       "      <td>[C]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>217.92</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46597227-A-C</td>\n",
       "      <td>46597227</td>\n",
       "      <td>46597227</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46597227A&gt;C</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.781</td>\n",
       "      <td>[{'transcript': 'ENST00000291700.9', 'source':...</td>\n",
       "      <td>151</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.85</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.574</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46597227</td>\n",
       "      <td>A</td>\n",
       "      <td>[C]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>217.92</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46597227-A-C</td>\n",
       "      <td>46597227</td>\n",
       "      <td>46597227</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46597227A&gt;C</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.781</td>\n",
       "      <td>[{'transcript': 'ENST00000291700.9', 'source':...</td>\n",
       "      <td>151</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.85</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.574</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46643510</td>\n",
       "      <td>T</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>211.58</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46643510-T-A</td>\n",
       "      <td>46643510</td>\n",
       "      <td>46643510</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46643510T&gt;A</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.603</td>\n",
       "      <td>[{'transcript': 'ENST00000451211.6', 'source':...</td>\n",
       "      <td>150</td>\n",
       "      <td>[{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>141118.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>39060.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>14018.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000232</td>\n",
       "      <td>4306.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>7762.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>65358.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3364.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4110.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1968.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68068.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>73050.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000108</td>\n",
       "      <td>27846.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[rs1601921720]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.58</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.622</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46643510</td>\n",
       "      <td>T</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>211.58</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46643510-T-A</td>\n",
       "      <td>46643510</td>\n",
       "      <td>46643510</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46643510T&gt;A</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.603</td>\n",
       "      <td>[{'transcript': 'ENST00000451211.6', 'source':...</td>\n",
       "      <td>150</td>\n",
       "      <td>[{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>141118.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>39060.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>14018.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000232</td>\n",
       "      <td>4306.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>7762.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>65358.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3364.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4110.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1968.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68068.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>73050.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000108</td>\n",
       "      <td>27846.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[rs1601921720]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.58</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.622</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2472 rows × 99 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome  position refAllele altAlleles   \n",
       "0       chr21   5228221         G        [T]  \\\n",
       "0       chr21   5232869         T        [G]   \n",
       "0       chr21   5243664         C        [T]   \n",
       "0       chr21   5243812         C        [T]   \n",
       "0       chr21   6367612         -    [CAATG]   \n",
       "..        ...       ...       ...        ...   \n",
       "0       chr21  46597114         C        [A]   \n",
       "0       chr21  46597227         A        [C]   \n",
       "0       chr21  46597227         A        [C]   \n",
       "0       chr21  46643510         T        [A]   \n",
       "0       chr21  46643510         T        [A]   \n",
       "\n",
       "                                              filters  mappingQuality   \n",
       "0                                              [PASS]           75.75  \\\n",
       "0                       [base_quality, weak_evidence]           58.10   \n",
       "0   [mapping_quality, no_reliable_supporting_read,...           58.13   \n",
       "0                                              [PASS]          108.19   \n",
       "0                                    [filtered_reads]           21.70   \n",
       "..                                                ...             ...   \n",
       "0                                              [PASS]          219.26   \n",
       "0                       [base_quality, weak_evidence]          217.92   \n",
       "0                       [base_quality, weak_evidence]          217.92   \n",
       "0                       [base_quality, weak_evidence]          211.58   \n",
       "0                       [base_quality, weak_evidence]          211.58   \n",
       "\n",
       "   cyatogeneticBand                 vid     begin       end altAllele   \n",
       "0             21p12      21-5228221-G-T   5228221   5228221         T  \\\n",
       "0             21p12      21-5232869-T-G   5232869   5232869         G   \n",
       "0             21p12      21-5243664-C-T   5243664   5243664         T   \n",
       "0             21p12      21-5243812-C-T   5243812   5243812         T   \n",
       "0             21p12  21-6367612-C-CAATG   6367613   6367612      AATG   \n",
       "..              ...                 ...       ...       ...       ...   \n",
       "0           21q22.3     21-46597114-C-A  46597114  46597114         A   \n",
       "0           21q22.3     21-46597227-A-C  46597227  46597227         C   \n",
       "0           21q22.3     21-46597227-A-C  46597227  46597227         C   \n",
       "0           21q22.3     21-46643510-T-A  46643510  46643510         A   \n",
       "0           21q22.3     21-46643510-T-A  46643510  46643510         A   \n",
       "\n",
       "   variantType                             hgvsg phylopScore   \n",
       "0          SNV          NC_000021.9:g.5228221G>T         0.2  \\\n",
       "0          SNV          NC_000021.9:g.5232869T>G         0.5   \n",
       "0          SNV          NC_000021.9:g.5243664C>T         0.6   \n",
       "0          SNV          NC_000021.9:g.5243812C>T         0.4   \n",
       "0    insertion  NC_000021.9:g.6367616_6367619dup         NaN   \n",
       "..         ...                               ...         ...   \n",
       "0          SNV         NC_000021.9:g.46597114C>A         0.5   \n",
       "0          SNV         NC_000021.9:g.46597227A>C         0.6   \n",
       "0          SNV         NC_000021.9:g.46597227A>C         0.6   \n",
       "0          SNV         NC_000021.9:g.46643510T>A           0   \n",
       "0          SNV         NC_000021.9:g.46643510T>A           0   \n",
       "\n",
       "   phyloPPrimateScore                                        transcripts   \n",
       "0                   0  [{'transcript': 'ENST00000623753.1', 'source':...  \\\n",
       "0                 NaN  [{'transcript': 'ENST00000623753.1', 'source':...   \n",
       "0               0.074  [{'transcript': 'ENST00000623753.1', 'source':...   \n",
       "0               0.074  [{'transcript': 'ENST00000623753.1', 'source':...   \n",
       "0                 NaN  [{'transcript': 'ENST00000615262.1', 'source':...   \n",
       "..                ...                                                ...   \n",
       "0              -0.072  [{'transcript': 'ENST00000291700.9', 'source':...   \n",
       "0               0.781  [{'transcript': 'ENST00000291700.9', 'source':...   \n",
       "0               0.781  [{'transcript': 'ENST00000291700.9', 'source':...   \n",
       "0              -0.603  [{'transcript': 'ENST00000451211.6', 'source':...   \n",
       "0              -0.603  [{'transcript': 'ENST00000451211.6', 'source':...   \n",
       "\n",
       "   vcfInfo.DP                                  regulatoryRegions   \n",
       "0        1004                                                NaN  \\\n",
       "0         799                                                NaN   \n",
       "0         644  [{'id': 'ENSR00000140073', 'type': 'TF_binding...   \n",
       "0         514  [{'id': 'ENSR00000140073', 'type': 'TF_binding...   \n",
       "0         375                                                NaN   \n",
       "..        ...                                                ...   \n",
       "0         143                                                NaN   \n",
       "0         151                                                NaN   \n",
       "0         151                                                NaN   \n",
       "0         150  [{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...   \n",
       "0         150  [{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...   \n",
       "\n",
       "    gnomad.coverage gnomad.failedFilter gnomad.allAf  gnomad.allAn   \n",
       "0               NaN                 NaN          NaN           NaN  \\\n",
       "0               NaN                 NaN          NaN           NaN   \n",
       "0               0.0                True            0      152310.0   \n",
       "0               0.0                True            0      152296.0   \n",
       "0              11.0                True     0.139262       20860.0   \n",
       "..              ...                 ...          ...           ...   \n",
       "0               NaN                 NaN          NaN           NaN   \n",
       "0               NaN                 NaN          NaN           NaN   \n",
       "0               NaN                 NaN          NaN           NaN   \n",
       "0               0.0                True     0.000021      141118.0   \n",
       "0               0.0                True     0.000021      141118.0   \n",
       "\n",
       "    gnomad.allAc  gnomad.allHc gnomad.afrAf  gnomad.afrAn  gnomad.afrAc   \n",
       "0            NaN           NaN          NaN           NaN           NaN  \\\n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            0.0           0.0            0       41488.0           0.0   \n",
       "0            0.0           0.0            0       41480.0           0.0   \n",
       "0         2905.0           3.0     0.065393        2722.0         178.0   \n",
       "..           ...           ...          ...           ...           ...   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            3.0           0.0     0.000051       39060.0           2.0   \n",
       "0            3.0           0.0     0.000051       39060.0           2.0   \n",
       "\n",
       "    gnomad.afrHc gnomad.amrAf  gnomad.amrAn  gnomad.amrAc  gnomad.amrHc   \n",
       "0            NaN          NaN           NaN           NaN           NaN  \\\n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            0.0            0       15294.0           0.0           0.0   \n",
       "0            0.0            0       15294.0           0.0           0.0   \n",
       "0            0.0     0.110881        1930.0         214.0           1.0   \n",
       "..           ...          ...           ...           ...           ...   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            0.0            0       14018.0           0.0           0.0   \n",
       "0            0.0            0       14018.0           0.0           0.0   \n",
       "\n",
       "   gnomad.easAf  gnomad.easAn  gnomad.easAc  gnomad.easHc gnomad.finAf   \n",
       "0           NaN           NaN           NaN           NaN          NaN  \\\n",
       "0           NaN           NaN           NaN           NaN          NaN   \n",
       "0             0        5208.0           0.0           0.0            0   \n",
       "0             0        5206.0           0.0           0.0            0   \n",
       "0      0.108384         978.0         106.0           0.0     0.132432   \n",
       "..          ...           ...           ...           ...          ...   \n",
       "0           NaN           NaN           NaN           NaN          NaN   \n",
       "0           NaN           NaN           NaN           NaN          NaN   \n",
       "0           NaN           NaN           NaN           NaN          NaN   \n",
       "0      0.000232        4306.0           1.0           0.0            0   \n",
       "0      0.000232        4306.0           1.0           0.0            0   \n",
       "\n",
       "    gnomad.finAn  gnomad.finAc  gnomad.finHc gnomad.nfeAf  gnomad.nfeAn   \n",
       "0            NaN           NaN           NaN          NaN           NaN  \\\n",
       "0            NaN           NaN           NaN          NaN           NaN   \n",
       "0        10632.0           0.0           0.0            0       68056.0   \n",
       "0        10628.0           0.0           0.0            0       68058.0   \n",
       "0          370.0          49.0           0.0     0.172472       12518.0   \n",
       "..           ...           ...           ...          ...           ...   \n",
       "0            NaN           NaN           NaN          NaN           NaN   \n",
       "0            NaN           NaN           NaN          NaN           NaN   \n",
       "0            NaN           NaN           NaN          NaN           NaN   \n",
       "0         7762.0           0.0           0.0            0       65358.0   \n",
       "0         7762.0           0.0           0.0            0       65358.0   \n",
       "\n",
       "    gnomad.nfeAc  gnomad.nfeHc gnomad.asjAf  gnomad.asjAn  gnomad.asjAc   \n",
       "0            NaN           NaN          NaN           NaN           NaN  \\\n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            0.0           0.0            0        3472.0           0.0   \n",
       "0            0.0           0.0            0        3472.0           0.0   \n",
       "0         2159.0           2.0     0.087379         824.0          72.0   \n",
       "..           ...           ...          ...           ...           ...   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            0.0           0.0            0        3364.0           0.0   \n",
       "0            0.0           0.0            0        3364.0           0.0   \n",
       "\n",
       "    gnomad.asjHc gnomad.sasAf  gnomad.sasAn  gnomad.sasAc  gnomad.sasHc   \n",
       "0            NaN          NaN           NaN           NaN           NaN  \\\n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            0.0            0        4838.0           0.0           0.0   \n",
       "0            0.0            0        4836.0           0.0           0.0   \n",
       "0            0.0     0.059756         820.0          49.0           0.0   \n",
       "..           ...          ...           ...           ...           ...   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            0.0            0        4110.0           0.0           0.0   \n",
       "0            0.0            0        4110.0           0.0           0.0   \n",
       "\n",
       "   gnomad.othAf  gnomad.othAn  gnomad.othAc  gnomad.othHc gnomad.maleAf   \n",
       "0           NaN           NaN           NaN           NaN           NaN  \\\n",
       "0           NaN           NaN           NaN           NaN           NaN   \n",
       "0             0        2094.0           0.0           0.0             0   \n",
       "0             0        2094.0           0.0           0.0             0   \n",
       "0          0.16         300.0          48.0           0.0      0.127778   \n",
       "..          ...           ...           ...           ...           ...   \n",
       "0           NaN           NaN           NaN           NaN           NaN   \n",
       "0           NaN           NaN           NaN           NaN           NaN   \n",
       "0           NaN           NaN           NaN           NaN           NaN   \n",
       "0             0        1968.0           0.0           0.0             0   \n",
       "0             0        1968.0           0.0           0.0             0   \n",
       "\n",
       "    gnomad.maleAn  gnomad.maleAc  gnomad.maleHc gnomad.femaleAf   \n",
       "0             NaN            NaN            NaN             NaN  \\\n",
       "0             NaN            NaN            NaN             NaN   \n",
       "0         74416.0            0.0            0.0               0   \n",
       "0         74406.0            0.0            0.0               0   \n",
       "0          8640.0         1104.0            0.0        0.147381   \n",
       "..            ...            ...            ...             ...   \n",
       "0             NaN            NaN            NaN             NaN   \n",
       "0             NaN            NaN            NaN             NaN   \n",
       "0             NaN            NaN            NaN             NaN   \n",
       "0         68068.0            0.0            0.0        0.000041   \n",
       "0         68068.0            0.0            0.0        0.000041   \n",
       "\n",
       "    gnomad.femaleAn  gnomad.femaleAc  gnomad.femaleHc gnomad.controlsAllAf   \n",
       "0               NaN              NaN              NaN                  NaN  \\\n",
       "0               NaN              NaN              NaN                  NaN   \n",
       "0           77894.0              0.0              0.0                    0   \n",
       "0           77890.0              0.0              0.0                    0   \n",
       "0           12220.0           1801.0              3.0             0.123314   \n",
       "..              ...              ...              ...                  ...   \n",
       "0               NaN              NaN              NaN                  NaN   \n",
       "0               NaN              NaN              NaN                  NaN   \n",
       "0               NaN              NaN              NaN                  NaN   \n",
       "0           73050.0              3.0              0.0             0.000108   \n",
       "0           73050.0              3.0              0.0             0.000108   \n",
       "\n",
       "    gnomad.controlsAllAn  gnomad.controlsAllAc           dbsnp topmed.allAf   \n",
       "0                    NaN                   NaN             NaN          NaN  \\\n",
       "0                    NaN                   NaN             NaN          NaN   \n",
       "0                32928.0                   0.0             NaN          NaN   \n",
       "0                32922.0                   0.0             NaN          NaN   \n",
       "0                 5190.0                 640.0  [rs1234944247]     0.078897   \n",
       "..                   ...                   ...             ...          ...   \n",
       "0                    NaN                   NaN             NaN          NaN   \n",
       "0                    NaN                   NaN             NaN          NaN   \n",
       "0                    NaN                   NaN             NaN          NaN   \n",
       "0                27846.0                   3.0  [rs1601921720]          NaN   \n",
       "0                27846.0                   3.0  [rs1601921720]          NaN   \n",
       "\n",
       "    topmed.allAn  topmed.allAc  topmed.allHc topmed.failedFilter   \n",
       "0            NaN           NaN           NaN                 NaN  \\\n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0       125568.0        9907.0           0.0                True   \n",
       "..           ...           ...           ...                 ...   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "\n",
       "   inLowComplexityRegion dannScore oneKg.allAf oneKg.afrAf oneKg.amrAf   \n",
       "0                    NaN       NaN         NaN         NaN         NaN  \\\n",
       "0                    NaN       NaN         NaN         NaN         NaN   \n",
       "0                    NaN       NaN         NaN         NaN         NaN   \n",
       "0                    NaN       NaN         NaN         NaN         NaN   \n",
       "0                    NaN       NaN         NaN         NaN         NaN   \n",
       "..                   ...       ...         ...         ...         ...   \n",
       "0                    NaN      0.75         NaN         NaN         NaN   \n",
       "0                    NaN      0.85         NaN         NaN         NaN   \n",
       "0                    NaN      0.85         NaN         NaN         NaN   \n",
       "0                    NaN      0.58         NaN         NaN         NaN   \n",
       "0                    NaN      0.58         NaN         NaN         NaN   \n",
       "\n",
       "   oneKg.easAf oneKg.eurAf oneKg.sasAf  oneKg.allAn  oneKg.afrAn  oneKg.amrAn   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN  \\\n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "..         ...         ...         ...          ...          ...          ...   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "\n",
       "    oneKg.easAn  oneKg.eurAn  oneKg.sasAn  oneKg.allAc  oneKg.afrAc   \n",
       "0           NaN          NaN          NaN          NaN          NaN  \\\n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "..          ...          ...          ...          ...          ...   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "\n",
       "    oneKg.amrAc  oneKg.easAc  oneKg.eurAc  oneKg.sasAc cosmic gerpScore   \n",
       "0           NaN          NaN          NaN          NaN    NaN       NaN  \\\n",
       "0           NaN          NaN          NaN          NaN    NaN       NaN   \n",
       "0           NaN          NaN          NaN          NaN    NaN       NaN   \n",
       "0           NaN          NaN          NaN          NaN    NaN       NaN   \n",
       "0           NaN          NaN          NaN          NaN    NaN       NaN   \n",
       "..          ...          ...          ...          ...    ...       ...   \n",
       "0           NaN          NaN          NaN          NaN    NaN    -0.635   \n",
       "0           NaN          NaN          NaN          NaN    NaN    -0.574   \n",
       "0           NaN          NaN          NaN          NaN    NaN    -0.574   \n",
       "0           NaN          NaN          NaN          NaN    NaN     0.622   \n",
       "0           NaN          NaN          NaN          NaN    NaN     0.622   \n",
       "\n",
       "   primateAI-3D revel.score clinvar spliceAI  \n",
       "0           NaN         NaN     NaN      NaN  \n",
       "0           NaN         NaN     NaN      NaN  \n",
       "0           NaN         NaN     NaN      NaN  \n",
       "0           NaN         NaN     NaN      NaN  \n",
       "0           NaN         NaN     NaN      NaN  \n",
       "..          ...         ...     ...      ...  \n",
       "0           NaN         NaN     NaN      NaN  \n",
       "0           NaN         NaN     NaN      NaN  \n",
       "0           NaN         NaN     NaN      NaN  \n",
       "0           NaN         NaN     NaN      NaN  \n",
       "0           NaN         NaN     NaN      NaN  \n",
       "\n",
       "[2472 rows x 99 columns]"
      ]
     },
     "execution_count": 244,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Position.positions_to_df(positions, \"variants\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>variants.transcripts</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5228221</td>\n",
       "      <td>G</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[PASS]</td>\n",
       "      <td>75.75</td>\n",
       "      <td>21p12</td>\n",
       "      <td>None</td>\n",
       "      <td>1004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles filters  mappingQuality   \n",
       "0      chr21   5228221         G        [T]  [PASS]           75.75  \\\n",
       "\n",
       "  cyatogeneticBand variants.transcripts vcfInfo.DP  \n",
       "0            21p12                 None       1004  "
      ]
     },
     "execution_count": 245,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions[0].to_df(\"variants.transcripts\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'transcript': 'ENST00000623753.1',\n",
       "  'source': 'Ensembl',\n",
       "  'bioType': 'lncRNA',\n",
       "  'geneId': 'ENSG00000279669',\n",
       "  'hgnc': 'ENSG00000279669',\n",
       "  'consequence': ['downstream_gene_variant'],\n",
       "  'impact': 'modifier',\n",
       "  'isCanonical': True}]"
      ]
     },
     "execution_count": 246,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions[0].variants[0].get(\"transcripts\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter by Consequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = Parser(annotated_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(147,\n",
       " {'chromosome': 'chr21',\n",
       "  'position': 5232869,\n",
       "  'refAllele': 'T',\n",
       "  'altAlleles': ['G'],\n",
       "  'filters': ['base_quality', 'weak_evidence'],\n",
       "  'mappingQuality': 58.1,\n",
       "  'cytogeneticBand': '21p12',\n",
       "  'vcfInfo': {'DP': '799'},\n",
       "  'samples': [{'genotype': '0/0',\n",
       "    'variantFrequencies': [0.0552],\n",
       "    'totalDepth': 181,\n",
       "    'alleleDepths': [171, 10],\n",
       "    'somaticQuality': 0,\n",
       "    'vcfSampleInfo': {'F1R2': '96,6', 'F2R1': '75,4'}},\n",
       "   {'genotype': '0/1',\n",
       "    'variantFrequencies': [0.0677],\n",
       "    'totalDepth': 591,\n",
       "    'alleleDepths': [551, 40],\n",
       "    'somaticQuality': 8.6,\n",
       "    'vcfSampleInfo': {'F1R2': '280,23', 'F2R1': '271,17'}}],\n",
       "  'variants': [{'vid': '21-5232869-T-G',\n",
       "    'chromosome': 'chr21',\n",
       "    'begin': 5232869,\n",
       "    'end': 5232869,\n",
       "    'refAllele': 'T',\n",
       "    'altAllele': 'G',\n",
       "    'variantType': 'SNV',\n",
       "    'hgvsg': 'NC_000021.9:g.5232869T>G',\n",
       "    'phylopScore': 0.5,\n",
       "    'transcripts': [{'transcript': 'ENST00000623753.1',\n",
       "      'source': 'Ensembl',\n",
       "      'bioType': 'lncRNA',\n",
       "      'cdnaPos': '187',\n",
       "      'exons': '2/2',\n",
       "      'geneId': 'ENSG00000279669',\n",
       "      'hgnc': 'ENSG00000279669',\n",
       "      'consequence': ['non_coding_transcript_exon_variant'],\n",
       "      'impact': 'modifier',\n",
       "      'hgvsc': 'ENST00000623753.1:n.187A>C',\n",
       "      'isCanonical': True}]}]})"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = parser.filter_transcripts_by_consequence(\n",
    "    include=[\"non_coding_transcript_exon_variant\"]\n",
    ")\n",
    "len(positions), positions[0].model_dump()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(11723,\n",
       " {'chromosome': 'chr21',\n",
       "  'position': 5232869,\n",
       "  'refAllele': 'T',\n",
       "  'altAlleles': ['G'],\n",
       "  'filters': ['base_quality', 'weak_evidence'],\n",
       "  'mappingQuality': 58.1,\n",
       "  'cytogeneticBand': '21p12',\n",
       "  'vcfInfo': {'DP': '799'},\n",
       "  'samples': [{'genotype': '0/0',\n",
       "    'variantFrequencies': [0.0552],\n",
       "    'totalDepth': 181,\n",
       "    'alleleDepths': [171, 10],\n",
       "    'somaticQuality': 0,\n",
       "    'vcfSampleInfo': {'F1R2': '96,6', 'F2R1': '75,4'}},\n",
       "   {'genotype': '0/1',\n",
       "    'variantFrequencies': [0.0677],\n",
       "    'totalDepth': 591,\n",
       "    'alleleDepths': [551, 40],\n",
       "    'somaticQuality': 8.6,\n",
       "    'vcfSampleInfo': {'F1R2': '280,23', 'F2R1': '271,17'}}],\n",
       "  'variants': [{'vid': '21-5232869-T-G',\n",
       "    'chromosome': 'chr21',\n",
       "    'begin': 5232869,\n",
       "    'end': 5232869,\n",
       "    'refAllele': 'T',\n",
       "    'altAllele': 'G',\n",
       "    'variantType': 'SNV',\n",
       "    'hgvsg': 'NC_000021.9:g.5232869T>G',\n",
       "    'phylopScore': 0.5,\n",
       "    'transcripts': [{'transcript': 'ENST00000623753.1',\n",
       "      'source': 'Ensembl',\n",
       "      'bioType': 'lncRNA',\n",
       "      'cdnaPos': '187',\n",
       "      'exons': '2/2',\n",
       "      'geneId': 'ENSG00000279669',\n",
       "      'hgnc': 'ENSG00000279669',\n",
       "      'consequence': ['non_coding_transcript_exon_variant'],\n",
       "      'impact': 'modifier',\n",
       "      'hgvsc': 'ENST00000623753.1:n.187A>C',\n",
       "      'isCanonical': True}]}]})"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = parser.filter_transcripts_by_consequence(\n",
    "    exclude=[\"downstream_gene_variant\", \"upstream_gene_variant\"]\n",
    ")\n",
    "len(positions), positions[0].model_dump()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter by gnomad frequency\n",
    "Possible values\n",
    "\n",
    "'coverage', 'failedFilter', 'allAf', 'allAn', 'allAc', 'allHc', 'afrAf', 'afrAn', 'afrAc', 'afrHc', 'amrAf', 'amrAn', 'amrAc', 'amrHc', 'easAf', 'easAn', 'easAc', 'easHc', 'finAf', 'finAn', 'finAc', 'finHc', 'nfeAf', 'nfeAn', 'nfeAc', 'nfeHc', 'asjAf', 'asjAn', 'asjAc', 'asjHc', 'sasAf', 'sasAn', 'sasAc', 'sasHc', 'othAf', 'othAn', 'othAc', 'othHc', 'maleAf', 'maleAn', 'maleAc', 'maleHc', 'femaleAf', 'femaleAn', 'femaleAc', 'femaleHc', 'controlsAllAf', 'controlsAllAn', 'controlsAllAc'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = Parser(annotated_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(851,\n",
       " {'chromosome': 'chr21',\n",
       "  'position': 5222289,\n",
       "  'refAllele': 'C',\n",
       "  'altAlleles': ['T'],\n",
       "  'filters': ['mapping_quality', 'weak_evidence'],\n",
       "  'mappingQuality': 49.7,\n",
       "  'cytogeneticBand': '21p12',\n",
       "  'vcfInfo': {'DP': '749'},\n",
       "  'samples': [{'genotype': '0/0',\n",
       "    'variantFrequencies': [0.0219],\n",
       "    'totalDepth': 137,\n",
       "    'alleleDepths': [134, 3],\n",
       "    'somaticQuality': 0,\n",
       "    'vcfSampleInfo': {'F1R2': '77,3', 'F2R1': '57,0'}},\n",
       "   {'genotype': '0/1',\n",
       "    'variantFrequencies': [0.0196],\n",
       "    'totalDepth': 562,\n",
       "    'alleleDepths': [551, 11],\n",
       "    'somaticQuality': 7,\n",
       "    'vcfSampleInfo': {'F1R2': '286,6', 'F2R1': '265,5'}}],\n",
       "  'variants': [{'vid': '21-5222289-C-T',\n",
       "    'chromosome': 'chr21',\n",
       "    'begin': 5222289,\n",
       "    'end': 5222289,\n",
       "    'refAllele': 'C',\n",
       "    'altAllele': 'T',\n",
       "    'variantType': 'SNV',\n",
       "    'hgvsg': 'NC_000021.9:g.5222289C>T',\n",
       "    'phylopScore': -0.2,\n",
       "    'dbsnp': ['rs1366179382'],\n",
       "    'gnomad': {'coverage': 3,\n",
       "     'allAf': 0.000502,\n",
       "     'allAn': 149274,\n",
       "     'allAc': 75,\n",
       "     'allHc': 0,\n",
       "     'afrAf': 0.001704,\n",
       "     'afrAn': 40502,\n",
       "     'afrAc': 69,\n",
       "     'afrHc': 0,\n",
       "     'amrAf': 6.7e-05,\n",
       "     'amrAn': 14966,\n",
       "     'amrAc': 1,\n",
       "     'amrHc': 0,\n",
       "     'easAf': 0,\n",
       "     'easAn': 5022,\n",
       "     'easAc': 0,\n",
       "     'easHc': 0,\n",
       "     'finAf': 0,\n",
       "     'finAn': 10430,\n",
       "     'finAc': 0,\n",
       "     'finHc': 0,\n",
       "     'nfeAf': 7.5e-05,\n",
       "     'nfeAn': 66950,\n",
       "     'nfeAc': 5,\n",
       "     'nfeHc': 0,\n",
       "     'asjAf': 0,\n",
       "     'asjAn': 3438,\n",
       "     'asjAc': 0,\n",
       "     'asjHc': 0,\n",
       "     'sasAf': 0,\n",
       "     'sasAn': 4734,\n",
       "     'sasAc': 0,\n",
       "     'sasHc': 0,\n",
       "     'othAf': 0,\n",
       "     'othAn': 2024,\n",
       "     'othAc': 0,\n",
       "     'othHc': 0,\n",
       "     'maleAf': 0.000411,\n",
       "     'maleAn': 73006,\n",
       "     'maleAc': 30,\n",
       "     'maleHc': 0,\n",
       "     'femaleAf': 0.00059,\n",
       "     'femaleAn': 76268,\n",
       "     'femaleAc': 45,\n",
       "     'femaleHc': 0,\n",
       "     'controlsAllAf': 0.000565,\n",
       "     'controlsAllAn': 31850,\n",
       "     'controlsAllAc': 18},\n",
       "    'topmed': {'allAf': 0.001091,\n",
       "     'allAn': 125568,\n",
       "     'allAc': 137,\n",
       "     'allHc': 0,\n",
       "     'failedFilter': True},\n",
       "    'phyloPPrimateScore': 0.074}]})"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = parser.get_variants_above_gnomad_freq(frequency_key=\"allAf\", frequency_threshold_high=0.1)\n",
    "len(positions), positions[0].model_dump()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(302,\n",
       " {'chromosome': 'chr21',\n",
       "  'position': 5227548,\n",
       "  'refAllele': 'C',\n",
       "  'altAlleles': ['T'],\n",
       "  'filters': ['PASS'],\n",
       "  'mappingQuality': 93.13,\n",
       "  'cytogeneticBand': '21p12',\n",
       "  'vcfInfo': {'DP': '314'},\n",
       "  'samples': [{'genotype': '0|0',\n",
       "    'variantFrequencies': [0.125],\n",
       "    'totalDepth': 40,\n",
       "    'alleleDepths': [35, 5],\n",
       "    'somaticQuality': 0,\n",
       "    'vcfSampleInfo': {'F1R2': '16,4', 'F2R1': '19,1'}},\n",
       "   {'genotype': '0|1',\n",
       "    'variantFrequencies': [0.3432],\n",
       "    'totalDepth': 169,\n",
       "    'alleleDepths': [111, 58],\n",
       "    'somaticQuality': 28.1,\n",
       "    'vcfSampleInfo': {'F1R2': '60,29', 'F2R1': '51,29'}}],\n",
       "  'variants': [{'vid': '21-5227548-C-T',\n",
       "    'chromosome': 'chr21',\n",
       "    'begin': 5227548,\n",
       "    'end': 5227548,\n",
       "    'refAllele': 'C',\n",
       "    'altAllele': 'T',\n",
       "    'variantType': 'SNV',\n",
       "    'hgvsg': 'NC_000021.9:g.5227548C>T',\n",
       "    'phylopScore': 0.2,\n",
       "    'inLowComplexityRegion': True,\n",
       "    'dbsnp': ['rs1219135472'],\n",
       "    'gnomad': {'coverage': 15,\n",
       "     'allAf': 0.348842,\n",
       "     'allAn': 65554,\n",
       "     'allAc': 22868,\n",
       "     'allHc': 4287,\n",
       "     'afrAf': 0.195536,\n",
       "     'afrAn': 16130,\n",
       "     'afrAc': 3154,\n",
       "     'afrHc': 157,\n",
       "     'amrAf': 0.38877,\n",
       "     'amrAn': 5628,\n",
       "     'amrAc': 2188,\n",
       "     'amrHc': 454,\n",
       "     'easAf': 0.258838,\n",
       "     'easAn': 2376,\n",
       "     'easAc': 615,\n",
       "     'easHc': 84,\n",
       "     'finAf': 0.511703,\n",
       "     'finAn': 2478,\n",
       "     'finAc': 1268,\n",
       "     'finHc': 372,\n",
       "     'nfeAf': 0.406416,\n",
       "     'nfeAn': 33980,\n",
       "     'nfeAc': 13810,\n",
       "     'nfeHc': 2911,\n",
       "     'asjAf': 0.357451,\n",
       "     'asjAn': 1852,\n",
       "     'asjAc': 662,\n",
       "     'asjHc': 100,\n",
       "     'sasAf': 0.371722,\n",
       "     'sasAn': 1754,\n",
       "     'sasAc': 652,\n",
       "     'sasHc': 107,\n",
       "     'othAf': 0.363636,\n",
       "     'othAn': 880,\n",
       "     'othAc': 320,\n",
       "     'othHc': 60,\n",
       "     'maleAf': 0.348586,\n",
       "     'maleAn': 30136,\n",
       "     'maleAc': 10505,\n",
       "     'maleHc': 2007,\n",
       "     'femaleAf': 0.34906,\n",
       "     'femaleAn': 35418,\n",
       "     'femaleAc': 12363,\n",
       "     'femaleHc': 2280,\n",
       "     'controlsAllAf': 0.338529,\n",
       "     'controlsAllAn': 11804,\n",
       "     'controlsAllAc': 3996},\n",
       "    'topmed': {'allAf': 0.499148,\n",
       "     'allAn': 125568,\n",
       "     'allAc': 62677,\n",
       "     'allHc': 0,\n",
       "     'failedFilter': True},\n",
       "    'phyloPPrimateScore': 0}]})"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = parser.get_variants_above_gnomad_freq(frequency_key=\"allAf\", frequency_threshold_low=0.1)\n",
    "len(positions), positions[0].model_dump()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(114,\n",
       " {'chromosome': 'chr21',\n",
       "  'position': 5278225,\n",
       "  'refAllele': 'A',\n",
       "  'altAlleles': ['T'],\n",
       "  'filters': ['alt_allele_in_normal',\n",
       "   'filtered_reads',\n",
       "   'mapping_quality',\n",
       "   'non_homref_normal',\n",
       "   'weak_evidence'],\n",
       "  'mappingQuality': 12.91,\n",
       "  'cytogeneticBand': '21p12',\n",
       "  'vcfInfo': {'DP': '113'},\n",
       "  'samples': [{'genotype': '0/0',\n",
       "    'variantFrequencies': [0.25],\n",
       "    'totalDepth': 12,\n",
       "    'alleleDepths': [9, 3],\n",
       "    'somaticQuality': 0.5,\n",
       "    'vcfSampleInfo': {'F1R2': '5,2', 'F2R1': '4,1'}},\n",
       "   {'genotype': '0/1',\n",
       "    'variantFrequencies': [0.7931],\n",
       "    'totalDepth': 29,\n",
       "    'alleleDepths': [6, 23],\n",
       "    'somaticQuality': 7.5,\n",
       "    'vcfSampleInfo': {'F1R2': '4,9', 'F2R1': '2,14'}}],\n",
       "  'variants': [{'vid': '21-5278225-A-T',\n",
       "    'chromosome': 'chr21',\n",
       "    'begin': 5278225,\n",
       "    'end': 5278225,\n",
       "    'refAllele': 'A',\n",
       "    'altAllele': 'T',\n",
       "    'variantType': 'SNV',\n",
       "    'hgvsg': 'NC_000021.9:g.5278225A>T',\n",
       "    'dbsnp': ['rs1171728286'],\n",
       "    'gnomad': {'coverage': 7,\n",
       "     'failedFilter': True,\n",
       "     'allAf': 0.148668,\n",
       "     'allAn': 46856,\n",
       "     'allAc': 6966,\n",
       "     'allHc': 80,\n",
       "     'afrAf': 0.181794,\n",
       "     'afrAn': 26244,\n",
       "     'afrAc': 4771,\n",
       "     'afrHc': 46,\n",
       "     'amrAf': 0.110039,\n",
       "     'amrAn': 2590,\n",
       "     'amrAc': 285,\n",
       "     'amrHc': 5,\n",
       "     'easAf': 0.000608,\n",
       "     'easAn': 1644,\n",
       "     'easAc': 1,\n",
       "     'easHc': 0,\n",
       "     'finAf': 0.050104,\n",
       "     'finAn': 958,\n",
       "     'finAc': 48,\n",
       "     'finHc': 0,\n",
       "     'nfeAf': 0.122751,\n",
       "     'nfeAn': 12448,\n",
       "     'nfeAc': 1528,\n",
       "     'nfeHc': 26,\n",
       "     'asjAf': 0.225636,\n",
       "     'asjAn': 944,\n",
       "     'asjAc': 213,\n",
       "     'asjHc': 2,\n",
       "     'sasAf': 0.022059,\n",
       "     'sasAn': 1224,\n",
       "     'sasAc': 27,\n",
       "     'sasHc': 0,\n",
       "     'othAf': 0.116838,\n",
       "     'othAn': 582,\n",
       "     'othAc': 68,\n",
       "     'othHc': 0,\n",
       "     'maleAf': 0.145535,\n",
       "     'maleAn': 22084,\n",
       "     'maleAc': 3214,\n",
       "     'maleHc': 34,\n",
       "     'femaleAf': 0.151461,\n",
       "     'femaleAn': 24772,\n",
       "     'femaleAc': 3752,\n",
       "     'femaleHc': 46,\n",
       "     'controlsAllAf': 0.128554,\n",
       "     'controlsAllAn': 9848,\n",
       "     'controlsAllAc': 1266},\n",
       "    'topmed': {'allAf': 0.084209,\n",
       "     'allAn': 125568,\n",
       "     'allAc': 10574,\n",
       "     'allHc': 0,\n",
       "     'failedFilter': True}}]})"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = parser.get_variants_above_gnomad_freq(frequency_key=\"allAf\", frequency_threshold_low=0.1, frequency_threshold_high=0.2)\n",
    "len(positions), positions[0].model_dump()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene Filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>clingenDosageSensitivityMap</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AATBC</td>\n",
       "      <td>51526.0</td>\n",
       "      <td>284837</td>\n",
       "      <td>ENSG00000215458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000291052</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>16022.0</td>\n",
       "      <td>150000</td>\n",
       "      <td>ENSG00000243064</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCG1</td>\n",
       "      <td>73.0</td>\n",
       "      <td>9619</td>\n",
       "      <td>ENSG00000160179</td>\n",
       "      <td>[{'mimNumber': 603076, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>{'pLi': 0.112, 'pRec': 0.888, 'pNull': 9.34e-0...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ADARB1</td>\n",
       "      <td>226.0</td>\n",
       "      <td>104</td>\n",
       "      <td>ENSG00000197381</td>\n",
       "      <td>[{'mimNumber': 601218, 'geneName': 'Adenosine ...</td>\n",
       "      <td>{'pLi': 0.803, 'pRec': 0.197, 'pNull': 8.32e-0...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>YBEY</td>\n",
       "      <td>1299.0</td>\n",
       "      <td>54059</td>\n",
       "      <td>ENSG00000182362</td>\n",
       "      <td>[{'mimNumber': 617461, 'geneName': 'YBEY metal...</td>\n",
       "      <td>{'pLi': 1.34e-07, 'pRec': 0.0625, 'pNull': 0.9...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>YRDCP3</td>\n",
       "      <td>39921.0</td>\n",
       "      <td>100861429</td>\n",
       "      <td>ENSG00000230859</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>ZBTB21</td>\n",
       "      <td>13083.0</td>\n",
       "      <td>49854</td>\n",
       "      <td>ENSG00000173276</td>\n",
       "      <td>[{'mimNumber': 616485, 'geneName': 'Zinc finge...</td>\n",
       "      <td>{'pLi': 0.998, 'pRec': 0.00222, 'pNull': 5.45e...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>ZNF295-AS1</td>\n",
       "      <td>23130.0</td>\n",
       "      <td>150142</td>\n",
       "      <td>ENSG00000237232</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>ZNF355P</td>\n",
       "      <td>17218.0</td>\n",
       "      <td>100505852</td>\n",
       "      <td>ENSG00000168122</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>442 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "0         AATBC  51526.0     284837  ENSG00000215458  \\\n",
       "1        ABCC13      NaN        NaN  ENSG00000291052   \n",
       "2        ABCC13  16022.0     150000  ENSG00000243064   \n",
       "3         ABCG1     73.0       9619  ENSG00000160179   \n",
       "4        ADARB1    226.0        104  ENSG00000197381   \n",
       "..          ...      ...        ...              ...   \n",
       "437        YBEY   1299.0      54059  ENSG00000182362   \n",
       "438      YRDCP3  39921.0  100861429  ENSG00000230859   \n",
       "439      ZBTB21  13083.0      49854  ENSG00000173276   \n",
       "440  ZNF295-AS1  23130.0     150142  ENSG00000237232   \n",
       "441     ZNF355P  17218.0  100505852  ENSG00000168122   \n",
       "\n",
       "                                                  omim   \n",
       "0                                                  NaN  \\\n",
       "1    [{'mimNumber': 608835, 'geneName': 'ATP-bindin...   \n",
       "2    [{'mimNumber': 608835, 'geneName': 'ATP-bindin...   \n",
       "3    [{'mimNumber': 603076, 'geneName': 'ATP-bindin...   \n",
       "4    [{'mimNumber': 601218, 'geneName': 'Adenosine ...   \n",
       "..                                                 ...   \n",
       "437  [{'mimNumber': 617461, 'geneName': 'YBEY metal...   \n",
       "438                                                NaN   \n",
       "439  [{'mimNumber': 616485, 'geneName': 'Zinc finge...   \n",
       "440                                                NaN   \n",
       "441                                                NaN   \n",
       "\n",
       "                                                gnomAD   \n",
       "0                                                  NaN  \\\n",
       "1                                                  NaN   \n",
       "2                                                  NaN   \n",
       "3    {'pLi': 0.112, 'pRec': 0.888, 'pNull': 9.34e-0...   \n",
       "4    {'pLi': 0.803, 'pRec': 0.197, 'pNull': 8.32e-0...   \n",
       "..                                                 ...   \n",
       "437  {'pLi': 1.34e-07, 'pRec': 0.0625, 'pNull': 0.9...   \n",
       "438                                                NaN   \n",
       "439  {'pLi': 0.998, 'pRec': 0.00222, 'pNull': 5.45e...   \n",
       "440                                                NaN   \n",
       "441                                                NaN   \n",
       "\n",
       "    clingenDosageSensitivityMap clingenGeneValidity cosmic  \n",
       "0                           NaN                 NaN    NaN  \n",
       "1                           NaN                 NaN    NaN  \n",
       "2                           NaN                 NaN    NaN  \n",
       "3                           NaN                 NaN    NaN  \n",
       "4                           NaN                 NaN    NaN  \n",
       "..                          ...                 ...    ...  \n",
       "437                         NaN                 NaN    NaN  \n",
       "438                         NaN                 NaN    NaN  \n",
       "439                         NaN                 NaN    NaN  \n",
       "440                         NaN                 NaN    NaN  \n",
       "441                         NaN                 NaN    NaN  \n",
       "\n",
       "[442 rows x 9 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>clingenDosageSensitivityMap</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000291052</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>16022.0</td>\n",
       "      <td>150000</td>\n",
       "      <td>ENSG00000243064</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "1  ABCC13      NaN        NaN  ENSG00000291052  \\\n",
       "2  ABCC13  16022.0     150000  ENSG00000243064   \n",
       "\n",
       "                                                omim gnomAD   \n",
       "1  [{'mimNumber': 608835, 'geneName': 'ATP-bindin...    NaN  \\\n",
       "2  [{'mimNumber': 608835, 'geneName': 'ATP-bindin...    NaN   \n",
       "\n",
       "  clingenDosageSensitivityMap clingenGeneValidity cosmic  \n",
       "1                         NaN                 NaN    NaN  \n",
       "2                         NaN                 NaN    NaN  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.genes[annotated_data.genes[\"name\"] == \"ABCC13\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>clingenDosageSensitivityMap</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>16022.0</td>\n",
       "      <td>150000</td>\n",
       "      <td>ENSG00000243064</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>ZBTB21</td>\n",
       "      <td>13083.0</td>\n",
       "      <td>49854</td>\n",
       "      <td>ENSG00000173276</td>\n",
       "      <td>[{'mimNumber': 616485, 'geneName': 'Zinc finge...</td>\n",
       "      <td>{'pLi': 0.998, 'pRec': 0.00222, 'pNull': 5.45e...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "2    ABCC13  16022.0     150000  ENSG00000243064  \\\n",
       "439  ZBTB21  13083.0      49854  ENSG00000173276   \n",
       "\n",
       "                                                  omim   \n",
       "2    [{'mimNumber': 608835, 'geneName': 'ATP-bindin...  \\\n",
       "439  [{'mimNumber': 616485, 'geneName': 'Zinc finge...   \n",
       "\n",
       "                                                gnomAD   \n",
       "2                                                  NaN  \\\n",
       "439  {'pLi': 0.998, 'pRec': 0.00222, 'pNull': 5.45e...   \n",
       "\n",
       "    clingenDosageSensitivityMap clingenGeneValidity cosmic  \n",
       "2                           NaN                 NaN    NaN  \n",
       "439                         NaN                 NaN    NaN  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.genes[annotated_data.genes[\"ensemblGeneId\"].isin([\"ENSG00000173276\", \"ENSG00000243064\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>clingenDosageSensitivityMap</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>ERG</td>\n",
       "      <td>3446.0</td>\n",
       "      <td>2078</td>\n",
       "      <td>ENSG00000157554</td>\n",
       "      <td>[{'mimNumber': 165080, 'geneName': 'ETS transc...</td>\n",
       "      <td>{'pLi': 0.964, 'pRec': 0.0359, 'pNull': 5.03e-...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{'roleInCancer': ['oncogene', 'fusion']}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>TMPRSS2</td>\n",
       "      <td>11876.0</td>\n",
       "      <td>7113</td>\n",
       "      <td>ENSG00000184012</td>\n",
       "      <td>[{'mimNumber': 602060, 'geneName': 'Transmembr...</td>\n",
       "      <td>{'pLi': 2.4e-10, 'pRec': 0.88, 'pNull': 0.12, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{'roleInCancer': ['fusion']}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "198      ERG   3446.0       2078  ENSG00000157554  \\\n",
       "418  TMPRSS2  11876.0       7113  ENSG00000184012   \n",
       "\n",
       "                                                  omim   \n",
       "198  [{'mimNumber': 165080, 'geneName': 'ETS transc...  \\\n",
       "418  [{'mimNumber': 602060, 'geneName': 'Transmembr...   \n",
       "\n",
       "                                                gnomAD   \n",
       "198  {'pLi': 0.964, 'pRec': 0.0359, 'pNull': 5.03e-...  \\\n",
       "418  {'pLi': 2.4e-10, 'pRec': 0.88, 'pNull': 0.12, ...   \n",
       "\n",
       "    clingenDosageSensitivityMap clingenGeneValidity   \n",
       "198                         NaN                 NaN  \\\n",
       "418                         NaN                 NaN   \n",
       "\n",
       "                                       cosmic  \n",
       "198  {'roleInCancer': ['oncogene', 'fusion']}  \n",
       "418              {'roleInCancer': ['fusion']}  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.genes.dropna(subset=[\"cosmic\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
