{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Parse Nirvana JSON output in python\n",
    "\n",
    "Nirvana outputs a single JSON annotation file for a single input VCF file. The output file contains a single [JSON object](https://www.w3schools.com/js/js_json_objects.asp) to represent the annotations of all input VCF variants. The JSON file format can be found in the [documentation](https://illumina.github.io/NirvanaDocumentation/file-formats/nirvana-json-file-format).\n",
    "\n",
    "A [sample JSON file](https://github.com/Illumina/NirvanaDocumentation/blob/master/static/files/ceph_trio_test.json.gz) of the CEPH trio (NA12878, NA12891, and NA12892) can be downloaded from the [NirvanaDocumention](https://github.com/Illumina/NirvanaDocumentation/tree/master/static/files) git repo.\n",
    "\n",
    "This notebook demonstrates how you can parse this sample JSON file in python and retrieves the annotation data.\n",
    "\n",
    "## Read Nirvana JSON output by lines\n",
    "\n",
    "Even though the Nirvana JSON output is a single JSON object, different JSON object fields are written in different lines for memory efficient reading.\n",
    "\n",
    "The first line in the JSON file is the `header` line:\n",
    "\n",
    "```json\n",
    "{\"header\":{\"annotator\":\"Nirvana 3.13.0\",\"creationTime\":\"2020-12-18 19:43:53\",\"genomeAssembly\":\"GRCh38\",\"schemaVersion\":6,\"dataVersion\":\"91.27.61\",\"dataSources\":[{\"name\":\"VEP\",\"version\":\"91\",\"description\":\"BothRefSeqAndEnsembl\",\"releaseDate\":\"2017-12-18\"},{\"name\":\"MultiZ100Way\",\"version\":\"20171006\",\"description\":\"Amino acid conservation scores calculated from MultiZ100Way alignments from UCSC.\",\"releaseDate\":\"2017-10-06\"},{\"name\":\"ClinVar\",\"version\":\"20200903\",\"description\":\"A freely accessible, public archive of reports of the relationships among human variations and phenotypes, with supporting evidence\",\"releaseDate\":\"2020-09-03\"},{\"name\":\"dbSNP\",\"version\":\"154\",\"description\":\"Identifiers for observed variants\",\"releaseDate\":\"2020-05-01\"},{\"name\":\"dbSNP\",\"version\":\"151\",\"description\":\"Identifiers for observed variants\",\"releaseDate\":\"2018-04-18\"},{\"name\":\"gnomAD\",\"version\":\"3.0\",\"description\":\"Allele frequencies from Genome Aggregation Database (gnomAD)\",\"releaseDate\":\"2019-10-16\"},{\"name\":\"MITOMAP\",\"version\":\"20200819\",\"description\":\"Small variants in the MITOMAP human mitochondrial genome database\",\"releaseDate\":\"2020-08-19\"},{\"name\":\"1000 Genomes Project\",\"version\":\"Phase 3 v3plus\",\"description\":\"A public catalogue of human variation and genotype data\",\"releaseDate\":\"2013-05-27\"},{\"name\":\"PrimateAI\",\"version\":\"0.2\",\"description\":\"PrimateAI percentile scores.\",\"releaseDate\":\"2018-11-07\"},{\"name\":\"REVEL\",\"version\":\"20200205\",\"description\":\"Pathogenicity scores of missense variants predicted by REVEL\",\"releaseDate\":\"2020-02-05\"},{\"name\":\"SpliceAI\",\"version\":\"1.3\",\"description\":\"SpliceAI scores, distances, etc.\",\"releaseDate\":\"2019-10-04\"},{\"name\":\"TOPMed\",\"version\":\"freeze_5\",\"description\":\"Allele frequencies from TOPMed data\",\"releaseDate\":\"2017-08-28\"},{\"name\":\"ClinGen\",\"version\":\"20160414\",\"releaseDate\":\"2016-04-14\"},{\"name\":\"ClinGen Dosage Sensitivity Map\",\"version\":\"20200913\",\"description\":\"Dosage sensitivity map from ClinGen (dbVar)\",\"releaseDate\":\"2020-09-13\"},{\"name\":\"MITOMAP_SV\",\"version\":\"20200819\",\"description\":\"Large structural variants in the MITOMAP human mitochondrial genome database\",\"releaseDate\":\"2020-08-19\"},{\"name\":\"1000 Genomes Project (SV)\",\"version\":\"Phase 3 v5a\",\"description\":\"A public catalogue of human variation and genotype data\",\"releaseDate\":\"2013-05-27\"},{\"name\":\"ClinGen disease validity curations\",\"version\":\"20200914\",\"description\":\"Disease validity curations from ClinGen (dbVar)\",\"releaseDate\":\"2020-09-14\"},{\"name\":\"gnomAD_gene_scores\",\"version\":\"2.1\",\"description\":\"Gene Loss of function scores from Genome Aggregation Database (gnomAD)\",\"releaseDate\":\"2018-10-17\"},{\"name\":\"OMIM\",\"version\":\"20200915\",\"description\":\"An Online Catalog of Human Genes and Genetic Disorders\",\"releaseDate\":\"2020-09-15\"},{\"name\":\"phyloP\",\"version\":\"hg38\",\"description\":\"20 way conservation score between humans and 19 other vertebrates\",\"releaseDate\":\"2015-04-17\"},{\"name\":\"gnomAD_LCR\",\"version\":\"2.1\",\"description\":\"Low complexity regions from gnomAD.\",\"releaseDate\":\"2019-04-10\"},{\"name\":\"MitochondrialHeteroplasmy\",\"version\":\"20180410\",\"description\":\"Variant read frequency percentiles for the Mitochondrial reference\",\"releaseDate\":\"2020-05-21\"}],\"samples\":[\"Novaseq_TSPF450-NA12878-1-HFHWJDMXX_S1_L001\",\"Novaseq_TSPF450-NA12891-1-HFHWJDMXX_S3_L001\",\"Novaseq_TSPF450-NA12892-1-HFHWJDMXX_S2_L001\"]},\"positions\":[\n",
    "```\n",
    "Followed by the `positions` lines (two position lines in this example):\n",
    "\n",
    "```json\n",
    "{\"chromosome\":\"chr21\",\"position\":9975027,\"refAllele\":\"C\",\"altAlleles\":[\"G\"],\"quality\":102.47,\"filters\":[\"PASS\"],\"fisherStrandBias\":0.727,\"mappingQuality\":43.11,\"cytogeneticBand\":\"21p11.2\",\"samples\":[{\"genotype\":\"0/1\",\"variantFrequencies\":[0.4474],\"totalDepth\":38,\"genotypeQuality\":32,\"alleleDepths\":[21,17]},{\"genotype\":\"0/1\",\"variantFrequencies\":[0.5147],\"totalDepth\":68,\"genotypeQuality\":36,\"alleleDepths\":[33,35]},{\"genotype\":\"0/0\",\"variantFrequencies\":[0.1667],\"totalDepth\":72,\"genotypeQuality\":0,\"alleleDepths\":[60,12],\"failedFilter\":true}],\"variants\":[{\"vid\":\"21-9975027-C-G\",\"chromosome\":\"chr21\",\"begin\":9975027,\"end\":9975027,\"refAllele\":\"C\",\"altAllele\":\"G\",\"variantType\":\"SNV\",\"hgvsg\":\"NC_000021.9:g.9975027C>G\",\"phylopScore\":-1.7,\"dbsnp\":[\"rs75310125\"],\"gnomad\":{\"coverage\":102,\"failedFilter\":true,\"allAf\":0.464664,\"allAn\":125536,\"allAc\":58332,\"allHc\":34,\"afrAf\":0.396789,\"afrAn\":31886,\"afrAc\":12652,\"afrHc\":0,\"amrAf\":0.489333,\"amrAn\":12562,\"amrAc\":6147,\"amrHc\":9,\"easAf\":0.490489,\"easAn\":2944,\"easAc\":1444,\"easHc\":0,\"finAf\":0.492007,\"finAn\":9758,\"finAc\":4801,\"finHc\":3,\"nfeAf\":0.486377,\"nfeAn\":59604,\"nfeAc\":28990,\"nfeHc\":12,\"asjAf\":0.491335,\"asjAn\":3116,\"asjAc\":1531,\"asjHc\":4,\"sasAf\":0.494874,\"sasAn\":2926,\"sasAc\":1448,\"sasHc\":6,\"othAf\":0.477083,\"othAn\":1920,\"othAc\":916,\"othHc\":0,\"maleAf\":0.466374,\"maleAn\":61114,\"maleAc\":28502,\"maleHc\":24,\"femaleAf\":0.463041,\"femaleAn\":64422,\"femaleAc\":29830,\"femaleHc\":10},\"topmed\":{\"allAf\":0.447447,\"allAn\":125568,\"allAc\":56185,\"allHc\":137,\"failedFilter\":true},\"transcripts\":[{\"transcript\":\"ENST00000625012.1\",\"source\":\"Ensembl\",\"bioType\":\"processed_transcript\",\"cdnaPos\":\"1475\",\"exons\":\"11/11\",\"geneId\":\"ENSG00000270533\",\"hgnc\":\"CR382285.1\",\"consequence\":[\"non_coding_transcript_exon_variant\"],\"hgvsc\":\"ENST00000625012.1:n.1475G>C\",\"isCanonical\":true}]}]},\n",
    "{\"chromosome\":\"chr21\",\"position\":9975061,\"refAllele\":\"A\",\"altAlleles\":[\"G\"],\"quality\":190.73,\"filters\":[\"PASS\"],\"fisherStrandBias\":3.698,\"mappingQuality\":44.4,\"cytogeneticBand\":\"21p11.2\",\"samples\":[{\"genotype\":\"0/1\",\"variantFrequencies\":[0.5349],\"totalDepth\":43,\"genotypeQuality\":37,\"alleleDepths\":[20,23]},{\"genotype\":\"1/1\",\"variantFrequencies\":[0.7727],\"totalDepth\":66,\"genotypeQuality\":3,\"alleleDepths\":[15,51]},{\"genotype\":\"0/1\",\"variantFrequencies\":[0.5833],\"totalDepth\":72,\"genotypeQuality\":38,\"alleleDepths\":[30,42]}],\"variants\":[{\"vid\":\"21-9975061-A-G\",\"chromosome\":\"chr21\",\"begin\":9975061,\"end\":9975061,\"refAllele\":\"A\",\"altAllele\":\"G\",\"variantType\":\"SNV\",\"hgvsg\":\"NC_000021.9:g.9975061A>G\",\"phylopScore\":-0.1,\"dbsnp\":[\"rs75428092\"],\"gnomad\":{\"coverage\":93,\"failedFilter\":true,\"allAf\":0.539865,\"allAn\":142030,\"allAc\":76677,\"allHc\":5731,\"afrAf\":0.594262,\"afrAn\":41586,\"afrAc\":24713,\"afrHc\":3944,\"amrAf\":0.518277,\"amrAn\":13514,\"amrAc\":7004,\"amrHc\":254,\"easAf\":0.581298,\"easAn\":3112,\"easAc\":1809,\"easHc\":253,\"finAf\":0.517912,\"finAn\":10384,\"finAc\":5378,\"finHc\":188,\"nfeAf\":0.511456,\"nfeAn\":64074,\"nfeAc\":32771,\"nfeHc\":769,\"asjAf\":0.524864,\"asjAn\":3298,\"asjAc\":1731,\"asjHc\":83,\"sasAf\":0.549736,\"sasAn\":3036,\"sasAc\":1669,\"sasHc\":151,\"othAf\":0.539869,\"othAn\":2132,\"othAc\":1151,\"othHc\":85,\"maleAf\":0.539736,\"maleAn\":68804,\"maleAc\":37136,\"maleHc\":2762,\"femaleAf\":0.539986,\"femaleAn\":73226,\"femaleAc\":39541,\"femaleHc\":2969},\"topmed\":{\"allAf\":0.567087,\"allAn\":125568,\"allAc\":71208,\"allHc\":8558,\"failedFilter\":true},\"transcripts\":[{\"transcript\":\"ENST00000625012.1\",\"source\":\"Ensembl\",\"bioType\":\"processed_transcript\",\"cdnaPos\":\"1441\",\"exons\":\"11/11\",\"geneId\":\"ENSG00000270533\",\"hgnc\":\"CR382285.1\",\"consequence\":[\"non_coding_transcript_exon_variant\"],\"hgvsc\":\"ENST00000625012.1:n.1441T>C\",\"isCanonical\":true}]}]}\n",
    "```\n",
    "\n",
    "After the `positions` lines, there are the `genes` lines, which are optional if there is no overlapping gene of the input VCF variants (3 lines for 2 genes in the example):\n",
    "\n",
    "```json\n",
    "],\"genes\":[\n",
    "{\"name\":\"ABCC13\",\"omim\":[{\"mimNumber\":608835,\"geneName\":\"ATP-binding cassette, subfamily C, member 13\",\"description\":\"ABCC13 belongs to a large family of ATP-binding cassette (ABC) transporters that play important roles as membrane transporters or ion channel modulators. However, ABCC13 is a truncated protein that lacks critical ATP-binding motifs and is unlikely to be a functional transporter (Yabuuchi et al., 2002).\"}]},\n",
    "{\"name\":\"USP25\",\"gnomAD\":{\"pLi\":7.52e-1,\"pRec\":2.48e-1,\"pNull\":1.85e-12,\"synZ\":2.56e-1,\"misZ\":1.24e0,\"loeuf\":3.30e-1},\"omim\":[{\"mimNumber\":604736,\"description\":\"Ubiquitin is a highly conserved 76-amino acid protein involved in regulation of intracellular protein breakdown, cell cycle regulation, and stress response. Ubiquitin is released from degraded proteins by disassembly of the polyubiquitin chains, which is mediated by ubiquitin-specific proteases (USPs), such as USP25 (Valero et al., 1999).\"}]}\n",
    "```\n",
    "\n",
    "Finally, the last line of the JSON file are two brackets to complete the JSON object structure:\n",
    "\n",
    "```json\n",
    "]}\n",
    "```\n",
    "\n",
    "Therefore, parsing the file requires some trimming and handling for each line to make them correctly formatted JSON objects or arrays."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "header object: {\"annotator\":\"Nirvana 3.13.0\",\"creationTime\":\"2020-12-18 19:43:53\",\"genomeAssembly\":\"GRCh38\",\"schemaVersion\":6,\"dataVersion\":\"91.27.61\",\"dataSources\":[{\"name\":\"VEP\",\"version\":\"91\",\"description\":\"BothRefSeqAndEnsembl\",\"releaseDate\":\"2017-12-18\"},{\"name\":\"MultiZ100Way\",\"version\":\"20171006\",\"description\":\"Amino acid conservation scores calculated from MultiZ100Way alignments from UCSC.\",\"releaseDate\":\"2017-10-06\"},{\"name\":\"ClinVar\",\"version\":\"20200903\",\"description\":\"A freely accessible, public archive of reports of the relationships among human variations and phenotypes, with supporting evidence\",\"releaseDate\":\"2020-09-03\"},{\"name\":\"dbSNP\",\"version\":\"154\",\"description\":\"Identifiers for observed variants\",\"releaseDate\":\"2020-05-01\"},{\"name\":\"dbSNP\",\"version\":\"151\",\"description\":\"Identifiers for observed variants\",\"releaseDate\":\"2018-04-18\"},{\"name\":\"gnomAD\",\"version\":\"3.0\",\"description\":\"Allele frequencies from Genome Aggregation Database (gnomAD)\",\"releaseDate\":\"2019-10-16\"},{\"name\":\"MITOMAP\",\"version\":\"20200819\",\"description\":\"Small variants in the MITOMAP human mitochondrial genome database\",\"releaseDate\":\"2020-08-19\"},{\"name\":\"1000 Genomes Project\",\"version\":\"Phase 3 v3plus\",\"description\":\"A public catalogue of human variation and genotype data\",\"releaseDate\":\"2013-05-27\"},{\"name\":\"PrimateAI\",\"version\":\"0.2\",\"description\":\"PrimateAI percentile scores.\",\"releaseDate\":\"2018-11-07\"},{\"name\":\"REVEL\",\"version\":\"20200205\",\"description\":\"Pathogenicity scores of missense variants predicted by REVEL\",\"releaseDate\":\"2020-02-05\"},{\"name\":\"SpliceAI\",\"version\":\"1.3\",\"description\":\"SpliceAI scores, distances, etc.\",\"releaseDate\":\"2019-10-04\"},{\"name\":\"TOPMed\",\"version\":\"freeze_5\",\"description\":\"Allele frequencies from TOPMed data\",\"releaseDate\":\"2017-08-28\"},{\"name\":\"ClinGen\",\"version\":\"20160414\",\"releaseDate\":\"2016-04-14\"},{\"name\":\"ClinGen Dosage Sensitivity Map\",\"version\":\"20200913\",\"description\":\"Dosage sensitivity map from ClinGen (dbVar)\",\"releaseDate\":\"2020-09-13\"},{\"name\":\"MITOMAP_SV\",\"version\":\"20200819\",\"description\":\"Large structural variants in the MITOMAP human mitochondrial genome database\",\"releaseDate\":\"2020-08-19\"},{\"name\":\"1000 Genomes Project (SV)\",\"version\":\"Phase 3 v5a\",\"description\":\"A public catalogue of human variation and genotype data\",\"releaseDate\":\"2013-05-27\"},{\"name\":\"ClinGen disease validity curations\",\"version\":\"20200914\",\"description\":\"Disease validity curations from ClinGen (dbVar)\",\"releaseDate\":\"2020-09-14\"},{\"name\":\"gnomAD_gene_scores\",\"version\":\"2.1\",\"description\":\"Gene Loss of function scores from Genome Aggregation Database (gnomAD)\",\"releaseDate\":\"2018-10-17\"},{\"name\":\"OMIM\",\"version\":\"20200915\",\"description\":\"An Online Catalog of Human Genes and Genetic Disorders\",\"releaseDate\":\"2020-09-15\"},{\"name\":\"phyloP\",\"version\":\"hg38\",\"description\":\"20 way conservation score between humans and 19 other vertebrates\",\"releaseDate\":\"2015-04-17\"},{\"name\":\"gnomAD_LCR\",\"version\":\"2.1\",\"description\":\"Low complexity regions from gnomAD.\",\"releaseDate\":\"2019-04-10\"},{\"name\":\"MitochondrialHeteroplasmy\",\"version\":\"20180410\",\"description\":\"Variant read frequency percentiles for the Mitochondrial reference\",\"releaseDate\":\"2020-05-21\"}],\"samples\":[\"Novaseq_TSPF450-NA12878-1-HFHWJDMXX_S1_L001\",\"Novaseq_TSPF450-NA12891-1-HFHWJDMXX_S3_L001\",\"Novaseq_TSPF450-NA12892-1-HFHWJDMXX_S2_L001\"]}\n",
      "number of positions: 17384\n",
      "number of genes: 12\n"
     ]
    }
   ],
   "source": [
    "import gzip\n",
    "\n",
    "header = ''\n",
    "positions = []\n",
    "genes = []\n",
    "is_header_line = True\n",
    "is_position_line = False\n",
    "is_gene_line = False\n",
    "gene_section_line = '],\"genes\":['\n",
    "end_line = ']}'\n",
    "with gzip.open('/directory/to/test/file/ceph_trio_test.json.gz', 'rt') as f:\n",
    "    position_count = 0\n",
    "    gene_count = 0\n",
    "    for line in f:\n",
    "        trim_line = line.strip()\n",
    "        if is_header_line:\n",
    "            ## only keep the \"header\" field content from the line\n",
    "            header = trim_line[10:-14]\n",
    "            is_header_line = False\n",
    "            is_position_line = True\n",
    "            continue\n",
    "        if trim_line == gene_section_line:\n",
    "            is_gene_line = True\n",
    "            is_position_line = False\n",
    "            continue\n",
    "        elif trim_line == end_line:\n",
    "            break\n",
    "        else:\n",
    "            if is_position_line:\n",
    "                ## remove the trailing ',' if there is\n",
    "                positions.append(trim_line.rstrip(','))\n",
    "                position_count += 1\n",
    "            if is_gene_line:\n",
    "                ## remove the trailing ',' if there is\n",
    "                genes.append(trim_line.rstrip(','))\n",
    "                gene_count += 1\n",
    "\n",
    "print ('header object:', header)\n",
    "print ('number of positions:', position_count)\n",
    "print ('number of genes:', gene_count)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Retrieve variants under a gnomAD allele frequency threshold\n",
    "\n",
    "The code block below demonstrates a use case which retrieves all variants under a given gnomAD allele frequency threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>gnomAD_allele_freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>21-10044005-G-C</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>21-10057340-G-A</td>\n",
       "      <td>0.000054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>21-10057468-C-A</td>\n",
       "      <td>0.000041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>21-10068145-T-C</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>21-10068147-A-T</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>21-16859389-ACT-A</td>\n",
       "      <td>0.000055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>21-16859389-ACTC-A</td>\n",
       "      <td>0.000008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>21-16993322-C-T</td>\n",
       "      <td>0.000098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>21-17085263-TAGAGAGAG-T</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>21-17130182-C-T</td>\n",
       "      <td>0.000007</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  variant_id  gnomAD_allele_freq\n",
       "0            21-10044005-G-C            0.000000\n",
       "1            21-10057340-G-A            0.000054\n",
       "2            21-10057468-C-A            0.000041\n",
       "3            21-10068145-T-C            0.000000\n",
       "4            21-10068147-A-T            0.000000\n",
       "..                       ...                 ...\n",
       "148        21-16859389-ACT-A            0.000055\n",
       "149       21-16859389-ACTC-A            0.000008\n",
       "150          21-16993322-C-T            0.000098\n",
       "151  21-17085263-TAGAGAGAG-T            0.000000\n",
       "152          21-17130182-C-T            0.000007\n",
       "\n",
       "[153 rows x 2 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import json\n",
    "\n",
    "variants_field = 'variants'\n",
    "gnomad_field = 'gnomad'\n",
    "freq_threshold = 0.0001\n",
    "freq_data = {'variant_id': [], 'gnomAD_allele_freq': []}\n",
    "for position in positions:\n",
    "    position_dict = json.loads(position)\n",
    "    if variants_field in position_dict:\n",
    "        for variant_dict in position_dict[variants_field]:\n",
    "            if gnomad_field in variant_dict:\n",
    "                freq = variant_dict[gnomad_field]['allAf']\n",
    "                if freq < freq_threshold:\n",
    "                    freq_data['variant_id'].append(variant_dict['vid'])\n",
    "                    freq_data['gnomAD_allele_freq'].append(freq)\n",
    "\n",
    "freq_df = pd.DataFrame(data=freq_data)\n",
    "freq_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Retrieve all relevant genes and their OMIM gene names\n",
    "The code block below utilizes the `genes` section of the annotation JSON file and retreives relevant OMIM annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>OMIM_gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>ATP-binding cassette, subfamily C, member 13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AF130351.1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BAGE2</td>\n",
       "      <td>BAGE family, member 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HSPA13</td>\n",
       "      <td>Stress 70 protein chaperone, microsome-associa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LIPI</td>\n",
       "      <td>Lipase I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MIR99AHG</td>\n",
       "      <td>MIR99A-LET7C cluster host gene, noncoding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NRIP1</td>\n",
       "      <td>Nuclear receptor interacting protein 1 (recept...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>POTED</td>\n",
       "      <td>POTE ankyrin domain family, member D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>RBM11</td>\n",
       "      <td>RNA-binding motif protein 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SAMSN1</td>\n",
       "      <td>SAM domain, SH3 domain, and nuclear localizati...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>TPTE</td>\n",
       "      <td>Transmembrane phosphatase with tensin homology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>USP25</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene                                     OMIM_gene_name\n",
       "0       ABCC13       ATP-binding cassette, subfamily C, member 13\n",
       "1   AF130351.1                                                   \n",
       "2        BAGE2                              BAGE family, member 2\n",
       "3       HSPA13  Stress 70 protein chaperone, microsome-associa...\n",
       "4         LIPI                                           Lipase I\n",
       "5     MIR99AHG          MIR99A-LET7C cluster host gene, noncoding\n",
       "6        NRIP1  Nuclear receptor interacting protein 1 (recept...\n",
       "7        POTED               POTE ankyrin domain family, member D\n",
       "8        RBM11                       RNA-binding motif protein 11\n",
       "9       SAMSN1  SAM domain, SH3 domain, and nuclear localizati...\n",
       "10        TPTE     Transmembrane phosphatase with tensin homology\n",
       "11       USP25                                                   "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_data = {'gene': [], 'OMIM_gene_name': []}\n",
    "for gene in genes:\n",
    "    gene_dict = json.loads(gene)\n",
    "    gene_data['gene'].append(gene_dict['name'])\n",
    "    omim_gene_name = ''\n",
    "    if 'omim' in gene_dict:\n",
    "        omim_dict = gene_dict['omim'][0]\n",
    "        if 'geneName' in omim_dict:\n",
    "            omim_gene_name = omim_dict['geneName']\n",
    "    gene_data['OMIM_gene_name'].append(omim_gene_name)\n",
    "\n",
    "gene_df=pd.DataFrame(data=gene_data)\n",
    "gene_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
