{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Parse Nirvana JSON output in python\n",
    "\n",
    "Nirvana outputs a single JSON annotation file for a single input VCF file. The output file contains a single [JSON object](https://www.w3schools.com/js/js_json_objects.asp) to represent the annotations of all input VCF variants. The JSON file format can be found in the [documentation](https://illumina.github.io/IlluminaConnectedAnnotationsDocumentation/file-formats/illumina-annotator-json-file-format).\n",
    "\n",
    "\n",
    "This notebook demonstrates how you can parse the JSON file in python and perform filtering operations.\n",
    "\n",
    "## Read Nirvana JSON output by lines\n",
    "\n",
    "Even though the Nirvana JSON output is a single JSON object, different JSON object fields are written in different lines for memory efficient reading.\n",
    "\n",
    "The first line in the JSON file is the `header` line:\n",
    "\n",
    "```json\n",
    "{\"header\":{\"annotator\":\"Nirvana\n",
    "...\n",
    ",\"positions\":[\n",
    "```\n",
    "Followed by the `positions` lines:\n",
    "\n",
    "```json\n",
    "{\"chromosome\":\"chr21\",\"position\":9975027,\"refAllele\":\"C\",\"altAlleles\":[\"G\"],\"quality\":102.47,\"filters\":[\"PASS\"],\"fisherStrandBias\":0.727,\"mappingQuality\":43.11,\"cytogeneticBand\":\"21p11.2\",\"samples\":,\n",
    "...\n",
    "```\n",
    "\n",
    "After the `positions` lines, there are the `genes` lines, which are optional if there is no overlapping gene of the input VCF variants (3 lines for 2 genes in the example):\n",
    "\n",
    "```json\n",
    "],\"genes\":[\n",
    "{\"name\":\"ABCC13\",\"omim\":[{\"mimNumber\":608835,\"geneName\":\"ATP-binding cassette, subfamily C, member 13\",\"description\":\"ABCC13 belongs to a large family of ATP-binding cassette (ABC) transporters that play important roles as membrane transporters or ion channel modulators. However, ABCC13 is a truncated protein that lacks critical ATP-binding motifs and is unlikely to be a functional transporter (Yabuuchi et al., 2002).\"}]},\n",
    "```\n",
    "\n",
    "Finally, the last line of the JSON file are two brackets to complete the JSON object structure:\n",
    "\n",
    "```json\n",
    "]}\n",
    "```\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Install requirements"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pip3 install pydantic ijson pandas\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pydantic objects\n",
    "Here we use pydantic objects to deserialize the json.\n",
    "Pandas is also used to display the data neatly.\n",
    "\n",
    "### Note: Memory efficiency\n",
    "In these examples, ijson is used to minimize the footprint. Python generators are used where possible to minimize memory footprint. This also means that the file is scanned each time which might be time consuming for very large file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "import gzip\n",
    "from typing import Any, Dict, Generator, List, Optional\n",
    "\n",
    "import ijson\n",
    "import pandas as pd\n",
    "import pydantic\n",
    "\n",
    "pd.set_option('display.max_columns', None)\n",
    "\n",
    "\n",
    "class BaseClass(pydantic.BaseModel):\n",
    "    model_config = pydantic.ConfigDict(extra='allow')\n",
    "\n",
    "    def get_top_level(self) -> pd.DataFrame:\n",
    "        return pd.json_normalize(self.get_top_level_dict())\n",
    "\n",
    "    def get_top_level_dict(self) -> Dict[str, Any]:\n",
    "        raise NotImplementedError\n",
    "\n",
    "    def to_df(self, key: str = \"\") -> pd.DataFrame:\n",
    "        if not key:\n",
    "            return pd.json_normalize(self.model_dump())\n",
    "\n",
    "        values = self.model_dump().get(key)\n",
    "\n",
    "        if isinstance(values, list):\n",
    "            merged = [self.get_top_level_dict() | value for value in values]\n",
    "        else:\n",
    "            merged = [self.get_top_level_dict() | {key: values}]\n",
    "\n",
    "        return pd.json_normalize(merged)\n",
    "\n",
    "\n",
    "class Transcript(BaseClass):\n",
    "    transcript: str\n",
    "    source: str\n",
    "    bioType: Optional[str] = None\n",
    "    geneId: Optional[str] = None\n",
    "    hgnc: Optional[str] = None\n",
    "    consequence: Optional[List[str]] = None\n",
    "    impact: Optional[str] = None\n",
    "    isCanonical: Optional[bool] = None\n",
    "\n",
    "    def get_top_level_dict(self) -> Dict[str, Any]:\n",
    "        return dict(zip((\"transcript\", \"isCanonical\"), (self.transcript, self.isCanonical)))\n",
    "\n",
    "\n",
    "class Variant(BaseClass):\n",
    "    vid: str\n",
    "    chromosome: str\n",
    "    begin: int\n",
    "    end: int\n",
    "    refAllele: str\n",
    "    altAllele: str\n",
    "    variantType: Optional[str] = None\n",
    "    hgvsg: Optional[str] = None\n",
    "    phylopScore: Optional[float] = None\n",
    "    phyloPPrimateScore: Optional[float] = None\n",
    "    transcripts: Optional[List[Transcript]] = None\n",
    "\n",
    "    def get_top_level_dict(self) -> Dict[str, Any]:\n",
    "        return dict(\n",
    "            zip(\n",
    "                (\"chromosome\", \"begin\", \"end\", \"refAllele\", \"altAllele\", \"hgvsg\"),\n",
    "                (self.chromosome, self.begin, self.end, self.refAllele, self.altAllele, self.hgvsg),\n",
    "            )\n",
    "        )\n",
    "\n",
    "\n",
    "class Position(BaseClass):\n",
    "    chromosome: str\n",
    "    position: int\n",
    "    refAllele: str\n",
    "    altAlleles: List[str]\n",
    "    filters: List[str]\n",
    "    mappingQuality: float\n",
    "    cytogeneticBand: str\n",
    "    vcfInfo: Dict[str, Any]\n",
    "    samples: List[Dict[str, Any]]\n",
    "    variants: List[Variant]\n",
    "\n",
    "    def get_top_level_dict(self) -> Dict[str, Any]:\n",
    "        return dict(\n",
    "            zip(\n",
    "                (\n",
    "                    \"chromosome\",\n",
    "                    \"position\",\n",
    "                    \"refAllele\",\n",
    "                    \"altAlleles\",\n",
    "                    \"filters\",\n",
    "                    \"mappingQuality\",\n",
    "                    \"cyatogeneticBand\",\n",
    "                    \"vcfInfo\",\n",
    "                ),\n",
    "                (\n",
    "                    self.chromosome,\n",
    "                    self.position,\n",
    "                    self.refAllele,\n",
    "                    self.altAlleles,\n",
    "                    self.filters,\n",
    "                    self.mappingQuality,\n",
    "                    self.cytogeneticBand,\n",
    "                    self.vcfInfo,\n",
    "                ),\n",
    "            )\n",
    "        )\n",
    "\n",
    "\n",
    "class AnnotatedData:\n",
    "    def __init__(self, filename: str):\n",
    "        self._filename = filename\n",
    "\n",
    "        for key in (\"annotator\", \"genomeAssembly\", \"creationTime\"):\n",
    "            print(f\"{key}: {self.header[key]}\")\n",
    "\n",
    "    @property\n",
    "    def header(self) -> Dict[str, Any]:\n",
    "        with gzip.open(self._filename, 'r') as f:\n",
    "            return next(ijson.items(f, \"header\"))\n",
    "\n",
    "    @property\n",
    "    def data_sources(self) -> pd.DataFrame:\n",
    "        return pd.DataFrame(self.header[\"dataSources\"]).set_index(\"name\").sort_index()\n",
    "\n",
    "    @property\n",
    "    def genes(self) -> pd.DataFrame:\n",
    "        with gzip.open(self._filename, 'r') as f:\n",
    "            return pd.json_normalize(ijson.items(f, \"genes.item\"))\n",
    "\n",
    "    @property\n",
    "    def positions(self) -> Any:\n",
    "        f = gzip.open(self._filename, 'r')\n",
    "        return ijson.items(f, \"positions.item\")\n",
    "\n",
    "    def get_annotation(self, chromosome: str, position: int) -> Dict[str, Any]:\n",
    "        return next(\n",
    "            (\n",
    "                position_item\n",
    "                for position_item in self.positions\n",
    "                if chromosome == position_item.get(\"chromosome\") and position == position_item.get(\"position\")\n",
    "            ),\n",
    "            {},\n",
    "        )\n",
    "\n",
    "    def get_annotation_range(self, chromosome: str, position: int, end: int) -> Generator[Any, Any, None]:\n",
    "        return (\n",
    "            position_item\n",
    "            for position_item in self.positions\n",
    "            if chromosome == position_item.get(\"chromosome\") and position <= position_item.get(\"position\") <= end\n",
    "        )\n",
    "\n",
    "    @staticmethod\n",
    "    def multiple_to_df(items: List[BaseClass], key: str = \"\") -> pd.DataFrame:\n",
    "        return pd.concat((item.to_df(key) for item in items))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filename"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "filename = \"\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Header"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotator: Illumina Connected Annotations 3.22.0\n",
      "genomeAssembly: GRCh38\n",
      "creationTime: 2023-12-07 14:15:54\n"
     ]
    }
   ],
   "source": [
    "annotated_data = AnnotatedData(filename=filename)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>version</th>\n",
       "      <th>description</th>\n",
       "      <th>releaseDate</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1000 Genomes Project</th>\n",
       "      <td>Phase 3 v3plus</td>\n",
       "      <td>A public catalogue of human variation and geno...</td>\n",
       "      <td>2013-05-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000 Genomes Project (SV)</th>\n",
       "      <td>Phase 3 v5a</td>\n",
       "      <td>A public catalogue of human variation and geno...</td>\n",
       "      <td>2013-05-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COSMIC</th>\n",
       "      <td>96</td>\n",
       "      <td>resource for exploring the impact of somatic m...</td>\n",
       "      <td>2022-05-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COSMIC gene fusions</th>\n",
       "      <td>96</td>\n",
       "      <td>manually curated somatic gene fusions</td>\n",
       "      <td>2023-11-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CancerHotspots</th>\n",
       "      <td>2017</td>\n",
       "      <td>A resouce for statistically significant mutati...</td>\n",
       "      <td>2017-01-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ClinGen</th>\n",
       "      <td>20160414</td>\n",
       "      <td>None</td>\n",
       "      <td>2016-04-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ClinGen Dosage Sensitivity Map</th>\n",
       "      <td>20231105</td>\n",
       "      <td>Dosage sensitivity map from ClinGen (dbVar)</td>\n",
       "      <td>2023-11-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ClinGen disease validity curations</th>\n",
       "      <td>20231105</td>\n",
       "      <td>Disease validity curations from ClinGen (dbVar)</td>\n",
       "      <td>2023-11-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ClinVar</th>\n",
       "      <td>20231028</td>\n",
       "      <td>A freely accessible, public archive of reports...</td>\n",
       "      <td>2023-11-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cosmic Cancer Gene Census</th>\n",
       "      <td>97</td>\n",
       "      <td>Cosmic Cancer Gene Census catalogs genes with ...</td>\n",
       "      <td>2022-11-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DANN</th>\n",
       "      <td>20200205</td>\n",
       "      <td>Pathogenicity scores of missense variants pred...</td>\n",
       "      <td>2020-02-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DECIPHER</th>\n",
       "      <td>201509</td>\n",
       "      <td>Common copy-number variants and their frequenc...</td>\n",
       "      <td>2015-09-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ensembl</th>\n",
       "      <td>110</td>\n",
       "      <td>Ensembl GRCh38 Release</td>\n",
       "      <td>2023-04-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FusionCatcher</th>\n",
       "      <td>1.33</td>\n",
       "      <td>known germline and somatic gene fusions</td>\n",
       "      <td>2020-12-22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GME</th>\n",
       "      <td>20160618</td>\n",
       "      <td>Greater Middle Eastern Variome allele frequenc...</td>\n",
       "      <td>2016-06-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gerp</th>\n",
       "      <td>20110522</td>\n",
       "      <td>GERP identifies constrained elements in multip...</td>\n",
       "      <td>2011-05-22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MITOMAP</th>\n",
       "      <td>20200819</td>\n",
       "      <td>Small variants in the MITOMAP human mitochondr...</td>\n",
       "      <td>2020-08-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MITOMAP_SV</th>\n",
       "      <td>20200819</td>\n",
       "      <td>Large structural variants in the MITOMAP human...</td>\n",
       "      <td>2020-08-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MitochondrialHeteroplasmy</th>\n",
       "      <td>20180410</td>\n",
       "      <td>Variant read frequency percentiles for the Mit...</td>\n",
       "      <td>2020-05-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OMIM</th>\n",
       "      <td>20231105</td>\n",
       "      <td>An Online Catalog of Human Genes and Genetic D...</td>\n",
       "      <td>2023-11-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PhyloPPrimate</th>\n",
       "      <td>1.0</td>\n",
       "      <td>PhyloPPrimate identifies primate-specific DNA ...</td>\n",
       "      <td>2023-11-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PrimateAI-3D</th>\n",
       "      <td>1.0</td>\n",
       "      <td>PrimateAI-3D is a deep-learning network that i...</td>\n",
       "      <td>2023-06-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REVEL</th>\n",
       "      <td>20200205</td>\n",
       "      <td>Pathogenicity scores of missense variants pred...</td>\n",
       "      <td>2020-02-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RefSeq</th>\n",
       "      <td>GCF_000001405.40-RS_2023_03</td>\n",
       "      <td>NCBI Homo sapiens Annotation Release</td>\n",
       "      <td>2023-03-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SpliceAI</th>\n",
       "      <td>1.3</td>\n",
       "      <td>SpliceAI scores, distances, etc.</td>\n",
       "      <td>2019-10-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TOPMed</th>\n",
       "      <td>freeze_5</td>\n",
       "      <td>Allele frequencies from TOPMed data</td>\n",
       "      <td>2017-08-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbSNP</th>\n",
       "      <td>156</td>\n",
       "      <td>Identifiers for observed variants</td>\n",
       "      <td>2023-09-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gnomAD</th>\n",
       "      <td>3.1.2</td>\n",
       "      <td>Allele frequencies from Genome Aggregation Dat...</td>\n",
       "      <td>2021-10-22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gnomAD_LCR</th>\n",
       "      <td>2.1</td>\n",
       "      <td>Low complexity regions from gnomAD.</td>\n",
       "      <td>2019-04-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gnomAD_SV</th>\n",
       "      <td>2.1</td>\n",
       "      <td>A structural variation reference from Genome A...</td>\n",
       "      <td>2020-08-20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gnomAD_gene_scores</th>\n",
       "      <td>2.1</td>\n",
       "      <td>Gene Loss of function scores from Genome Aggre...</td>\n",
       "      <td>2018-10-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phyloP</th>\n",
       "      <td>hg38</td>\n",
       "      <td>20 way conservation score between humans and 1...</td>\n",
       "      <td>2015-04-17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        version   \n",
       "name                                                              \n",
       "1000 Genomes Project                             Phase 3 v3plus  \\\n",
       "1000 Genomes Project (SV)                           Phase 3 v5a   \n",
       "COSMIC                                                       96   \n",
       "COSMIC gene fusions                                          96   \n",
       "CancerHotspots                                             2017   \n",
       "ClinGen                                                20160414   \n",
       "ClinGen Dosage Sensitivity Map                         20231105   \n",
       "ClinGen disease validity curations                     20231105   \n",
       "ClinVar                                                20231028   \n",
       "Cosmic Cancer Gene Census                                    97   \n",
       "DANN                                                   20200205   \n",
       "DECIPHER                                                 201509   \n",
       "Ensembl                                                     110   \n",
       "FusionCatcher                                              1.33   \n",
       "GME                                                    20160618   \n",
       "Gerp                                                   20110522   \n",
       "MITOMAP                                                20200819   \n",
       "MITOMAP_SV                                             20200819   \n",
       "MitochondrialHeteroplasmy                              20180410   \n",
       "OMIM                                                   20231105   \n",
       "PhyloPPrimate                                               1.0   \n",
       "PrimateAI-3D                                                1.0   \n",
       "REVEL                                                  20200205   \n",
       "RefSeq                              GCF_000001405.40-RS_2023_03   \n",
       "SpliceAI                                                    1.3   \n",
       "TOPMed                                                 freeze_5   \n",
       "dbSNP                                                       156   \n",
       "gnomAD                                                    3.1.2   \n",
       "gnomAD_LCR                                                  2.1   \n",
       "gnomAD_SV                                                   2.1   \n",
       "gnomAD_gene_scores                                          2.1   \n",
       "phyloP                                                     hg38   \n",
       "\n",
       "                                                                          description   \n",
       "name                                                                                    \n",
       "1000 Genomes Project                A public catalogue of human variation and geno...  \\\n",
       "1000 Genomes Project (SV)           A public catalogue of human variation and geno...   \n",
       "COSMIC                              resource for exploring the impact of somatic m...   \n",
       "COSMIC gene fusions                             manually curated somatic gene fusions   \n",
       "CancerHotspots                      A resouce for statistically significant mutati...   \n",
       "ClinGen                                                                          None   \n",
       "ClinGen Dosage Sensitivity Map            Dosage sensitivity map from ClinGen (dbVar)   \n",
       "ClinGen disease validity curations    Disease validity curations from ClinGen (dbVar)   \n",
       "ClinVar                             A freely accessible, public archive of reports...   \n",
       "Cosmic Cancer Gene Census           Cosmic Cancer Gene Census catalogs genes with ...   \n",
       "DANN                                Pathogenicity scores of missense variants pred...   \n",
       "DECIPHER                            Common copy-number variants and their frequenc...   \n",
       "Ensembl                                                        Ensembl GRCh38 Release   \n",
       "FusionCatcher                                 known germline and somatic gene fusions   \n",
       "GME                                 Greater Middle Eastern Variome allele frequenc...   \n",
       "Gerp                                GERP identifies constrained elements in multip...   \n",
       "MITOMAP                             Small variants in the MITOMAP human mitochondr...   \n",
       "MITOMAP_SV                          Large structural variants in the MITOMAP human...   \n",
       "MitochondrialHeteroplasmy           Variant read frequency percentiles for the Mit...   \n",
       "OMIM                                An Online Catalog of Human Genes and Genetic D...   \n",
       "PhyloPPrimate                       PhyloPPrimate identifies primate-specific DNA ...   \n",
       "PrimateAI-3D                        PrimateAI-3D is a deep-learning network that i...   \n",
       "REVEL                               Pathogenicity scores of missense variants pred...   \n",
       "RefSeq                                           NCBI Homo sapiens Annotation Release   \n",
       "SpliceAI                                             SpliceAI scores, distances, etc.   \n",
       "TOPMed                                            Allele frequencies from TOPMed data   \n",
       "dbSNP                                               Identifiers for observed variants   \n",
       "gnomAD                              Allele frequencies from Genome Aggregation Dat...   \n",
       "gnomAD_LCR                                        Low complexity regions from gnomAD.   \n",
       "gnomAD_SV                           A structural variation reference from Genome A...   \n",
       "gnomAD_gene_scores                  Gene Loss of function scores from Genome Aggre...   \n",
       "phyloP                              20 way conservation score between humans and 1...   \n",
       "\n",
       "                                   releaseDate  \n",
       "name                                            \n",
       "1000 Genomes Project                2013-05-27  \n",
       "1000 Genomes Project (SV)           2013-05-27  \n",
       "COSMIC                              2022-05-31  \n",
       "COSMIC gene fusions                 2023-11-07  \n",
       "CancerHotspots                      2017-01-01  \n",
       "ClinGen                             2016-04-14  \n",
       "ClinGen Dosage Sensitivity Map      2023-11-05  \n",
       "ClinGen disease validity curations  2023-11-05  \n",
       "ClinVar                             2023-11-05  \n",
       "Cosmic Cancer Gene Census           2022-11-29  \n",
       "DANN                                2020-02-05  \n",
       "DECIPHER                            2015-09-01  \n",
       "Ensembl                             2023-04-27  \n",
       "FusionCatcher                       2020-12-22  \n",
       "GME                                 2016-06-18  \n",
       "Gerp                                2011-05-22  \n",
       "MITOMAP                             2020-08-19  \n",
       "MITOMAP_SV                          2020-08-19  \n",
       "MitochondrialHeteroplasmy           2020-05-21  \n",
       "OMIM                                2023-11-05  \n",
       "PhyloPPrimate                       2023-11-29  \n",
       "PrimateAI-3D                        2023-06-02  \n",
       "REVEL                               2020-02-05  \n",
       "RefSeq                              2023-03-21  \n",
       "SpliceAI                            2019-10-04  \n",
       "TOPMed                              2017-08-28  \n",
       "dbSNP                               2023-09-28  \n",
       "gnomAD                              2021-10-22  \n",
       "gnomAD_LCR                          2019-04-10  \n",
       "gnomAD_SV                           2020-08-20  \n",
       "gnomAD_gene_scores                  2018-10-17  \n",
       "phyloP                              2015-04-17  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.data_sources"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD.pLi</th>\n",
       "      <th>gnomAD.pRec</th>\n",
       "      <th>gnomAD.pNull</th>\n",
       "      <th>gnomAD.synZ</th>\n",
       "      <th>gnomAD.misZ</th>\n",
       "      <th>gnomAD.loeuf</th>\n",
       "      <th>clingenDosageSensitivityMap.haploinsufficiency</th>\n",
       "      <th>clingenDosageSensitivityMap.triplosensitivity</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic.roleInCancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AATBC</td>\n",
       "      <td>51526.0</td>\n",
       "      <td>284837</td>\n",
       "      <td>ENSG00000215458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000291052</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>16022.0</td>\n",
       "      <td>150000</td>\n",
       "      <td>ENSG00000243064</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCG1</td>\n",
       "      <td>73.0</td>\n",
       "      <td>9619</td>\n",
       "      <td>ENSG00000160179</td>\n",
       "      <td>[{'mimNumber': 603076, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>0.112</td>\n",
       "      <td>0.888</td>\n",
       "      <td>0.00000934</td>\n",
       "      <td>0.423</td>\n",
       "      <td>2.14</td>\n",
       "      <td>0.461</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ADARB1</td>\n",
       "      <td>226.0</td>\n",
       "      <td>104</td>\n",
       "      <td>ENSG00000197381</td>\n",
       "      <td>[{'mimNumber': 601218, 'geneName': 'Adenosine ...</td>\n",
       "      <td>0.803</td>\n",
       "      <td>0.197</td>\n",
       "      <td>0.00000832</td>\n",
       "      <td>0.603</td>\n",
       "      <td>3.52</td>\n",
       "      <td>0.399</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>YBEY</td>\n",
       "      <td>1299.0</td>\n",
       "      <td>54059</td>\n",
       "      <td>ENSG00000182362</td>\n",
       "      <td>[{'mimNumber': 617461, 'geneName': 'YBEY metal...</td>\n",
       "      <td>1.34E-7</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.937</td>\n",
       "      <td>1.13</td>\n",
       "      <td>0.0149</td>\n",
       "      <td>1.92</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>YRDCP3</td>\n",
       "      <td>39921.0</td>\n",
       "      <td>100861429</td>\n",
       "      <td>ENSG00000230859</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>ZBTB21</td>\n",
       "      <td>13083.0</td>\n",
       "      <td>49854</td>\n",
       "      <td>ENSG00000173276</td>\n",
       "      <td>[{'mimNumber': 616485, 'geneName': 'Zinc finge...</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.00222</td>\n",
       "      <td>5.45E-10</td>\n",
       "      <td>-1.59</td>\n",
       "      <td>0.512</td>\n",
       "      <td>0.249</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>ZNF295-AS1</td>\n",
       "      <td>23130.0</td>\n",
       "      <td>150142</td>\n",
       "      <td>ENSG00000237232</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>ZNF355P</td>\n",
       "      <td>17218.0</td>\n",
       "      <td>100505852</td>\n",
       "      <td>ENSG00000168122</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>442 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "0         AATBC  51526.0     284837  ENSG00000215458  \\\n",
       "1        ABCC13      NaN        NaN  ENSG00000291052   \n",
       "2        ABCC13  16022.0     150000  ENSG00000243064   \n",
       "3         ABCG1     73.0       9619  ENSG00000160179   \n",
       "4        ADARB1    226.0        104  ENSG00000197381   \n",
       "..          ...      ...        ...              ...   \n",
       "437        YBEY   1299.0      54059  ENSG00000182362   \n",
       "438      YRDCP3  39921.0  100861429  ENSG00000230859   \n",
       "439      ZBTB21  13083.0      49854  ENSG00000173276   \n",
       "440  ZNF295-AS1  23130.0     150142  ENSG00000237232   \n",
       "441     ZNF355P  17218.0  100505852  ENSG00000168122   \n",
       "\n",
       "                                                  omim gnomAD.pLi gnomAD.pRec   \n",
       "0                                                  NaN        NaN         NaN  \\\n",
       "1    [{'mimNumber': 608835, 'geneName': 'ATP-bindin...        NaN         NaN   \n",
       "2    [{'mimNumber': 608835, 'geneName': 'ATP-bindin...        NaN         NaN   \n",
       "3    [{'mimNumber': 603076, 'geneName': 'ATP-bindin...      0.112       0.888   \n",
       "4    [{'mimNumber': 601218, 'geneName': 'Adenosine ...      0.803       0.197   \n",
       "..                                                 ...        ...         ...   \n",
       "437  [{'mimNumber': 617461, 'geneName': 'YBEY metal...    1.34E-7      0.0625   \n",
       "438                                                NaN        NaN         NaN   \n",
       "439  [{'mimNumber': 616485, 'geneName': 'Zinc finge...      0.998     0.00222   \n",
       "440                                                NaN        NaN         NaN   \n",
       "441                                                NaN        NaN         NaN   \n",
       "\n",
       "    gnomAD.pNull gnomAD.synZ gnomAD.misZ gnomAD.loeuf   \n",
       "0            NaN         NaN         NaN          NaN  \\\n",
       "1            NaN         NaN         NaN          NaN   \n",
       "2            NaN         NaN         NaN          NaN   \n",
       "3     0.00000934       0.423        2.14        0.461   \n",
       "4     0.00000832       0.603        3.52        0.399   \n",
       "..           ...         ...         ...          ...   \n",
       "437        0.937        1.13      0.0149         1.92   \n",
       "438          NaN         NaN         NaN          NaN   \n",
       "439     5.45E-10       -1.59       0.512        0.249   \n",
       "440          NaN         NaN         NaN          NaN   \n",
       "441          NaN         NaN         NaN          NaN   \n",
       "\n",
       "    clingenDosageSensitivityMap.haploinsufficiency   \n",
       "0                                              NaN  \\\n",
       "1                                              NaN   \n",
       "2                                              NaN   \n",
       "3                                              NaN   \n",
       "4                                              NaN   \n",
       "..                                             ...   \n",
       "437                                            NaN   \n",
       "438                                            NaN   \n",
       "439                                            NaN   \n",
       "440                                            NaN   \n",
       "441                                            NaN   \n",
       "\n",
       "    clingenDosageSensitivityMap.triplosensitivity clingenGeneValidity   \n",
       "0                                             NaN                 NaN  \\\n",
       "1                                             NaN                 NaN   \n",
       "2                                             NaN                 NaN   \n",
       "3                                             NaN                 NaN   \n",
       "4                                             NaN                 NaN   \n",
       "..                                            ...                 ...   \n",
       "437                                           NaN                 NaN   \n",
       "438                                           NaN                 NaN   \n",
       "439                                           NaN                 NaN   \n",
       "440                                           NaN                 NaN   \n",
       "441                                           NaN                 NaN   \n",
       "\n",
       "    cosmic.roleInCancer  \n",
       "0                   NaN  \n",
       "1                   NaN  \n",
       "2                   NaN  \n",
       "3                   NaN  \n",
       "4                   NaN  \n",
       "..                  ...  \n",
       "437                 NaN  \n",
       "438                 NaN  \n",
       "439                 NaN  \n",
       "440                 NaN  \n",
       "441                 NaN  \n",
       "\n",
       "[442 rows x 15 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Positions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cytogeneticBand</th>\n",
       "      <th>samples</th>\n",
       "      <th>variants</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5228221</td>\n",
       "      <td>G</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[PASS]</td>\n",
       "      <td>75.75</td>\n",
       "      <td>21p12</td>\n",
       "      <td>[{'genotype': '0/0', 'variantFrequencies': [0....</td>\n",
       "      <td>[{'vid': '21-5228221-G-T', 'chromosome': 'chr2...</td>\n",
       "      <td>1004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles filters  mappingQuality   \n",
       "0      chr21   5228221         G        [T]  [PASS]           75.75  \\\n",
       "\n",
       "  cytogeneticBand                                            samples   \n",
       "0           21p12  [{'genotype': '0/0', 'variantFrequencies': [0....  \\\n",
       "\n",
       "                                            variants vcfInfo.DP  \n",
       "0  [{'vid': '21-5228221-G-T', 'chromosome': 'chr2...       1004  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position = Position.model_validate(annotated_data.get_annotation(\"chr21\", 5228221))\n",
    "position.to_df()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>vid</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>phylopScore</th>\n",
       "      <th>phyloPPrimateScore</th>\n",
       "      <th>transcripts</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5228221</td>\n",
       "      <td>G</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[PASS]</td>\n",
       "      <td>75.75</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5228221-G-T</td>\n",
       "      <td>5228221</td>\n",
       "      <td>5228221</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5228221G&gt;T</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[{'transcript': 'ENST00000623753.1', 'source':...</td>\n",
       "      <td>1004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles filters  mappingQuality   \n",
       "0      chr21   5228221         G        [T]  [PASS]           75.75  \\\n",
       "\n",
       "  cyatogeneticBand             vid    begin      end altAllele variantType   \n",
       "0            21p12  21-5228221-G-T  5228221  5228221         T         SNV  \\\n",
       "\n",
       "                      hgvsg  phylopScore  phyloPPrimateScore   \n",
       "0  NC_000021.9:g.5228221G>T          0.2                 0.0  \\\n",
       "\n",
       "                                         transcripts vcfInfo.DP  \n",
       "0  [{'transcript': 'ENST00000623753.1', 'source':...       1004  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position.to_df(key=\"variants\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>transcript</th>\n",
       "      <th>source</th>\n",
       "      <th>bioType</th>\n",
       "      <th>geneId</th>\n",
       "      <th>hgnc</th>\n",
       "      <th>consequence</th>\n",
       "      <th>impact</th>\n",
       "      <th>isCanonical</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5228221</td>\n",
       "      <td>5228221</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000021.9:g.5228221G&gt;T</td>\n",
       "      <td>ENST00000623753.1</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENSG00000279669</td>\n",
       "      <td>ENSG00000279669</td>\n",
       "      <td>[downstream_gene_variant]</td>\n",
       "      <td>modifier</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome    begin      end refAllele altAllele                     hgvsg   \n",
       "0      chr21  5228221  5228221         G         T  NC_000021.9:g.5228221G>T  \\\n",
       "\n",
       "          transcript   source bioType           geneId             hgnc   \n",
       "0  ENST00000623753.1  Ensembl  lncRNA  ENSG00000279669  ENSG00000279669  \\\n",
       "\n",
       "                 consequence    impact  isCanonical  \n",
       "0  [downstream_gene_variant]  modifier         True  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position.variants[0].to_df(\"transcripts\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>transcript</th>\n",
       "      <th>source</th>\n",
       "      <th>bioType</th>\n",
       "      <th>geneId</th>\n",
       "      <th>hgnc</th>\n",
       "      <th>consequence</th>\n",
       "      <th>impact</th>\n",
       "      <th>isCanonical</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5228221</td>\n",
       "      <td>5228221</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000021.9:g.5228221G&gt;T</td>\n",
       "      <td>ENST00000623753.1</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENSG00000279669</td>\n",
       "      <td>ENSG00000279669</td>\n",
       "      <td>[downstream_gene_variant]</td>\n",
       "      <td>modifier</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome    begin      end refAllele altAllele                     hgvsg   \n",
       "0      chr21  5228221  5228221         G         T  NC_000021.9:g.5228221G>T  \\\n",
       "\n",
       "          transcript   source bioType           geneId             hgnc   \n",
       "0  ENST00000623753.1  Ensembl  lncRNA  ENSG00000279669  ENSG00000279669  \\\n",
       "\n",
       "                 consequence    impact  isCanonical  \n",
       "0  [downstream_gene_variant]  modifier         True  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AnnotatedData.multiple_to_df(position.variants, key=\"transcripts\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5222289</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[mapping_quality, weak_evidence]</td>\n",
       "      <td>49.7</td>\n",
       "      <td>21p12</td>\n",
       "      <td>749</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles                           filters   \n",
       "0      chr21   5222289         C        [T]  [mapping_quality, weak_evidence]  \\\n",
       "\n",
       "   mappingQuality cyatogeneticBand vcfInfo.DP  \n",
       "0            49.7            21p12        749  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position = Position.model_validate(annotated_data.get_annotation(\"chr21\", 5222289))\n",
    "position.get_top_level()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cytogeneticBand</th>\n",
       "      <th>samples</th>\n",
       "      <th>variants</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5222289</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[mapping_quality, weak_evidence]</td>\n",
       "      <td>49.7</td>\n",
       "      <td>21p12</td>\n",
       "      <td>[{'genotype': '0/0', 'variantFrequencies': [0....</td>\n",
       "      <td>[{'vid': '21-5222289-C-T', 'chromosome': 'chr2...</td>\n",
       "      <td>749</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles                           filters   \n",
       "0      chr21   5222289         C        [T]  [mapping_quality, weak_evidence]  \\\n",
       "\n",
       "   mappingQuality cytogeneticBand   \n",
       "0            49.7           21p12  \\\n",
       "\n",
       "                                             samples   \n",
       "0  [{'genotype': '0/0', 'variantFrequencies': [0....  \\\n",
       "\n",
       "                                            variants vcfInfo.DP  \n",
       "0  [{'vid': '21-5222289-C-T', 'chromosome': 'chr2...        749  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position.to_df()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>cytogeneticBand</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5222289</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[mapping_quality, weak_evidence]</td>\n",
       "      <td>49.7</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21p12</td>\n",
       "      <td>749</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles                           filters   \n",
       "0      chr21   5222289         C        [T]  [mapping_quality, weak_evidence]  \\\n",
       "\n",
       "   mappingQuality cyatogeneticBand cytogeneticBand vcfInfo.DP  \n",
       "0            49.7            21p12           21p12        749  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position.to_df(\"cytogeneticBand\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>vid</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>phylopScore</th>\n",
       "      <th>phyloPPrimateScore</th>\n",
       "      <th>transcripts</th>\n",
       "      <th>dbsnp</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "      <th>gnomad.coverage</th>\n",
       "      <th>gnomad.allAf</th>\n",
       "      <th>gnomad.allAn</th>\n",
       "      <th>gnomad.allAc</th>\n",
       "      <th>gnomad.allHc</th>\n",
       "      <th>gnomad.afrAf</th>\n",
       "      <th>gnomad.afrAn</th>\n",
       "      <th>gnomad.afrAc</th>\n",
       "      <th>gnomad.afrHc</th>\n",
       "      <th>gnomad.amrAf</th>\n",
       "      <th>gnomad.amrAn</th>\n",
       "      <th>gnomad.amrAc</th>\n",
       "      <th>gnomad.amrHc</th>\n",
       "      <th>gnomad.easAf</th>\n",
       "      <th>gnomad.easAn</th>\n",
       "      <th>gnomad.easAc</th>\n",
       "      <th>gnomad.easHc</th>\n",
       "      <th>gnomad.finAf</th>\n",
       "      <th>gnomad.finAn</th>\n",
       "      <th>gnomad.finAc</th>\n",
       "      <th>gnomad.finHc</th>\n",
       "      <th>gnomad.nfeAf</th>\n",
       "      <th>gnomad.nfeAn</th>\n",
       "      <th>gnomad.nfeAc</th>\n",
       "      <th>gnomad.nfeHc</th>\n",
       "      <th>gnomad.asjAf</th>\n",
       "      <th>gnomad.asjAn</th>\n",
       "      <th>gnomad.asjAc</th>\n",
       "      <th>gnomad.asjHc</th>\n",
       "      <th>gnomad.sasAf</th>\n",
       "      <th>gnomad.sasAn</th>\n",
       "      <th>gnomad.sasAc</th>\n",
       "      <th>gnomad.sasHc</th>\n",
       "      <th>gnomad.othAf</th>\n",
       "      <th>gnomad.othAn</th>\n",
       "      <th>gnomad.othAc</th>\n",
       "      <th>gnomad.othHc</th>\n",
       "      <th>gnomad.maleAf</th>\n",
       "      <th>gnomad.maleAn</th>\n",
       "      <th>gnomad.maleAc</th>\n",
       "      <th>gnomad.maleHc</th>\n",
       "      <th>gnomad.femaleAf</th>\n",
       "      <th>gnomad.femaleAn</th>\n",
       "      <th>gnomad.femaleAc</th>\n",
       "      <th>gnomad.femaleHc</th>\n",
       "      <th>gnomad.controlsAllAf</th>\n",
       "      <th>gnomad.controlsAllAn</th>\n",
       "      <th>gnomad.controlsAllAc</th>\n",
       "      <th>topmed.allAf</th>\n",
       "      <th>topmed.allAn</th>\n",
       "      <th>topmed.allAc</th>\n",
       "      <th>topmed.allHc</th>\n",
       "      <th>topmed.failedFilter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5222289</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[mapping_quality, weak_evidence]</td>\n",
       "      <td>49.7</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5222289-C-T</td>\n",
       "      <td>5222289</td>\n",
       "      <td>5222289</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5222289C&gt;T</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.074</td>\n",
       "      <td>None</td>\n",
       "      <td>[rs1366179382]</td>\n",
       "      <td>749</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000502</td>\n",
       "      <td>149274</td>\n",
       "      <td>75</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001704</td>\n",
       "      <td>40502</td>\n",
       "      <td>69</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>14966</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5022</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10430</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000075</td>\n",
       "      <td>66950</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3438</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4734</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2024</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000411</td>\n",
       "      <td>73006</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00059</td>\n",
       "      <td>76268</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000565</td>\n",
       "      <td>31850</td>\n",
       "      <td>18</td>\n",
       "      <td>0.001091</td>\n",
       "      <td>125568</td>\n",
       "      <td>137</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles                           filters   \n",
       "0      chr21   5222289         C        [T]  [mapping_quality, weak_evidence]  \\\n",
       "\n",
       "   mappingQuality cyatogeneticBand             vid    begin      end   \n",
       "0            49.7            21p12  21-5222289-C-T  5222289  5222289  \\\n",
       "\n",
       "  altAllele variantType                     hgvsg  phylopScore   \n",
       "0         T         SNV  NC_000021.9:g.5222289C>T         -0.2  \\\n",
       "\n",
       "   phyloPPrimateScore transcripts           dbsnp vcfInfo.DP  gnomad.coverage   \n",
       "0               0.074        None  [rs1366179382]        749                3  \\\n",
       "\n",
       "  gnomad.allAf  gnomad.allAn  gnomad.allAc  gnomad.allHc gnomad.afrAf   \n",
       "0     0.000502        149274            75             0     0.001704  \\\n",
       "\n",
       "   gnomad.afrAn  gnomad.afrAc  gnomad.afrHc gnomad.amrAf  gnomad.amrAn   \n",
       "0         40502            69             0     0.000067         14966  \\\n",
       "\n",
       "   gnomad.amrAc  gnomad.amrHc  gnomad.easAf  gnomad.easAn  gnomad.easAc   \n",
       "0             1             0             0          5022             0  \\\n",
       "\n",
       "   gnomad.easHc  gnomad.finAf  gnomad.finAn  gnomad.finAc  gnomad.finHc   \n",
       "0             0             0         10430             0             0  \\\n",
       "\n",
       "  gnomad.nfeAf  gnomad.nfeAn  gnomad.nfeAc  gnomad.nfeHc  gnomad.asjAf   \n",
       "0     0.000075         66950             5             0             0  \\\n",
       "\n",
       "   gnomad.asjAn  gnomad.asjAc  gnomad.asjHc  gnomad.sasAf  gnomad.sasAn   \n",
       "0          3438             0             0             0          4734  \\\n",
       "\n",
       "   gnomad.sasAc  gnomad.sasHc  gnomad.othAf  gnomad.othAn  gnomad.othAc   \n",
       "0             0             0             0          2024             0  \\\n",
       "\n",
       "   gnomad.othHc gnomad.maleAf  gnomad.maleAn  gnomad.maleAc  gnomad.maleHc   \n",
       "0             0      0.000411          73006             30              0  \\\n",
       "\n",
       "  gnomad.femaleAf  gnomad.femaleAn  gnomad.femaleAc  gnomad.femaleHc   \n",
       "0         0.00059            76268               45                0  \\\n",
       "\n",
       "  gnomad.controlsAllAf  gnomad.controlsAllAn  gnomad.controlsAllAc   \n",
       "0             0.000565                 31850                    18  \\\n",
       "\n",
       "  topmed.allAf  topmed.allAn  topmed.allAc  topmed.allHc  topmed.failedFilter  \n",
       "0     0.001091        125568           137             0                 True  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "position.to_df(\"variants\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Parsing and Filtering Positions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "class Parser:\n",
    "    def __init__(self, annotated_data: AnnotatedData):\n",
    "        self.annotated_data = annotated_data\n",
    "\n",
    "    def get_variants_above_gnomad_freq(\n",
    "        self,\n",
    "        frequency_key: str,\n",
    "        frequency_threshold_low: float = float(\"-inf\"),\n",
    "        frequency_threshold_high: float = float(\"inf\"),\n",
    "    ) -> Generator[Any, Any, None]:\n",
    "        positions = (\n",
    "            Position.model_validate(position)\n",
    "            for position in self.annotated_data.positions\n",
    "            for variant in position.get(\"variants\", {})\n",
    "            if (freq := variant.get(\"gnomad\", {}).get(frequency_key, None))\n",
    "            and frequency_threshold_low < freq < frequency_threshold_high\n",
    "        )\n",
    "        return positions\n",
    "\n",
    "    def get_positions_with_cannonical_transcripts(self) -> Generator[Any, Any, None]:\n",
    "        positions = (\n",
    "            Position.model_validate(position)\n",
    "            for position in self.annotated_data.positions\n",
    "            for variant in position.get(\"variants\", {})\n",
    "            for transcript in variant.get(\"transcripts\", [])\n",
    "            if transcript.get(\"isCanonical\")\n",
    "        )\n",
    "\n",
    "        return positions\n",
    "\n",
    "    def filter_transcripts_by_consequence(\n",
    "        self, include: Optional[List[str]] = None, exclude: Optional[List[str]] = None\n",
    "    ) -> Generator[Any, Any, None]:\n",
    "        if not exclude:\n",
    "            exclude = []\n",
    "\n",
    "        if not include:\n",
    "            include = []\n",
    "\n",
    "        positions = (\n",
    "            Position.model_validate(position)\n",
    "            for position in self.annotated_data.positions\n",
    "            for variant in position.get(\"variants\", {})\n",
    "            for transcript in variant.get(\"transcripts\", [])\n",
    "            for consequence in transcript.get(\"consequence\", [])\n",
    "            if (not bool(include) or consequence in include) and consequence not in exclude\n",
    "        )\n",
    "        return positions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Positions with Cannonical Transcripts only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = Parser(annotated_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2472"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = list(parser.get_positions_with_cannonical_transcripts())\n",
    "len(positions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# AnnotatedData.multiple_to_df(positions, \"variants\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>vid</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>phylopScore</th>\n",
       "      <th>phyloPPrimateScore</th>\n",
       "      <th>transcripts</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5228221</td>\n",
       "      <td>G</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[PASS]</td>\n",
       "      <td>75.75</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5228221-G-T</td>\n",
       "      <td>5228221</td>\n",
       "      <td>5228221</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5228221G&gt;T</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[{'transcript': 'ENST00000623753.1', 'source':...</td>\n",
       "      <td>1004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles filters  mappingQuality   \n",
       "0      chr21   5228221         G        [T]  [PASS]           75.75  \\\n",
       "\n",
       "  cyatogeneticBand             vid    begin      end altAllele variantType   \n",
       "0            21p12  21-5228221-G-T  5228221  5228221         T         SNV  \\\n",
       "\n",
       "                      hgvsg  phylopScore  phyloPPrimateScore   \n",
       "0  NC_000021.9:g.5228221G>T          0.2                 0.0  \\\n",
       "\n",
       "                                         transcripts vcfInfo.DP  \n",
       "0  [{'transcript': 'ENST00000623753.1', 'source':...       1004  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions[0].to_df(\"variants\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>transcript</th>\n",
       "      <th>source</th>\n",
       "      <th>bioType</th>\n",
       "      <th>geneId</th>\n",
       "      <th>hgnc</th>\n",
       "      <th>consequence</th>\n",
       "      <th>impact</th>\n",
       "      <th>isCanonical</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5228221</td>\n",
       "      <td>5228221</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000021.9:g.5228221G&gt;T</td>\n",
       "      <td>ENST00000623753.1</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENSG00000279669</td>\n",
       "      <td>ENSG00000279669</td>\n",
       "      <td>[downstream_gene_variant]</td>\n",
       "      <td>modifier</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome    begin      end refAllele altAllele                     hgvsg   \n",
       "0      chr21  5228221  5228221         G         T  NC_000021.9:g.5228221G>T  \\\n",
       "\n",
       "          transcript   source bioType           geneId             hgnc   \n",
       "0  ENST00000623753.1  Ensembl  lncRNA  ENSG00000279669  ENSG00000279669  \\\n",
       "\n",
       "                 consequence    impact  isCanonical  \n",
       "0  [downstream_gene_variant]  modifier         True  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions[0].variants[0].to_df(\"transcripts\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter by Consequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = Parser(annotated_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "147"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = list(parser.filter_transcripts_by_consequence(\n",
    "    include=[\"non_coding_transcript_exon_variant\"]\n",
    "))\n",
    "len(positions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>vid</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>phylopScore</th>\n",
       "      <th>phyloPPrimateScore</th>\n",
       "      <th>transcripts</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "      <th>regulatoryRegions</th>\n",
       "      <th>gnomad.coverage</th>\n",
       "      <th>gnomad.failedFilter</th>\n",
       "      <th>gnomad.allAf</th>\n",
       "      <th>gnomad.allAn</th>\n",
       "      <th>gnomad.allAc</th>\n",
       "      <th>gnomad.allHc</th>\n",
       "      <th>gnomad.afrAf</th>\n",
       "      <th>gnomad.afrAn</th>\n",
       "      <th>gnomad.afrAc</th>\n",
       "      <th>gnomad.afrHc</th>\n",
       "      <th>gnomad.amrAf</th>\n",
       "      <th>gnomad.amrAn</th>\n",
       "      <th>gnomad.amrAc</th>\n",
       "      <th>gnomad.amrHc</th>\n",
       "      <th>gnomad.easAf</th>\n",
       "      <th>gnomad.easAn</th>\n",
       "      <th>gnomad.easAc</th>\n",
       "      <th>gnomad.easHc</th>\n",
       "      <th>gnomad.finAf</th>\n",
       "      <th>gnomad.finAn</th>\n",
       "      <th>gnomad.finAc</th>\n",
       "      <th>gnomad.finHc</th>\n",
       "      <th>gnomad.nfeAf</th>\n",
       "      <th>gnomad.nfeAn</th>\n",
       "      <th>gnomad.nfeAc</th>\n",
       "      <th>gnomad.nfeHc</th>\n",
       "      <th>gnomad.asjAf</th>\n",
       "      <th>gnomad.asjAn</th>\n",
       "      <th>gnomad.asjAc</th>\n",
       "      <th>gnomad.asjHc</th>\n",
       "      <th>gnomad.sasAf</th>\n",
       "      <th>gnomad.sasAn</th>\n",
       "      <th>gnomad.sasAc</th>\n",
       "      <th>gnomad.sasHc</th>\n",
       "      <th>gnomad.othAf</th>\n",
       "      <th>gnomad.othAn</th>\n",
       "      <th>gnomad.othAc</th>\n",
       "      <th>gnomad.othHc</th>\n",
       "      <th>gnomad.maleAf</th>\n",
       "      <th>gnomad.maleAn</th>\n",
       "      <th>gnomad.maleAc</th>\n",
       "      <th>gnomad.maleHc</th>\n",
       "      <th>gnomad.femaleAf</th>\n",
       "      <th>gnomad.femaleAn</th>\n",
       "      <th>gnomad.femaleAc</th>\n",
       "      <th>gnomad.femaleHc</th>\n",
       "      <th>gnomad.controlsAllAf</th>\n",
       "      <th>gnomad.controlsAllAn</th>\n",
       "      <th>gnomad.controlsAllAc</th>\n",
       "      <th>dbsnp</th>\n",
       "      <th>topmed.allAf</th>\n",
       "      <th>topmed.allAn</th>\n",
       "      <th>topmed.allAc</th>\n",
       "      <th>topmed.allHc</th>\n",
       "      <th>topmed.failedFilter</th>\n",
       "      <th>dannScore</th>\n",
       "      <th>gerpScore</th>\n",
       "      <th>inLowComplexityRegion</th>\n",
       "      <th>oneKg.allAf</th>\n",
       "      <th>oneKg.afrAf</th>\n",
       "      <th>oneKg.amrAf</th>\n",
       "      <th>oneKg.easAf</th>\n",
       "      <th>oneKg.eurAf</th>\n",
       "      <th>oneKg.sasAf</th>\n",
       "      <th>oneKg.allAn</th>\n",
       "      <th>oneKg.afrAn</th>\n",
       "      <th>oneKg.amrAn</th>\n",
       "      <th>oneKg.easAn</th>\n",
       "      <th>oneKg.eurAn</th>\n",
       "      <th>oneKg.sasAn</th>\n",
       "      <th>oneKg.allAc</th>\n",
       "      <th>oneKg.afrAc</th>\n",
       "      <th>oneKg.amrAc</th>\n",
       "      <th>oneKg.easAc</th>\n",
       "      <th>oneKg.eurAc</th>\n",
       "      <th>oneKg.sasAc</th>\n",
       "      <th>primateAI-3D</th>\n",
       "      <th>revel.score</th>\n",
       "      <th>cosmic</th>\n",
       "      <th>spliceAI</th>\n",
       "      <th>clinvar</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5232869</td>\n",
       "      <td>T</td>\n",
       "      <td>[G]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>58.10</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5232869-T-G</td>\n",
       "      <td>5232869</td>\n",
       "      <td>5232869</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5232869T&gt;G</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'transcript': 'ENST00000623753.1', 'source':...</td>\n",
       "      <td>799</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5243812</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[PASS]</td>\n",
       "      <td>108.19</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5243812-C-T</td>\n",
       "      <td>5243812</td>\n",
       "      <td>5243812</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5243812C&gt;T</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.074</td>\n",
       "      <td>[{'transcript': 'ENST00000623753.1', 'source':...</td>\n",
       "      <td>514</td>\n",
       "      <td>[{'id': 'ENSR00000140073', 'type': 'TF_binding...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>152296.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>41480.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>15294.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>5206.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>10628.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68058.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3472.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4836.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2094.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>74406.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>77890.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>32922.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>8209019</td>\n",
       "      <td>-</td>\n",
       "      <td>[GCGC]</td>\n",
       "      <td>[filtered_reads, mapping_quality, no_reliable_...</td>\n",
       "      <td>9.94</td>\n",
       "      <td>21p11.2</td>\n",
       "      <td>21-8209019-G-GCGC</td>\n",
       "      <td>8209020</td>\n",
       "      <td>8209019</td>\n",
       "      <td>CGC</td>\n",
       "      <td>insertion</td>\n",
       "      <td>NC_000021.9:g.8209022_8209024dup</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'transcript': 'ENST00000623664.1', 'source':...</td>\n",
       "      <td>1317</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000379</td>\n",
       "      <td>145094.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>40532.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000205</td>\n",
       "      <td>14652.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>5082.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000367</td>\n",
       "      <td>8170.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000582</td>\n",
       "      <td>65286.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002097</td>\n",
       "      <td>3338.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4814.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2000.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000311</td>\n",
       "      <td>70634.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000443</td>\n",
       "      <td>74460.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000297</td>\n",
       "      <td>30260.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>[rs1555877791]</td>\n",
       "      <td>0.00039</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>8213090</td>\n",
       "      <td>T</td>\n",
       "      <td>[G]</td>\n",
       "      <td>[base_quality, filtered_reads, mapping_quality...</td>\n",
       "      <td>21.85</td>\n",
       "      <td>21p11.2</td>\n",
       "      <td>21-8213090-T-G</td>\n",
       "      <td>8213090</td>\n",
       "      <td>8213090</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.8213090T&gt;G</td>\n",
       "      <td>-1.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'transcript': 'ENST00000623664.1', 'source':...</td>\n",
       "      <td>1841</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>8213222</td>\n",
       "      <td>A</td>\n",
       "      <td>[C]</td>\n",
       "      <td>[filtered_reads, mapping_quality, no_reliable_...</td>\n",
       "      <td>18.39</td>\n",
       "      <td>21p11.2</td>\n",
       "      <td>21-8213222-A-C</td>\n",
       "      <td>8213222</td>\n",
       "      <td>8213222</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.8213222A&gt;C</td>\n",
       "      <td>0.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'transcript': 'ENST00000623664.1', 'source':...</td>\n",
       "      <td>1173</td>\n",
       "      <td>NaN</td>\n",
       "      <td>178.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.154747</td>\n",
       "      <td>118316.0</td>\n",
       "      <td>18309.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.076287</td>\n",
       "      <td>34816.0</td>\n",
       "      <td>2656.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.117568</td>\n",
       "      <td>11942.0</td>\n",
       "      <td>1404.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.003146</td>\n",
       "      <td>5086.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.164856</td>\n",
       "      <td>7734.0</td>\n",
       "      <td>1275.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.228802</td>\n",
       "      <td>49816.0</td>\n",
       "      <td>11398.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.276284</td>\n",
       "      <td>2454.0</td>\n",
       "      <td>678.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.098278</td>\n",
       "      <td>3948.0</td>\n",
       "      <td>388.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.182048</td>\n",
       "      <td>1582.0</td>\n",
       "      <td>288.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.148306</td>\n",
       "      <td>57840.0</td>\n",
       "      <td>8578.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.160907</td>\n",
       "      <td>60476.0</td>\n",
       "      <td>9731.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.126071</td>\n",
       "      <td>25922.0</td>\n",
       "      <td>3268.0</td>\n",
       "      <td>[rs1460148663]</td>\n",
       "      <td>0.210945</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>26488.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46193167</td>\n",
       "      <td>T</td>\n",
       "      <td>[C]</td>\n",
       "      <td>[base_quality, mapping_quality, non_homref_nor...</td>\n",
       "      <td>56.43</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46193167-T-C</td>\n",
       "      <td>46193167</td>\n",
       "      <td>46193167</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46193167T&gt;C</td>\n",
       "      <td>-2.0</td>\n",
       "      <td>-3.537</td>\n",
       "      <td>[{'transcript': 'ENST00000415026.1', 'source':...</td>\n",
       "      <td>130</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002211</td>\n",
       "      <td>94994.0</td>\n",
       "      <td>210.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.003126</td>\n",
       "      <td>24314.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001071</td>\n",
       "      <td>9340.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>2846.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001511</td>\n",
       "      <td>5958.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001844</td>\n",
       "      <td>45550.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2332.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002682</td>\n",
       "      <td>2610.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00219</td>\n",
       "      <td>1370.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002105</td>\n",
       "      <td>46556.0</td>\n",
       "      <td>98.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002312</td>\n",
       "      <td>48438.0</td>\n",
       "      <td>112.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.003282</td>\n",
       "      <td>17368.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>[rs112603162]</td>\n",
       "      <td>0.123869</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>15554.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.007</td>\n",
       "      <td>[{'id': 'COSV62702051', 'numSamples': 2, 'refA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46286556</td>\n",
       "      <td>T</td>\n",
       "      <td>[G]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>213.21</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46286556-T-G</td>\n",
       "      <td>46286556</td>\n",
       "      <td>46286556</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46286556T&gt;G</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>-1.756</td>\n",
       "      <td>[{'transcript': 'ENST00000291688.6', 'source':...</td>\n",
       "      <td>93</td>\n",
       "      <td>[{'id': 'ENSR00000143372', 'type': 'promoter',...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46325257</td>\n",
       "      <td>A</td>\n",
       "      <td>[C]</td>\n",
       "      <td>[base_quality, non_homref_normal, weak_evidence]</td>\n",
       "      <td>221.46</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46325257-A-C</td>\n",
       "      <td>46325257</td>\n",
       "      <td>46325257</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46325257A&gt;C</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>-2.122</td>\n",
       "      <td>[{'transcript': 'ENST00000397683.5', 'source':...</td>\n",
       "      <td>144</td>\n",
       "      <td>[{'id': 'ENSR00000143378', 'type': 'promoter',...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.54</td>\n",
       "      <td>-0.51</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46436858</td>\n",
       "      <td>T</td>\n",
       "      <td>[G]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>216.57</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46436858-T-G</td>\n",
       "      <td>46436858</td>\n",
       "      <td>46436858</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46436858T&gt;G</td>\n",
       "      <td>0.1</td>\n",
       "      <td>-1.329</td>\n",
       "      <td>[{'transcript': 'ENST00000703224.1', 'source':...</td>\n",
       "      <td>149</td>\n",
       "      <td>[{'id': 'ENSR00000665397', 'type': 'enhancer',...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.67</td>\n",
       "      <td>-0.837</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'hgnc': 'PCNT', 'donorGainScore': 0.6, 'dono...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46643510</td>\n",
       "      <td>T</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>211.58</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46643510-T-A</td>\n",
       "      <td>46643510</td>\n",
       "      <td>46643510</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46643510T&gt;A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.603</td>\n",
       "      <td>[{'transcript': 'ENST00000451211.6', 'source':...</td>\n",
       "      <td>150</td>\n",
       "      <td>[{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>141118.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>39060.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>14018.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000232</td>\n",
       "      <td>4306.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>7762.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>65358.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3364.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4110.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1968.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68068.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>73050.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000108</td>\n",
       "      <td>27846.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[rs1601921720]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.58</td>\n",
       "      <td>0.622</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>147 rows × 99 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome  position refAllele altAlleles   \n",
       "0       chr21   5232869         T        [G]  \\\n",
       "0       chr21   5243812         C        [T]   \n",
       "0       chr21   8209019         -     [GCGC]   \n",
       "0       chr21   8213090         T        [G]   \n",
       "0       chr21   8213222         A        [C]   \n",
       "..        ...       ...       ...        ...   \n",
       "0       chr21  46193167         T        [C]   \n",
       "0       chr21  46286556         T        [G]   \n",
       "0       chr21  46325257         A        [C]   \n",
       "0       chr21  46436858         T        [G]   \n",
       "0       chr21  46643510         T        [A]   \n",
       "\n",
       "                                              filters  mappingQuality   \n",
       "0                       [base_quality, weak_evidence]           58.10  \\\n",
       "0                                              [PASS]          108.19   \n",
       "0   [filtered_reads, mapping_quality, no_reliable_...            9.94   \n",
       "0   [base_quality, filtered_reads, mapping_quality...           21.85   \n",
       "0   [filtered_reads, mapping_quality, no_reliable_...           18.39   \n",
       "..                                                ...             ...   \n",
       "0   [base_quality, mapping_quality, non_homref_nor...           56.43   \n",
       "0                       [base_quality, weak_evidence]          213.21   \n",
       "0    [base_quality, non_homref_normal, weak_evidence]          221.46   \n",
       "0                       [base_quality, weak_evidence]          216.57   \n",
       "0                       [base_quality, weak_evidence]          211.58   \n",
       "\n",
       "   cyatogeneticBand                vid     begin       end altAllele   \n",
       "0             21p12     21-5232869-T-G   5232869   5232869         G  \\\n",
       "0             21p12     21-5243812-C-T   5243812   5243812         T   \n",
       "0           21p11.2  21-8209019-G-GCGC   8209020   8209019       CGC   \n",
       "0           21p11.2     21-8213090-T-G   8213090   8213090         G   \n",
       "0           21p11.2     21-8213222-A-C   8213222   8213222         C   \n",
       "..              ...                ...       ...       ...       ...   \n",
       "0           21q22.3    21-46193167-T-C  46193167  46193167         C   \n",
       "0           21q22.3    21-46286556-T-G  46286556  46286556         G   \n",
       "0           21q22.3    21-46325257-A-C  46325257  46325257         C   \n",
       "0           21q22.3    21-46436858-T-G  46436858  46436858         G   \n",
       "0           21q22.3    21-46643510-T-A  46643510  46643510         A   \n",
       "\n",
       "   variantType                             hgvsg  phylopScore   \n",
       "0          SNV          NC_000021.9:g.5232869T>G          0.5  \\\n",
       "0          SNV          NC_000021.9:g.5243812C>T          0.4   \n",
       "0    insertion  NC_000021.9:g.8209022_8209024dup          NaN   \n",
       "0          SNV          NC_000021.9:g.8213090T>G         -1.8   \n",
       "0          SNV          NC_000021.9:g.8213222A>C          0.2   \n",
       "..         ...                               ...          ...   \n",
       "0          SNV         NC_000021.9:g.46193167T>C         -2.0   \n",
       "0          SNV         NC_000021.9:g.46286556T>G         -0.2   \n",
       "0          SNV         NC_000021.9:g.46325257A>C         -0.5   \n",
       "0          SNV         NC_000021.9:g.46436858T>G          0.1   \n",
       "0          SNV         NC_000021.9:g.46643510T>A          0.0   \n",
       "\n",
       "    phyloPPrimateScore                                        transcripts   \n",
       "0                  NaN  [{'transcript': 'ENST00000623753.1', 'source':...  \\\n",
       "0                0.074  [{'transcript': 'ENST00000623753.1', 'source':...   \n",
       "0                  NaN  [{'transcript': 'ENST00000623664.1', 'source':...   \n",
       "0                  NaN  [{'transcript': 'ENST00000623664.1', 'source':...   \n",
       "0                  NaN  [{'transcript': 'ENST00000623664.1', 'source':...   \n",
       "..                 ...                                                ...   \n",
       "0               -3.537  [{'transcript': 'ENST00000415026.1', 'source':...   \n",
       "0               -1.756  [{'transcript': 'ENST00000291688.6', 'source':...   \n",
       "0               -2.122  [{'transcript': 'ENST00000397683.5', 'source':...   \n",
       "0               -1.329  [{'transcript': 'ENST00000703224.1', 'source':...   \n",
       "0               -0.603  [{'transcript': 'ENST00000451211.6', 'source':...   \n",
       "\n",
       "   vcfInfo.DP                                  regulatoryRegions   \n",
       "0         799                                                NaN  \\\n",
       "0         514  [{'id': 'ENSR00000140073', 'type': 'TF_binding...   \n",
       "0        1317                                                NaN   \n",
       "0        1841                                                NaN   \n",
       "0        1173                                                NaN   \n",
       "..        ...                                                ...   \n",
       "0         130                                                NaN   \n",
       "0          93  [{'id': 'ENSR00000143372', 'type': 'promoter',...   \n",
       "0         144  [{'id': 'ENSR00000143378', 'type': 'promoter',...   \n",
       "0         149  [{'id': 'ENSR00000665397', 'type': 'enhancer',...   \n",
       "0         150  [{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...   \n",
       "\n",
       "    gnomad.coverage gnomad.failedFilter gnomad.allAf  gnomad.allAn   \n",
       "0               NaN                 NaN          NaN           NaN  \\\n",
       "0               0.0                True            0      152296.0   \n",
       "0               1.0                 NaN     0.000379      145094.0   \n",
       "0               NaN                 NaN          NaN           NaN   \n",
       "0             178.0                 NaN     0.154747      118316.0   \n",
       "..              ...                 ...          ...           ...   \n",
       "0               1.0                 NaN     0.002211       94994.0   \n",
       "0               NaN                 NaN          NaN           NaN   \n",
       "0               NaN                 NaN          NaN           NaN   \n",
       "0               NaN                 NaN          NaN           NaN   \n",
       "0               0.0                True     0.000021      141118.0   \n",
       "\n",
       "    gnomad.allAc  gnomad.allHc gnomad.afrAf  gnomad.afrAn  gnomad.afrAc   \n",
       "0            NaN           NaN          NaN           NaN           NaN  \\\n",
       "0            0.0           0.0            0       41480.0           0.0   \n",
       "0           55.0           0.0     0.000049       40532.0           2.0   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0        18309.0           4.0     0.076287       34816.0        2656.0   \n",
       "..           ...           ...          ...           ...           ...   \n",
       "0          210.0           0.0     0.003126       24314.0          76.0   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            3.0           0.0     0.000051       39060.0           2.0   \n",
       "\n",
       "    gnomad.afrHc gnomad.amrAf  gnomad.amrAn  gnomad.amrAc  gnomad.amrHc   \n",
       "0            NaN          NaN           NaN           NaN           NaN  \\\n",
       "0            0.0            0       15294.0           0.0           0.0   \n",
       "0            0.0     0.000205       14652.0           3.0           0.0   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            1.0     0.117568       11942.0        1404.0           0.0   \n",
       "..           ...          ...           ...           ...           ...   \n",
       "0            0.0     0.001071        9340.0          10.0           0.0   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            0.0            0       14018.0           0.0           0.0   \n",
       "\n",
       "   gnomad.easAf  gnomad.easAn  gnomad.easAc  gnomad.easHc gnomad.finAf   \n",
       "0           NaN           NaN           NaN           NaN          NaN  \\\n",
       "0             0        5206.0           0.0           0.0            0   \n",
       "0             0        5082.0           0.0           0.0     0.000367   \n",
       "0           NaN           NaN           NaN           NaN          NaN   \n",
       "0      0.003146        5086.0          16.0           0.0     0.164856   \n",
       "..          ...           ...           ...           ...          ...   \n",
       "0      0.005271        2846.0          15.0           0.0     0.001511   \n",
       "0           NaN           NaN           NaN           NaN          NaN   \n",
       "0           NaN           NaN           NaN           NaN          NaN   \n",
       "0           NaN           NaN           NaN           NaN          NaN   \n",
       "0      0.000232        4306.0           1.0           0.0            0   \n",
       "\n",
       "    gnomad.finAn  gnomad.finAc  gnomad.finHc gnomad.nfeAf  gnomad.nfeAn   \n",
       "0            NaN           NaN           NaN          NaN           NaN  \\\n",
       "0        10628.0           0.0           0.0            0       68058.0   \n",
       "0         8170.0           3.0           0.0     0.000582       65286.0   \n",
       "0            NaN           NaN           NaN          NaN           NaN   \n",
       "0         7734.0        1275.0           1.0     0.228802       49816.0   \n",
       "..           ...           ...           ...          ...           ...   \n",
       "0         5958.0           9.0           0.0     0.001844       45550.0   \n",
       "0            NaN           NaN           NaN          NaN           NaN   \n",
       "0            NaN           NaN           NaN          NaN           NaN   \n",
       "0            NaN           NaN           NaN          NaN           NaN   \n",
       "0         7762.0           0.0           0.0            0       65358.0   \n",
       "\n",
       "    gnomad.nfeAc  gnomad.nfeHc gnomad.asjAf  gnomad.asjAn  gnomad.asjAc   \n",
       "0            NaN           NaN          NaN           NaN           NaN  \\\n",
       "0            0.0           0.0            0        3472.0           0.0   \n",
       "0           38.0           0.0     0.002097        3338.0           7.0   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0        11398.0           2.0     0.276284        2454.0         678.0   \n",
       "..           ...           ...          ...           ...           ...   \n",
       "0           84.0           0.0            0        2332.0           0.0   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            NaN           NaN          NaN           NaN           NaN   \n",
       "0            0.0           0.0            0        3364.0           0.0   \n",
       "\n",
       "    gnomad.asjHc gnomad.sasAf  gnomad.sasAn  gnomad.sasAc  gnomad.sasHc   \n",
       "0            NaN          NaN           NaN           NaN           NaN  \\\n",
       "0            0.0            0        4836.0           0.0           0.0   \n",
       "0            0.0            0        4814.0           0.0           0.0   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            0.0     0.098278        3948.0         388.0           0.0   \n",
       "..           ...          ...           ...           ...           ...   \n",
       "0            0.0     0.002682        2610.0           7.0           0.0   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            NaN          NaN           NaN           NaN           NaN   \n",
       "0            0.0            0        4110.0           0.0           0.0   \n",
       "\n",
       "   gnomad.othAf  gnomad.othAn  gnomad.othAc  gnomad.othHc gnomad.maleAf   \n",
       "0           NaN           NaN           NaN           NaN           NaN  \\\n",
       "0             0        2094.0           0.0           0.0             0   \n",
       "0             0        2000.0           0.0           0.0      0.000311   \n",
       "0           NaN           NaN           NaN           NaN           NaN   \n",
       "0      0.182048        1582.0         288.0           0.0      0.148306   \n",
       "..          ...           ...           ...           ...           ...   \n",
       "0       0.00219        1370.0           3.0           0.0      0.002105   \n",
       "0           NaN           NaN           NaN           NaN           NaN   \n",
       "0           NaN           NaN           NaN           NaN           NaN   \n",
       "0           NaN           NaN           NaN           NaN           NaN   \n",
       "0             0        1968.0           0.0           0.0             0   \n",
       "\n",
       "    gnomad.maleAn  gnomad.maleAc  gnomad.maleHc gnomad.femaleAf   \n",
       "0             NaN            NaN            NaN             NaN  \\\n",
       "0         74406.0            0.0            0.0               0   \n",
       "0         70634.0           22.0            0.0        0.000443   \n",
       "0             NaN            NaN            NaN             NaN   \n",
       "0         57840.0         8578.0            1.0        0.160907   \n",
       "..            ...            ...            ...             ...   \n",
       "0         46556.0           98.0            0.0        0.002312   \n",
       "0             NaN            NaN            NaN             NaN   \n",
       "0             NaN            NaN            NaN             NaN   \n",
       "0             NaN            NaN            NaN             NaN   \n",
       "0         68068.0            0.0            0.0        0.000041   \n",
       "\n",
       "    gnomad.femaleAn  gnomad.femaleAc  gnomad.femaleHc gnomad.controlsAllAf   \n",
       "0               NaN              NaN              NaN                  NaN  \\\n",
       "0           77890.0              0.0              0.0                    0   \n",
       "0           74460.0             33.0              0.0             0.000297   \n",
       "0               NaN              NaN              NaN                  NaN   \n",
       "0           60476.0           9731.0              3.0             0.126071   \n",
       "..              ...              ...              ...                  ...   \n",
       "0           48438.0            112.0              0.0             0.003282   \n",
       "0               NaN              NaN              NaN                  NaN   \n",
       "0               NaN              NaN              NaN                  NaN   \n",
       "0               NaN              NaN              NaN                  NaN   \n",
       "0           73050.0              3.0              0.0             0.000108   \n",
       "\n",
       "    gnomad.controlsAllAn  gnomad.controlsAllAc           dbsnp topmed.allAf   \n",
       "0                    NaN                   NaN             NaN          NaN  \\\n",
       "0                32922.0                   0.0             NaN          NaN   \n",
       "0                30260.0                   9.0  [rs1555877791]      0.00039   \n",
       "0                    NaN                   NaN             NaN          NaN   \n",
       "0                25922.0                3268.0  [rs1460148663]     0.210945   \n",
       "..                   ...                   ...             ...          ...   \n",
       "0                17368.0                  57.0   [rs112603162]     0.123869   \n",
       "0                    NaN                   NaN             NaN          NaN   \n",
       "0                    NaN                   NaN             NaN          NaN   \n",
       "0                    NaN                   NaN             NaN          NaN   \n",
       "0                27846.0                   3.0  [rs1601921720]          NaN   \n",
       "\n",
       "    topmed.allAn  topmed.allAc  topmed.allHc topmed.failedFilter dannScore   \n",
       "0            NaN           NaN           NaN                 NaN       NaN  \\\n",
       "0            NaN           NaN           NaN                 NaN       NaN   \n",
       "0       125568.0          49.0           0.0                True       NaN   \n",
       "0            NaN           NaN           NaN                 NaN       NaN   \n",
       "0       125568.0       26488.0           1.0                True       NaN   \n",
       "..           ...           ...           ...                 ...       ...   \n",
       "0       125568.0       15554.0           0.0                True      0.32   \n",
       "0            NaN           NaN           NaN                 NaN      0.32   \n",
       "0            NaN           NaN           NaN                 NaN      0.54   \n",
       "0            NaN           NaN           NaN                 NaN      0.67   \n",
       "0            NaN           NaN           NaN                 NaN      0.58   \n",
       "\n",
       "   gerpScore inLowComplexityRegion oneKg.allAf oneKg.afrAf  oneKg.amrAf   \n",
       "0        NaN                   NaN         NaN         NaN          NaN  \\\n",
       "0        NaN                   NaN         NaN         NaN          NaN   \n",
       "0        NaN                   NaN         NaN         NaN          NaN   \n",
       "0        NaN                   NaN         NaN         NaN          NaN   \n",
       "0        NaN                   NaN         NaN         NaN          NaN   \n",
       "..       ...                   ...         ...         ...          ...   \n",
       "0        NaN                   NaN         NaN         NaN          NaN   \n",
       "0        NaN                   NaN         NaN         NaN          NaN   \n",
       "0      -0.51                   NaN         NaN         NaN          NaN   \n",
       "0     -0.837                   NaN         NaN         NaN          NaN   \n",
       "0      0.622                   NaN         NaN         NaN          NaN   \n",
       "\n",
       "   oneKg.easAf oneKg.eurAf oneKg.sasAf  oneKg.allAn  oneKg.afrAn  oneKg.amrAn   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN  \\\n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "..         ...         ...         ...          ...          ...          ...   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "\n",
       "    oneKg.easAn  oneKg.eurAn  oneKg.sasAn  oneKg.allAc  oneKg.afrAc   \n",
       "0           NaN          NaN          NaN          NaN          NaN  \\\n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "..          ...          ...          ...          ...          ...   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "\n",
       "    oneKg.amrAc  oneKg.easAc  oneKg.eurAc  oneKg.sasAc primateAI-3D   \n",
       "0           NaN          NaN          NaN          NaN          NaN  \\\n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "..          ...          ...          ...          ...          ...   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "\n",
       "   revel.score                                             cosmic   \n",
       "0          NaN                                                NaN  \\\n",
       "0          NaN                                                NaN   \n",
       "0          NaN                                                NaN   \n",
       "0          NaN                                                NaN   \n",
       "0          NaN                                                NaN   \n",
       "..         ...                                                ...   \n",
       "0        0.007  [{'id': 'COSV62702051', 'numSamples': 2, 'refA...   \n",
       "0          NaN                                                NaN   \n",
       "0          NaN                                                NaN   \n",
       "0          NaN                                                NaN   \n",
       "0          NaN                                                NaN   \n",
       "\n",
       "                                             spliceAI clinvar  \n",
       "0                                                 NaN     NaN  \n",
       "0                                                 NaN     NaN  \n",
       "0                                                 NaN     NaN  \n",
       "0                                                 NaN     NaN  \n",
       "0                                                 NaN     NaN  \n",
       "..                                                ...     ...  \n",
       "0                                                 NaN     NaN  \n",
       "0                                                 NaN     NaN  \n",
       "0                                                 NaN     NaN  \n",
       "0   [{'hgnc': 'PCNT', 'donorGainScore': 0.6, 'dono...     NaN  \n",
       "0                                                 NaN     NaN  \n",
       "\n",
       "[147 rows x 99 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AnnotatedData.multiple_to_df(positions, \"variants\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11723"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = list(parser.filter_transcripts_by_consequence(\n",
    "    exclude=[\"downstream_gene_variant\", \"upstream_gene_variant\"]\n",
    "))\n",
    "len(positions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>vid</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>phylopScore</th>\n",
       "      <th>phyloPPrimateScore</th>\n",
       "      <th>transcripts</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "      <th>regulatoryRegions</th>\n",
       "      <th>gnomad.coverage</th>\n",
       "      <th>gnomad.failedFilter</th>\n",
       "      <th>gnomad.allAf</th>\n",
       "      <th>gnomad.allAn</th>\n",
       "      <th>gnomad.allAc</th>\n",
       "      <th>gnomad.allHc</th>\n",
       "      <th>gnomad.afrAf</th>\n",
       "      <th>gnomad.afrAn</th>\n",
       "      <th>gnomad.afrAc</th>\n",
       "      <th>gnomad.afrHc</th>\n",
       "      <th>gnomad.amrAf</th>\n",
       "      <th>gnomad.amrAn</th>\n",
       "      <th>gnomad.amrAc</th>\n",
       "      <th>gnomad.amrHc</th>\n",
       "      <th>gnomad.easAf</th>\n",
       "      <th>gnomad.easAn</th>\n",
       "      <th>gnomad.easAc</th>\n",
       "      <th>gnomad.easHc</th>\n",
       "      <th>gnomad.finAf</th>\n",
       "      <th>gnomad.finAn</th>\n",
       "      <th>gnomad.finAc</th>\n",
       "      <th>gnomad.finHc</th>\n",
       "      <th>gnomad.nfeAf</th>\n",
       "      <th>gnomad.nfeAn</th>\n",
       "      <th>gnomad.nfeAc</th>\n",
       "      <th>gnomad.nfeHc</th>\n",
       "      <th>gnomad.asjAf</th>\n",
       "      <th>gnomad.asjAn</th>\n",
       "      <th>gnomad.asjAc</th>\n",
       "      <th>gnomad.asjHc</th>\n",
       "      <th>gnomad.sasAf</th>\n",
       "      <th>gnomad.sasAn</th>\n",
       "      <th>gnomad.sasAc</th>\n",
       "      <th>gnomad.sasHc</th>\n",
       "      <th>gnomad.othAf</th>\n",
       "      <th>gnomad.othAn</th>\n",
       "      <th>gnomad.othAc</th>\n",
       "      <th>gnomad.othHc</th>\n",
       "      <th>gnomad.maleAf</th>\n",
       "      <th>gnomad.maleAn</th>\n",
       "      <th>gnomad.maleAc</th>\n",
       "      <th>gnomad.maleHc</th>\n",
       "      <th>gnomad.femaleAf</th>\n",
       "      <th>gnomad.femaleAn</th>\n",
       "      <th>gnomad.femaleAc</th>\n",
       "      <th>gnomad.femaleHc</th>\n",
       "      <th>gnomad.controlsAllAf</th>\n",
       "      <th>gnomad.controlsAllAn</th>\n",
       "      <th>gnomad.controlsAllAc</th>\n",
       "      <th>dbsnp</th>\n",
       "      <th>topmed.allAf</th>\n",
       "      <th>topmed.allAn</th>\n",
       "      <th>topmed.allAc</th>\n",
       "      <th>topmed.allHc</th>\n",
       "      <th>topmed.failedFilter</th>\n",
       "      <th>inLowComplexityRegion</th>\n",
       "      <th>dannScore</th>\n",
       "      <th>cosmic</th>\n",
       "      <th>oneKg.allAf</th>\n",
       "      <th>oneKg.afrAf</th>\n",
       "      <th>oneKg.amrAf</th>\n",
       "      <th>oneKg.easAf</th>\n",
       "      <th>oneKg.eurAf</th>\n",
       "      <th>oneKg.sasAf</th>\n",
       "      <th>oneKg.allAn</th>\n",
       "      <th>oneKg.afrAn</th>\n",
       "      <th>oneKg.amrAn</th>\n",
       "      <th>oneKg.easAn</th>\n",
       "      <th>oneKg.eurAn</th>\n",
       "      <th>oneKg.sasAn</th>\n",
       "      <th>oneKg.allAc</th>\n",
       "      <th>oneKg.afrAc</th>\n",
       "      <th>oneKg.amrAc</th>\n",
       "      <th>oneKg.easAc</th>\n",
       "      <th>oneKg.eurAc</th>\n",
       "      <th>oneKg.sasAc</th>\n",
       "      <th>gerpScore</th>\n",
       "      <th>primateAI-3D</th>\n",
       "      <th>revel.score</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>spliceAI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5232869</td>\n",
       "      <td>T</td>\n",
       "      <td>[G]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>58.10</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5232869-T-G</td>\n",
       "      <td>5232869</td>\n",
       "      <td>5232869</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5232869T&gt;G</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'transcript': 'ENST00000623753.1', 'source':...</td>\n",
       "      <td>799</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5243664</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[mapping_quality, no_reliable_supporting_read,...</td>\n",
       "      <td>58.13</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5243664-C-T</td>\n",
       "      <td>5243664</td>\n",
       "      <td>5243664</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5243664C&gt;T</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.074</td>\n",
       "      <td>[{'transcript': 'ENST00000623753.1', 'source':...</td>\n",
       "      <td>644</td>\n",
       "      <td>[{'id': 'ENSR00000140073', 'type': 'TF_binding...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>152310.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>41488.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>15294.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>5208.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>10632.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68056.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3472.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4838.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2094.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>74416.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>77894.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>32928.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5243664</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[mapping_quality, no_reliable_supporting_read,...</td>\n",
       "      <td>58.13</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5243664-C-T</td>\n",
       "      <td>5243664</td>\n",
       "      <td>5243664</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5243664C&gt;T</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.074</td>\n",
       "      <td>[{'transcript': 'ENST00000623753.1', 'source':...</td>\n",
       "      <td>644</td>\n",
       "      <td>[{'id': 'ENSR00000140073', 'type': 'TF_binding...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>152310.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>41488.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>15294.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>5208.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>10632.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68056.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3472.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4838.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2094.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>74416.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>77894.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>32928.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5243812</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[PASS]</td>\n",
       "      <td>108.19</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5243812-C-T</td>\n",
       "      <td>5243812</td>\n",
       "      <td>5243812</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5243812C&gt;T</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.074</td>\n",
       "      <td>[{'transcript': 'ENST00000623753.1', 'source':...</td>\n",
       "      <td>514</td>\n",
       "      <td>[{'id': 'ENSR00000140073', 'type': 'TF_binding...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>152296.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>41480.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>15294.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>5206.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>10628.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68058.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3472.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4836.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2094.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>74406.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>77890.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>32922.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>8209019</td>\n",
       "      <td>-</td>\n",
       "      <td>[GCGC]</td>\n",
       "      <td>[filtered_reads, mapping_quality, no_reliable_...</td>\n",
       "      <td>9.94</td>\n",
       "      <td>21p11.2</td>\n",
       "      <td>21-8209019-G-GCGC</td>\n",
       "      <td>8209020</td>\n",
       "      <td>8209019</td>\n",
       "      <td>CGC</td>\n",
       "      <td>insertion</td>\n",
       "      <td>NC_000021.9:g.8209022_8209024dup</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'transcript': 'ENST00000623664.1', 'source':...</td>\n",
       "      <td>1317</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000379</td>\n",
       "      <td>145094.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>40532.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000205</td>\n",
       "      <td>14652.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>5082.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000367</td>\n",
       "      <td>8170.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000582</td>\n",
       "      <td>65286.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002097</td>\n",
       "      <td>3338.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4814.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2000.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000311</td>\n",
       "      <td>70634.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000443</td>\n",
       "      <td>74460.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000297</td>\n",
       "      <td>30260.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>[rs1555877791]</td>\n",
       "      <td>0.00039</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46643510</td>\n",
       "      <td>T</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>211.58</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46643510-T-A</td>\n",
       "      <td>46643510</td>\n",
       "      <td>46643510</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46643510T&gt;A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.603</td>\n",
       "      <td>[{'transcript': 'ENST00000451211.6', 'source':...</td>\n",
       "      <td>150</td>\n",
       "      <td>[{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>141118.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>39060.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>14018.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000232</td>\n",
       "      <td>4306.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>7762.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>65358.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3364.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4110.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1968.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68068.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>73050.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000108</td>\n",
       "      <td>27846.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[rs1601921720]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.58</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.622</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46643510</td>\n",
       "      <td>T</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>211.58</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46643510-T-A</td>\n",
       "      <td>46643510</td>\n",
       "      <td>46643510</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46643510T&gt;A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.603</td>\n",
       "      <td>[{'transcript': 'ENST00000451211.6', 'source':...</td>\n",
       "      <td>150</td>\n",
       "      <td>[{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>141118.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>39060.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>14018.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000232</td>\n",
       "      <td>4306.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>7762.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>65358.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3364.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4110.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1968.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68068.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>73050.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000108</td>\n",
       "      <td>27846.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[rs1601921720]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.58</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.622</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46643510</td>\n",
       "      <td>T</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>211.58</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46643510-T-A</td>\n",
       "      <td>46643510</td>\n",
       "      <td>46643510</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46643510T&gt;A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.603</td>\n",
       "      <td>[{'transcript': 'ENST00000451211.6', 'source':...</td>\n",
       "      <td>150</td>\n",
       "      <td>[{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>141118.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>39060.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>14018.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000232</td>\n",
       "      <td>4306.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>7762.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>65358.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3364.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4110.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1968.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68068.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>73050.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000108</td>\n",
       "      <td>27846.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[rs1601921720]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.58</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.622</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46643510</td>\n",
       "      <td>T</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>211.58</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46643510-T-A</td>\n",
       "      <td>46643510</td>\n",
       "      <td>46643510</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46643510T&gt;A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.603</td>\n",
       "      <td>[{'transcript': 'ENST00000451211.6', 'source':...</td>\n",
       "      <td>150</td>\n",
       "      <td>[{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>141118.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>39060.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>14018.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000232</td>\n",
       "      <td>4306.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>7762.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>65358.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3364.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4110.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1968.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68068.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>73050.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000108</td>\n",
       "      <td>27846.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[rs1601921720]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.58</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.622</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46643510</td>\n",
       "      <td>T</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[base_quality, weak_evidence]</td>\n",
       "      <td>211.58</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46643510-T-A</td>\n",
       "      <td>46643510</td>\n",
       "      <td>46643510</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46643510T&gt;A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.603</td>\n",
       "      <td>[{'transcript': 'ENST00000451211.6', 'source':...</td>\n",
       "      <td>150</td>\n",
       "      <td>[{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>141118.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>39060.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>14018.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000232</td>\n",
       "      <td>4306.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>7762.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>65358.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3364.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4110.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1968.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>68068.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>73050.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000108</td>\n",
       "      <td>27846.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[rs1601921720]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.58</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.622</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11723 rows × 99 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome  position refAllele altAlleles   \n",
       "0       chr21   5232869         T        [G]  \\\n",
       "0       chr21   5243664         C        [T]   \n",
       "0       chr21   5243664         C        [T]   \n",
       "0       chr21   5243812         C        [T]   \n",
       "0       chr21   8209019         -     [GCGC]   \n",
       "..        ...       ...       ...        ...   \n",
       "0       chr21  46643510         T        [A]   \n",
       "0       chr21  46643510         T        [A]   \n",
       "0       chr21  46643510         T        [A]   \n",
       "0       chr21  46643510         T        [A]   \n",
       "0       chr21  46643510         T        [A]   \n",
       "\n",
       "                                              filters  mappingQuality   \n",
       "0                       [base_quality, weak_evidence]           58.10  \\\n",
       "0   [mapping_quality, no_reliable_supporting_read,...           58.13   \n",
       "0   [mapping_quality, no_reliable_supporting_read,...           58.13   \n",
       "0                                              [PASS]          108.19   \n",
       "0   [filtered_reads, mapping_quality, no_reliable_...            9.94   \n",
       "..                                                ...             ...   \n",
       "0                       [base_quality, weak_evidence]          211.58   \n",
       "0                       [base_quality, weak_evidence]          211.58   \n",
       "0                       [base_quality, weak_evidence]          211.58   \n",
       "0                       [base_quality, weak_evidence]          211.58   \n",
       "0                       [base_quality, weak_evidence]          211.58   \n",
       "\n",
       "   cyatogeneticBand                vid     begin       end altAllele   \n",
       "0             21p12     21-5232869-T-G   5232869   5232869         G  \\\n",
       "0             21p12     21-5243664-C-T   5243664   5243664         T   \n",
       "0             21p12     21-5243664-C-T   5243664   5243664         T   \n",
       "0             21p12     21-5243812-C-T   5243812   5243812         T   \n",
       "0           21p11.2  21-8209019-G-GCGC   8209020   8209019       CGC   \n",
       "..              ...                ...       ...       ...       ...   \n",
       "0           21q22.3    21-46643510-T-A  46643510  46643510         A   \n",
       "0           21q22.3    21-46643510-T-A  46643510  46643510         A   \n",
       "0           21q22.3    21-46643510-T-A  46643510  46643510         A   \n",
       "0           21q22.3    21-46643510-T-A  46643510  46643510         A   \n",
       "0           21q22.3    21-46643510-T-A  46643510  46643510         A   \n",
       "\n",
       "   variantType                             hgvsg  phylopScore   \n",
       "0          SNV          NC_000021.9:g.5232869T>G          0.5  \\\n",
       "0          SNV          NC_000021.9:g.5243664C>T          0.6   \n",
       "0          SNV          NC_000021.9:g.5243664C>T          0.6   \n",
       "0          SNV          NC_000021.9:g.5243812C>T          0.4   \n",
       "0    insertion  NC_000021.9:g.8209022_8209024dup          NaN   \n",
       "..         ...                               ...          ...   \n",
       "0          SNV         NC_000021.9:g.46643510T>A          0.0   \n",
       "0          SNV         NC_000021.9:g.46643510T>A          0.0   \n",
       "0          SNV         NC_000021.9:g.46643510T>A          0.0   \n",
       "0          SNV         NC_000021.9:g.46643510T>A          0.0   \n",
       "0          SNV         NC_000021.9:g.46643510T>A          0.0   \n",
       "\n",
       "    phyloPPrimateScore                                        transcripts   \n",
       "0                  NaN  [{'transcript': 'ENST00000623753.1', 'source':...  \\\n",
       "0                0.074  [{'transcript': 'ENST00000623753.1', 'source':...   \n",
       "0                0.074  [{'transcript': 'ENST00000623753.1', 'source':...   \n",
       "0                0.074  [{'transcript': 'ENST00000623753.1', 'source':...   \n",
       "0                  NaN  [{'transcript': 'ENST00000623664.1', 'source':...   \n",
       "..                 ...                                                ...   \n",
       "0               -0.603  [{'transcript': 'ENST00000451211.6', 'source':...   \n",
       "0               -0.603  [{'transcript': 'ENST00000451211.6', 'source':...   \n",
       "0               -0.603  [{'transcript': 'ENST00000451211.6', 'source':...   \n",
       "0               -0.603  [{'transcript': 'ENST00000451211.6', 'source':...   \n",
       "0               -0.603  [{'transcript': 'ENST00000451211.6', 'source':...   \n",
       "\n",
       "   vcfInfo.DP                                  regulatoryRegions   \n",
       "0         799                                                NaN  \\\n",
       "0         644  [{'id': 'ENSR00000140073', 'type': 'TF_binding...   \n",
       "0         644  [{'id': 'ENSR00000140073', 'type': 'TF_binding...   \n",
       "0         514  [{'id': 'ENSR00000140073', 'type': 'TF_binding...   \n",
       "0        1317                                                NaN   \n",
       "..        ...                                                ...   \n",
       "0         150  [{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...   \n",
       "0         150  [{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...   \n",
       "0         150  [{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...   \n",
       "0         150  [{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...   \n",
       "0         150  [{'id': 'ENSR00001057246', 'type': 'CTCF_bindi...   \n",
       "\n",
       "    gnomad.coverage gnomad.failedFilter gnomad.allAf  gnomad.allAn   \n",
       "0               NaN                 NaN          NaN           NaN  \\\n",
       "0               0.0                True            0      152310.0   \n",
       "0               0.0                True            0      152310.0   \n",
       "0               0.0                True            0      152296.0   \n",
       "0               1.0                 NaN     0.000379      145094.0   \n",
       "..              ...                 ...          ...           ...   \n",
       "0               0.0                True     0.000021      141118.0   \n",
       "0               0.0                True     0.000021      141118.0   \n",
       "0               0.0                True     0.000021      141118.0   \n",
       "0               0.0                True     0.000021      141118.0   \n",
       "0               0.0                True     0.000021      141118.0   \n",
       "\n",
       "    gnomad.allAc  gnomad.allHc gnomad.afrAf  gnomad.afrAn  gnomad.afrAc   \n",
       "0            NaN           NaN          NaN           NaN           NaN  \\\n",
       "0            0.0           0.0            0       41488.0           0.0   \n",
       "0            0.0           0.0            0       41488.0           0.0   \n",
       "0            0.0           0.0            0       41480.0           0.0   \n",
       "0           55.0           0.0     0.000049       40532.0           2.0   \n",
       "..           ...           ...          ...           ...           ...   \n",
       "0            3.0           0.0     0.000051       39060.0           2.0   \n",
       "0            3.0           0.0     0.000051       39060.0           2.0   \n",
       "0            3.0           0.0     0.000051       39060.0           2.0   \n",
       "0            3.0           0.0     0.000051       39060.0           2.0   \n",
       "0            3.0           0.0     0.000051       39060.0           2.0   \n",
       "\n",
       "    gnomad.afrHc gnomad.amrAf  gnomad.amrAn  gnomad.amrAc  gnomad.amrHc   \n",
       "0            NaN          NaN           NaN           NaN           NaN  \\\n",
       "0            0.0            0       15294.0           0.0           0.0   \n",
       "0            0.0            0       15294.0           0.0           0.0   \n",
       "0            0.0            0       15294.0           0.0           0.0   \n",
       "0            0.0     0.000205       14652.0           3.0           0.0   \n",
       "..           ...          ...           ...           ...           ...   \n",
       "0            0.0            0       14018.0           0.0           0.0   \n",
       "0            0.0            0       14018.0           0.0           0.0   \n",
       "0            0.0            0       14018.0           0.0           0.0   \n",
       "0            0.0            0       14018.0           0.0           0.0   \n",
       "0            0.0            0       14018.0           0.0           0.0   \n",
       "\n",
       "   gnomad.easAf  gnomad.easAn  gnomad.easAc  gnomad.easHc gnomad.finAf   \n",
       "0           NaN           NaN           NaN           NaN          NaN  \\\n",
       "0             0        5208.0           0.0           0.0            0   \n",
       "0             0        5208.0           0.0           0.0            0   \n",
       "0             0        5206.0           0.0           0.0            0   \n",
       "0             0        5082.0           0.0           0.0     0.000367   \n",
       "..          ...           ...           ...           ...          ...   \n",
       "0      0.000232        4306.0           1.0           0.0            0   \n",
       "0      0.000232        4306.0           1.0           0.0            0   \n",
       "0      0.000232        4306.0           1.0           0.0            0   \n",
       "0      0.000232        4306.0           1.0           0.0            0   \n",
       "0      0.000232        4306.0           1.0           0.0            0   \n",
       "\n",
       "    gnomad.finAn  gnomad.finAc  gnomad.finHc gnomad.nfeAf  gnomad.nfeAn   \n",
       "0            NaN           NaN           NaN          NaN           NaN  \\\n",
       "0        10632.0           0.0           0.0            0       68056.0   \n",
       "0        10632.0           0.0           0.0            0       68056.0   \n",
       "0        10628.0           0.0           0.0            0       68058.0   \n",
       "0         8170.0           3.0           0.0     0.000582       65286.0   \n",
       "..           ...           ...           ...          ...           ...   \n",
       "0         7762.0           0.0           0.0            0       65358.0   \n",
       "0         7762.0           0.0           0.0            0       65358.0   \n",
       "0         7762.0           0.0           0.0            0       65358.0   \n",
       "0         7762.0           0.0           0.0            0       65358.0   \n",
       "0         7762.0           0.0           0.0            0       65358.0   \n",
       "\n",
       "    gnomad.nfeAc  gnomad.nfeHc gnomad.asjAf  gnomad.asjAn  gnomad.asjAc   \n",
       "0            NaN           NaN          NaN           NaN           NaN  \\\n",
       "0            0.0           0.0            0        3472.0           0.0   \n",
       "0            0.0           0.0            0        3472.0           0.0   \n",
       "0            0.0           0.0            0        3472.0           0.0   \n",
       "0           38.0           0.0     0.002097        3338.0           7.0   \n",
       "..           ...           ...          ...           ...           ...   \n",
       "0            0.0           0.0            0        3364.0           0.0   \n",
       "0            0.0           0.0            0        3364.0           0.0   \n",
       "0            0.0           0.0            0        3364.0           0.0   \n",
       "0            0.0           0.0            0        3364.0           0.0   \n",
       "0            0.0           0.0            0        3364.0           0.0   \n",
       "\n",
       "    gnomad.asjHc gnomad.sasAf  gnomad.sasAn  gnomad.sasAc  gnomad.sasHc   \n",
       "0            NaN          NaN           NaN           NaN           NaN  \\\n",
       "0            0.0            0        4838.0           0.0           0.0   \n",
       "0            0.0            0        4838.0           0.0           0.0   \n",
       "0            0.0            0        4836.0           0.0           0.0   \n",
       "0            0.0            0        4814.0           0.0           0.0   \n",
       "..           ...          ...           ...           ...           ...   \n",
       "0            0.0            0        4110.0           0.0           0.0   \n",
       "0            0.0            0        4110.0           0.0           0.0   \n",
       "0            0.0            0        4110.0           0.0           0.0   \n",
       "0            0.0            0        4110.0           0.0           0.0   \n",
       "0            0.0            0        4110.0           0.0           0.0   \n",
       "\n",
       "   gnomad.othAf  gnomad.othAn  gnomad.othAc  gnomad.othHc gnomad.maleAf   \n",
       "0           NaN           NaN           NaN           NaN           NaN  \\\n",
       "0             0        2094.0           0.0           0.0             0   \n",
       "0             0        2094.0           0.0           0.0             0   \n",
       "0             0        2094.0           0.0           0.0             0   \n",
       "0             0        2000.0           0.0           0.0      0.000311   \n",
       "..          ...           ...           ...           ...           ...   \n",
       "0             0        1968.0           0.0           0.0             0   \n",
       "0             0        1968.0           0.0           0.0             0   \n",
       "0             0        1968.0           0.0           0.0             0   \n",
       "0             0        1968.0           0.0           0.0             0   \n",
       "0             0        1968.0           0.0           0.0             0   \n",
       "\n",
       "    gnomad.maleAn  gnomad.maleAc  gnomad.maleHc gnomad.femaleAf   \n",
       "0             NaN            NaN            NaN             NaN  \\\n",
       "0         74416.0            0.0            0.0               0   \n",
       "0         74416.0            0.0            0.0               0   \n",
       "0         74406.0            0.0            0.0               0   \n",
       "0         70634.0           22.0            0.0        0.000443   \n",
       "..            ...            ...            ...             ...   \n",
       "0         68068.0            0.0            0.0        0.000041   \n",
       "0         68068.0            0.0            0.0        0.000041   \n",
       "0         68068.0            0.0            0.0        0.000041   \n",
       "0         68068.0            0.0            0.0        0.000041   \n",
       "0         68068.0            0.0            0.0        0.000041   \n",
       "\n",
       "    gnomad.femaleAn  gnomad.femaleAc  gnomad.femaleHc gnomad.controlsAllAf   \n",
       "0               NaN              NaN              NaN                  NaN  \\\n",
       "0           77894.0              0.0              0.0                    0   \n",
       "0           77894.0              0.0              0.0                    0   \n",
       "0           77890.0              0.0              0.0                    0   \n",
       "0           74460.0             33.0              0.0             0.000297   \n",
       "..              ...              ...              ...                  ...   \n",
       "0           73050.0              3.0              0.0             0.000108   \n",
       "0           73050.0              3.0              0.0             0.000108   \n",
       "0           73050.0              3.0              0.0             0.000108   \n",
       "0           73050.0              3.0              0.0             0.000108   \n",
       "0           73050.0              3.0              0.0             0.000108   \n",
       "\n",
       "    gnomad.controlsAllAn  gnomad.controlsAllAc           dbsnp topmed.allAf   \n",
       "0                    NaN                   NaN             NaN          NaN  \\\n",
       "0                32928.0                   0.0             NaN          NaN   \n",
       "0                32928.0                   0.0             NaN          NaN   \n",
       "0                32922.0                   0.0             NaN          NaN   \n",
       "0                30260.0                   9.0  [rs1555877791]      0.00039   \n",
       "..                   ...                   ...             ...          ...   \n",
       "0                27846.0                   3.0  [rs1601921720]          NaN   \n",
       "0                27846.0                   3.0  [rs1601921720]          NaN   \n",
       "0                27846.0                   3.0  [rs1601921720]          NaN   \n",
       "0                27846.0                   3.0  [rs1601921720]          NaN   \n",
       "0                27846.0                   3.0  [rs1601921720]          NaN   \n",
       "\n",
       "    topmed.allAn  topmed.allAc  topmed.allHc topmed.failedFilter   \n",
       "0            NaN           NaN           NaN                 NaN  \\\n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0       125568.0          49.0           0.0                True   \n",
       "..           ...           ...           ...                 ...   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "0            NaN           NaN           NaN                 NaN   \n",
       "\n",
       "   inLowComplexityRegion dannScore cosmic oneKg.allAf oneKg.afrAf oneKg.amrAf   \n",
       "0                    NaN       NaN    NaN         NaN         NaN         NaN  \\\n",
       "0                    NaN       NaN    NaN         NaN         NaN         NaN   \n",
       "0                    NaN       NaN    NaN         NaN         NaN         NaN   \n",
       "0                    NaN       NaN    NaN         NaN         NaN         NaN   \n",
       "0                    NaN       NaN    NaN         NaN         NaN         NaN   \n",
       "..                   ...       ...    ...         ...         ...         ...   \n",
       "0                    NaN      0.58    NaN         NaN         NaN         NaN   \n",
       "0                    NaN      0.58    NaN         NaN         NaN         NaN   \n",
       "0                    NaN      0.58    NaN         NaN         NaN         NaN   \n",
       "0                    NaN      0.58    NaN         NaN         NaN         NaN   \n",
       "0                    NaN      0.58    NaN         NaN         NaN         NaN   \n",
       "\n",
       "   oneKg.easAf oneKg.eurAf oneKg.sasAf  oneKg.allAn  oneKg.afrAn  oneKg.amrAn   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN  \\\n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "..         ...         ...         ...          ...          ...          ...   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "0          NaN         NaN         NaN          NaN          NaN          NaN   \n",
       "\n",
       "    oneKg.easAn  oneKg.eurAn  oneKg.sasAn  oneKg.allAc  oneKg.afrAc   \n",
       "0           NaN          NaN          NaN          NaN          NaN  \\\n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "..          ...          ...          ...          ...          ...   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "\n",
       "    oneKg.amrAc  oneKg.easAc  oneKg.eurAc  oneKg.sasAc gerpScore primateAI-3D   \n",
       "0           NaN          NaN          NaN          NaN       NaN          NaN  \\\n",
       "0           NaN          NaN          NaN          NaN       NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN       NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN       NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN       NaN          NaN   \n",
       "..          ...          ...          ...          ...       ...          ...   \n",
       "0           NaN          NaN          NaN          NaN     0.622          NaN   \n",
       "0           NaN          NaN          NaN          NaN     0.622          NaN   \n",
       "0           NaN          NaN          NaN          NaN     0.622          NaN   \n",
       "0           NaN          NaN          NaN          NaN     0.622          NaN   \n",
       "0           NaN          NaN          NaN          NaN     0.622          NaN   \n",
       "\n",
       "   revel.score clinvar spliceAI  \n",
       "0          NaN     NaN      NaN  \n",
       "0          NaN     NaN      NaN  \n",
       "0          NaN     NaN      NaN  \n",
       "0          NaN     NaN      NaN  \n",
       "0          NaN     NaN      NaN  \n",
       "..         ...     ...      ...  \n",
       "0          NaN     NaN      NaN  \n",
       "0          NaN     NaN      NaN  \n",
       "0          NaN     NaN      NaN  \n",
       "0          NaN     NaN      NaN  \n",
       "0          NaN     NaN      NaN  \n",
       "\n",
       "[11723 rows x 99 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AnnotatedData.multiple_to_df(positions, \"variants\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter by gnomad frequency\n",
    "Possible values\n",
    "\n",
    "'coverage', 'failedFilter', 'allAf', 'allAn', 'allAc', 'allHc', 'afrAf', 'afrAn', 'afrAc', 'afrHc', 'amrAf', 'amrAn', 'amrAc', 'amrHc', 'easAf', 'easAn', 'easAc', 'easHc', 'finAf', 'finAn', 'finAc', 'finHc', 'nfeAf', 'nfeAn', 'nfeAc', 'nfeHc', 'asjAf', 'asjAn', 'asjAc', 'asjHc', 'sasAf', 'sasAn', 'sasAc', 'sasHc', 'othAf', 'othAn', 'othAc', 'othHc', 'maleAf', 'maleAn', 'maleAc', 'maleHc', 'femaleAf', 'femaleAn', 'femaleAc', 'femaleHc', 'controlsAllAf', 'controlsAllAn', 'controlsAllAc'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = Parser(annotated_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "851"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = list(parser.get_variants_above_gnomad_freq(\n",
    "    frequency_key=\"allAf\", \n",
    "    frequency_threshold_high=0.1\n",
    "))\n",
    "len(positions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "302"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = list(parser.get_variants_above_gnomad_freq(\n",
    "    frequency_key=\"allAf\",\n",
    "    frequency_threshold_low=0.1\n",
    "))\n",
    "len(positions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "114"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "positions = list(parser.get_variants_above_gnomad_freq(\n",
    "    frequency_key=\"allAf\", \n",
    "    frequency_threshold_low=0.1,\n",
    "    frequency_threshold_high=0.2\n",
    "))\n",
    "len(positions)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Further filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>vid</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>phylopScore</th>\n",
       "      <th>phyloPPrimateScore</th>\n",
       "      <th>transcripts</th>\n",
       "      <th>dbsnp</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "      <th>gnomad.coverage</th>\n",
       "      <th>gnomad.failedFilter</th>\n",
       "      <th>gnomad.allAf</th>\n",
       "      <th>gnomad.allAn</th>\n",
       "      <th>gnomad.allAc</th>\n",
       "      <th>gnomad.allHc</th>\n",
       "      <th>gnomad.afrAf</th>\n",
       "      <th>gnomad.afrAn</th>\n",
       "      <th>gnomad.afrAc</th>\n",
       "      <th>gnomad.afrHc</th>\n",
       "      <th>gnomad.amrAf</th>\n",
       "      <th>gnomad.amrAn</th>\n",
       "      <th>gnomad.amrAc</th>\n",
       "      <th>gnomad.amrHc</th>\n",
       "      <th>gnomad.easAf</th>\n",
       "      <th>gnomad.easAn</th>\n",
       "      <th>gnomad.easAc</th>\n",
       "      <th>gnomad.easHc</th>\n",
       "      <th>gnomad.finAf</th>\n",
       "      <th>gnomad.finAn</th>\n",
       "      <th>gnomad.finAc</th>\n",
       "      <th>gnomad.finHc</th>\n",
       "      <th>gnomad.nfeAf</th>\n",
       "      <th>gnomad.nfeAn</th>\n",
       "      <th>gnomad.nfeAc</th>\n",
       "      <th>gnomad.nfeHc</th>\n",
       "      <th>gnomad.asjAf</th>\n",
       "      <th>gnomad.asjAn</th>\n",
       "      <th>gnomad.asjAc</th>\n",
       "      <th>gnomad.asjHc</th>\n",
       "      <th>gnomad.sasAf</th>\n",
       "      <th>gnomad.sasAn</th>\n",
       "      <th>gnomad.sasAc</th>\n",
       "      <th>gnomad.sasHc</th>\n",
       "      <th>gnomad.othAf</th>\n",
       "      <th>gnomad.othAn</th>\n",
       "      <th>gnomad.othAc</th>\n",
       "      <th>gnomad.othHc</th>\n",
       "      <th>gnomad.maleAf</th>\n",
       "      <th>gnomad.maleAn</th>\n",
       "      <th>gnomad.maleAc</th>\n",
       "      <th>gnomad.maleHc</th>\n",
       "      <th>gnomad.femaleAf</th>\n",
       "      <th>gnomad.femaleAn</th>\n",
       "      <th>gnomad.femaleAc</th>\n",
       "      <th>gnomad.femaleHc</th>\n",
       "      <th>gnomad.controlsAllAf</th>\n",
       "      <th>gnomad.controlsAllAn</th>\n",
       "      <th>gnomad.controlsAllAc</th>\n",
       "      <th>topmed.allAf</th>\n",
       "      <th>topmed.allAn</th>\n",
       "      <th>topmed.allAc</th>\n",
       "      <th>topmed.allHc</th>\n",
       "      <th>topmed.failedFilter</th>\n",
       "      <th>dannScore</th>\n",
       "      <th>oneKg.allAf</th>\n",
       "      <th>oneKg.afrAf</th>\n",
       "      <th>oneKg.amrAf</th>\n",
       "      <th>oneKg.easAf</th>\n",
       "      <th>oneKg.eurAf</th>\n",
       "      <th>oneKg.sasAf</th>\n",
       "      <th>oneKg.allAn</th>\n",
       "      <th>oneKg.afrAn</th>\n",
       "      <th>oneKg.amrAn</th>\n",
       "      <th>oneKg.easAn</th>\n",
       "      <th>oneKg.eurAn</th>\n",
       "      <th>oneKg.sasAn</th>\n",
       "      <th>oneKg.allAc</th>\n",
       "      <th>oneKg.afrAc</th>\n",
       "      <th>oneKg.amrAc</th>\n",
       "      <th>oneKg.easAc</th>\n",
       "      <th>oneKg.eurAc</th>\n",
       "      <th>oneKg.sasAc</th>\n",
       "      <th>cosmic</th>\n",
       "      <th>inLowComplexityRegion</th>\n",
       "      <th>gerpScore</th>\n",
       "      <th>regulatoryRegions</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5278225</td>\n",
       "      <td>A</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[alt_allele_in_normal, filtered_reads, mapping...</td>\n",
       "      <td>12.91</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5278225-A-T</td>\n",
       "      <td>5278225</td>\n",
       "      <td>5278225</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5278225A&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>[rs1171728286]</td>\n",
       "      <td>113</td>\n",
       "      <td>7</td>\n",
       "      <td>True</td>\n",
       "      <td>0.148668</td>\n",
       "      <td>46856</td>\n",
       "      <td>6966</td>\n",
       "      <td>80</td>\n",
       "      <td>0.181794</td>\n",
       "      <td>26244</td>\n",
       "      <td>4771</td>\n",
       "      <td>46</td>\n",
       "      <td>0.110039</td>\n",
       "      <td>2590</td>\n",
       "      <td>285</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000608</td>\n",
       "      <td>1644</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.050104</td>\n",
       "      <td>958</td>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>0.122751</td>\n",
       "      <td>12448</td>\n",
       "      <td>1528</td>\n",
       "      <td>26</td>\n",
       "      <td>0.225636</td>\n",
       "      <td>944</td>\n",
       "      <td>213</td>\n",
       "      <td>2</td>\n",
       "      <td>0.022059</td>\n",
       "      <td>1224</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>0.116838</td>\n",
       "      <td>582</td>\n",
       "      <td>68</td>\n",
       "      <td>0</td>\n",
       "      <td>0.145535</td>\n",
       "      <td>22084</td>\n",
       "      <td>3214</td>\n",
       "      <td>34</td>\n",
       "      <td>0.151461</td>\n",
       "      <td>24772</td>\n",
       "      <td>3752</td>\n",
       "      <td>46</td>\n",
       "      <td>0.128554</td>\n",
       "      <td>9848</td>\n",
       "      <td>1266</td>\n",
       "      <td>0.084209</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>10574.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5291458</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[mapping_quality, no_reliable_supporting_read,...</td>\n",
       "      <td>106.93</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5291458-C-T</td>\n",
       "      <td>5291458</td>\n",
       "      <td>5291458</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5291458C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>[rs1265206646]</td>\n",
       "      <td>220</td>\n",
       "      <td>26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.110935</td>\n",
       "      <td>70798</td>\n",
       "      <td>7854</td>\n",
       "      <td>78</td>\n",
       "      <td>0.022681</td>\n",
       "      <td>26322</td>\n",
       "      <td>597</td>\n",
       "      <td>0</td>\n",
       "      <td>0.079767</td>\n",
       "      <td>7196</td>\n",
       "      <td>574</td>\n",
       "      <td>2</td>\n",
       "      <td>0.075443</td>\n",
       "      <td>2598</td>\n",
       "      <td>196</td>\n",
       "      <td>0</td>\n",
       "      <td>0.102046</td>\n",
       "      <td>4204</td>\n",
       "      <td>429</td>\n",
       "      <td>0</td>\n",
       "      <td>0.210871</td>\n",
       "      <td>25480</td>\n",
       "      <td>5373</td>\n",
       "      <td>72</td>\n",
       "      <td>0.151239</td>\n",
       "      <td>1534</td>\n",
       "      <td>232</td>\n",
       "      <td>2</td>\n",
       "      <td>0.122714</td>\n",
       "      <td>2078</td>\n",
       "      <td>255</td>\n",
       "      <td>0</td>\n",
       "      <td>0.107527</td>\n",
       "      <td>930</td>\n",
       "      <td>100</td>\n",
       "      <td>0</td>\n",
       "      <td>0.102856</td>\n",
       "      <td>34660</td>\n",
       "      <td>3565</td>\n",
       "      <td>30</td>\n",
       "      <td>0.118684</td>\n",
       "      <td>36138</td>\n",
       "      <td>4289</td>\n",
       "      <td>48</td>\n",
       "      <td>0.074837</td>\n",
       "      <td>16302</td>\n",
       "      <td>1220</td>\n",
       "      <td>0.372101</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>46724.0</td>\n",
       "      <td>4255.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5317005</td>\n",
       "      <td>C</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[mapping_quality, non_homref_normal]</td>\n",
       "      <td>32.32</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5317005-C-A</td>\n",
       "      <td>5317005</td>\n",
       "      <td>5317005</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5317005C&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>[rs1399429098]</td>\n",
       "      <td>194</td>\n",
       "      <td>24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.187619</td>\n",
       "      <td>33520</td>\n",
       "      <td>6289</td>\n",
       "      <td>71</td>\n",
       "      <td>0.096655</td>\n",
       "      <td>18116</td>\n",
       "      <td>1751</td>\n",
       "      <td>1</td>\n",
       "      <td>0.237973</td>\n",
       "      <td>2328</td>\n",
       "      <td>554</td>\n",
       "      <td>10</td>\n",
       "      <td>0.275613</td>\n",
       "      <td>1386</td>\n",
       "      <td>382</td>\n",
       "      <td>2</td>\n",
       "      <td>0.122881</td>\n",
       "      <td>708</td>\n",
       "      <td>87</td>\n",
       "      <td>2</td>\n",
       "      <td>0.327003</td>\n",
       "      <td>8636</td>\n",
       "      <td>2824</td>\n",
       "      <td>47</td>\n",
       "      <td>0.306579</td>\n",
       "      <td>760</td>\n",
       "      <td>233</td>\n",
       "      <td>4</td>\n",
       "      <td>0.300403</td>\n",
       "      <td>992</td>\n",
       "      <td>298</td>\n",
       "      <td>3</td>\n",
       "      <td>0.234742</td>\n",
       "      <td>426</td>\n",
       "      <td>100</td>\n",
       "      <td>1</td>\n",
       "      <td>0.181277</td>\n",
       "      <td>15788</td>\n",
       "      <td>2862</td>\n",
       "      <td>24</td>\n",
       "      <td>0.193266</td>\n",
       "      <td>17732</td>\n",
       "      <td>3427</td>\n",
       "      <td>47</td>\n",
       "      <td>0.216009</td>\n",
       "      <td>7958</td>\n",
       "      <td>1719</td>\n",
       "      <td>0.508537</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>63856.0</td>\n",
       "      <td>3328.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5341568</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[weak_evidence]</td>\n",
       "      <td>30.38</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5341568-C-T</td>\n",
       "      <td>5341568</td>\n",
       "      <td>5341568</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5341568C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>[rs1362623940]</td>\n",
       "      <td>241</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.199399</td>\n",
       "      <td>94564</td>\n",
       "      <td>18856</td>\n",
       "      <td>550</td>\n",
       "      <td>0.039066</td>\n",
       "      <td>30282</td>\n",
       "      <td>1183</td>\n",
       "      <td>4</td>\n",
       "      <td>0.28162</td>\n",
       "      <td>8792</td>\n",
       "      <td>2476</td>\n",
       "      <td>39</td>\n",
       "      <td>0.320692</td>\n",
       "      <td>3296</td>\n",
       "      <td>1057</td>\n",
       "      <td>20</td>\n",
       "      <td>0.257278</td>\n",
       "      <td>4878</td>\n",
       "      <td>1255</td>\n",
       "      <td>27</td>\n",
       "      <td>0.283761</td>\n",
       "      <td>40298</td>\n",
       "      <td>11435</td>\n",
       "      <td>434</td>\n",
       "      <td>0.216651</td>\n",
       "      <td>2174</td>\n",
       "      <td>471</td>\n",
       "      <td>12</td>\n",
       "      <td>0.177857</td>\n",
       "      <td>2800</td>\n",
       "      <td>498</td>\n",
       "      <td>3</td>\n",
       "      <td>0.208531</td>\n",
       "      <td>1266</td>\n",
       "      <td>264</td>\n",
       "      <td>8</td>\n",
       "      <td>0.199067</td>\n",
       "      <td>45432</td>\n",
       "      <td>9044</td>\n",
       "      <td>245</td>\n",
       "      <td>0.199707</td>\n",
       "      <td>49132</td>\n",
       "      <td>9812</td>\n",
       "      <td>305</td>\n",
       "      <td>0.179372</td>\n",
       "      <td>19546</td>\n",
       "      <td>3506</td>\n",
       "      <td>0.22923</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>28784.0</td>\n",
       "      <td>977.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>5344000</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[no_reliable_supporting_read, weak_evidence]</td>\n",
       "      <td>24.00</td>\n",
       "      <td>21p12</td>\n",
       "      <td>21-5344000-C-T</td>\n",
       "      <td>5344000</td>\n",
       "      <td>5344000</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.5344000C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>[rs1156358732]</td>\n",
       "      <td>241</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.108729</td>\n",
       "      <td>99118</td>\n",
       "      <td>10777</td>\n",
       "      <td>218</td>\n",
       "      <td>0.01388</td>\n",
       "      <td>35374</td>\n",
       "      <td>491</td>\n",
       "      <td>2</td>\n",
       "      <td>0.178461</td>\n",
       "      <td>9016</td>\n",
       "      <td>1609</td>\n",
       "      <td>17</td>\n",
       "      <td>0.194145</td>\n",
       "      <td>2972</td>\n",
       "      <td>577</td>\n",
       "      <td>8</td>\n",
       "      <td>0.15375</td>\n",
       "      <td>5574</td>\n",
       "      <td>857</td>\n",
       "      <td>6</td>\n",
       "      <td>0.165372</td>\n",
       "      <td>38876</td>\n",
       "      <td>6429</td>\n",
       "      <td>177</td>\n",
       "      <td>0.113576</td>\n",
       "      <td>2254</td>\n",
       "      <td>256</td>\n",
       "      <td>3</td>\n",
       "      <td>0.096678</td>\n",
       "      <td>3010</td>\n",
       "      <td>291</td>\n",
       "      <td>0</td>\n",
       "      <td>0.107963</td>\n",
       "      <td>1306</td>\n",
       "      <td>141</td>\n",
       "      <td>2</td>\n",
       "      <td>0.108202</td>\n",
       "      <td>47892</td>\n",
       "      <td>5182</td>\n",
       "      <td>85</td>\n",
       "      <td>0.109222</td>\n",
       "      <td>51226</td>\n",
       "      <td>5595</td>\n",
       "      <td>133</td>\n",
       "      <td>0.103308</td>\n",
       "      <td>21344</td>\n",
       "      <td>2205</td>\n",
       "      <td>0.191402</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>24034.0</td>\n",
       "      <td>247.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>44260048</td>\n",
       "      <td>A</td>\n",
       "      <td>[C]</td>\n",
       "      <td>[PASS]</td>\n",
       "      <td>204.27</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-44260048-CA-C</td>\n",
       "      <td>44260049</td>\n",
       "      <td>44260049</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>NC_000021.9:g.44260071del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'transcript': 'ENST00000628202.3', 'source':...</td>\n",
       "      <td>[rs765722839]</td>\n",
       "      <td>121</td>\n",
       "      <td>18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.163523</td>\n",
       "      <td>63508</td>\n",
       "      <td>10385</td>\n",
       "      <td>235</td>\n",
       "      <td>0.241867</td>\n",
       "      <td>17460</td>\n",
       "      <td>4223</td>\n",
       "      <td>131</td>\n",
       "      <td>0.11847</td>\n",
       "      <td>5360</td>\n",
       "      <td>635</td>\n",
       "      <td>16</td>\n",
       "      <td>0.020419</td>\n",
       "      <td>1910</td>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>0.089506</td>\n",
       "      <td>1944</td>\n",
       "      <td>174</td>\n",
       "      <td>7</td>\n",
       "      <td>0.136704</td>\n",
       "      <td>32040</td>\n",
       "      <td>4380</td>\n",
       "      <td>49</td>\n",
       "      <td>0.189066</td>\n",
       "      <td>1756</td>\n",
       "      <td>332</td>\n",
       "      <td>8</td>\n",
       "      <td>0.229017</td>\n",
       "      <td>1668</td>\n",
       "      <td>382</td>\n",
       "      <td>13</td>\n",
       "      <td>0.157407</td>\n",
       "      <td>756</td>\n",
       "      <td>119</td>\n",
       "      <td>6</td>\n",
       "      <td>0.164289</td>\n",
       "      <td>29022</td>\n",
       "      <td>4768</td>\n",
       "      <td>103</td>\n",
       "      <td>0.162878</td>\n",
       "      <td>34486</td>\n",
       "      <td>5617</td>\n",
       "      <td>132</td>\n",
       "      <td>0.163979</td>\n",
       "      <td>11288</td>\n",
       "      <td>1851</td>\n",
       "      <td>0.110832</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>13917.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'id': 'ENSR00000664607', 'type': 'enhancer',...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>45654338</td>\n",
       "      <td>T</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[PASS]</td>\n",
       "      <td>205.80</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-45654338-AT-A</td>\n",
       "      <td>45654339</td>\n",
       "      <td>45654339</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>NC_000021.9:g.45654359del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'transcript': 'ENST00000465077.5', 'source':...</td>\n",
       "      <td>[rs398036499]</td>\n",
       "      <td>145</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.162227</td>\n",
       "      <td>108952</td>\n",
       "      <td>17675</td>\n",
       "      <td>674</td>\n",
       "      <td>0.215993</td>\n",
       "      <td>28200</td>\n",
       "      <td>6091</td>\n",
       "      <td>308</td>\n",
       "      <td>0.181828</td>\n",
       "      <td>9476</td>\n",
       "      <td>1723</td>\n",
       "      <td>86</td>\n",
       "      <td>0.060228</td>\n",
       "      <td>3852</td>\n",
       "      <td>232</td>\n",
       "      <td>1</td>\n",
       "      <td>0.08975</td>\n",
       "      <td>3844</td>\n",
       "      <td>345</td>\n",
       "      <td>9</td>\n",
       "      <td>0.149985</td>\n",
       "      <td>55012</td>\n",
       "      <td>8251</td>\n",
       "      <td>237</td>\n",
       "      <td>0.143775</td>\n",
       "      <td>2956</td>\n",
       "      <td>425</td>\n",
       "      <td>14</td>\n",
       "      <td>0.091275</td>\n",
       "      <td>3232</td>\n",
       "      <td>295</td>\n",
       "      <td>12</td>\n",
       "      <td>0.147222</td>\n",
       "      <td>1440</td>\n",
       "      <td>212</td>\n",
       "      <td>6</td>\n",
       "      <td>0.159572</td>\n",
       "      <td>50598</td>\n",
       "      <td>8074</td>\n",
       "      <td>306</td>\n",
       "      <td>0.16453</td>\n",
       "      <td>58354</td>\n",
       "      <td>9601</td>\n",
       "      <td>368</td>\n",
       "      <td>0.155962</td>\n",
       "      <td>19960</td>\n",
       "      <td>3113</td>\n",
       "      <td>0.163991</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>20592.0</td>\n",
       "      <td>276.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>45909790</td>\n",
       "      <td>C</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[weak_evidence]</td>\n",
       "      <td>107.77</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-45909790-AC-A</td>\n",
       "      <td>45909791</td>\n",
       "      <td>45909791</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>NC_000021.9:g.45909800del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'transcript': 'ENST00000400314.5', 'source':...</td>\n",
       "      <td>[rs376651490]</td>\n",
       "      <td>53</td>\n",
       "      <td>35</td>\n",
       "      <td>True</td>\n",
       "      <td>0.153031</td>\n",
       "      <td>9338</td>\n",
       "      <td>1429</td>\n",
       "      <td>128</td>\n",
       "      <td>0.31489</td>\n",
       "      <td>2458</td>\n",
       "      <td>774</td>\n",
       "      <td>92</td>\n",
       "      <td>0.086076</td>\n",
       "      <td>790</td>\n",
       "      <td>68</td>\n",
       "      <td>2</td>\n",
       "      <td>0.081818</td>\n",
       "      <td>220</td>\n",
       "      <td>18</td>\n",
       "      <td>2</td>\n",
       "      <td>0.079592</td>\n",
       "      <td>490</td>\n",
       "      <td>39</td>\n",
       "      <td>2</td>\n",
       "      <td>0.091773</td>\n",
       "      <td>4838</td>\n",
       "      <td>444</td>\n",
       "      <td>22</td>\n",
       "      <td>0.180952</td>\n",
       "      <td>210</td>\n",
       "      <td>38</td>\n",
       "      <td>3</td>\n",
       "      <td>0.131579</td>\n",
       "      <td>114</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>0.126984</td>\n",
       "      <td>126</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>0.154401</td>\n",
       "      <td>4158</td>\n",
       "      <td>642</td>\n",
       "      <td>64</td>\n",
       "      <td>0.151931</td>\n",
       "      <td>5180</td>\n",
       "      <td>787</td>\n",
       "      <td>64</td>\n",
       "      <td>0.158901</td>\n",
       "      <td>1674</td>\n",
       "      <td>266</td>\n",
       "      <td>0.373614</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>46914.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46145756</td>\n",
       "      <td>C</td>\n",
       "      <td>[G]</td>\n",
       "      <td>[weak_evidence]</td>\n",
       "      <td>205.79</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46145756-C-G</td>\n",
       "      <td>46145756</td>\n",
       "      <td>46145756</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.46145756C&gt;G</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>-1.657</td>\n",
       "      <td>[{'transcript': 'ENST00000494498.2', 'source':...</td>\n",
       "      <td>[rs191178818]</td>\n",
       "      <td>65</td>\n",
       "      <td>12</td>\n",
       "      <td>True</td>\n",
       "      <td>0.177827</td>\n",
       "      <td>53940</td>\n",
       "      <td>9592</td>\n",
       "      <td>0</td>\n",
       "      <td>0.077993</td>\n",
       "      <td>14232</td>\n",
       "      <td>1110</td>\n",
       "      <td>0</td>\n",
       "      <td>0.256837</td>\n",
       "      <td>5704</td>\n",
       "      <td>1465</td>\n",
       "      <td>0</td>\n",
       "      <td>0.139273</td>\n",
       "      <td>1156</td>\n",
       "      <td>161</td>\n",
       "      <td>0</td>\n",
       "      <td>0.153454</td>\n",
       "      <td>4314</td>\n",
       "      <td>662</td>\n",
       "      <td>0</td>\n",
       "      <td>0.222038</td>\n",
       "      <td>24784</td>\n",
       "      <td>5503</td>\n",
       "      <td>0</td>\n",
       "      <td>0.219403</td>\n",
       "      <td>1340</td>\n",
       "      <td>294</td>\n",
       "      <td>0</td>\n",
       "      <td>0.11203</td>\n",
       "      <td>1330</td>\n",
       "      <td>149</td>\n",
       "      <td>0</td>\n",
       "      <td>0.22093</td>\n",
       "      <td>688</td>\n",
       "      <td>152</td>\n",
       "      <td>0</td>\n",
       "      <td>0.183313</td>\n",
       "      <td>26332</td>\n",
       "      <td>4827</td>\n",
       "      <td>0</td>\n",
       "      <td>0.172595</td>\n",
       "      <td>27608</td>\n",
       "      <td>4765</td>\n",
       "      <td>0</td>\n",
       "      <td>0.155291</td>\n",
       "      <td>11018</td>\n",
       "      <td>1711</td>\n",
       "      <td>0.242052</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>30394.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>-0.675</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>46415365</td>\n",
       "      <td>TT</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[low_frac_info_reads, non_homref_normal]</td>\n",
       "      <td>210.96</td>\n",
       "      <td>21q22.3</td>\n",
       "      <td>21-46415365-ATT-A</td>\n",
       "      <td>46415366</td>\n",
       "      <td>46415367</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>NC_000021.9:g.46415394_46415395del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'transcript': 'ENST00000703224.1', 'source':...</td>\n",
       "      <td>[rs35983555]</td>\n",
       "      <td>122</td>\n",
       "      <td>20</td>\n",
       "      <td>True</td>\n",
       "      <td>0.122263</td>\n",
       "      <td>65760</td>\n",
       "      <td>8040</td>\n",
       "      <td>89</td>\n",
       "      <td>0.140013</td>\n",
       "      <td>15420</td>\n",
       "      <td>2159</td>\n",
       "      <td>21</td>\n",
       "      <td>0.155373</td>\n",
       "      <td>4132</td>\n",
       "      <td>642</td>\n",
       "      <td>25</td>\n",
       "      <td>0.099206</td>\n",
       "      <td>1764</td>\n",
       "      <td>175</td>\n",
       "      <td>0</td>\n",
       "      <td>0.062984</td>\n",
       "      <td>1032</td>\n",
       "      <td>65</td>\n",
       "      <td>2</td>\n",
       "      <td>0.107488</td>\n",
       "      <td>38674</td>\n",
       "      <td>4157</td>\n",
       "      <td>16</td>\n",
       "      <td>0.177374</td>\n",
       "      <td>2148</td>\n",
       "      <td>381</td>\n",
       "      <td>8</td>\n",
       "      <td>0.212605</td>\n",
       "      <td>1190</td>\n",
       "      <td>253</td>\n",
       "      <td>15</td>\n",
       "      <td>0.129156</td>\n",
       "      <td>782</td>\n",
       "      <td>101</td>\n",
       "      <td>1</td>\n",
       "      <td>0.127211</td>\n",
       "      <td>28386</td>\n",
       "      <td>3611</td>\n",
       "      <td>49</td>\n",
       "      <td>0.118505</td>\n",
       "      <td>37374</td>\n",
       "      <td>4429</td>\n",
       "      <td>40</td>\n",
       "      <td>0.132299</td>\n",
       "      <td>8700</td>\n",
       "      <td>1151</td>\n",
       "      <td>0.151727</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>19052.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>114 rows × 95 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome  position refAllele altAlleles   \n",
       "0       chr21   5278225         A        [T]  \\\n",
       "0       chr21   5291458         C        [T]   \n",
       "0       chr21   5317005         C        [A]   \n",
       "0       chr21   5341568         C        [T]   \n",
       "0       chr21   5344000         C        [T]   \n",
       "..        ...       ...       ...        ...   \n",
       "0       chr21  44260048         A        [C]   \n",
       "0       chr21  45654338         T        [A]   \n",
       "0       chr21  45909790         C        [A]   \n",
       "0       chr21  46145756         C        [G]   \n",
       "0       chr21  46415365        TT        [A]   \n",
       "\n",
       "                                              filters  mappingQuality   \n",
       "0   [alt_allele_in_normal, filtered_reads, mapping...           12.91  \\\n",
       "0   [mapping_quality, no_reliable_supporting_read,...          106.93   \n",
       "0                [mapping_quality, non_homref_normal]           32.32   \n",
       "0                                     [weak_evidence]           30.38   \n",
       "0        [no_reliable_supporting_read, weak_evidence]           24.00   \n",
       "..                                                ...             ...   \n",
       "0                                              [PASS]          204.27   \n",
       "0                                              [PASS]          205.80   \n",
       "0                                     [weak_evidence]          107.77   \n",
       "0                                     [weak_evidence]          205.79   \n",
       "0            [low_frac_info_reads, non_homref_normal]          210.96   \n",
       "\n",
       "   cyatogeneticBand                vid     begin       end altAllele   \n",
       "0             21p12     21-5278225-A-T   5278225   5278225         T  \\\n",
       "0             21p12     21-5291458-C-T   5291458   5291458         T   \n",
       "0             21p12     21-5317005-C-A   5317005   5317005         A   \n",
       "0             21p12     21-5341568-C-T   5341568   5341568         T   \n",
       "0             21p12     21-5344000-C-T   5344000   5344000         T   \n",
       "..              ...                ...       ...       ...       ...   \n",
       "0           21q22.3   21-44260048-CA-C  44260049  44260049         -   \n",
       "0           21q22.3   21-45654338-AT-A  45654339  45654339         -   \n",
       "0           21q22.3   21-45909790-AC-A  45909791  45909791         -   \n",
       "0           21q22.3    21-46145756-C-G  46145756  46145756         G   \n",
       "0           21q22.3  21-46415365-ATT-A  46415366  46415367         -   \n",
       "\n",
       "   variantType                               hgvsg  phylopScore   \n",
       "0          SNV            NC_000021.9:g.5278225A>T          NaN  \\\n",
       "0          SNV            NC_000021.9:g.5291458C>T          NaN   \n",
       "0          SNV            NC_000021.9:g.5317005C>A          NaN   \n",
       "0          SNV            NC_000021.9:g.5341568C>T          NaN   \n",
       "0          SNV            NC_000021.9:g.5344000C>T          NaN   \n",
       "..         ...                                 ...          ...   \n",
       "0     deletion           NC_000021.9:g.44260071del          NaN   \n",
       "0     deletion           NC_000021.9:g.45654359del          NaN   \n",
       "0     deletion           NC_000021.9:g.45909800del          NaN   \n",
       "0          SNV           NC_000021.9:g.46145756C>G         -0.2   \n",
       "0     deletion  NC_000021.9:g.46415394_46415395del          NaN   \n",
       "\n",
       "    phyloPPrimateScore                                        transcripts   \n",
       "0                  NaN                                               None  \\\n",
       "0                  NaN                                               None   \n",
       "0                  NaN                                               None   \n",
       "0                  NaN                                               None   \n",
       "0                  NaN                                               None   \n",
       "..                 ...                                                ...   \n",
       "0                  NaN  [{'transcript': 'ENST00000628202.3', 'source':...   \n",
       "0                  NaN  [{'transcript': 'ENST00000465077.5', 'source':...   \n",
       "0                  NaN  [{'transcript': 'ENST00000400314.5', 'source':...   \n",
       "0               -1.657  [{'transcript': 'ENST00000494498.2', 'source':...   \n",
       "0                  NaN  [{'transcript': 'ENST00000703224.1', 'source':...   \n",
       "\n",
       "             dbsnp vcfInfo.DP  gnomad.coverage gnomad.failedFilter   \n",
       "0   [rs1171728286]        113                7                True  \\\n",
       "0   [rs1265206646]        220               26                 NaN   \n",
       "0   [rs1399429098]        194               24                 NaN   \n",
       "0   [rs1362623940]        241               19                 NaN   \n",
       "0   [rs1156358732]        241               15                 NaN   \n",
       "..             ...        ...              ...                 ...   \n",
       "0    [rs765722839]        121               18                 NaN   \n",
       "0    [rs398036499]        145               15                 NaN   \n",
       "0    [rs376651490]         53               35                True   \n",
       "0    [rs191178818]         65               12                True   \n",
       "0     [rs35983555]        122               20                True   \n",
       "\n",
       "   gnomad.allAf  gnomad.allAn  gnomad.allAc  gnomad.allHc gnomad.afrAf   \n",
       "0      0.148668         46856          6966            80     0.181794  \\\n",
       "0      0.110935         70798          7854            78     0.022681   \n",
       "0      0.187619         33520          6289            71     0.096655   \n",
       "0      0.199399         94564         18856           550     0.039066   \n",
       "0      0.108729         99118         10777           218      0.01388   \n",
       "..          ...           ...           ...           ...          ...   \n",
       "0      0.163523         63508         10385           235     0.241867   \n",
       "0      0.162227        108952         17675           674     0.215993   \n",
       "0      0.153031          9338          1429           128      0.31489   \n",
       "0      0.177827         53940          9592             0     0.077993   \n",
       "0      0.122263         65760          8040            89     0.140013   \n",
       "\n",
       "    gnomad.afrAn  gnomad.afrAc  gnomad.afrHc gnomad.amrAf  gnomad.amrAn   \n",
       "0          26244          4771            46     0.110039          2590  \\\n",
       "0          26322           597             0     0.079767          7196   \n",
       "0          18116          1751             1     0.237973          2328   \n",
       "0          30282          1183             4      0.28162          8792   \n",
       "0          35374           491             2     0.178461          9016   \n",
       "..           ...           ...           ...          ...           ...   \n",
       "0          17460          4223           131      0.11847          5360   \n",
       "0          28200          6091           308     0.181828          9476   \n",
       "0           2458           774            92     0.086076           790   \n",
       "0          14232          1110             0     0.256837          5704   \n",
       "0          15420          2159            21     0.155373          4132   \n",
       "\n",
       "    gnomad.amrAc  gnomad.amrHc gnomad.easAf  gnomad.easAn  gnomad.easAc   \n",
       "0            285             5     0.000608          1644             1  \\\n",
       "0            574             2     0.075443          2598           196   \n",
       "0            554            10     0.275613          1386           382   \n",
       "0           2476            39     0.320692          3296          1057   \n",
       "0           1609            17     0.194145          2972           577   \n",
       "..           ...           ...          ...           ...           ...   \n",
       "0            635            16     0.020419          1910            39   \n",
       "0           1723            86     0.060228          3852           232   \n",
       "0             68             2     0.081818           220            18   \n",
       "0           1465             0     0.139273          1156           161   \n",
       "0            642            25     0.099206          1764           175   \n",
       "\n",
       "    gnomad.easHc gnomad.finAf  gnomad.finAn  gnomad.finAc  gnomad.finHc   \n",
       "0              0     0.050104           958            48             0  \\\n",
       "0              0     0.102046          4204           429             0   \n",
       "0              2     0.122881           708            87             2   \n",
       "0             20     0.257278          4878          1255            27   \n",
       "0              8      0.15375          5574           857             6   \n",
       "..           ...          ...           ...           ...           ...   \n",
       "0              1     0.089506          1944           174             7   \n",
       "0              1      0.08975          3844           345             9   \n",
       "0              2     0.079592           490            39             2   \n",
       "0              0     0.153454          4314           662             0   \n",
       "0              0     0.062984          1032            65             2   \n",
       "\n",
       "   gnomad.nfeAf  gnomad.nfeAn  gnomad.nfeAc  gnomad.nfeHc gnomad.asjAf   \n",
       "0      0.122751         12448          1528            26     0.225636  \\\n",
       "0      0.210871         25480          5373            72     0.151239   \n",
       "0      0.327003          8636          2824            47     0.306579   \n",
       "0      0.283761         40298         11435           434     0.216651   \n",
       "0      0.165372         38876          6429           177     0.113576   \n",
       "..          ...           ...           ...           ...          ...   \n",
       "0      0.136704         32040          4380            49     0.189066   \n",
       "0      0.149985         55012          8251           237     0.143775   \n",
       "0      0.091773          4838           444            22     0.180952   \n",
       "0      0.222038         24784          5503             0     0.219403   \n",
       "0      0.107488         38674          4157            16     0.177374   \n",
       "\n",
       "    gnomad.asjAn  gnomad.asjAc  gnomad.asjHc gnomad.sasAf  gnomad.sasAn   \n",
       "0            944           213             2     0.022059          1224  \\\n",
       "0           1534           232             2     0.122714          2078   \n",
       "0            760           233             4     0.300403           992   \n",
       "0           2174           471            12     0.177857          2800   \n",
       "0           2254           256             3     0.096678          3010   \n",
       "..           ...           ...           ...          ...           ...   \n",
       "0           1756           332             8     0.229017          1668   \n",
       "0           2956           425            14     0.091275          3232   \n",
       "0            210            38             3     0.131579           114   \n",
       "0           1340           294             0      0.11203          1330   \n",
       "0           2148           381             8     0.212605          1190   \n",
       "\n",
       "    gnomad.sasAc  gnomad.sasHc gnomad.othAf  gnomad.othAn  gnomad.othAc   \n",
       "0             27             0     0.116838           582            68  \\\n",
       "0            255             0     0.107527           930           100   \n",
       "0            298             3     0.234742           426           100   \n",
       "0            498             3     0.208531          1266           264   \n",
       "0            291             0     0.107963          1306           141   \n",
       "..           ...           ...          ...           ...           ...   \n",
       "0            382            13     0.157407           756           119   \n",
       "0            295            12     0.147222          1440           212   \n",
       "0             15             1     0.126984           126            16   \n",
       "0            149             0      0.22093           688           152   \n",
       "0            253            15     0.129156           782           101   \n",
       "\n",
       "    gnomad.othHc gnomad.maleAf  gnomad.maleAn  gnomad.maleAc  gnomad.maleHc   \n",
       "0              0      0.145535          22084           3214             34  \\\n",
       "0              0      0.102856          34660           3565             30   \n",
       "0              1      0.181277          15788           2862             24   \n",
       "0              8      0.199067          45432           9044            245   \n",
       "0              2      0.108202          47892           5182             85   \n",
       "..           ...           ...            ...            ...            ...   \n",
       "0              6      0.164289          29022           4768            103   \n",
       "0              6      0.159572          50598           8074            306   \n",
       "0              1      0.154401           4158            642             64   \n",
       "0              0      0.183313          26332           4827              0   \n",
       "0              1      0.127211          28386           3611             49   \n",
       "\n",
       "   gnomad.femaleAf  gnomad.femaleAn  gnomad.femaleAc  gnomad.femaleHc   \n",
       "0         0.151461            24772             3752               46  \\\n",
       "0         0.118684            36138             4289               48   \n",
       "0         0.193266            17732             3427               47   \n",
       "0         0.199707            49132             9812              305   \n",
       "0         0.109222            51226             5595              133   \n",
       "..             ...              ...              ...              ...   \n",
       "0         0.162878            34486             5617              132   \n",
       "0          0.16453            58354             9601              368   \n",
       "0         0.151931             5180              787               64   \n",
       "0         0.172595            27608             4765                0   \n",
       "0         0.118505            37374             4429               40   \n",
       "\n",
       "   gnomad.controlsAllAf  gnomad.controlsAllAn  gnomad.controlsAllAc   \n",
       "0              0.128554                  9848                  1266  \\\n",
       "0              0.074837                 16302                  1220   \n",
       "0              0.216009                  7958                  1719   \n",
       "0              0.179372                 19546                  3506   \n",
       "0              0.103308                 21344                  2205   \n",
       "..                  ...                   ...                   ...   \n",
       "0              0.163979                 11288                  1851   \n",
       "0              0.155962                 19960                  3113   \n",
       "0              0.158901                  1674                   266   \n",
       "0              0.155291                 11018                  1711   \n",
       "0              0.132299                  8700                  1151   \n",
       "\n",
       "   topmed.allAf  topmed.allAn  topmed.allAc  topmed.allHc topmed.failedFilter   \n",
       "0      0.084209      125568.0       10574.0           0.0                True  \\\n",
       "0      0.372101      125568.0       46724.0        4255.0                True   \n",
       "0      0.508537      125568.0       63856.0        3328.0                True   \n",
       "0       0.22923      125568.0       28784.0         977.0                True   \n",
       "0      0.191402      125568.0       24034.0         247.0                True   \n",
       "..          ...           ...           ...           ...                 ...   \n",
       "0      0.110832      125568.0       13917.0          51.0                True   \n",
       "0      0.163991      125568.0       20592.0         276.0                True   \n",
       "0      0.373614      125568.0       46914.0         125.0                True   \n",
       "0      0.242052      125568.0       30394.0           0.0                True   \n",
       "0      0.151727      125568.0       19052.0          26.0                True   \n",
       "\n",
       "   dannScore oneKg.allAf oneKg.afrAf oneKg.amrAf oneKg.easAf oneKg.eurAf   \n",
       "0        NaN         NaN         NaN         NaN         NaN         NaN  \\\n",
       "0        NaN         NaN         NaN         NaN         NaN         NaN   \n",
       "0        NaN         NaN         NaN         NaN         NaN         NaN   \n",
       "0        NaN         NaN         NaN         NaN         NaN         NaN   \n",
       "0        NaN         NaN         NaN         NaN         NaN         NaN   \n",
       "..       ...         ...         ...         ...         ...         ...   \n",
       "0        NaN         NaN         NaN         NaN         NaN         NaN   \n",
       "0        NaN         NaN         NaN         NaN         NaN         NaN   \n",
       "0        NaN         NaN         NaN         NaN         NaN         NaN   \n",
       "0        0.3         NaN         NaN         NaN         NaN         NaN   \n",
       "0        NaN         NaN         NaN         NaN         NaN         NaN   \n",
       "\n",
       "   oneKg.sasAf  oneKg.allAn  oneKg.afrAn  oneKg.amrAn  oneKg.easAn   \n",
       "0          NaN          NaN          NaN          NaN          NaN  \\\n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "..         ...          ...          ...          ...          ...   \n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "\n",
       "    oneKg.eurAn  oneKg.sasAn  oneKg.allAc  oneKg.afrAc  oneKg.amrAc   \n",
       "0           NaN          NaN          NaN          NaN          NaN  \\\n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "..          ...          ...          ...          ...          ...   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "0           NaN          NaN          NaN          NaN          NaN   \n",
       "\n",
       "    oneKg.easAc  oneKg.eurAc  oneKg.sasAc cosmic inLowComplexityRegion   \n",
       "0           NaN          NaN          NaN    NaN                   NaN  \\\n",
       "0           NaN          NaN          NaN    NaN                   NaN   \n",
       "0           NaN          NaN          NaN    NaN                   NaN   \n",
       "0           NaN          NaN          NaN    NaN                   NaN   \n",
       "0           NaN          NaN          NaN    NaN                   NaN   \n",
       "..          ...          ...          ...    ...                   ...   \n",
       "0           NaN          NaN          NaN    NaN                  True   \n",
       "0           NaN          NaN          NaN    NaN                  True   \n",
       "0           NaN          NaN          NaN    NaN                  True   \n",
       "0           NaN          NaN          NaN    NaN                  True   \n",
       "0           NaN          NaN          NaN    NaN                  True   \n",
       "\n",
       "   gerpScore                                  regulatoryRegions  \n",
       "0        NaN                                                NaN  \n",
       "0        NaN                                                NaN  \n",
       "0        NaN                                                NaN  \n",
       "0        NaN                                                NaN  \n",
       "0        NaN                                                NaN  \n",
       "..       ...                                                ...  \n",
       "0        NaN  [{'id': 'ENSR00000664607', 'type': 'enhancer',...  \n",
       "0        NaN                                                NaN  \n",
       "0        NaN                                                NaN  \n",
       "0     -0.675                                                NaN  \n",
       "0        NaN                                                NaN  \n",
       "\n",
       "[114 rows x 95 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = AnnotatedData.multiple_to_df(positions, \"variants\")\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>refAllele</th>\n",
       "      <th>altAlleles</th>\n",
       "      <th>filters</th>\n",
       "      <th>mappingQuality</th>\n",
       "      <th>cyatogeneticBand</th>\n",
       "      <th>vid</th>\n",
       "      <th>begin</th>\n",
       "      <th>end</th>\n",
       "      <th>altAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>hgvsg</th>\n",
       "      <th>phylopScore</th>\n",
       "      <th>phyloPPrimateScore</th>\n",
       "      <th>transcripts</th>\n",
       "      <th>dbsnp</th>\n",
       "      <th>vcfInfo.DP</th>\n",
       "      <th>gnomad.coverage</th>\n",
       "      <th>gnomad.failedFilter</th>\n",
       "      <th>gnomad.allAf</th>\n",
       "      <th>gnomad.allAn</th>\n",
       "      <th>gnomad.allAc</th>\n",
       "      <th>gnomad.allHc</th>\n",
       "      <th>gnomad.afrAf</th>\n",
       "      <th>gnomad.afrAn</th>\n",
       "      <th>gnomad.afrAc</th>\n",
       "      <th>gnomad.afrHc</th>\n",
       "      <th>gnomad.amrAf</th>\n",
       "      <th>gnomad.amrAn</th>\n",
       "      <th>gnomad.amrAc</th>\n",
       "      <th>gnomad.amrHc</th>\n",
       "      <th>gnomad.easAf</th>\n",
       "      <th>gnomad.easAn</th>\n",
       "      <th>gnomad.easAc</th>\n",
       "      <th>gnomad.easHc</th>\n",
       "      <th>gnomad.finAf</th>\n",
       "      <th>gnomad.finAn</th>\n",
       "      <th>gnomad.finAc</th>\n",
       "      <th>gnomad.finHc</th>\n",
       "      <th>gnomad.nfeAf</th>\n",
       "      <th>gnomad.nfeAn</th>\n",
       "      <th>gnomad.nfeAc</th>\n",
       "      <th>gnomad.nfeHc</th>\n",
       "      <th>gnomad.asjAf</th>\n",
       "      <th>gnomad.asjAn</th>\n",
       "      <th>gnomad.asjAc</th>\n",
       "      <th>gnomad.asjHc</th>\n",
       "      <th>gnomad.sasAf</th>\n",
       "      <th>gnomad.sasAn</th>\n",
       "      <th>gnomad.sasAc</th>\n",
       "      <th>gnomad.sasHc</th>\n",
       "      <th>gnomad.othAf</th>\n",
       "      <th>gnomad.othAn</th>\n",
       "      <th>gnomad.othAc</th>\n",
       "      <th>gnomad.othHc</th>\n",
       "      <th>gnomad.maleAf</th>\n",
       "      <th>gnomad.maleAn</th>\n",
       "      <th>gnomad.maleAc</th>\n",
       "      <th>gnomad.maleHc</th>\n",
       "      <th>gnomad.femaleAf</th>\n",
       "      <th>gnomad.femaleAn</th>\n",
       "      <th>gnomad.femaleAc</th>\n",
       "      <th>gnomad.femaleHc</th>\n",
       "      <th>gnomad.controlsAllAf</th>\n",
       "      <th>gnomad.controlsAllAn</th>\n",
       "      <th>gnomad.controlsAllAc</th>\n",
       "      <th>topmed.allAf</th>\n",
       "      <th>topmed.allAn</th>\n",
       "      <th>topmed.allAc</th>\n",
       "      <th>topmed.allHc</th>\n",
       "      <th>topmed.failedFilter</th>\n",
       "      <th>dannScore</th>\n",
       "      <th>oneKg.allAf</th>\n",
       "      <th>oneKg.afrAf</th>\n",
       "      <th>oneKg.amrAf</th>\n",
       "      <th>oneKg.easAf</th>\n",
       "      <th>oneKg.eurAf</th>\n",
       "      <th>oneKg.sasAf</th>\n",
       "      <th>oneKg.allAn</th>\n",
       "      <th>oneKg.afrAn</th>\n",
       "      <th>oneKg.amrAn</th>\n",
       "      <th>oneKg.easAn</th>\n",
       "      <th>oneKg.eurAn</th>\n",
       "      <th>oneKg.sasAn</th>\n",
       "      <th>oneKg.allAc</th>\n",
       "      <th>oneKg.afrAc</th>\n",
       "      <th>oneKg.amrAc</th>\n",
       "      <th>oneKg.easAc</th>\n",
       "      <th>oneKg.eurAc</th>\n",
       "      <th>oneKg.sasAc</th>\n",
       "      <th>cosmic</th>\n",
       "      <th>inLowComplexityRegion</th>\n",
       "      <th>gerpScore</th>\n",
       "      <th>regulatoryRegions</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>9325863</td>\n",
       "      <td>C</td>\n",
       "      <td>[G]</td>\n",
       "      <td>[filtered_reads, mapping_quality, non_homref_n...</td>\n",
       "      <td>8.65</td>\n",
       "      <td>21p11.2</td>\n",
       "      <td>21-9325863-C-G</td>\n",
       "      <td>9325863</td>\n",
       "      <td>9325863</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.9325863C&gt;G</td>\n",
       "      <td>-0.8</td>\n",
       "      <td>0.171</td>\n",
       "      <td>[{'transcript': 'ENST00000622961.3', 'source':...</td>\n",
       "      <td>[rs796282180]</td>\n",
       "      <td>306</td>\n",
       "      <td>30</td>\n",
       "      <td>True</td>\n",
       "      <td>0.118328</td>\n",
       "      <td>57146</td>\n",
       "      <td>6762</td>\n",
       "      <td>0</td>\n",
       "      <td>0.205042</td>\n",
       "      <td>13446</td>\n",
       "      <td>2757</td>\n",
       "      <td>0</td>\n",
       "      <td>0.088286</td>\n",
       "      <td>5822</td>\n",
       "      <td>514</td>\n",
       "      <td>0</td>\n",
       "      <td>0.040883</td>\n",
       "      <td>2446</td>\n",
       "      <td>100</td>\n",
       "      <td>0</td>\n",
       "      <td>0.150943</td>\n",
       "      <td>3710</td>\n",
       "      <td>560</td>\n",
       "      <td>0</td>\n",
       "      <td>0.089256</td>\n",
       "      <td>27382</td>\n",
       "      <td>2444</td>\n",
       "      <td>0</td>\n",
       "      <td>0.062331</td>\n",
       "      <td>1476</td>\n",
       "      <td>92</td>\n",
       "      <td>0</td>\n",
       "      <td>0.111321</td>\n",
       "      <td>1590</td>\n",
       "      <td>177</td>\n",
       "      <td>0</td>\n",
       "      <td>0.079012</td>\n",
       "      <td>810</td>\n",
       "      <td>64</td>\n",
       "      <td>0</td>\n",
       "      <td>0.119863</td>\n",
       "      <td>27498</td>\n",
       "      <td>3296</td>\n",
       "      <td>0</td>\n",
       "      <td>0.116905</td>\n",
       "      <td>29648</td>\n",
       "      <td>3466</td>\n",
       "      <td>0</td>\n",
       "      <td>0.121282</td>\n",
       "      <td>12170</td>\n",
       "      <td>1476</td>\n",
       "      <td>0.03997</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>5019.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.46</td>\n",
       "      <td>[{'id': 'ENSR00000140262', 'type': 'promoter',...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>9577267</td>\n",
       "      <td>C</td>\n",
       "      <td>[T]</td>\n",
       "      <td>[non_homref_normal, weak_evidence]</td>\n",
       "      <td>34.56</td>\n",
       "      <td>21p11.2</td>\n",
       "      <td>21-9577267-C-T</td>\n",
       "      <td>9577267</td>\n",
       "      <td>9577267</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.9577267C&gt;T</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.292</td>\n",
       "      <td>[{'transcript': 'ENST00000623408.1', 'source':...</td>\n",
       "      <td>[rs1414013851]</td>\n",
       "      <td>227</td>\n",
       "      <td>24</td>\n",
       "      <td>True</td>\n",
       "      <td>0.16528</td>\n",
       "      <td>104290</td>\n",
       "      <td>17237</td>\n",
       "      <td>1</td>\n",
       "      <td>0.131743</td>\n",
       "      <td>27918</td>\n",
       "      <td>3678</td>\n",
       "      <td>0</td>\n",
       "      <td>0.198579</td>\n",
       "      <td>10414</td>\n",
       "      <td>2068</td>\n",
       "      <td>1</td>\n",
       "      <td>0.168754</td>\n",
       "      <td>3354</td>\n",
       "      <td>566</td>\n",
       "      <td>0</td>\n",
       "      <td>0.170216</td>\n",
       "      <td>7420</td>\n",
       "      <td>1263</td>\n",
       "      <td>0</td>\n",
       "      <td>0.178455</td>\n",
       "      <td>47250</td>\n",
       "      <td>8432</td>\n",
       "      <td>0</td>\n",
       "      <td>0.193937</td>\n",
       "      <td>2408</td>\n",
       "      <td>467</td>\n",
       "      <td>0</td>\n",
       "      <td>0.129664</td>\n",
       "      <td>3270</td>\n",
       "      <td>424</td>\n",
       "      <td>0</td>\n",
       "      <td>0.150636</td>\n",
       "      <td>1414</td>\n",
       "      <td>213</td>\n",
       "      <td>0</td>\n",
       "      <td>0.163481</td>\n",
       "      <td>50954</td>\n",
       "      <td>8330</td>\n",
       "      <td>1</td>\n",
       "      <td>0.166998</td>\n",
       "      <td>53336</td>\n",
       "      <td>8907</td>\n",
       "      <td>0</td>\n",
       "      <td>0.16508</td>\n",
       "      <td>22486</td>\n",
       "      <td>3712</td>\n",
       "      <td>0.151583</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>19034.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.718</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>9732136</td>\n",
       "      <td>C</td>\n",
       "      <td>[A]</td>\n",
       "      <td>[weak_evidence]</td>\n",
       "      <td>71.44</td>\n",
       "      <td>21p11.2</td>\n",
       "      <td>21-9732136-C-A</td>\n",
       "      <td>9732136</td>\n",
       "      <td>9732136</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.9732136C&gt;A</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>0.019</td>\n",
       "      <td>None</td>\n",
       "      <td>[rs910381466]</td>\n",
       "      <td>393</td>\n",
       "      <td>59</td>\n",
       "      <td>True</td>\n",
       "      <td>0.181347</td>\n",
       "      <td>101518</td>\n",
       "      <td>18410</td>\n",
       "      <td>58</td>\n",
       "      <td>0.081754</td>\n",
       "      <td>29760</td>\n",
       "      <td>2433</td>\n",
       "      <td>2</td>\n",
       "      <td>0.231872</td>\n",
       "      <td>9764</td>\n",
       "      <td>2264</td>\n",
       "      <td>7</td>\n",
       "      <td>0.126896</td>\n",
       "      <td>3428</td>\n",
       "      <td>435</td>\n",
       "      <td>1</td>\n",
       "      <td>0.230513</td>\n",
       "      <td>6902</td>\n",
       "      <td>1591</td>\n",
       "      <td>5</td>\n",
       "      <td>0.23185</td>\n",
       "      <td>44214</td>\n",
       "      <td>10251</td>\n",
       "      <td>38</td>\n",
       "      <td>0.164279</td>\n",
       "      <td>2234</td>\n",
       "      <td>367</td>\n",
       "      <td>2</td>\n",
       "      <td>0.207627</td>\n",
       "      <td>3068</td>\n",
       "      <td>637</td>\n",
       "      <td>2</td>\n",
       "      <td>0.168666</td>\n",
       "      <td>1334</td>\n",
       "      <td>225</td>\n",
       "      <td>1</td>\n",
       "      <td>0.181859</td>\n",
       "      <td>49346</td>\n",
       "      <td>8974</td>\n",
       "      <td>31</td>\n",
       "      <td>0.180863</td>\n",
       "      <td>52172</td>\n",
       "      <td>9436</td>\n",
       "      <td>27</td>\n",
       "      <td>0.169482</td>\n",
       "      <td>22026</td>\n",
       "      <td>3733</td>\n",
       "      <td>0.22583</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>28357.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.22</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>27416167</td>\n",
       "      <td>T</td>\n",
       "      <td>[C]</td>\n",
       "      <td>[alt_allele_in_normal, non_homref_normal, weak...</td>\n",
       "      <td>194.46</td>\n",
       "      <td>21q21.3</td>\n",
       "      <td>21-27416167-T-C</td>\n",
       "      <td>27416167</td>\n",
       "      <td>27416167</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NC_000021.9:g.27416167T&gt;C</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.163</td>\n",
       "      <td>[{'transcript': 'ENST00000420186.2', 'source':...</td>\n",
       "      <td>[rs866572428]</td>\n",
       "      <td>113</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.161367</td>\n",
       "      <td>102834</td>\n",
       "      <td>16594</td>\n",
       "      <td>3980</td>\n",
       "      <td>0.072592</td>\n",
       "      <td>27262</td>\n",
       "      <td>1979</td>\n",
       "      <td>377</td>\n",
       "      <td>0.147874</td>\n",
       "      <td>8514</td>\n",
       "      <td>1259</td>\n",
       "      <td>386</td>\n",
       "      <td>0.246599</td>\n",
       "      <td>2940</td>\n",
       "      <td>725</td>\n",
       "      <td>200</td>\n",
       "      <td>0.08338</td>\n",
       "      <td>3550</td>\n",
       "      <td>296</td>\n",
       "      <td>115</td>\n",
       "      <td>0.210971</td>\n",
       "      <td>52538</td>\n",
       "      <td>11084</td>\n",
       "      <td>2603</td>\n",
       "      <td>0.178827</td>\n",
       "      <td>2796</td>\n",
       "      <td>500</td>\n",
       "      <td>101</td>\n",
       "      <td>0.107071</td>\n",
       "      <td>2998</td>\n",
       "      <td>321</td>\n",
       "      <td>90</td>\n",
       "      <td>0.174312</td>\n",
       "      <td>1308</td>\n",
       "      <td>228</td>\n",
       "      <td>59</td>\n",
       "      <td>0.148092</td>\n",
       "      <td>47700</td>\n",
       "      <td>7064</td>\n",
       "      <td>1751</td>\n",
       "      <td>0.172852</td>\n",
       "      <td>55134</td>\n",
       "      <td>9530</td>\n",
       "      <td>2229</td>\n",
       "      <td>0.112171</td>\n",
       "      <td>18026</td>\n",
       "      <td>2022</td>\n",
       "      <td>0.049694</td>\n",
       "      <td>125568.0</td>\n",
       "      <td>6240.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.46</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>0.266</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  position refAllele altAlleles   \n",
       "0      chr21   9325863         C        [G]  \\\n",
       "0      chr21   9577267         C        [T]   \n",
       "0      chr21   9732136         C        [A]   \n",
       "0      chr21  27416167         T        [C]   \n",
       "\n",
       "                                             filters  mappingQuality   \n",
       "0  [filtered_reads, mapping_quality, non_homref_n...            8.65  \\\n",
       "0                 [non_homref_normal, weak_evidence]           34.56   \n",
       "0                                    [weak_evidence]           71.44   \n",
       "0  [alt_allele_in_normal, non_homref_normal, weak...          194.46   \n",
       "\n",
       "  cyatogeneticBand              vid     begin       end altAllele variantType   \n",
       "0          21p11.2   21-9325863-C-G   9325863   9325863         G         SNV  \\\n",
       "0          21p11.2   21-9577267-C-T   9577267   9577267         T         SNV   \n",
       "0          21p11.2   21-9732136-C-A   9732136   9732136         A         SNV   \n",
       "0          21q21.3  21-27416167-T-C  27416167  27416167         C         SNV   \n",
       "\n",
       "                       hgvsg  phylopScore  phyloPPrimateScore   \n",
       "0   NC_000021.9:g.9325863C>G         -0.8               0.171  \\\n",
       "0   NC_000021.9:g.9577267C>T          0.2               0.292   \n",
       "0   NC_000021.9:g.9732136C>A         -0.7               0.019   \n",
       "0  NC_000021.9:g.27416167T>C          0.0               0.163   \n",
       "\n",
       "                                         transcripts           dbsnp   \n",
       "0  [{'transcript': 'ENST00000622961.3', 'source':...   [rs796282180]  \\\n",
       "0  [{'transcript': 'ENST00000623408.1', 'source':...  [rs1414013851]   \n",
       "0                                               None   [rs910381466]   \n",
       "0  [{'transcript': 'ENST00000420186.2', 'source':...   [rs866572428]   \n",
       "\n",
       "  vcfInfo.DP  gnomad.coverage gnomad.failedFilter gnomad.allAf  gnomad.allAn   \n",
       "0        306               30                True     0.118328         57146  \\\n",
       "0        227               24                True      0.16528        104290   \n",
       "0        393               59                True     0.181347        101518   \n",
       "0        113                5                 NaN     0.161367        102834   \n",
       "\n",
       "   gnomad.allAc  gnomad.allHc gnomad.afrAf  gnomad.afrAn  gnomad.afrAc   \n",
       "0          6762             0     0.205042         13446          2757  \\\n",
       "0         17237             1     0.131743         27918          3678   \n",
       "0         18410            58     0.081754         29760          2433   \n",
       "0         16594          3980     0.072592         27262          1979   \n",
       "\n",
       "   gnomad.afrHc gnomad.amrAf  gnomad.amrAn  gnomad.amrAc  gnomad.amrHc   \n",
       "0             0     0.088286          5822           514             0  \\\n",
       "0             0     0.198579         10414          2068             1   \n",
       "0             2     0.231872          9764          2264             7   \n",
       "0           377     0.147874          8514          1259           386   \n",
       "\n",
       "  gnomad.easAf  gnomad.easAn  gnomad.easAc  gnomad.easHc gnomad.finAf   \n",
       "0     0.040883          2446           100             0     0.150943  \\\n",
       "0     0.168754          3354           566             0     0.170216   \n",
       "0     0.126896          3428           435             1     0.230513   \n",
       "0     0.246599          2940           725           200      0.08338   \n",
       "\n",
       "   gnomad.finAn  gnomad.finAc  gnomad.finHc gnomad.nfeAf  gnomad.nfeAn   \n",
       "0          3710           560             0     0.089256         27382  \\\n",
       "0          7420          1263             0     0.178455         47250   \n",
       "0          6902          1591             5      0.23185         44214   \n",
       "0          3550           296           115     0.210971         52538   \n",
       "\n",
       "   gnomad.nfeAc  gnomad.nfeHc gnomad.asjAf  gnomad.asjAn  gnomad.asjAc   \n",
       "0          2444             0     0.062331          1476            92  \\\n",
       "0          8432             0     0.193937          2408           467   \n",
       "0         10251            38     0.164279          2234           367   \n",
       "0         11084          2603     0.178827          2796           500   \n",
       "\n",
       "   gnomad.asjHc gnomad.sasAf  gnomad.sasAn  gnomad.sasAc  gnomad.sasHc   \n",
       "0             0     0.111321          1590           177             0  \\\n",
       "0             0     0.129664          3270           424             0   \n",
       "0             2     0.207627          3068           637             2   \n",
       "0           101     0.107071          2998           321            90   \n",
       "\n",
       "  gnomad.othAf  gnomad.othAn  gnomad.othAc  gnomad.othHc gnomad.maleAf   \n",
       "0     0.079012           810            64             0      0.119863  \\\n",
       "0     0.150636          1414           213             0      0.163481   \n",
       "0     0.168666          1334           225             1      0.181859   \n",
       "0     0.174312          1308           228            59      0.148092   \n",
       "\n",
       "   gnomad.maleAn  gnomad.maleAc  gnomad.maleHc gnomad.femaleAf   \n",
       "0          27498           3296              0        0.116905  \\\n",
       "0          50954           8330              1        0.166998   \n",
       "0          49346           8974             31        0.180863   \n",
       "0          47700           7064           1751        0.172852   \n",
       "\n",
       "   gnomad.femaleAn  gnomad.femaleAc  gnomad.femaleHc gnomad.controlsAllAf   \n",
       "0            29648             3466                0             0.121282  \\\n",
       "0            53336             8907                0              0.16508   \n",
       "0            52172             9436               27             0.169482   \n",
       "0            55134             9530             2229             0.112171   \n",
       "\n",
       "   gnomad.controlsAllAn  gnomad.controlsAllAc topmed.allAf  topmed.allAn   \n",
       "0                 12170                  1476      0.03997      125568.0  \\\n",
       "0                 22486                  3712     0.151583      125568.0   \n",
       "0                 22026                  3733      0.22583      125568.0   \n",
       "0                 18026                  2022     0.049694      125568.0   \n",
       "\n",
       "   topmed.allAc  topmed.allHc topmed.failedFilter dannScore oneKg.allAf   \n",
       "0        5019.0           0.0                True      0.17         NaN  \\\n",
       "0       19034.0           0.0                True       NaN         NaN   \n",
       "0       28357.0          46.0                True       NaN         NaN   \n",
       "0        6240.0           0.0                True      0.46         NaN   \n",
       "\n",
       "  oneKg.afrAf oneKg.amrAf oneKg.easAf oneKg.eurAf oneKg.sasAf  oneKg.allAn   \n",
       "0         NaN         NaN         NaN         NaN         NaN          NaN  \\\n",
       "0         NaN         NaN         NaN         NaN         NaN          NaN   \n",
       "0         NaN         NaN         NaN         NaN         NaN          NaN   \n",
       "0         NaN         NaN         NaN         NaN         NaN          NaN   \n",
       "\n",
       "   oneKg.afrAn  oneKg.amrAn  oneKg.easAn  oneKg.eurAn  oneKg.sasAn   \n",
       "0          NaN          NaN          NaN          NaN          NaN  \\\n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "\n",
       "   oneKg.allAc  oneKg.afrAc  oneKg.amrAc  oneKg.easAc  oneKg.eurAc   \n",
       "0          NaN          NaN          NaN          NaN          NaN  \\\n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "0          NaN          NaN          NaN          NaN          NaN   \n",
       "\n",
       "   oneKg.sasAc cosmic inLowComplexityRegion gerpScore   \n",
       "0          NaN    NaN                   NaN      1.46  \\\n",
       "0          NaN    NaN                   NaN     0.718   \n",
       "0          NaN    NaN                   NaN      1.22   \n",
       "0          NaN    NaN                  True     0.266   \n",
       "\n",
       "                                   regulatoryRegions  \n",
       "0  [{'id': 'ENSR00000140262', 'type': 'promoter',...  \n",
       "0                                                NaN  \n",
       "0                                                NaN  \n",
       "0                                                NaN  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.dropna(subset= [\"phyloPPrimateScore\", \"gerpScore\"])\n",
    "df[(df[\"gerpScore\"]>0) & (df[\"phyloPPrimateScore\"]>0)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene Filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD.pLi</th>\n",
       "      <th>gnomAD.pRec</th>\n",
       "      <th>gnomAD.pNull</th>\n",
       "      <th>gnomAD.synZ</th>\n",
       "      <th>gnomAD.misZ</th>\n",
       "      <th>gnomAD.loeuf</th>\n",
       "      <th>clingenDosageSensitivityMap.haploinsufficiency</th>\n",
       "      <th>clingenDosageSensitivityMap.triplosensitivity</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic.roleInCancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AATBC</td>\n",
       "      <td>51526.0</td>\n",
       "      <td>284837</td>\n",
       "      <td>ENSG00000215458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000291052</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>16022.0</td>\n",
       "      <td>150000</td>\n",
       "      <td>ENSG00000243064</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCG1</td>\n",
       "      <td>73.0</td>\n",
       "      <td>9619</td>\n",
       "      <td>ENSG00000160179</td>\n",
       "      <td>[{'mimNumber': 603076, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>0.112</td>\n",
       "      <td>0.888</td>\n",
       "      <td>0.00000934</td>\n",
       "      <td>0.423</td>\n",
       "      <td>2.14</td>\n",
       "      <td>0.461</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ADARB1</td>\n",
       "      <td>226.0</td>\n",
       "      <td>104</td>\n",
       "      <td>ENSG00000197381</td>\n",
       "      <td>[{'mimNumber': 601218, 'geneName': 'Adenosine ...</td>\n",
       "      <td>0.803</td>\n",
       "      <td>0.197</td>\n",
       "      <td>0.00000832</td>\n",
       "      <td>0.603</td>\n",
       "      <td>3.52</td>\n",
       "      <td>0.399</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>YBEY</td>\n",
       "      <td>1299.0</td>\n",
       "      <td>54059</td>\n",
       "      <td>ENSG00000182362</td>\n",
       "      <td>[{'mimNumber': 617461, 'geneName': 'YBEY metal...</td>\n",
       "      <td>1.34E-7</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.937</td>\n",
       "      <td>1.13</td>\n",
       "      <td>0.0149</td>\n",
       "      <td>1.92</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>YRDCP3</td>\n",
       "      <td>39921.0</td>\n",
       "      <td>100861429</td>\n",
       "      <td>ENSG00000230859</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>ZBTB21</td>\n",
       "      <td>13083.0</td>\n",
       "      <td>49854</td>\n",
       "      <td>ENSG00000173276</td>\n",
       "      <td>[{'mimNumber': 616485, 'geneName': 'Zinc finge...</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.00222</td>\n",
       "      <td>5.45E-10</td>\n",
       "      <td>-1.59</td>\n",
       "      <td>0.512</td>\n",
       "      <td>0.249</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>ZNF295-AS1</td>\n",
       "      <td>23130.0</td>\n",
       "      <td>150142</td>\n",
       "      <td>ENSG00000237232</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>ZNF355P</td>\n",
       "      <td>17218.0</td>\n",
       "      <td>100505852</td>\n",
       "      <td>ENSG00000168122</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>442 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "0         AATBC  51526.0     284837  ENSG00000215458  \\\n",
       "1        ABCC13      NaN        NaN  ENSG00000291052   \n",
       "2        ABCC13  16022.0     150000  ENSG00000243064   \n",
       "3         ABCG1     73.0       9619  ENSG00000160179   \n",
       "4        ADARB1    226.0        104  ENSG00000197381   \n",
       "..          ...      ...        ...              ...   \n",
       "437        YBEY   1299.0      54059  ENSG00000182362   \n",
       "438      YRDCP3  39921.0  100861429  ENSG00000230859   \n",
       "439      ZBTB21  13083.0      49854  ENSG00000173276   \n",
       "440  ZNF295-AS1  23130.0     150142  ENSG00000237232   \n",
       "441     ZNF355P  17218.0  100505852  ENSG00000168122   \n",
       "\n",
       "                                                  omim gnomAD.pLi gnomAD.pRec   \n",
       "0                                                  NaN        NaN         NaN  \\\n",
       "1    [{'mimNumber': 608835, 'geneName': 'ATP-bindin...        NaN         NaN   \n",
       "2    [{'mimNumber': 608835, 'geneName': 'ATP-bindin...        NaN         NaN   \n",
       "3    [{'mimNumber': 603076, 'geneName': 'ATP-bindin...      0.112       0.888   \n",
       "4    [{'mimNumber': 601218, 'geneName': 'Adenosine ...      0.803       0.197   \n",
       "..                                                 ...        ...         ...   \n",
       "437  [{'mimNumber': 617461, 'geneName': 'YBEY metal...    1.34E-7      0.0625   \n",
       "438                                                NaN        NaN         NaN   \n",
       "439  [{'mimNumber': 616485, 'geneName': 'Zinc finge...      0.998     0.00222   \n",
       "440                                                NaN        NaN         NaN   \n",
       "441                                                NaN        NaN         NaN   \n",
       "\n",
       "    gnomAD.pNull gnomAD.synZ gnomAD.misZ gnomAD.loeuf   \n",
       "0            NaN         NaN         NaN          NaN  \\\n",
       "1            NaN         NaN         NaN          NaN   \n",
       "2            NaN         NaN         NaN          NaN   \n",
       "3     0.00000934       0.423        2.14        0.461   \n",
       "4     0.00000832       0.603        3.52        0.399   \n",
       "..           ...         ...         ...          ...   \n",
       "437        0.937        1.13      0.0149         1.92   \n",
       "438          NaN         NaN         NaN          NaN   \n",
       "439     5.45E-10       -1.59       0.512        0.249   \n",
       "440          NaN         NaN         NaN          NaN   \n",
       "441          NaN         NaN         NaN          NaN   \n",
       "\n",
       "    clingenDosageSensitivityMap.haploinsufficiency   \n",
       "0                                              NaN  \\\n",
       "1                                              NaN   \n",
       "2                                              NaN   \n",
       "3                                              NaN   \n",
       "4                                              NaN   \n",
       "..                                             ...   \n",
       "437                                            NaN   \n",
       "438                                            NaN   \n",
       "439                                            NaN   \n",
       "440                                            NaN   \n",
       "441                                            NaN   \n",
       "\n",
       "    clingenDosageSensitivityMap.triplosensitivity clingenGeneValidity   \n",
       "0                                             NaN                 NaN  \\\n",
       "1                                             NaN                 NaN   \n",
       "2                                             NaN                 NaN   \n",
       "3                                             NaN                 NaN   \n",
       "4                                             NaN                 NaN   \n",
       "..                                            ...                 ...   \n",
       "437                                           NaN                 NaN   \n",
       "438                                           NaN                 NaN   \n",
       "439                                           NaN                 NaN   \n",
       "440                                           NaN                 NaN   \n",
       "441                                           NaN                 NaN   \n",
       "\n",
       "    cosmic.roleInCancer  \n",
       "0                   NaN  \n",
       "1                   NaN  \n",
       "2                   NaN  \n",
       "3                   NaN  \n",
       "4                   NaN  \n",
       "..                  ...  \n",
       "437                 NaN  \n",
       "438                 NaN  \n",
       "439                 NaN  \n",
       "440                 NaN  \n",
       "441                 NaN  \n",
       "\n",
       "[442 rows x 15 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD.pLi</th>\n",
       "      <th>gnomAD.pRec</th>\n",
       "      <th>gnomAD.pNull</th>\n",
       "      <th>gnomAD.synZ</th>\n",
       "      <th>gnomAD.misZ</th>\n",
       "      <th>gnomAD.loeuf</th>\n",
       "      <th>clingenDosageSensitivityMap.haploinsufficiency</th>\n",
       "      <th>clingenDosageSensitivityMap.triplosensitivity</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic.roleInCancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000291052</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>16022.0</td>\n",
       "      <td>150000</td>\n",
       "      <td>ENSG00000243064</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "1  ABCC13      NaN        NaN  ENSG00000291052  \\\n",
       "2  ABCC13  16022.0     150000  ENSG00000243064   \n",
       "\n",
       "                                                omim gnomAD.pLi gnomAD.pRec   \n",
       "1  [{'mimNumber': 608835, 'geneName': 'ATP-bindin...        NaN         NaN  \\\n",
       "2  [{'mimNumber': 608835, 'geneName': 'ATP-bindin...        NaN         NaN   \n",
       "\n",
       "  gnomAD.pNull gnomAD.synZ gnomAD.misZ gnomAD.loeuf   \n",
       "1          NaN         NaN         NaN          NaN  \\\n",
       "2          NaN         NaN         NaN          NaN   \n",
       "\n",
       "  clingenDosageSensitivityMap.haploinsufficiency   \n",
       "1                                            NaN  \\\n",
       "2                                            NaN   \n",
       "\n",
       "  clingenDosageSensitivityMap.triplosensitivity clingenGeneValidity   \n",
       "1                                           NaN                 NaN  \\\n",
       "2                                           NaN                 NaN   \n",
       "\n",
       "  cosmic.roleInCancer  \n",
       "1                 NaN  \n",
       "2                 NaN  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.genes[annotated_data.genes[\"name\"] == \"ABCC13\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD.pLi</th>\n",
       "      <th>gnomAD.pRec</th>\n",
       "      <th>gnomAD.pNull</th>\n",
       "      <th>gnomAD.synZ</th>\n",
       "      <th>gnomAD.misZ</th>\n",
       "      <th>gnomAD.loeuf</th>\n",
       "      <th>clingenDosageSensitivityMap.haploinsufficiency</th>\n",
       "      <th>clingenDosageSensitivityMap.triplosensitivity</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic.roleInCancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCC13</td>\n",
       "      <td>16022.0</td>\n",
       "      <td>150000</td>\n",
       "      <td>ENSG00000243064</td>\n",
       "      <td>[{'mimNumber': 608835, 'geneName': 'ATP-bindin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>ZBTB21</td>\n",
       "      <td>13083.0</td>\n",
       "      <td>49854</td>\n",
       "      <td>ENSG00000173276</td>\n",
       "      <td>[{'mimNumber': 616485, 'geneName': 'Zinc finge...</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.00222</td>\n",
       "      <td>5.45E-10</td>\n",
       "      <td>-1.59</td>\n",
       "      <td>0.512</td>\n",
       "      <td>0.249</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "2    ABCC13  16022.0     150000  ENSG00000243064  \\\n",
       "439  ZBTB21  13083.0      49854  ENSG00000173276   \n",
       "\n",
       "                                                  omim gnomAD.pLi gnomAD.pRec   \n",
       "2    [{'mimNumber': 608835, 'geneName': 'ATP-bindin...        NaN         NaN  \\\n",
       "439  [{'mimNumber': 616485, 'geneName': 'Zinc finge...      0.998     0.00222   \n",
       "\n",
       "    gnomAD.pNull gnomAD.synZ gnomAD.misZ gnomAD.loeuf   \n",
       "2            NaN         NaN         NaN          NaN  \\\n",
       "439     5.45E-10       -1.59       0.512        0.249   \n",
       "\n",
       "    clingenDosageSensitivityMap.haploinsufficiency   \n",
       "2                                              NaN  \\\n",
       "439                                            NaN   \n",
       "\n",
       "    clingenDosageSensitivityMap.triplosensitivity clingenGeneValidity   \n",
       "2                                             NaN                 NaN  \\\n",
       "439                                           NaN                 NaN   \n",
       "\n",
       "    cosmic.roleInCancer  \n",
       "2                   NaN  \n",
       "439                 NaN  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "required_gene_ids = [\"ENSG00000173276\", \"ENSG00000243064\"]\n",
    "annotated_data.genes[annotated_data.genes[\"ensemblGeneId\"].isin(required_gene_ids)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>hgncId</th>\n",
       "      <th>ncbiGeneId</th>\n",
       "      <th>ensemblGeneId</th>\n",
       "      <th>omim</th>\n",
       "      <th>gnomAD.pLi</th>\n",
       "      <th>gnomAD.pRec</th>\n",
       "      <th>gnomAD.pNull</th>\n",
       "      <th>gnomAD.synZ</th>\n",
       "      <th>gnomAD.misZ</th>\n",
       "      <th>gnomAD.loeuf</th>\n",
       "      <th>clingenDosageSensitivityMap.haploinsufficiency</th>\n",
       "      <th>clingenDosageSensitivityMap.triplosensitivity</th>\n",
       "      <th>clingenGeneValidity</th>\n",
       "      <th>cosmic.roleInCancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>ERG</td>\n",
       "      <td>3446.0</td>\n",
       "      <td>2078</td>\n",
       "      <td>ENSG00000157554</td>\n",
       "      <td>[{'mimNumber': 165080, 'geneName': 'ETS transc...</td>\n",
       "      <td>0.964</td>\n",
       "      <td>0.0359</td>\n",
       "      <td>5.03E-7</td>\n",
       "      <td>0.228</td>\n",
       "      <td>2.53</td>\n",
       "      <td>0.329</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[oncogene, fusion]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>TMPRSS2</td>\n",
       "      <td>11876.0</td>\n",
       "      <td>7113</td>\n",
       "      <td>ENSG00000184012</td>\n",
       "      <td>[{'mimNumber': 602060, 'geneName': 'Transmembr...</td>\n",
       "      <td>2.40E-10</td>\n",
       "      <td>0.880</td>\n",
       "      <td>0.120</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.399</td>\n",
       "      <td>0.942</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[fusion]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        name   hgncId ncbiGeneId    ensemblGeneId   \n",
       "198      ERG   3446.0       2078  ENSG00000157554  \\\n",
       "418  TMPRSS2  11876.0       7113  ENSG00000184012   \n",
       "\n",
       "                                                  omim gnomAD.pLi gnomAD.pRec   \n",
       "198  [{'mimNumber': 165080, 'geneName': 'ETS transc...      0.964      0.0359  \\\n",
       "418  [{'mimNumber': 602060, 'geneName': 'Transmembr...   2.40E-10       0.880   \n",
       "\n",
       "    gnomAD.pNull gnomAD.synZ gnomAD.misZ gnomAD.loeuf   \n",
       "198      5.03E-7       0.228        2.53        0.329  \\\n",
       "418        0.120       0.758       0.399        0.942   \n",
       "\n",
       "    clingenDosageSensitivityMap.haploinsufficiency   \n",
       "198                                            NaN  \\\n",
       "418                                            NaN   \n",
       "\n",
       "    clingenDosageSensitivityMap.triplosensitivity clingenGeneValidity   \n",
       "198                                           NaN                 NaN  \\\n",
       "418                                           NaN                 NaN   \n",
       "\n",
       "    cosmic.roleInCancer  \n",
       "198  [oncogene, fusion]  \n",
       "418            [fusion]  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_data.genes.dropna(subset=[\"cosmic.roleInCancer\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
